<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html">
<link rel="next" href="maintaining-your-bioconductor-package.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#for-everyone"><i class="fa fa-check"></i><b>1.1</b> For Everyone</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#for-workshop-authors"><i class="fa fa-check"></i><b>1.2</b> For Workshop Authors</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#description"><i class="fa fa-check"></i><b>1.2.1</b> DESCRIPTION</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#classifying-your-workshop"><i class="fa fa-check"></i><b>1.2.2</b> Classifying your workshop</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#bookdown"><i class="fa fa-check"></i><b>1.2.3</b> Bookdown</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#deadlines-for-bioc2018"><i class="fa fa-check"></i><b>1.3</b> Deadlines for Bioc2018</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html"><i class="fa fa-check"></i><b>2</b> <em>R</em> and <em>Bioconductor</em> for everyone: an introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#description-1"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="2.1.7" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-goals"><i class="fa fa-check"></i><b>2.1.7</b> Learning goals</a></li>
<li class="chapter" data-level="2.1.8" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-objectives"><i class="fa fa-check"></i><b>2.1.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-r"><i class="fa fa-check"></i><b>2.2</b> Introduction to <em>R</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#rstudio-orientation"><i class="fa fa-check"></i><b>2.2.1</b> <em>RStudio</em> orientation</a></li>
<li class="chapter" data-level="2.2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#very-basics"><i class="fa fa-check"></i><b>2.2.2</b> Very basics</a></li>
<li class="chapter" data-level="2.2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#data-input-and-manipulation"><i class="fa fa-check"></i><b>2.2.3</b> Data input and manipulation</a></li>
<li class="chapter" data-level="2.2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#writing-r-scripts"><i class="fa fa-check"></i><b>2.2.4</b> Writing <em>R</em> scripts</a></li>
<li class="chapter" data-level="2.2.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#using-packages"><i class="fa fa-check"></i><b>2.2.5</b> Using packages</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-bioconductor"><i class="fa fa-check"></i><b>2.3</b> Introduction to <em>Bioconductor</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#discovering-installing-and-learning-how-to-use-bioconductor-packages"><i class="fa fa-check"></i><b>2.3.1</b> Discovering, installing, and learning how to use <em>Bioconductor</em> packages</a></li>
<li class="chapter" data-level="2.3.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-genomic-ranges"><i class="fa fa-check"></i><b>2.3.2</b> Working with Genomic Ranges</a></li>
<li class="chapter" data-level="2.3.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-summarized-experimental-data"><i class="fa fa-check"></i><b>2.3.3</b> Working with summarized experimental data</a></li>
<li class="chapter" data-level="2.3.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#down-stream-analysis"><i class="fa fa-check"></i><b>2.3.4</b> Down-stream analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>3</b> RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview-1"><i class="fa fa-check"></i><b>3.1</b> Overview</a><ul>
<li class="chapter" data-level="3.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-2"><i class="fa fa-check"></i><b>3.1.1</b> Description</a></li>
<li class="chapter" data-level="3.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites-1"><i class="fa fa-check"></i><b>3.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="3.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation-1"><i class="fa fa-check"></i><b>3.1.3</b> Participation</a></li>
<li class="chapter" data-level="3.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="3.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline-1"><i class="fa fa-check"></i><b>3.1.5</b> Time outline</a></li>
<li class="chapter" data-level="3.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>3.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="3.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>3.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="3.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>3.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="3.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>3.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="3.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>3.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>3.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="3.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>3.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="3.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>3.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="3.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>3.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>3.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="3.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>3.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="3.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>3.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="3.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>3.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis"><i class="fa fa-check"></i><b>3.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="3.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>3.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="3.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>3.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub"><i class="fa fa-check"></i><b>3.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="3.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub"><i class="fa fa-check"></i><b>3.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="3.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>3.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>3.7</b> Building reports</a><ul>
<li class="chapter" data-level="3.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>3.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="3.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>3.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>3.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="3.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>3.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="3.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>3.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="3.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>3.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="3.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>3.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="3.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>3.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="3.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>3.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>4</b> Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-2"><i class="fa fa-check"></i><b>4.1</b> Overview</a><ul>
<li class="chapter" data-level="4.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-3"><i class="fa fa-check"></i><b>4.1.1</b> Description</a></li>
<li class="chapter" data-level="4.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-2"><i class="fa fa-check"></i><b>4.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-2"><i class="fa fa-check"></i><b>4.1.3</b> Participation</a></li>
<li class="chapter" data-level="4.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-2"><i class="fa fa-check"></i><b>4.1.5</b> Time outline</a></li>
<li class="chapter" data-level="4.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started"><i class="fa fa-check"></i><b>4.2</b> Getting started</a><ul>
<li class="chapter" data-level="4.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>4.2.1</b> The data</a></li>
<li class="chapter" data-level="4.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>4.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>4.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="4.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>4.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="4.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>4.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>4.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="4.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>4.5.1</b> Normalization</a></li>
<li class="chapter" data-level="4.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>4.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>4.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="4.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>4.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="4.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>4.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>5</b> Cytoscape Automation in R using Rcy3</a><ul>
<li class="chapter" data-level="5.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-3"><i class="fa fa-check"></i><b>5.1</b> Overview</a><ul>
<li class="chapter" data-level="5.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructors-names-and-contact-information"><i class="fa fa-check"></i><b>5.1.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="5.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description"><i class="fa fa-check"></i><b>5.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="5.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>5.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="5.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>5.1.4</b> Background:</a></li>
<li class="chapter" data-level="5.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation"><i class="fa fa-check"></i><b>5.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>5.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="5.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-3"><i class="fa fa-check"></i><b>5.1.7</b> Time outline</a></li>
<li class="chapter" data-level="5.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>5.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>5.2</b> Background</a><ul>
<li class="chapter" data-level="5.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>5.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="5.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>5.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="5.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>5.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>5.3</b> Translating biological data into Cytoscape using RCy3</a></li>
<li class="chapter" data-level="5.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>5.4</b> Set up</a><ul>
<li class="chapter" data-level="5.4.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started-1"><i class="fa fa-check"></i><b>5.4.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>5.5</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="5.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>5.6</b> Example Data Set</a></li>
<li class="chapter" data-level="5.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>5.7</b> Finding Network Data</a></li>
<li class="chapter" data-level="5.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>5.8</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="5.9" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>5.9</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="5.10" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>5.10</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
<li class="chapter" data-level="5.11" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#references"><i class="fa fa-check"></i><b>5.11</b> References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><i class="fa fa-check"></i><b>6</b> Effectively using the DelayedArray framework to support the analysis of large datasets</a><ul>
<li class="chapter" data-level="6.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-4"><i class="fa fa-check"></i><b>6.1</b> Overview</a><ul>
<li class="chapter" data-level="6.1.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#description-4"><i class="fa fa-check"></i><b>6.1.1</b> Description</a></li>
<li class="chapter" data-level="6.1.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.1.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#participation-3"><i class="fa fa-check"></i><b>6.1.3</b> Participation</a></li>
<li class="chapter" data-level="6.1.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>6.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.1.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#time-outline-4"><i class="fa fa-check"></i><b>6.1.5</b> Time outline</a></li>
<li class="chapter" data-level="6.1.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>6.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#introductory-material"><i class="fa fa-check"></i><b>6.2</b> Introductory material</a><ul>
<li class="chapter" data-level="6.2.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#discussion"><i class="fa fa-check"></i><b>6.2.1</b> Discussion</a></li>
<li class="chapter" data-level="6.2.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#why-learn-delayedarray"><i class="fa fa-check"></i><b>6.2.2</b> Why learn DelayedArray?</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-of-delayedarray-framework"><i class="fa fa-check"></i><b>6.3</b> Overview of DelayedArray framework</a><ul>
<li class="chapter" data-level="6.3.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-package"><i class="fa fa-check"></i><b>6.3.1</b> The <strong>DelayedArray</strong> package</a></li>
<li class="chapter" data-level="6.3.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#packages-that-extend-delayedarray"><i class="fa fa-check"></i><b>6.3.2</b> Packages that extend <strong>DelayedArray</strong></a></li>
<li class="chapter" data-level="6.3.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#packages-that-use-delayedarray"><i class="fa fa-check"></i><b>6.3.3</b> Packages that use <strong>DelayedArray</strong></a></li>
<li class="chapter" data-level="6.3.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-class"><i class="fa fa-check"></i><b>6.3.4</b> The <em>DelayedArray</em> class</a></li>
<li class="chapter" data-level="6.3.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fundamental-concepts"><i class="fa fa-check"></i><b>6.3.5</b> Fundamental concepts</a></li>
<li class="chapter" data-level="6.3.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#whats-out-there-already"><i class="fa fa-check"></i><b>6.3.6</b> Whatâ€™s out there already?</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#incoporating-delayedarray-into-a-package"><i class="fa fa-check"></i><b>6.4</b> Incoporating DelayedArray into a package</a><ul>
<li class="chapter" data-level="6.4.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#writing-algorithms-to-process-delayedarray-instances"><i class="fa fa-check"></i><b>6.4.1</b> Writing algorithms to process <em>DelayedArray</em> instances</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#questions-and-discussion"><i class="fa fa-check"></i><b>6.5</b> Questions and discussion</a></li>
<li class="chapter" data-level="6.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#todos"><i class="fa fa-check"></i><b>6.6</b> TODOs</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><i class="fa fa-check"></i><b>7</b> Variant Functional Annotation using StatePaintR, FunciVar &amp; MotifBreakR</a><ul>
<li class="chapter" data-level="7.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i><b>7.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="7.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-description-1"><i class="fa fa-check"></i><b>7.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="7.2.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.2.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-participation-1"><i class="fa fa-check"></i><b>7.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="7.2.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>7.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.2.4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#time-outline-5"><i class="fa fa-check"></i><b>7.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>7.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="7.3.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#key-concepts"><i class="fa fa-check"></i><b>7.3.1</b> Key Concepts</a></li>
<li class="chapter" data-level="7.3.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#learning-objectives-4"><i class="fa fa-check"></i><b>7.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>8</b> Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="8.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructors-names-and-contact-information-2"><i class="fa fa-check"></i><b>8.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="8.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-2"><i class="fa fa-check"></i><b>8.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="8.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-2"><i class="fa fa-check"></i><b>8.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="8.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>8.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-6"><i class="fa fa-check"></i><b>8.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>8.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="8.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop"><i class="fa fa-check"></i><b>8.4</b> Workshop</a></li>
<li class="chapter" data-level="8.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>8.5</b> Where does it all come from?</a></li>
<li class="chapter" data-level="8.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>8.6</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="8.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>8.7</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="8.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>8.8</b> Data types</a></li>
<li class="chapter" data-level="8.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>8.9</b> Differential expression analysis</a></li>
<li class="chapter" data-level="8.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>8.10</b> Gene sets</a></li>
<li class="chapter" data-level="8.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>8.11</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="8.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>8.12</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="8.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>8.13</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="8.14" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>8.14</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>9</b> RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="9.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>9.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-3"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-6"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-3"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-7"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>9.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="9.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals-4"><i class="fa fa-check"></i><b>9.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="9.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-5"><i class="fa fa-check"></i><b>9.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-1"><i class="fa fa-check"></i><b>9.4</b> Introduction</a></li>
<li class="chapter" data-level="9.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>9.5</b> Data packaging</a><ul>
<li class="chapter" data-level="9.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>9.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="9.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>9.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="9.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>9.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>9.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="9.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>9.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="9.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>9.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="9.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>9.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="9.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>9.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>9.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="9.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>9.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="9.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>9.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="9.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>9.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="9.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>9.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="9.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>9.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="9.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>9.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>10</b> Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="10.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructors-names-and-contact-information-3"><i class="fa fa-check"></i><b>10.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-4"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-7"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-4"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>10.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-8"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-5"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-6"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>10.4</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="10.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>10.5</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>10.6</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>10.7</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>10.8</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#grangeslist-groups-of-genomic-ranges"><i class="fa fa-check"></i><b>10.9</b> <em>GRangesList</em>: Groups of Genomic Ranges</a></li>
<li class="chapter" data-level="10.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-grangeslist-accessors"><i class="fa fa-check"></i><b>10.10</b> Basic <em>GRangesList</em> accessors</a></li>
<li class="chapter" data-level="10.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#combining-grangeslist-objects"><i class="fa fa-check"></i><b>10.11</b> Combining <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-grangeslist-objects"><i class="fa fa-check"></i><b>10.12</b> Basic interval operations for <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-grangeslist-objects"><i class="fa fa-check"></i><b>10.13</b> Subsetting <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#looping-over-grangeslist-objects"><i class="fa fa-check"></i><b>10.14</b> Looping over <em>GRangesList</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>11</b> Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="11.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructors-names-and-contact-information-4"><i class="fa fa-check"></i><b>11.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="11.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-5"><i class="fa fa-check"></i><b>11.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="11.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-8"><i class="fa fa-check"></i><b>11.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="11.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-5"><i class="fa fa-check"></i><b>11.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-9"><i class="fa fa-check"></i><b>11.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="11.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-9"><i class="fa fa-check"></i><b>11.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="11.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-6"><i class="fa fa-check"></i><b>11.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="11.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-7"><i class="fa fa-check"></i><b>11.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-1"><i class="fa fa-check"></i><b>11.4</b> Workshop</a></li>
<li class="chapter" data-level="11.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-2"><i class="fa fa-check"></i><b>11.5</b> Introduction</a><ul>
<li class="chapter" data-level="11.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>11.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="11.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>11.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="11.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>11.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup"><i class="fa fa-check"></i><b>11.6</b> Setup</a></li>
<li class="chapter" data-level="11.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>11.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="11.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>11.8</b> The Grammar</a><ul>
<li class="chapter" data-level="11.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>11.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="11.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>11.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="11.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises"><i class="fa fa-check"></i><b>11.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>11.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="11.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>11.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="11.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>11.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="11.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>11.10</b> Whatâ€™s next?</a></li>
<li class="chapter" data-level="11.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>11.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>12</b> Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="12.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>12.1</b> Instructors</a></li>
<li class="chapter" data-level="12.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description-6"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>12.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation-6"><i class="fa fa-check"></i><b>12.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-10"><i class="fa fa-check"></i><b>12.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-7"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-8"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>12.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="12.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>12.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="12.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>12.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="12.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>12.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="12.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>12.4.4</b> Data containers</a></li>
<li class="chapter" data-level="12.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>12.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="12.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>12.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="12.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>12.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="12.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>12.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="12.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>12.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="12.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>12.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="12.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>12.4.11</b> Simple example</a></li>
<li class="chapter" data-level="12.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>12.4.12</b> Questions!</a></li>
<li class="chapter" data-level="12.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>12.4.13</b> More questions!</a></li>
<li class="chapter" data-level="12.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>12.4.14</b> Another example</a></li>
<li class="chapter" data-level="12.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>12.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="12.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>12.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="12.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>12.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="12.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>12.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="12.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>12.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="12.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>12.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="12.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>12.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="12.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>12.4.22</b> GRanges</a></li>
<li class="chapter" data-level="12.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>12.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="12.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>12.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="12.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>12.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="12.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>12.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="12.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>12.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="12.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>12.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="12.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>12.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="12.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>12.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="12.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>12.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="12.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>12.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="12.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr"><i class="fa fa-check"></i><b>12.4.33</b> Organism.dplyr</a></li>
<li class="chapter" data-level="12.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-1"><i class="fa fa-check"></i><b>12.4.34</b> Organism.dplyr</a></li>
<li class="chapter" data-level="12.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-2"><i class="fa fa-check"></i><b>12.4.35</b> Organism.dplyr</a></li>
<li class="chapter" data-level="12.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>12.4.36</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="12.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>12.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="12.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>12.4.38</b> BSgenome packages</a></li>
<li class="chapter" data-level="12.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>12.4.39</b> BSgenome packages</a></li>
<li class="chapter" data-level="12.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>12.4.40</b> BSgenome exercises</a></li>
<li class="chapter" data-level="12.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-1"><i class="fa fa-check"></i><b>12.4.41</b> AnnotationHub</a></li>
<li class="chapter" data-level="12.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>12.4.42</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="12.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>12.4.43</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="12.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>12.4.44</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="12.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>12.4.45</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="12.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>12.4.46</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="12.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>12.4.47</b> AnnotationHub query</a></li>
<li class="chapter" data-level="12.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>12.4.48</b> AnnotationHub query</a></li>
<li class="chapter" data-level="12.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>12.4.49</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="12.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>12.4.50</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="12.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>12.4.51</b> biomaRt</a></li>
<li class="chapter" data-level="12.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>12.4.52</b> biomaRt data sets</a></li>
<li class="chapter" data-level="12.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>12.4.53</b> biomaRt queries</a></li>
<li class="chapter" data-level="12.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>12.4.54</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="12.4.55" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>12.4.55</b> biomaRt query</a></li>
<li class="chapter" data-level="12.4.56" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>12.4.56</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>13</b> Workflow for Multi-omics Analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>13.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-7"><i class="fa fa-check"></i><b>13.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="13.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-9"><i class="fa fa-check"></i><b>13.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-7"><i class="fa fa-check"></i><b>13.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>13.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="13.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-10"><i class="fa fa-check"></i><b>13.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="13.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-8"><i class="fa fa-check"></i><b>13.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="13.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-9"><i class="fa fa-check"></i><b>13.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>13.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="13.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>13.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="13.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>13.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="13.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>13.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>13.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="13.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>13.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="13.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>13.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="13.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>13.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>13.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="13.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>13.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="13.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>13.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="13.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>13.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="13.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>13.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="13.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>13.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="13.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>13.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>13.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="13.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>13.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="13.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>13.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="13.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>13.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>13.9</b> Complete cases</a></li>
<li class="chapter" data-level="13.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>13.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="13.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>13.11</b> Extraction</a><ul>
<li class="chapter" data-level="13.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>13.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="13.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>13.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="13.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>13.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="13.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>13.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="13.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>13.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="13.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>13.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="13.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>13.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="13.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>13.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="13.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>13.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="13.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>13.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="13.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>13.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="13.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>13.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="13.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>13.16.1</b> â€œSimplificationâ€ of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="13.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>13.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="13.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>13.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="13.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>13.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="13.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>13.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="13.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>13.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="13.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>13.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="13.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>13.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="13.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>13.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="13.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>13.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="13.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>13.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="13.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>13.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>14</b> Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="14.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>14.1</b> Instructors:</a></li>
<li class="chapter" data-level="14.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-8"><i class="fa fa-check"></i><b>14.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-10"><i class="fa fa-check"></i><b>14.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-8"><i class="fa fa-check"></i><b>14.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="14.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-11"><i class="fa fa-check"></i><b>14.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="14.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-11"><i class="fa fa-check"></i><b>14.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-11"><i class="fa fa-check"></i><b>14.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="14.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-9"><i class="fa fa-check"></i><b>14.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="14.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-10"><i class="fa fa-check"></i><b>14.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>15</b> Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="15.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>15.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="15.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-9"><i class="fa fa-check"></i><b>15.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="15.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-11"><i class="fa fa-check"></i><b>15.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="15.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-4"><i class="fa fa-check"></i><b>15.2.2</b> Participation</a></li>
<li class="chapter" data-level="15.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>15.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="15.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-12"><i class="fa fa-check"></i><b>15.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-4"><i class="fa fa-check"></i><b>15.3</b> Introduction</a></li>
<li class="chapter" data-level="15.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>15.4</b> Why maintain your package</a></li>
<li class="chapter" data-level="15.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>15.5</b> Infrastructure used for maintenance</a></li>
<li class="chapter" data-level="15.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>15.6</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="15.6.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>15.6.1</b> Build report</a></li>
<li class="chapter" data-level="15.6.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>15.6.2</b> Landing page badges</a></li>
<li class="chapter" data-level="15.6.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>15.6.3</b> Download statistics</a></li>
<li class="chapter" data-level="15.6.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>15.6.4</b> RSS feeds</a></li>
<li class="chapter" data-level="15.6.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>15.6.5</b> Support site, Bioc-devel mailing list</a></li>
<li class="chapter" data-level="15.6.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github"><i class="fa fa-check"></i><b>15.6.6</b> GitHub</a></li>
</ul></li>
<li class="chapter" data-level="15.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>15.7</b> Round up of resources available</a></li>
<li class="chapter" data-level="15.8" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#acknowledgements"><i class="fa fa-check"></i><b>15.8</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>16</b> Public Data Resources and Bioconductor</a><ul>
<li class="chapter" data-level="16.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>16.1</b> Instructor name(s) and contact information</a></li>
<li class="chapter" data-level="16.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>16.2</b> Syllabus</a><ul>
<li class="chapter" data-level="16.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-10"><i class="fa fa-check"></i><b>16.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="16.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-12"><i class="fa fa-check"></i><b>16.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="16.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-9"><i class="fa fa-check"></i><b>16.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="16.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>16.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="16.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-12"><i class="fa fa-check"></i><b>16.2.5</b> Time outline</a></li>
<li class="chapter" data-level="16.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-13"><i class="fa fa-check"></i><b>16.2.6</b> workshop goals and objectives</a></li>
<li class="chapter" data-level="16.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-10"><i class="fa fa-check"></i><b>16.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="16.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-11"><i class="fa fa-check"></i><b>16.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-5"><i class="fa fa-check"></i><b>16.3</b> Overview</a></li>
<li class="chapter" data-level="16.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>16.4</b> GEOquery</a><ul>
<li class="chapter" data-level="16.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>16.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="16.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>16.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="16.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>16.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="16.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>16.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>16.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="16.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>16.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="16.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>16.5.2</b> Basic design</a></li>
<li class="chapter" data-level="16.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>16.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="16.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>16.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="16.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>16.6.1</b> Authentication</a></li>
<li class="chapter" data-level="16.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>16.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>16.7</b> Sequence Read Archive</a></li>
<li class="chapter" data-level="16.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>16.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="16.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>16.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="16.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>16.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>16.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="16.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>16.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="16.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>16.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="16.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>16.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="16.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>16.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="16.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>16.12</b> Pharmacogenomics</a></li>
<li class="chapter" data-level="16.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>16.13</b> Bibliography</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="biomarker-discovery-from-large-pharmacogenomics-datasets" class="section level1">
<h1><span class="header-section-number">14</span> Biomarker discovery from large pharmacogenomics datasets</h1>
<div id="instructors-1" class="section level2">
<h2><span class="header-section-number">14.1</span> Instructors:</h2>
<ul>
<li>Zhaleh Safikhani (<a href="mailto:zhaleh.safikhani@utoront.ca">zhaleh.safikhani@utoront.ca</a>)</li>
<li>Petr Smirnov (<a href="mailto:petr.smirnov@mail.utoronto.ca">petr.smirnov@mail.utoronto.ca</a>)</li>
<li>Benjamin Haib-Kains (<a href="mailto:benjamin.haibe.kains@utoronto.ca">benjamin.haibe.kains@utoronto.ca</a>)</li>
</ul>
</div>
<div id="workshop-description-8" class="section level2">
<h2><span class="header-section-number">14.2</span> Workshop Description</h2>
<p>This workshop will focus on the challenges encountered when applying machine learning techniques in complex, high dimensional biological data. In particular, we will focus on biomarker discovery from pharmacogenomic data, which consists of developing predictors of response of cancer cell lines to chemical compounds based on their genomic features. From a methodological viewpoint, biomarker discovery is strongly linked to variable selection, through methods such as Supervised Learning with sparsity inducing norms (e.g., ElasticNet) or techniques accounting for the complex correlation structure of biological features (e.g., mRMR). Yet, the main focus of this talk will be on sound use of such methods in a pharmacogenomics context, their validation and correct interpretation of the produced results. We will discuss how to assess the quality of both the input and output data. We will illustrate the importance of unified analytical platforms, data and code sharing in bioinformatics and biomedical research, as the data generation process becomes increasingly complex and requires high level of replication to achieve robust results. This is particularly relevant as our portfolio of machine learning techniques is ever enlarging, with its set of hyperparameters that can be tuning in a multitude of ways, increasing the risk of overfitting when developing multivariate predictors of drug response.</p>
<div id="pre-requisites-10" class="section level3">
<h3><span class="header-section-number">14.2.1</span> Pre-requisites</h3>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Familiarity with the machine learning concept and at least a few approaches</li>
</ul>
<p>Following resources might be useful to read:</p>
<ul>
<li><a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btv723" class="uri">https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btv723</a></li>
<li><a href="https://academic.oup.com/nar/article/46/D1/D994/4372597" class="uri">https://academic.oup.com/nar/article/46/D1/D994/4372597</a></li>
<li><a href="https://web.stanford.edu/~hastie/Papers/ESLII.pdf" class="uri">https://web.stanford.edu/~hastie/Papers/ESLII.pdf</a></li>
</ul>
</div>
<div id="workshop-participation-8" class="section level3">
<h3><span class="header-section-number">14.2.2</span> Workshop Participation</h3>
<p>Participants expected to have the following required packages installed on their machines to be able to run the commands along with the instructors. * PharmacoGx and Biobase from Bioconductor * mRMRe, caret, glmnet, randomForest from cran * bhklab/mci and bhklab/PharmacoGx-ML from github</p>
</div>
<div id="r-bioconductor-packages-used-11" class="section level3">
<h3><span class="header-section-number">14.2.3</span> <em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/PharmacoGx.html" class="uri">https://bioconductor.org/packages/release/bioc/html/PharmacoGx.html</a></li>
</ul>
</div>
<div id="time-outline-11" class="section level3">
<h3><span class="header-section-number">14.2.4</span> Time outline</h3>
<p>An example for a 45-minute workshop:</p>
<table>
<thead>
<tr class="header">
<th>Activity</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Introduction</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Basic functionalities of PharmacoGx</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>Consistency assessment between datasets</td>
<td>15m</td>
</tr>
<tr class="even">
<td>Machine learning and biomarker discovery</td>
<td>20m</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="workshop-goals-and-objectives-11" class="section level2">
<h2><span class="header-section-number">14.3</span> Workshop goals and objectives</h2>
<div id="learning-goals-9" class="section level3">
<h3><span class="header-section-number">14.3.1</span> Learning goals</h3>
<ul>
<li>describe the pharmacogenomic datasets and their usefulness</li>
<li>learn how to extract information from these datasets and to intersect them over their common features</li>
<li>identify functionalities available in PharmacoGx package to work with the high dimensional pharmacogenomics data</li>
<li>assess reproducibility and replication of pharmacogenomics studies</li>
<li>understand how to handle the biomarker discovery as a pattern recognition problem in the domain of pharmacogenomics studies</li>
</ul>
</div>
<div id="learning-objectives-10" class="section level3">
<h3><span class="header-section-number">14.3.2</span> Learning objectives</h3>
<ul>
<li>list available standardized pharmacogenomic datasets and download them</li>
<li>understand the structure of these darasest and how to access the features and response quantifications</li>
<li>create drug-dose response plots</li>
<li>Measure the consistency across multiple datasets and how to improve such measurements</li>
<li>Assess whether known biomarkers are reproduced within these datasets</li>
<li>Predict new biomarkers by applying different machine learning methods</li>
</ul>

<p>This course will focus on the challenges encountered when applying machine learning techniques in complex, high dimensional biological data. In particular, we will focus on biomarker discovery from pharmacogenomic data, which consists of developing predictors of response of cancer cell lines to chemical compounds based on their genomic features. From a methodological viewpoint, biomarker discovery is strongly linked to variable selection, through methods such as Supervised Learning with sparsity inducing norms (e.g., ElasticNet) or techniques accounting for the complex correlation structure of biological features (e.g., mRMR). Yet, the main focus of this talk will be on sound use of such methods in a pharmacogenomics context, their validation and correct interpretation of the produced results. We will discuss how to assess the quality of both the input and output data. We will illustrate the importance of unified analytical platforms, data and code sharing in bioinformatics and biomedical research, as the data generation process becomes increasingly complex and requires high level of replication to achieve robust results. This is particularly relevant as our portfolio of machine learning techniques is ever enlarging, with its set of hyperparameters that can be tuning in a multitude of ways, increasing the risk of overfitting when developing multivariate predictors of drug response.</p>

<p>Pharmacogenomics holds much potential to aid in discovering drug response biomarkers and developing novel targeted therapies, leading to development of precision medicine and working towards the goal of personalized therapy. Several large experiments have been conducted, both to molecularly characterize drug dose response across many cell lines, and to examine the molecular response to drug administration. However, the experiments lack a standardization of protocols and annotations, hindering meta-analysis across several experiments.</p>
<p><em>PharmacoGx</em> was developed to address these challenges, by providing a unified framework for downloading and analyzing large pharmacogenomic datasets which are extensively curated to ensure maximum overlap and consistency.</p>
<p><em>PharmacoGx</em> is based on a level of abstraction from the raw experimental data, and allows bioinformaticians and biologists to work with data at the level of genes, drugs and cell lines. This provides a more intuitive interface and, in combination with unified curation, simplifies analyses between multiple datasets.</p>
<p>Load <em>PharamacoGx</em> into your current workspace:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(PharmacoGx, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; Warning in fun(libname, pkgname): couldn&#39;t connect to display &quot;:0&quot;</span>
<span class="kw">library</span>(mCI, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="kw">library</span>(PharmacoGxML, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="kw">library</span>(Biobase, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;BiocGenerics&#39;</span>
<span class="co">#&gt; The following objects are masked from &#39;package:parallel&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class="co">#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
<span class="co">#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB</span>
<span class="co">#&gt; The following objects are masked from &#39;package:stats&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from &#39;package:base&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind,</span>
<span class="co">#&gt;     colMeans, colnames, colSums, dirname, do.call, duplicated,</span>
<span class="co">#&gt;     eval, evalq, Filter, Find, get, grep, grepl, intersect,</span>
<span class="co">#&gt;     is.unsorted, lapply, lengths, Map, mapply, match, mget, order,</span>
<span class="co">#&gt;     paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind,</span>
<span class="co">#&gt;     Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,</span>
<span class="co">#&gt;     table, tapply, union, unique, unsplit, which, which.max,</span>
<span class="co">#&gt;     which.min</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     &#39;browseVignettes()&#39;. To cite Bioconductor, see</span>
<span class="co">#&gt;     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</span></code></pre></div>

<p>We have made the PharmacoSet objects of the curated datasets available for download using functions provided in the package. A table of available PharmacoSet objects can be obtained by using the <em>availablePSets</em> function. Any of the PharmacoSets in the table can then be downloaded by calling <em>downloadPSet</em>, which saves the datasets into a directory of the users choice, and returns the data into the R session.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  <span class="kw">availablePSets</span>(<span class="dt">saveDir=</span><span class="kw">file.path</span>(<span class="st">&quot;.&quot;</span>, <span class="st">&quot;Safikhani_Pharmacogenomics&quot;</span>))
  GDSC &lt;-<span class="st"> </span><span class="kw">downloadPSet</span>(<span class="st">&quot;GDSC&quot;</span>, <span class="dt">saveDir=</span><span class="kw">file.path</span>(<span class="st">&quot;.&quot;</span>, <span class="st">&quot;Safikhani_Pharmacogenomics&quot;</span>)) 
  CCLE &lt;-<span class="st"> </span><span class="kw">downloadPSet</span>(<span class="st">&quot;CCLE&quot;</span>, <span class="dt">saveDir=</span><span class="kw">file.path</span>(<span class="st">&quot;.&quot;</span>, <span class="st">&quot;Safikhani_Pharmacogenomics&quot;</span>))</code></pre></div>

<p><em>PharmacoGx</em> can be used to process pharmacogenomic datasets. First we want to check the heterogenity of cell lines in one of the available psets, CCLE.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mycol &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;#8dd3c7&quot;</span>,<span class="st">&quot;#ffffb3&quot;</span>,<span class="st">&quot;#bebada&quot;</span>,<span class="st">&quot;#fb8072&quot;</span>,<span class="st">&quot;#80b1d3&quot;</span>,<span class="st">&quot;#fdb462&quot;</span>,
           <span class="st">&quot;#b3de69&quot;</span>,<span class="st">&quot;#fccde5&quot;</span>,<span class="st">&quot;#d9d9d9&quot;</span>,<span class="st">&quot;#bc80bd&quot;</span>,<span class="st">&quot;#ccebc5&quot;</span>,<span class="st">&quot;#ffed6f&quot;</span>,
           <span class="st">&quot;#a6cee3&quot;</span>,<span class="st">&quot;#1f78b4&quot;</span>,<span class="st">&quot;#b2df8a&quot;</span>,<span class="st">&quot;#33a02c&quot;</span>,<span class="st">&quot;#fb9a99&quot;</span>,<span class="st">&quot;#e31a1c&quot;</span>,
           <span class="st">&quot;#fdbf6f&quot;</span>,<span class="st">&quot;#ff7f00&quot;</span>,<span class="st">&quot;#cab2d6&quot;</span>,<span class="st">&quot;#6a3d9a&quot;</span>,<span class="st">&quot;#ffff99&quot;</span>,<span class="st">&quot;#b15928&quot;</span>)
<span class="kw">pie</span>(<span class="kw">table</span>(CCLE<span class="op">@</span>cell[,<span class="st">&quot;tissueid&quot;</span>]), 
    <span class="dt">col=</span>mycol, 
    <span class="dt">main=</span><span class="st">&quot;Tissue types&quot;</span>, 
    <span class="dt">radius=</span><span class="dv">1</span>, 
    <span class="dt">cex=</span><span class="fl">0.8</span>)</code></pre></div>
<div class="figure">
<img src="Safikhani_Pharmacogenomics_files/figure-html/pie_chart-1.png" alt="Tissue of origin of cell lines in CCLE study" width="672" />
<p class="caption">
(#fig:pie_chart)Tissue of origin of cell lines in CCLE study
</p>
</div>

<p>Drug-Dose response data included in the PharmacoSet objects can be conviniently plotted using the <em>drugDoseResponseCurve</em> function. Given a list of PharmacoSets, a drug name and a cell name, it will plot the drug dose response curves for the given cell-drug combination in each dataset, allowing direct comparisons of data between datasets.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">CCLE.auc &lt;-<span class="st"> </span><span class="kw">summarizeSensitivityProfiles</span>(
                <span class="dt">pSet=</span>CCLE,
                <span class="dt">sensitivity.measure=</span><span class="st">&quot;auc_published&quot;</span>, 
                <span class="dt">summary.stat=</span><span class="st">&quot;median&quot;</span>,
                <span class="dt">verbose=</span><span class="ot">FALSE</span>)

lapatinib.aac &lt;-<span class="st"> </span>CCLE.auc[<span class="st">&quot;lapatinib&quot;</span>,]
cells &lt;-<span class="st"> </span><span class="kw">names</span>(lapatinib.aac)[
                <span class="kw">c</span>(<span class="kw">which.min</span>(lapatinib.aac), 
                  <span class="kw">which</span>((lapatinib.aac <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.2</span>) <span class="op">&amp;</span><span class="st"> </span>(lapatinib.aac <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.4</span>))[<span class="dv">1</span>],
                  <span class="kw">which.max</span>(lapatinib.aac))]
<span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>))
<span class="kw">drugDoseResponseCurve</span>(<span class="dt">drug=</span><span class="st">&quot;lapatinib&quot;</span>, <span class="dt">cellline=</span>cells[<span class="dv">1</span>], 
                      <span class="dt">pSets=</span>CCLE, <span class="dt">plot.type=</span><span class="st">&quot;Fitted&quot;</span>, 
                      <span class="dt">legends.label=</span><span class="st">&quot;auc_published&quot;</span>)
<span class="kw">drugDoseResponseCurve</span>(<span class="dt">drug=</span><span class="st">&quot;lapatinib&quot;</span>, <span class="dt">cellline=</span>cells[<span class="dv">2</span>], 
                      <span class="dt">pSets=</span>CCLE, <span class="dt">plot.type=</span><span class="st">&quot;Fitted&quot;</span>, 
                      <span class="dt">legends.label=</span><span class="st">&quot;auc_published&quot;</span>)
<span class="kw">drugDoseResponseCurve</span>(<span class="dt">drug=</span><span class="st">&quot;lapatinib&quot;</span>, <span class="dt">cellline=</span>cells[<span class="dv">3</span>], 
                      <span class="dt">pSets=</span>CCLE, <span class="dt">plot.type=</span><span class="st">&quot;Fitted&quot;</span>, 
                      <span class="dt">legends.label=</span><span class="st">&quot;auc_published&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:curves"></span>
<img src="Safikhani_Pharmacogenomics_files/figure-html/curves-1.png" alt="Cells response to lapatinib in CCLE" width="672" />
<p class="caption">
Figure 4.14: Cells response to lapatinib in CCLE
</p>
</div>

<p>In pharmacogenomic studies Cells were also tested for their response to increasing concentrations of various compounds, and form this the IC50 and AUC were computed. These pharmacological profiles are available for all the psets in <em>PharmacoGx</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggplot2, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="kw">library</span>(reshape2, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
melted_data &lt;-<span class="st"> </span><span class="kw">melt</span>(CCLE.auc)
NA_rows &lt;-<span class="st"> </span><span class="kw">unique</span>(<span class="kw">which</span>(<span class="kw">is.na</span>(melted_data), <span class="dt">arr.ind=</span>T)[,<span class="dv">1</span>])
melted_data &lt;-<span class="st"> </span>melted_data[<span class="op">-</span>NA_rows,]
<span class="kw">ggplot</span>(melted_data, <span class="kw">aes</span>(<span class="dt">x=</span>Var1,<span class="dt">y=</span>value)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">fill=</span><span class="st">&quot;gray&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">theme</span>(<span class="dt">axis.text.x=</span><span class="kw">element_text</span>(<span class="dt">angle=</span><span class="dv">90</span>,<span class="dt">hjust=</span><span class="dv">1</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="st">&quot;Drugs&quot;</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;AAC&quot;</span>)</code></pre></div>
<div class="figure">
<img src="Safikhani_Pharmacogenomics_files/figure-html/ccle_auc-1.png" alt="Cells response to drugs in CCLE" width="672" />
<p class="caption">
(#fig:ccle_auc)Cells response to drugs in CCLE
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#hist(CCLE.auc[&quot;lapatinib&quot;,], xlab=&quot;Cells response to lapatinib(AAC)&quot;, </span>
<span class="co">#     col=&quot;gray&quot;, main=&quot;&quot;)</span></code></pre></div>

<p>In this section we will investigate the consistency between the GDSC and CCLE datasets. In both CCLE and GDSC, the transcriptome of cells was profiled using an Affymatrix microarray chip. Cells were also tested for their response to increasing concentrations of various compounds, and form this the IC50 and AUC were computed. However, the cell and drugs names used between the two datasets were not consistent. Furthermore, two different microarray platforms were used. However, <em>PharmacoGx</em> allows us to overcome these differences to do a comparative study between these two datasets.</p>
<p>GDSC was profiled using the hgu133a platform, while CCLE was profiled with the expanded hgu133plus2 platform. While in this case the hgu133a is almost a strict subset of hgu133plus2 platform, the expression information in <em>PharmacoSet</em> objects is summarized by Ensemble Gene Ids, allowing datasets with different platforms to be directly compared. The probe to gene mapping is done using the BrainArray customCDF for each platform .</p>
<p>To begin, you would load the datasets from disk or download them using the <em>downloadPSet</em> function above.</p>
<p>We want to investigate the consistency of the data between the two datasets. The common intersection between the datasets can then be found using <em>intersectPSet</em>. We create a summary of the gene expression and drug sensitivity measures for both datasets, so we are left with one gene expression profile and one sensitivity profile per cell line within each dataset. We can then compare the gene expression and sensitivity measures between the datasets using a standard correlation coefficient.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">common &lt;-<span class="st"> </span><span class="kw">intersectPSet</span>(<span class="dt">pSets =</span> <span class="kw">list</span>(<span class="st">&quot;CCLE&quot;</span>=CCLE, <span class="st">&quot;GDSC&quot;</span>=GDSC), 
                        <span class="dt">intersectOn =</span> <span class="kw">c</span>(<span class="st">&quot;cell.lines&quot;</span>, <span class="st">&quot;drugs&quot;</span>), 
                        <span class="dt">strictIntersect =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; Intersecting large PSets may take a long time ...</span>
drugs &lt;-<span class="st"> </span><span class="kw">drugNames</span>(common<span class="op">$</span>CCLE)

##Example of concordant and discordant drug curves
cases &lt;-<span class="st"> </span><span class="kw">rbind</span>(
  <span class="kw">c</span>(<span class="st">&quot;CAL-85-1&quot;</span>, <span class="st">&quot;17-AAG&quot;</span>),
  <span class="kw">c</span>(<span class="st">&quot;HT-29&quot;</span>, <span class="st">&quot;PLX4720&quot;</span>),
  <span class="kw">c</span>(<span class="st">&quot;COLO-320-HSR&quot;</span>, <span class="st">&quot;AZD6244&quot;</span>),
  <span class="kw">c</span>(<span class="st">&quot;HT-1080&quot;</span>, <span class="st">&quot;PD-0332991&quot;</span>))

<span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>))
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">nrow</span>(cases)) {
  <span class="kw">drugDoseResponseCurve</span>(<span class="dt">pSets=</span>common, 
                        <span class="dt">drug=</span>cases[i,<span class="dv">2</span>], 
                        <span class="dt">cellline=</span>cases[i,<span class="dv">1</span>], 
                        <span class="dt">legends.label=</span><span class="st">&quot;ic50_published&quot;</span>, 
                        <span class="dt">plot.type=</span><span class="st">&quot;Fitted&quot;</span>, 
                        <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">130</span>))
}</code></pre></div>
<div class="figure">
<img src="Safikhani_Pharmacogenomics_files/figure-html/Replication_curves-1.png" alt="Consistency of drug response curves across studies" width="672" />
<p class="caption">
(#fig:Replication_curves)Consistency of drug response curves across studies
</p>
</div>

<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  
  ##AAC scatter plot 
GDSC.aac &lt;-<span class="st"> </span><span class="kw">summarizeSensitivityProfiles</span>(
    <span class="dt">pSet=</span>common<span class="op">$</span>GDSC,
    <span class="dt">sensitivity.measure=</span><span class="st">&#39;auc_recomputed&#39;</span>, 
    <span class="dt">summary.stat=</span><span class="st">&quot;median&quot;</span>,
    <span class="dt">verbose=</span><span class="ot">FALSE</span>)
CCLE.aac &lt;-<span class="st"> </span><span class="kw">summarizeSensitivityProfiles</span>(
  <span class="dt">pSet=</span>common<span class="op">$</span>CCLE,
  <span class="dt">sensitivity.measure=</span><span class="st">&#39;auc_recomputed&#39;</span>, 
  <span class="dt">summary.stat=</span><span class="st">&quot;median&quot;</span>,
  <span class="dt">verbose=</span><span class="ot">FALSE</span>)

GDSC.ic50 &lt;-<span class="st"> </span><span class="kw">summarizeSensitivityProfiles</span>(
    <span class="dt">pSet=</span>common<span class="op">$</span>GDSC, 
    <span class="dt">sensitivity.measure=</span><span class="st">&#39;ic50_recomputed&#39;</span>, 
    <span class="dt">summary.stat=</span><span class="st">&quot;median&quot;</span>,
    <span class="dt">verbose=</span><span class="ot">FALSE</span>)
CCLE.ic50 &lt;-<span class="st"> </span><span class="kw">summarizeSensitivityProfiles</span>(
  <span class="dt">pSet=</span>common<span class="op">$</span>CCLE, 
  <span class="dt">sensitivity.measure=</span><span class="st">&#39;ic50_recomputed&#39;</span>, 
  <span class="dt">summary.stat=</span><span class="st">&quot;median&quot;</span>,
  <span class="dt">verbose=</span><span class="ot">FALSE</span>)

drug &lt;-<span class="st"> &quot;lapatinib&quot;</span>
<span class="co">#par(mfrow=c(1, 2))</span>
<span class="kw">myScatterPlot</span>(<span class="dt">x=</span>GDSC.aac[drug,], 
              <span class="dt">y=</span>CCLE.aac[drug,], 
              <span class="dt">method=</span><span class="kw">c</span>(<span class="st">&quot;transparent&quot;</span>), 
              <span class="dt">transparency=</span><span class="fl">0.8</span>, <span class="dt">pch=</span><span class="dv">16</span>, <span class="dt">minp=</span><span class="dv">50</span>, 
              <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(<span class="kw">max</span>(GDSC.aac[drug,], <span class="dt">na.rm=</span>T), <span class="kw">max</span>(CCLE.aac[drug,], <span class="dt">na.rm=</span>T))), 
              <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">max</span>(<span class="kw">max</span>(GDSC.aac[drug,], <span class="dt">na.rm=</span>T), <span class="kw">max</span>(CCLE.aac[drug,], <span class="dt">na.rm=</span>T))), 
              <span class="dt">main=</span><span class="st">&quot;cells response to lapatinib&quot;</span>, 
              <span class="dt">cex.sub=</span><span class="fl">0.7</span>, 
              <span class="dt">xlab=</span><span class="st">&quot;AAC in GDSC&quot;</span>, 
              <span class="dt">ylab=</span><span class="st">&quot;AAC in CCLE&quot;</span>)
<span class="kw">legend</span>(<span class="st">&quot;topright&quot;</span>,
         <span class="dt">legend=</span><span class="kw">sprintf</span>(<span class="st">&quot;r=%s</span><span class="ch">\n</span><span class="st">rs=%s</span><span class="ch">\n</span><span class="st">CI=%s&quot;</span>, 
                 <span class="kw">round</span>(<span class="kw">cor</span>(GDSC.aac[drug,], 
                           CCLE.aac[drug,], 
                           <span class="dt">method=</span><span class="st">&quot;pearson&quot;</span>,
                           <span class="dt">use=</span><span class="st">&quot;pairwise.complete.obs&quot;</span>), 
                       <span class="dt">digits=</span><span class="dv">2</span>),
                 <span class="kw">round</span>(<span class="kw">cor</span>(GDSC.aac[drug,], 
                           CCLE.aac[drug,], 
                           <span class="dt">method=</span><span class="st">&quot;spearman&quot;</span>,
                           <span class="dt">use=</span><span class="st">&quot;pairwise.complete.obs&quot;</span>), 
                       <span class="dt">digits=</span><span class="dv">2</span>),
                 <span class="kw">round</span>(<span class="kw">paired.concordance.index</span>(GDSC.aac[drug,],
                                                CCLE.aac[drug,],
                                                <span class="dt">delta.pred=</span><span class="dv">0</span>,
                                                <span class="dt">delta.obs=</span><span class="dv">0</span>)<span class="op">$</span>cindex, 
                       <span class="dt">digits=</span><span class="dv">2</span>)), 
         <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>)</code></pre></div>
<div class="figure">
<img src="Safikhani_Pharmacogenomics_files/figure-html/sensitivity_scatter_plots-1.png" alt="Concordance of AAC values" width="960" />
<p class="caption">
(#fig:sensitivity_scatter_plots)Concordance of AAC values
</p>
</div>

<p>To better assess the concordance of multiple pharmacogenomic studies we introduced the modified concordance index (mCI). Recognizing that the noise in the drug screening assays is high and may yield to inaccurate sensitive-based ranking of cell lines with close AAC values, the mCI only considers cell line pairs with drug sensitivity (AAC) difference greater than <span class="math inline">\(\delta\)</span> .</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">c_index &lt;-<span class="st">  </span>mc_index &lt;-<span class="st"> </span><span class="ot">NULL</span>
<span class="cf">for</span>(drug <span class="cf">in</span> drugs){
  tt &lt;-<span class="st"> </span>mCI<span class="op">::</span><span class="kw">paired.concordance.index</span>(GDSC.aac[drug,], CCLE.aac[drug,], <span class="dt">delta.pred=</span><span class="dv">0</span>, <span class="dt">delta.obs=</span><span class="dv">0</span>, <span class="dt">alternative=</span><span class="st">&quot;greater&quot;</span>)
  c_index &lt;-<span class="st"> </span><span class="kw">c</span>(c_index, tt<span class="op">$</span>cindex)
  tt &lt;-<span class="st"> </span>mCI<span class="op">::</span><span class="kw">paired.concordance.index</span>(GDSC.aac[drug,], CCLE.aac[drug,], <span class="dt">delta.pred=</span><span class="fl">0.2</span>, <span class="dt">delta.obs=</span><span class="fl">0.2</span>, <span class="dt">alternative=</span><span class="st">&quot;greater&quot;</span>, <span class="dt">logic.operator=</span><span class="st">&quot;or&quot;</span>)
  mc_index &lt;-<span class="st"> </span><span class="kw">c</span>(mc_index, tt<span class="op">$</span>cindex)
}
mp &lt;-<span class="st"> </span><span class="kw">barplot</span>(<span class="kw">as.vector</span>(<span class="kw">rbind</span>(c_index, mc_index)), <span class="dt">beside=</span><span class="ot">TRUE</span>, <span class="dt">col=</span><span class="kw">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>), <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="dt">ylab=</span><span class="st">&quot;concordance index&quot;</span>, <span class="dt">space=</span><span class="kw">c</span>(.<span class="dv">15</span>,.<span class="dv">85</span>), <span class="dt">border=</span><span class="ot">NA</span>, <span class="dt">main=</span><span class="st">&quot;mCI&quot;</span>)
<span class="kw">text</span>(mp, <span class="kw">par</span>(<span class="st">&quot;usr&quot;</span>)[<span class="dv">3</span>], <span class="dt">labels=</span><span class="kw">as.vector</span>(<span class="kw">rbind</span>(drugs, <span class="kw">rep</span>(<span class="st">&quot;&quot;</span>, <span class="dv">15</span>))), <span class="dt">srt=</span><span class="dv">45</span>, <span class="dt">adj=</span><span class="kw">c</span>(<span class="fl">1.1</span>,<span class="fl">1.1</span>), <span class="dt">xpd=</span><span class="ot">TRUE</span>, <span class="dt">cex=</span>.<span class="dv">8</span>)
<span class="kw">abline</span>(<span class="dt">h=</span>.<span class="dv">7</span>, <span class="dt">lty=</span><span class="dv">2</span>)</code></pre></div>
<p><img src="Safikhani_Pharmacogenomics_files/figure-html/mci-1.png" width="672" /></p>

<p>The association between molecular features and response to a given drug is modelled using a linear regression model adjusted for tissue source: <span class="math display">\[Y = \beta_{0} + \beta_{i}G_i + \beta_{t}T + \beta_{b}B\]</span> where <span class="math inline">\(Y\)</span> denotes the drug sensitivity variable, <span class="math inline">\(G_i\)</span>, <span class="math inline">\(T\)</span> and <span class="math inline">\(B\)</span> denote the expression of gene <span class="math inline">\(i\)</span>, the tissue source and the experimental batch respectively, and <span class="math inline">\(\beta\)</span>s are the regression coefficients. The strength of gene-drug association is quantified by <span class="math inline">\(\beta_i\)</span>, above and beyond the relationship between drug sensitivity and tissue source. The variables <span class="math inline">\(Y\)</span> and <span class="math inline">\(G\)</span> are scaled (standard deviation equals to 1) to estimate standardized coefficients from the linear model. Significance of the gene-drug association is estimated by the statistical significance of <span class="math inline">\(\beta_i\)</span> (two-sided t test). P-values are then corrected for multiple testing using the false discovery rate (FDR) approach.</p>
<p>As an example of the reproducibility of biomarker discovery across pharmacogenomic studies, we can model the significance of the association between two drugs and their known biomarkers in CCLE and GDSC. We examine the association between drug <em>17-AAG</em> and gene <em>NQO1</em>, as well as drug <em>PD-0325901</em> and gene <em>BRAF</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  features &lt;-<span class="st"> </span>PharmacoGx<span class="op">::</span><span class="kw">fNames</span>(CCLE, <span class="st">&quot;rna&quot;</span>)[
    <span class="kw">which</span>(<span class="kw">featureInfo</span>(CCLE,
                      <span class="st">&quot;rna&quot;</span>)<span class="op">$</span>Symbol <span class="op">==</span><span class="st"> &quot;NQO1&quot;</span>)]
    ccle.sig.rna &lt;-<span class="st"> </span><span class="kw">drugSensitivitySig</span>(<span class="dt">pSet=</span>CCLE, 
                                  <span class="dt">mDataType=</span><span class="st">&quot;rna&quot;</span>, 
                                  <span class="dt">drugs=</span><span class="kw">c</span>(<span class="st">&quot;17-AAG&quot;</span>), 
                                  <span class="dt">features=</span>features, 
                                  <span class="dt">sensitivity.measure=</span><span class="st">&quot;auc_published&quot;</span>, 
                                  <span class="dt">molecular.summary.stat=</span><span class="st">&quot;median&quot;</span>, 
                                  <span class="dt">sensitivity.summary.stat=</span><span class="st">&quot;median&quot;</span>,
                                  <span class="dt">verbose=</span><span class="ot">FALSE</span>)
    gdsc.sig.rna &lt;-<span class="st"> </span><span class="kw">drugSensitivitySig</span>(<span class="dt">pSet=</span>GDSC, 
                                  <span class="dt">mDataType=</span><span class="st">&quot;rna&quot;</span>, 
                                  <span class="dt">drugs=</span><span class="kw">c</span>(<span class="st">&quot;17-AAG&quot;</span>), 
                                  <span class="dt">features=</span>features, 
                                  <span class="dt">sensitivity.measure=</span><span class="st">&quot;auc_published&quot;</span>, 
                                  <span class="dt">molecular.summary.stat=</span><span class="st">&quot;median&quot;</span>, 
                                  <span class="dt">sensitivity.summary.stat=</span><span class="st">&quot;median&quot;</span>,
                                  <span class="dt">verbose=</span><span class="ot">FALSE</span>)
    ccle.sig.mut &lt;-<span class="st"> </span><span class="kw">drugSensitivitySig</span>(<span class="dt">pSet=</span>CCLE, 
                                  <span class="dt">mDataType=</span><span class="st">&quot;mutation&quot;</span>, 
                                  <span class="dt">drugs=</span><span class="kw">c</span>(<span class="st">&quot;PD-0325901&quot;</span>), 
                                  <span class="dt">features=</span><span class="st">&quot;BRAF&quot;</span>, 
                                  <span class="dt">sensitivity.measure=</span><span class="st">&quot;auc_published&quot;</span>, 
                                  <span class="dt">molecular.summary.stat=</span><span class="st">&quot;and&quot;</span>, 
                                  <span class="dt">sensitivity.summary.stat=</span><span class="st">&quot;median&quot;</span>,
                                  <span class="dt">verbose=</span><span class="ot">FALSE</span>)
    gdsc.sig.mut &lt;-<span class="st"> </span><span class="kw">drugSensitivitySig</span>(<span class="dt">pSet=</span>GDSC, 
                                  <span class="dt">mDataType=</span><span class="st">&quot;mutation&quot;</span>, 
                                  <span class="dt">drugs=</span><span class="kw">c</span>(<span class="st">&quot;PD-0325901&quot;</span>), 
                                  <span class="dt">features=</span><span class="st">&quot;BRAF&quot;</span>, 
                                  <span class="dt">sensitivity.measure=</span><span class="st">&quot;auc_published&quot;</span>, 
                                  <span class="dt">molecular.summary.stat=</span><span class="st">&quot;and&quot;</span>, 
                                  <span class="dt">sensitivity.summary.stat=</span><span class="st">&quot;median&quot;</span>,
                                  <span class="dt">verbose=</span><span class="ot">FALSE</span>)
    ccle.sig &lt;-<span class="st"> </span><span class="kw">rbind</span>(ccle.sig.rna, ccle.sig.mut)
    gdsc.sig &lt;-<span class="st"> </span><span class="kw">rbind</span>(gdsc.sig.rna, gdsc.sig.mut)
    known.biomarkers &lt;-<span class="st"> </span><span class="kw">cbind</span>(<span class="st">&quot;GDSC effect size&quot;</span>=gdsc.sig[,<span class="dv">1</span>], 
                              <span class="st">&quot;GDSC pvalue&quot;</span>=gdsc.sig[,<span class="dv">6</span>], 
                              <span class="st">&quot;CCLE effect size&quot;</span>=ccle.sig[,<span class="dv">1</span>], 
                              <span class="st">&quot;CCLE pvalue&quot;</span>=ccle.sig[,<span class="dv">6</span>])
    <span class="kw">rownames</span>(known.biomarkers) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;17-AAG + NQO1&quot;</span>,<span class="st">&quot;PD-0325901 + BRAF&quot;</span>)
    <span class="kw">library</span>(xtable, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
    <span class="kw">xtable</span>(known.biomarkers, <span class="dt">digits=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="op">-</span><span class="dv">1</span>, <span class="dv">2</span>, <span class="op">-</span><span class="dv">1</span>), <span class="dt">caption=</span><span class="st">&#39;Concordance of biomarkers across stuudies&#39;</span>)
    <span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">2</span>))
    CCLE_expr &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">exprs</span>(<span class="kw">summarizeMolecularProfiles</span>(CCLE, <span class="dt">mDataType=</span><span class="st">&quot;rna&quot;</span>, <span class="dt">fill.missing=</span><span class="ot">FALSE</span>)))
    CCLE_cells &lt;-<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">rownames</span>(CCLE_expr), <span class="kw">colnames</span>(CCLE.aac))
    <span class="kw">plot</span>(CCLE.aac[<span class="st">&quot;17-AAG&quot;</span>, CCLE_cells], CCLE_expr[CCLE_cells, features],
     <span class="dt">main=</span><span class="st">&quot;CCLE + 17-AAG + NQO1&quot;</span>,
     <span class="dt">cex.main=</span><span class="dv">1</span>, <span class="dt">ylab=</span><span class="st">&quot;Predictions&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;drug sensitivity&quot;</span>, <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;gray40&quot;</span>)

    GDSC_expr &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">exprs</span>(<span class="kw">summarizeMolecularProfiles</span>(GDSC, <span class="dt">mDataType=</span><span class="st">&quot;rna&quot;</span>, <span class="dt">fill.missing=</span><span class="ot">FALSE</span>)))
<span class="co">#&gt; Summarizing rna molecular data for:  GDSC</span>
    GDSC_cells &lt;-<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">rownames</span>(GDSC_expr), <span class="kw">colnames</span>(GDSC.aac))
    <span class="kw">plot</span>(GDSC.aac[<span class="st">&quot;17-AAG&quot;</span>, GDSC_cells], GDSC_expr[GDSC_cells, features],
     <span class="dt">main=</span><span class="st">&quot;GDSC + 17-AAG + NQO1&quot;</span>,
     <span class="dt">cex.main=</span><span class="dv">1</span>, <span class="dt">ylab=</span><span class="st">&quot;Predictions&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;drug sensitivity&quot;</span>, <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">col=</span><span class="st">&quot;gray40&quot;</span>)
    
    CCLE_mut &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">exprs</span>(<span class="kw">summarizeMolecularProfiles</span>(CCLE, <span class="dt">mDataType=</span><span class="st">&quot;mutation&quot;</span>, <span class="dt">fill.missing=</span><span class="ot">FALSE</span>, <span class="dt">summary.stat=</span><span class="st">&quot;or&quot;</span>)))
CCLE_cells &lt;-<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">rownames</span>(CCLE_mut), <span class="kw">colnames</span>(CCLE.aac))
<span class="kw">boxplot</span>(CCLE.aac[<span class="st">&quot;PD-0325901&quot;</span>, CCLE_cells]<span class="op">~</span><span class="st"> </span>CCLE_mut[CCLE_cells, <span class="st">&quot;BRAF&quot;</span>], <span class="dt">col=</span><span class="st">&quot;gray80&quot;</span>, <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">main=</span><span class="st">&quot;CCLE + PD-0325901 + BRAF&quot;</span>,
        <span class="dt">cex.main=</span><span class="dv">1</span>, <span class="dt">xlab=</span><span class="st">&quot;mutation&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;drug sensitivity&quot;</span>)

GDSC_mut &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">exprs</span>(<span class="kw">summarizeMolecularProfiles</span>(GDSC, <span class="dt">mDataType=</span><span class="st">&quot;mutation&quot;</span>, <span class="dt">fill.missing=</span><span class="ot">FALSE</span>, <span class="dt">summary.stat=</span><span class="st">&quot;or&quot;</span>)))
GDSC_cells &lt;-<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">rownames</span>(GDSC_mut), <span class="kw">colnames</span>(GDSC.aac))
<span class="kw">boxplot</span>(GDSC.aac[<span class="st">&quot;PD-0325901&quot;</span>, GDSC_cells]<span class="op">~</span><span class="st"> </span>GDSC_mut[GDSC_cells, <span class="st">&quot;BRAF&quot;</span>], <span class="dt">col=</span><span class="st">&quot;gray80&quot;</span>, <span class="dt">pch=</span><span class="dv">20</span>, <span class="dt">main=</span><span class="st">&quot;GDSC + PD-0325901 + BRAF&quot;</span>,
        <span class="dt">cex.main=</span><span class="dv">1</span>, <span class="dt">xlab=</span><span class="st">&quot;mutation&quot;</span>, <span class="dt">ylab=</span><span class="st">&quot;drug sensitivity&quot;</span>)</code></pre></div>
<p><img src="Safikhani_Pharmacogenomics_files/figure-html/biomarker_discovery-1.png" width="672" /></p>

<p>Some of the widely used multivariate machine learning methods such as elastic net, Random Forest (RF) and Support Vector Machine (SVM) have been already implemented in the MLWorkshop. It optimizes hyperparameters of these methods in the training phase. To assess the performance of the predictive models, it implements <em>m</em> number of sampling with <em>n-fold</em> cross validations (CV). The performance will then be assessed by multiple metrics including pearson correlation coefficient, concordance index and modified concordance index.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(mRMRe, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; Loading required package: survival</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;survival&#39;</span>
<span class="co">#&gt; The following object is masked from &#39;package:PharmacoGxML&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     ridge</span>
<span class="co">#&gt; Loading required package: igraph</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;igraph&#39;</span>
<span class="co">#&gt; The following objects are masked from &#39;package:BiocGenerics&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     normalize, path, union</span>
<span class="co">#&gt; The following objects are masked from &#39;package:stats&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     decompose, spectrum</span>
<span class="co">#&gt; The following object is masked from &#39;package:base&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     union</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;mRMRe&#39;</span>
<span class="co">#&gt; The following objects are masked from &#39;package:Biobase&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     featureData, featureNames, sampleNames</span>
<span class="kw">library</span>(Biobase, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="kw">library</span>(Hmisc, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; Loading required package: lattice</span>
<span class="co">#&gt; Loading required package: Formula</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;Hmisc&#39;</span>
<span class="co">#&gt; The following objects are masked from &#39;package:xtable&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     label, label&lt;-</span>
<span class="co">#&gt; The following object is masked from &#39;package:Biobase&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     contents</span>
<span class="co">#&gt; The following objects are masked from &#39;package:base&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     format.pval, units</span>
<span class="kw">library</span>(glmnet, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Loaded glmnet 2.0-16</span>
<span class="kw">library</span>(caret, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;caret&#39;</span>
<span class="co">#&gt; The following object is masked from &#39;package:survival&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     cluster</span>
<span class="kw">library</span>(randomForest, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; randomForest 4.6-14</span>
<span class="co">#&gt; Type rfNews() to see new features/changes/bug fixes.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;randomForest&#39;</span>
<span class="co">#&gt; The following object is masked from &#39;package:ggplot2&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     margin</span>
<span class="co">#&gt; The following object is masked from &#39;package:Biobase&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine</span>
<span class="co">#&gt; The following object is masked from &#39;package:BiocGenerics&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine</span>

##Preparing trainig dataset
train_expr &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">exprs</span>(<span class="kw">summarizeMolecularProfiles</span>(GDSC, <span class="dt">mDataType=</span><span class="st">&quot;rna&quot;</span>, <span class="dt">fill.missing=</span><span class="ot">FALSE</span>, <span class="dt">verbose=</span><span class="ot">FALSE</span>)))
aac &lt;-<span class="st"> </span><span class="kw">summarizeSensitivityProfiles</span>(GDSC, <span class="dt">sensitivity.measure=</span><span class="st">&quot;auc_recomputed&quot;</span>, <span class="dt">drug=</span><span class="st">&quot;lapatinib&quot;</span>, <span class="dt">fill.missing=</span><span class="ot">FALSE</span>, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
cells &lt;-<span class="st"> </span><span class="kw">intersect</span>(<span class="kw">rownames</span>(train_expr), <span class="kw">names</span>(aac))
df &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">cbind</span>(train_expr[cells,], <span class="st">&quot;lapatinib&quot;</span>=aac[cells]))


##Preparing validation dataset
validation_expr &lt;-<span class="st"> </span><span class="kw">summarizeMolecularProfiles</span>(CCLE, <span class="dt">mDataType=</span><span class="st">&quot;rna&quot;</span>, <span class="dt">fill.missing=</span><span class="ot">FALSE</span>, <span class="dt">verbose=</span><span class="ot">FALSE</span>)
actual_labels &lt;-<span class="st"> </span><span class="kw">summarizeSensitivityProfiles</span>(CCLE, <span class="dt">sensitivity.measure=</span><span class="st">&quot;auc_recomputed&quot;</span>, <span class="dt">drug=</span><span class="st">&quot;lapatinib&quot;</span>, <span class="dt">fill.missing=</span><span class="ot">FALSE</span>, <span class="dt">verbose=</span><span class="ot">FALSE</span>)


<span class="cf">for</span>(method <span class="cf">in</span> <span class="kw">c</span>(<span class="st">&quot;ridge&quot;</span>, <span class="st">&quot;lasso&quot;</span>, <span class="st">&quot;random_forest&quot;</span>, <span class="st">&quot;svm&quot;</span>)){
  <span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))
  res &lt;-<span class="st"> </span><span class="kw">optimization</span>(<span class="dt">train=</span>df[, <span class="op">-</span><span class="kw">ncol</span>(df), <span class="dt">drop=</span>F],
                      <span class="dt">labels=</span><span class="kw">t</span>(df[, <span class="kw">ncol</span>(df), <span class="dt">drop=</span>F]),
                      <span class="dt">method=</span>method,
                      <span class="dt">folds.no=</span><span class="dv">5</span>,
                      <span class="dt">sampling.no=</span><span class="dv">1</span>,
                      <span class="dt">features.no=</span><span class="dv">10</span>,
                      <span class="dt">feature.selection=</span><span class="st">&quot;mRMR&quot;</span>,
                      <span class="dt">assessment=</span><span class="kw">c</span>(<span class="st">&quot;corr&quot;</span>, <span class="st">&quot;mCI&quot;</span>))
  
  validation_labels &lt;-<span class="st"> </span><span class="kw">validation</span>(<span class="dt">model=</span>res<span class="op">$</span>model<span class="op">$</span>lapatinib,
                                  <span class="dt">validation.set=</span><span class="kw">t</span>(<span class="kw">exprs</span>(validation_expr)),
                                  <span class="dt">validation.labels=</span>actual_labels,
                                  <span class="dt">method=</span>method,
                                  <span class="dt">assessment=</span><span class="st">&quot;mCI&quot;</span>)

}</code></pre></div>
<p><img src="Safikhani_Pharmacogenomics_files/figure-html/machine_learning-1.png" width="672" /><img src="Safikhani_Pharmacogenomics_files/figure-html/machine_learning-2.png" width="672" /><img src="Safikhani_Pharmacogenomics_files/figure-html/machine_learning-3.png" width="672" /><img src="Safikhani_Pharmacogenomics_files/figure-html/machine_learning-4.png" width="672" /> </p>

<p>This document was generated with the following R version and packages loaded:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  <span class="kw">sessionInfo</span>()
<span class="co">#&gt; R version 3.5.0 (2018-04-23)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Debian GNU/Linux 9 (stretch)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS: /usr/lib/openblas-base/libblas.so.3</span>
<span class="co">#&gt; LAPACK: /usr/lib/libopenblasp-r0.2.19.so</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] randomForest_4.6-14 caret_6.0-80        glmnet_2.0-16      </span>
<span class="co">#&gt;  [4] foreach_1.4.4       Matrix_1.2-14       Hmisc_4.1-1        </span>
<span class="co">#&gt;  [7] Formula_1.2-3       lattice_0.20-35     mRMRe_2.0.7        </span>
<span class="co">#&gt; [10] igraph_1.2.1        survival_2.42-6     xtable_1.8-2       </span>
<span class="co">#&gt; [13] reshape2_1.4.3      ggplot2_3.0.0       Biobase_2.41.1     </span>
<span class="co">#&gt; [16] BiocGenerics_0.27.1 PharmacoGxML_0.1.0  mCI_0.1.0          </span>
<span class="co">#&gt; [19] PharmacoGx_1.11.0  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] fgsea_1.7.1         colorspace_1.3-2    class_7.3-14       </span>
<span class="co">#&gt;   [4] rprojroot_1.3-2     htmlTable_1.12      lsa_0.73.1         </span>
<span class="co">#&gt;   [7] pls_2.6-0           base64enc_0.1-3     rstudioapi_0.7     </span>
<span class="co">#&gt;  [10] DRR_0.0.3           SnowballC_0.5.1     lubridate_1.7.4    </span>
<span class="co">#&gt;  [13] prodlim_2018.04.18  codetools_0.2-15    splines_3.5.0      </span>
<span class="co">#&gt;  [16] mnormt_1.5-5        robustbase_0.93-1   knitr_1.20         </span>
<span class="co">#&gt;  [19] RcppRoll_0.3.0      magicaxis_2.0.3     broom_0.4.5        </span>
<span class="co">#&gt;  [22] ddalpha_1.3.4       kernlab_0.9-26      cluster_2.0.7-1    </span>
<span class="co">#&gt;  [25] sfsmisc_1.1-2       mapproj_1.2.6       compiler_3.5.0     </span>
<span class="co">#&gt;  [28] backports_1.1.2     assertthat_0.2.0    lazyeval_0.2.1     </span>
<span class="co">#&gt;  [31] limma_3.37.3        acepack_1.4.1       htmltools_0.3.6    </span>
<span class="co">#&gt;  [34] tools_3.5.0         bindrcpp_0.2.2      gtable_0.2.0       </span>
<span class="co">#&gt;  [37] glue_1.2.0          RANN_2.5.1          dplyr_0.7.6        </span>
<span class="co">#&gt;  [40] maps_3.3.0          fastmatch_1.1-0     Rcpp_0.12.17       </span>
<span class="co">#&gt;  [43] slam_0.1-43         gdata_2.18.0        nlme_3.1-137       </span>
<span class="co">#&gt;  [46] iterators_1.0.10    psych_1.8.4         timeDate_3043.102  </span>
<span class="co">#&gt;  [49] gower_0.1.2         xfun_0.3            stringr_1.3.1      </span>
<span class="co">#&gt;  [52] gtools_3.8.1        DEoptimR_1.0-8      MASS_7.3-50        </span>
<span class="co">#&gt;  [55] scales_0.5.0        ipred_0.9-6         relations_0.6-8    </span>
<span class="co">#&gt;  [58] RColorBrewer_1.1-2  sets_1.0-18         yaml_2.1.19        </span>
<span class="co">#&gt;  [61] gridExtra_2.3       downloader_0.4      rpart_4.1-13       </span>
<span class="co">#&gt;  [64] latticeExtra_0.6-28 stringi_1.2.3       highr_0.7          </span>
<span class="co">#&gt;  [67] NISTunits_1.0.1     plotrix_3.7-2       checkmate_1.8.5    </span>
<span class="co">#&gt;  [70] caTools_1.17.1      BiocParallel_1.15.7 lava_1.6.2         </span>
<span class="co">#&gt;  [73] geometry_0.3-6      rlang_0.2.1         pkgconfig_2.0.1    </span>
<span class="co">#&gt;  [76] bitops_1.0-6        pracma_2.1.4        evaluate_0.10.1    </span>
<span class="co">#&gt;  [79] purrr_0.2.5         bindr_0.1.1         recipes_0.1.3      </span>
<span class="co">#&gt;  [82] htmlwidgets_1.2     labeling_0.3        CVST_0.2-2         </span>
<span class="co">#&gt;  [85] tidyselect_0.2.4    plyr_1.8.4          magrittr_1.5       </span>
<span class="co">#&gt;  [88] bookdown_0.7        R6_2.2.2            gplots_3.0.1       </span>
<span class="co">#&gt;  [91] dimRed_0.1.0        sm_2.2-5.5          pillar_1.3.0       </span>
<span class="co">#&gt;  [94] foreign_0.8-70      withr_2.1.2         abind_1.4-5        </span>
<span class="co">#&gt;  [97] nnet_7.3-12         tibble_1.4.2        crayon_1.3.4       </span>
<span class="co">#&gt; [100] KernSmooth_2.23-15  rmarkdown_1.10      grid_3.5.0         </span>
<span class="co">#&gt; [103] data.table_1.11.4   marray_1.59.0       piano_1.21.0       </span>
<span class="co">#&gt; [106] ModelMetrics_1.1.0  digest_0.6.15       tidyr_0.8.1        </span>
<span class="co">#&gt; [109] stats4_3.5.0        munsell_0.5.0       celestial_1.4.1    </span>
<span class="co">#&gt; [112] magic_1.5-8         tcltk_3.5.0</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="maintaining-your-bioconductor-package.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
