<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="rna-seq-data-analysis-with-deseq2.html">
<link rel="next" href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#for-everyone"><i class="fa fa-check"></i><b>1.1</b> For Everyone</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#for-workshop-authors"><i class="fa fa-check"></i><b>1.2</b> For Workshop Authors</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#description"><i class="fa fa-check"></i><b>1.2.1</b> DESCRIPTION</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#classifying-your-workshop"><i class="fa fa-check"></i><b>1.2.2</b> Classifying your workshop</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#bookdown"><i class="fa fa-check"></i><b>1.2.3</b> Bookdown</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#deadlines-for-bioc2018"><i class="fa fa-check"></i><b>1.3</b> Deadlines for Bioc2018</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>2</b> RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-1"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>2.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>2.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="2.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>2.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="2.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>2.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="2.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>2.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>2.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>2.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="2.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>2.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="2.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>2.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>2.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="2.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>2.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="2.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>2.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="2.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>2.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis"><i class="fa fa-check"></i><b>2.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="2.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>2.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="2.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>2.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub"><i class="fa fa-check"></i><b>2.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="2.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub"><i class="fa fa-check"></i><b>2.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="2.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>2.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>2.7</b> Building reports</a><ul>
<li class="chapter" data-level="2.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>2.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="2.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>2.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>2.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="2.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>2.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="2.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>2.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="2.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>2.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="2.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>2.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="2.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>2.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="2.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>2.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>3</b> Cytoscape Automation in R using Rcy3</a><ul>
<li class="chapter" data-level="3.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-1"><i class="fa fa-check"></i><b>3.1</b> Overview</a><ul>
<li class="chapter" data-level="3.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructors-names-and-contact-information"><i class="fa fa-check"></i><b>3.1.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="3.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description"><i class="fa fa-check"></i><b>3.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="3.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>3.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="3.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>3.1.4</b> Background:</a></li>
<li class="chapter" data-level="3.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation"><i class="fa fa-check"></i><b>3.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="3.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="3.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-1"><i class="fa fa-check"></i><b>3.1.7</b> Time outline</a></li>
<li class="chapter" data-level="3.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>3.2</b> Background</a><ul>
<li class="chapter" data-level="3.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>3.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="3.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>3.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="3.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>3.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>3.3</b> Translating biological data into Cytoscape using RCy3</a><ul>
<li class="chapter" data-level="3.3.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>3.3.1</b> Set up</a></li>
<li class="chapter" data-level="3.3.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started"><i class="fa fa-check"></i><b>3.3.2</b> Getting started</a></li>
<li class="chapter" data-level="3.3.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>3.3.3</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="3.3.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>3.3.4</b> Example Data Set</a></li>
<li class="chapter" data-level="3.3.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>3.3.5</b> Finding Network Data</a></li>
<li class="chapter" data-level="3.3.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>3.3.6</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="3.3.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>3.3.7</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="3.3.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>3.3.8</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#references"><i class="fa fa-check"></i><b>3.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><i class="fa fa-check"></i><b>4</b> Variant Functional Annotation using StatePaintR, FunciVar &amp; MotifBreakR</a><ul>
<li class="chapter" data-level="4.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i><b>4.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="4.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-description-1"><i class="fa fa-check"></i><b>4.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="4.2.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#pre-requisites-1"><i class="fa fa-check"></i><b>4.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.2.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-participation-1"><i class="fa fa-check"></i><b>4.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="4.2.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.2.4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#time-outline-2"><i class="fa fa-check"></i><b>4.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="4.3.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#key-concepts"><i class="fa fa-check"></i><b>4.3.1</b> Key Concepts</a></li>
<li class="chapter" data-level="4.3.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#learning-objectives"><i class="fa fa-check"></i><b>4.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>5</b> Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="5.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructors-names-and-contact-information-2"><i class="fa fa-check"></i><b>5.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="5.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-2"><i class="fa fa-check"></i><b>5.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="5.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-2"><i class="fa fa-check"></i><b>5.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="5.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-2"><i class="fa fa-check"></i><b>5.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>5.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="5.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-3"><i class="fa fa-check"></i><b>5.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>5.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="5.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop"><i class="fa fa-check"></i><b>5.4</b> Workshop</a></li>
<li class="chapter" data-level="5.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>5.5</b> Where does it all come from?</a></li>
<li class="chapter" data-level="5.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>5.6</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="5.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>5.7</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="5.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>5.8</b> Data types</a></li>
<li class="chapter" data-level="5.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>5.9</b> Differential expression analysis</a></li>
<li class="chapter" data-level="5.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>5.10</b> Gene sets</a></li>
<li class="chapter" data-level="5.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>5.11</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="5.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>5.12</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="5.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>5.13</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="5.14" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>5.14</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>6</b> RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>6.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-3"><i class="fa fa-check"></i><b>6.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="6.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-3"><i class="fa fa-check"></i><b>6.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="6.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>6.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-4"><i class="fa fa-check"></i><b>6.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>6.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="6.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals"><i class="fa fa-check"></i><b>6.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="6.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-1"><i class="fa fa-check"></i><b>6.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-1"><i class="fa fa-check"></i><b>6.4</b> Introduction</a></li>
<li class="chapter" data-level="6.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>6.5</b> Data packaging</a><ul>
<li class="chapter" data-level="6.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>6.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="6.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>6.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="6.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>6.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>6.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="6.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>6.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="6.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>6.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="6.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>6.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="6.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>6.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>6.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="6.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>6.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="6.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>6.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="6.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>6.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="6.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>6.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="6.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>6.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="6.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>6.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>7</b> Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="7.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructors-names-and-contact-information-3"><i class="fa fa-check"></i><b>7.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="7.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-4"><i class="fa fa-check"></i><b>7.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="7.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-4"><i class="fa fa-check"></i><b>7.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="7.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>7.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-5"><i class="fa fa-check"></i><b>7.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>7.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="7.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-1"><i class="fa fa-check"></i><b>7.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="7.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-2"><i class="fa fa-check"></i><b>7.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>7.4</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="7.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>7.5</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="7.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>7.6</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="7.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>7.7</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="7.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>7.8</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="7.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#grangeslist-groups-of-genomic-ranges"><i class="fa fa-check"></i><b>7.9</b> <em>GRangesList</em>: Groups of Genomic Ranges</a></li>
<li class="chapter" data-level="7.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-grangeslist-accessors"><i class="fa fa-check"></i><b>7.10</b> Basic <em>GRangesList</em> accessors</a></li>
<li class="chapter" data-level="7.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#combining-grangeslist-objects"><i class="fa fa-check"></i><b>7.11</b> Combining <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="7.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-grangeslist-objects"><i class="fa fa-check"></i><b>7.12</b> Basic interval operations for <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="7.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-grangeslist-objects"><i class="fa fa-check"></i><b>7.13</b> Subsetting <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="7.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#looping-over-grangeslist-objects"><i class="fa fa-check"></i><b>7.14</b> Looping over <em>GRangesList</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>8</b> Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructors-names-and-contact-information-4"><i class="fa fa-check"></i><b>8.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-5"><i class="fa fa-check"></i><b>8.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="8.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-5"><i class="fa fa-check"></i><b>8.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="8.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>8.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-6"><i class="fa fa-check"></i><b>8.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>8.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="8.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-2"><i class="fa fa-check"></i><b>8.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="8.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-3"><i class="fa fa-check"></i><b>8.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-1"><i class="fa fa-check"></i><b>8.4</b> Workshop</a></li>
<li class="chapter" data-level="8.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-2"><i class="fa fa-check"></i><b>8.5</b> Introduction</a><ul>
<li class="chapter" data-level="8.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>8.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="8.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>8.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="8.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>8.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup"><i class="fa fa-check"></i><b>8.6</b> Setup</a></li>
<li class="chapter" data-level="8.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>8.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="8.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>8.8</b> The Grammar</a><ul>
<li class="chapter" data-level="8.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>8.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="8.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>8.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="8.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises"><i class="fa fa-check"></i><b>8.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="8.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>8.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="8.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>8.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="8.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>8.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>8.10</b> Whatâ€™s next?</a></li>
<li class="chapter" data-level="8.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>8.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>9</b> Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="9.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>9.1</b> Instructors</a></li>
<li class="chapter" data-level="9.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description-6"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>9.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation-6"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>9.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="9.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-3"><i class="fa fa-check"></i><b>9.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="9.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-4"><i class="fa fa-check"></i><b>9.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>9.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="9.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>9.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="9.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>9.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="9.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>9.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="9.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>9.4.4</b> Data containers</a></li>
<li class="chapter" data-level="9.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>9.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="9.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>9.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="9.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>9.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="9.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>9.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="9.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>9.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="9.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>9.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="9.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>9.4.11</b> Simple example</a></li>
<li class="chapter" data-level="9.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>9.4.12</b> Questions!</a></li>
<li class="chapter" data-level="9.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>9.4.13</b> More questions!</a></li>
<li class="chapter" data-level="9.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>9.4.14</b> Another example</a></li>
<li class="chapter" data-level="9.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>9.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="9.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>9.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="9.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>9.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="9.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>9.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="9.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>9.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="9.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>9.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="9.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>9.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="9.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>9.4.22</b> GRanges</a></li>
<li class="chapter" data-level="9.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>9.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="9.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>9.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="9.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>9.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="9.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>9.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="9.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>9.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="9.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>9.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="9.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>9.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="9.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>9.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="9.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>9.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="9.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>9.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="9.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dply-example"><i class="fa fa-check"></i><b>9.4.33</b> Organism.dply example</a></li>
<li class="chapter" data-level="9.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>9.4.34</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="9.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>9.4.35</b> BSgenome packages</a></li>
<li class="chapter" data-level="9.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>9.4.36</b> BSgenome packages</a></li>
<li class="chapter" data-level="9.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>9.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="9.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>9.4.38</b> BSgenome exercises</a></li>
<li class="chapter" data-level="9.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-1"><i class="fa fa-check"></i><b>9.4.39</b> AnnotationHub</a></li>
<li class="chapter" data-level="9.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>9.4.40</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="9.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>9.4.41</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="9.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>9.4.42</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="9.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>9.4.43</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="9.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>9.4.44</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="9.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>9.4.45</b> AnnotationHub query</a></li>
<li class="chapter" data-level="9.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>9.4.46</b> AnnotationHub query</a></li>
<li class="chapter" data-level="9.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>9.4.47</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="9.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>9.4.48</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="9.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>9.4.49</b> biomaRt</a></li>
<li class="chapter" data-level="9.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>9.4.50</b> biomaRt data sets</a></li>
<li class="chapter" data-level="9.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>9.4.51</b> biomaRt queries</a></li>
<li class="chapter" data-level="9.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>9.4.52</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="9.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>9.4.53</b> biomaRt query</a></li>
<li class="chapter" data-level="9.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>9.4.54</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>10</b> Workflow for Multi-omics Analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="10.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>10.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-7"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-6"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-7"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>10.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-7"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-4"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-5"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>10.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="10.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="10.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>10.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="10.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>10.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>10.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="10.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>10.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="10.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>10.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="10.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>10.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>10.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="10.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>10.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="10.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>10.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="10.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>10.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="10.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>10.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="10.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>10.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>10.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="10.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>10.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="10.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>10.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="10.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>10.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>10.9</b> Complete cases</a></li>
<li class="chapter" data-level="10.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>10.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="10.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>10.11</b> Extraction</a><ul>
<li class="chapter" data-level="10.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>10.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="10.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>10.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="10.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>10.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="10.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>10.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="10.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>10.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="10.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>10.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="10.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>10.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="10.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>10.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="10.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>10.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>10.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="10.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>10.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="10.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>10.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="10.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>10.16.1</b> â€œSimplificationâ€ of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="10.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>10.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="10.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>10.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="10.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>10.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="10.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>10.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="10.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>10.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="10.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>10.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="10.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>10.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="10.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>10.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="10.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>10.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="10.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>10.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="10.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>10.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>11</b> Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="11.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>11.1</b> Instructors:</a></li>
<li class="chapter" data-level="11.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-8"><i class="fa fa-check"></i><b>11.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="11.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-7"><i class="fa fa-check"></i><b>11.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="11.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-8"><i class="fa fa-check"></i><b>11.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>11.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="11.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-8"><i class="fa fa-check"></i><b>11.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="11.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-5"><i class="fa fa-check"></i><b>11.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="11.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-6"><i class="fa fa-check"></i><b>11.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>12</b> Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="12.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>12.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="12.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-9"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-8"><i class="fa fa-check"></i><b>12.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-1"><i class="fa fa-check"></i><b>12.2.2</b> Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>12.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="12.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-4"><i class="fa fa-check"></i><b>12.3</b> Introduction</a></li>
<li class="chapter" data-level="12.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>12.4</b> Why maintain your package</a></li>
<li class="chapter" data-level="12.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>12.5</b> Infrastructure used for maintenance</a></li>
<li class="chapter" data-level="12.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>12.6</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="12.6.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>12.6.1</b> Build report</a></li>
<li class="chapter" data-level="12.6.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>12.6.2</b> Landing page badges</a></li>
<li class="chapter" data-level="12.6.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>12.6.3</b> RSS feeds</a></li>
<li class="chapter" data-level="12.6.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>12.6.4</b> Download statistics</a></li>
<li class="chapter" data-level="12.6.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-github-issues-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>12.6.5</b> Support site, Github issues, Bioc-devel mailing list</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#sync-package-with-github-bioconductor-before-every-change"><i class="fa fa-check"></i><b>12.7</b> Sync package with Github / Bioconductor before every change</a><ul>
<li class="chapter" data-level="12.7.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github-webhooks"><i class="fa fa-check"></i><b>12.7.1</b> GitHub + webhooks</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>12.8</b> Round up of resources available</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>13</b> Public Data Resources and Bioconductor</a><ul>
<li class="chapter" data-level="13.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>13.1</b> Instructor name(s) and contact information</a></li>
<li class="chapter" data-level="13.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>13.2</b> Syllabus</a><ul>
<li class="chapter" data-level="13.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-10"><i class="fa fa-check"></i><b>13.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="13.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-9"><i class="fa fa-check"></i><b>13.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-9"><i class="fa fa-check"></i><b>13.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>13.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="13.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-9"><i class="fa fa-check"></i><b>13.2.5</b> Time outline</a></li>
<li class="chapter" data-level="13.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.2.6</b> workshop goals and objectives</a></li>
<li class="chapter" data-level="13.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-6"><i class="fa fa-check"></i><b>13.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="13.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-7"><i class="fa fa-check"></i><b>13.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-2"><i class="fa fa-check"></i><b>13.3</b> Overview</a></li>
<li class="chapter" data-level="13.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>13.4</b> GEOquery</a><ul>
<li class="chapter" data-level="13.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>13.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="13.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>13.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="13.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>13.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="13.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>13.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>13.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="13.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>13.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="13.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>13.5.2</b> Basic design</a></li>
<li class="chapter" data-level="13.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>13.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>13.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="13.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>13.6.1</b> Authentication</a></li>
<li class="chapter" data-level="13.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>13.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="13.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>13.7</b> Sequence Read Archive</a></li>
<li class="chapter" data-level="13.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>13.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="13.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>13.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="13.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>13.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>13.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="13.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>13.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="13.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>13.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="13.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>13.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="13.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>13.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="13.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>13.12</b> Pharmacogenomics</a></li>
<li class="chapter" data-level="13.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>13.13</b> Bibliography</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="cytoscape-automation-in-r-using-rcy3" class="section level1">
<h1><span class="header-section-number">3</span> Cytoscape Automation in R using Rcy3</h1>
<div id="overview-1" class="section level2">
<h2><span class="header-section-number">3.1</span> Overview</h2>
<div id="instructors-names-and-contact-information" class="section level3">
<h3><span class="header-section-number">3.1.1</span> Instructor(s) name(s) and contact information</h3>
<ul>
<li>Ruth Isserlin - ruth dot isserlin (at) utoronto (dot) ca</li>
<li>Brendan Innes - brendan (dot) innes (at) mail (dot) utoronto (dot) ca</li>
<li>Jeff Wong - jvwong (at) gmail (dot) com</li>
<li>Gary Bader - gary (dot) bader (at) utoronto (dot) ca</li>
</ul>
</div>
<div id="workshop-description" class="section level3">
<h3><span class="header-section-number">3.1.2</span> Workshop Description</h3>
<p>Cytoscape(www.cytoscape.org) is one of the most popular applications for network analysis and visualization. In this workshop, we will demonstrate new capabilities to integrate Cytoscape into programmatic workflows and pipelines using R. We will begin with an overview of network biology themes and concepts, and then we will translate these into Cytoscape terms for practical applications. The bulk of the workshop will be a hands-on demonstration of accessing and controlling Cytoscape from R to perform a network analysis of tumor expression data.</p>
</div>
<div id="workshop-prerequisites" class="section level3">
<h3><span class="header-section-number">3.1.3</span> Workshop prerequisites:</h3>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Basic knowledge of Cytoscape software</li>
<li>Familiarity with network biology concepts</li>
</ul>
</div>
<div id="background" class="section level3">
<h3><span class="header-section-number">3.1.4</span> Background:</h3>
<ul>
<li>â€œHow to visually interpret biological data using networks.â€ Merico D, Gfeller D, Bader GD. Nature Biotechnology 2009 Oct 27, 921-924 - <a href="http://baderlab.org/Publications?action=AttachFile&amp;do=view&amp;target=2009_Merico_Primer_NatBiotech_Oct.pdf" class="uri">http://baderlab.org/Publications?action=AttachFile&amp;do=view&amp;target=2009_Merico_Primer_NatBiotech_Oct.pdf</a></li>
<li>â€œCyREST: Turbocharging Cytoscape Access for External Tools via a RESTful APIâ€. Keiichiro Ono, Tanja Muetze, Georgi Kolishovski, Paul Shannon, Barry Demchak.F1000Res. 2015 Aug 5;4:478. - <a href="https://f1000research.com/articles/4-478/v1" class="uri">https://f1000research.com/articles/4-478/v1</a></li>
</ul>
</div>
<div id="workshop-participation" class="section level3">
<h3><span class="header-section-number">3.1.5</span> Workshop Participation</h3>
<p>Participants are required to bring a laptop with Cytoscape, R, and RStudio installed. Installation instructions will be provided in the weeks preceding the workshop. The workshop will consist of a lecture and lab.</p>
</div>
<div id="r-bioconductor-packages-used-1" class="section level3">
<h3><span class="header-section-number">3.1.6</span> R / Bioconductor packages used</h3>
<ul>
<li>RCy3</li>
<li>gProfileR</li>
<li>RCurl</li>
<li>EnrichmentBrowser</li>
</ul>
</div>
<div id="time-outline-1" class="section level3">
<h3><span class="header-section-number">3.1.7</span> Time outline</h3>
<table>
<thead>
<tr class="header">
<th>Activity</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Introduction</td>
<td>15m</td>
</tr>
<tr class="even">
<td>Driving Cytoscape from R</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>Creating, retrieving and manipulating networks</td>
<td>15m</td>
</tr>
<tr class="even">
<td>Summary</td>
<td>10m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives-1" class="section level3">
<h3><span class="header-section-number">3.1.8</span> Workshop goals and objectives</h3>
<p>Learning goals</p>
<ul>
<li>Know when and how to use Cytoscape in your research area</li>
<li>Generalize network analysis methods to multiple problem domains</li>
<li>Integrate Cytoscape into your bioinformatics pipelines</li>
</ul>
<p>Learning objectives</p>
<ul>
<li>Programmatic control over Cytoscape from R</li>
<li>Publish, share and export networks</li>
</ul>
</div>
</div>
<div id="background-1" class="section level2">
<h2><span class="header-section-number">3.2</span> Background</h2>
<div id="cytoscape" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Cytoscape</h3>
<div id="left_logo">
<img src="230_Isserlin_RCy3_intro/images/cy3logoOrange.png" alt="Cytoscape Apps" width="100" />
</div>
<ul>
<li><a href="www.cytoscape.org">Cytoscape</a><span class="citation">(Shannon et al. 2003)</span> is a freely available open-source, cross platform network analysis software.<br />
</li>
<li><a href="www.cytoscape.org">Cytoscape</a><span class="citation">(Shannon et al. 2003)</span> can visualize complex networks and help integrate them with any other data type.<br />
</li>
<li><a href="www.cytoscape.org">Cytoscape</a><span class="citation">(Shannon et al. 2003)</span> has an active developer and user base with more than <strong>300</strong> community created apps available from the (Cytoscape App Store)[apps.cytoscape.org].</li>
<li>Check out some of the tasks you can do with Cytoscape in our online tutorial guide - tutorials.cytoscape.org</li>
</ul>
<div class="figure"><span id="fig:cytoapps"></span>
<img src="230_Isserlin_RCy3_intro/images/Apps-4.png" alt="Available Cytoscape Apps" width="1554" />
<p class="caption">
Figure 3.1: Available Cytoscape Apps
</p>
</div>
</div>
<div id="overview-of-network-biology-themes-and-concepts" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Overview of network biology themes and concepts</h3>
<div id="why-networks" class="section level4">
<h4><span class="header-section-number">3.2.2.1</span> Why Networks?</h4>
<p>Networks are everywhereâ€¦</p>
<ul>
<li>Molecular Networks</li>
<li>Cell-Cell communication Networks</li>
<li>Computer networks</li>
<li>Social Networks</li>
<li>Internet</li>
</ul>
<p>Networks are powerful toolsâ€¦</p>
<ul>
<li>Reduce complexity</li>
<li>More efficient than tables</li>
<li>Great for data integration</li>
<li>Intuitive visualization</li>
</ul>
<p>Often data in our pipelines are represented as data.frames, tables, matrices, vectors or lists. Sometimes we represent this data as heatmaps, or plots in efforts to summarize the results visually. Network visualization offers an additional method that readily incorporates many different data types and variables into a single picture.</p>
<div class="figure"><span id="fig:table2network"></span>
<img src="230_Isserlin_RCy3_intro/images/tables2networks.png" alt="Translating data to networks" width="792" />
<p class="caption">
Figure 3.2: Translating data to networks
</p>
</div>
<p>In order to translate your data into a network it is important to define the entities and their relationships. Entities and relationships can be anything. They can be user defined or they can be queried from a database.</p>
<p>Examples of Networks and their associated entities:</p>
<ul>
<li><strong>Protein - Protein interaction network</strong> - is a directed or undirected network where nodes in the network are proteins or genes and edges represent how those proteins interact.<br />
</li>
<li><strong>Gene - gene interaction network</strong> - nodes in the network are genes and edges can represent synthetic lethality i.e.Â two genes have a connection if deleting both of them cause a decrease in fitness.<br />
</li>
<li><strong>Coexpression network</strong> - nodes in the network are genes or proteins and edges represent the degree of co-expression the two genes have. Often the edges are associated with a correlation score (i.e.Â pearson correlation) and edges are filtered by a defined threshold. If no threshold is specified all genes will be connected to all other genes creating a hairball.</li>
<li><strong>Enrichment Map</strong> - nodes in the networks are groups of genes from pathways or functions (i.e.Â genesets) and edges represent pathway crosstalk ( genes in common).</li>
<li><strong>Social network</strong> - nodes in the network are individuals and edges are some sort of social interaction between two individuals, for example, friends on Facebook, linked in LinkedIN, â€¦</li>
<li><strong>Copublication network</strong> - a specialization of the social network for research purposes. Nodes in the network are individuals and edges between any two individuals for those who have co-authored a publication together.</li>
</ul>
</div>
</div>
<div id="networks-as-tools" class="section level3">
<h3><span class="header-section-number">3.2.3</span> Networks as Tools</h3>
<p>Networks can be used for two main purposes but often go hand in hand.</p>
<p><strong>Analysis</strong></p>
<ul>
<li>Topological properties - including number of nodes, number of edges, node degree, average clustering coefficients, shortest path lengths, density, and many more. Topological properties can help gain insights into the structure and organization of the resulting biological networks as well as help highlight specific node or regions of the network.<br />
</li>
<li>Hubs and subnetworks - a hub is generally a highly connected node in a scale-free network. Removal of hubs cause rapid breakdown of the underlying network. Subnetworks are interconnected regions of the network and can be defined by any user defined parameter.<br />
</li>
<li>Cluster, classify, and diffuse<br />
</li>
<li>Data integration</li>
</ul>
<p><strong>Visualization</strong></p>
<ul>
<li>Data overlays</li>
<li>Layouts and animation</li>
<li>Exploratory analysis</li>
<li>Context and interpretation</li>
</ul>
</div>
</div>
<div id="translating-biological-data-into-cytoscape-using-rcy3" class="section level2">
<h2><span class="header-section-number">3.3</span> Translating biological data into Cytoscape using RCy3</h2>
<p>Networks offer us a useful way to represent our biological data. But how do we seamlessly translate our data from R into Cytoscape?</p>
<div class="figure"><span id="fig:cytoscapeRcy3"></span>
<img src="230_Isserlin_RCy3_intro/images/CytoscapeAutomation_3.png" alt="Different ways to communicate with Cytoscape" width="500" />
<p class="caption">
Figure 3.3: Different ways to communicate with Cytoscape
</p>
</div>
<p>There are multiple ways to communicate with Cytoscape programmatically. There are two main complementary portals,<strong>cyRest</strong><span class="citation">(Ono et al. 2015)</span> and <strong>Commands</strong>, that form the foundation. cyRest transforms Cytoscape in to a REST (Representational State Transfer) enabled service where it essentially listens for events through a predefined port (by default port 1234). The cyRest functionality started as an app add in but has now been incorporated into the main release. Commands, on the other hand, offer a mechanism whereby app developers can expose their functionality to other apps or to user through the command interface. Prior to the implementation of cyRest many of the basic network functions were first available as commands so there is some overlap between the two different methods. <a href="cytoscape-automation-in-r-using-rcy3.html#fig:cytoscapeRcy3">3.3</a> shows the different ways you can call Cytoscape.</p>
<div id="set-up" class="section level3">
<h3><span class="header-section-number">3.3.1</span> Set up</h3>
<p>In order to create networks in Cytoscape from R you need:</p>
<ul>
<li><strong>RCy3</strong> - a biocondutor package</li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="op">!</span><span class="st">&quot;RCy3&quot;</span> <span class="op">%in%</span><span class="st"> </span><span class="kw">installed.packages</span>()){
    <span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)
    BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;RCy3&quot;</span>)
}
<span class="kw">library</span>(RCy3)</code></pre></div>
<ul>
<li><strong>Cytoscape</strong> - Download and install <a href="cytoscape.org/download.php">Cytoscape 3.6.1.</a> or higher. Java 9 in not supported. Please make sure that Java 8 is installed.</li>
<li>Install additional cytoscape apps that will be used in this workshop. If using cytoscape 3.6.1 or older the apps need to manually installed through the <a href="http://manual.cytoscape.org/en/latest/App_Manager.html#installing-apps">app manager</a> in Cytoscape or through your <a href="https://apps.cytoscape.org/help/getstarted_app_install">web browser</a>. (click on the method to see detailed instructions)
<ul>
<li><a href="http://apps.cytoscape.org/apps/functionalenrichmentcollection">Functional Enrichment Collection</a> -a collection of apps to retrieve networks and pathways, integrate and explore the data, perform functional enrichment analysis, and interpret and display your results.</li>
<li><a href="http://apps.cytoscape.org/apps/enrichmentmappipelinecollection">EnrichmentMap Pipeline Collection</a> - a collection of apps including EnrichmentMap, AutoAnnotate, WordCloud and clusterMaker2 used to visualize and analysis enrichment results.</li>
</ul></li>
</ul>
<p>If you are using Cytoscape 3.7 or higher (Cytoscape 3.7 will be released in August 2018) then apps can be installed directly from R.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#list of cytoscape apps to install</span>
installation_responses &lt;-<span class="st"> </span><span class="kw">c</span>()

<span class="co">#list of app to install</span>
cyto_app_toinstall &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;clustermaker2&quot;</span>, <span class="st">&quot;enrichmentmap&quot;</span>, <span class="st">&quot;autoannotate&quot;</span>, <span class="st">&quot;wordcloud&quot;</span>, <span class="st">&quot;stringapp&quot;</span>, <span class="st">&quot;aMatReader&quot;</span>)

cytoscape_version &lt;-<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">strsplit</span>( <span class="kw">cytoscapeVersionInfo</span>()[<span class="st">&#39;cytoscapeVersion&#39;</span>],<span class="dt">split =</span> <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>))
<span class="cf">if</span>(<span class="kw">length</span>(cytoscape_version) <span class="op">==</span><span class="st"> </span><span class="dv">3</span> <span class="op">&amp;&amp;</span><span class="st"> </span><span class="kw">as.numeric</span>(cytoscape_version[<span class="dv">1</span>]<span class="op">&gt;=</span><span class="dv">3</span>) 
   <span class="op">&amp;&amp;</span><span class="st"> </span><span class="kw">as.numeric</span>(cytoscape_version[<span class="dv">2</span>]<span class="op">&gt;=</span><span class="dv">7</span>)){
  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(cyto_app_toinstall)){
    <span class="co">#check to see if the app is installed.  Only install it if it hasn&#39;t been installed</span>
    <span class="cf">if</span>(<span class="op">!</span><span class="kw">grep</span>(<span class="kw">commandsGET</span>(<span class="kw">paste</span>(<span class="st">&quot;apps status app=</span><span class="ch">\&quot;</span><span class="st">&quot;</span>, cyto_app_toinstall[i],<span class="st">&quot;</span><span class="ch">\&quot;</span><span class="st">&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)), 
             <span class="dt">pattern =</span> <span class="st">&quot;status: Installed&quot;</span>)){
      installation_response &lt;-<span class="kw">commandsGET</span>(<span class="kw">paste</span>(<span class="st">&quot;apps install app=</span><span class="ch">\&quot;</span><span class="st">&quot;</span>, 
                                                cyto_app_toinstall[i],<span class="st">&quot;</span><span class="ch">\&quot;</span><span class="st">&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>))
      installation_responses &lt;-<span class="st"> </span><span class="kw">c</span>(installation_responses,installation_response)
    } <span class="cf">else</span>{
      installation_responses &lt;-<span class="st"> </span><span class="kw">c</span>(installation_responses,<span class="st">&quot;already installed&quot;</span>)
    }
  }
  installation_summary &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">name =</span> cyto_app_toinstall, 
                                     <span class="dt">status =</span> installation_responses)
  
  knitr<span class="op">::</span><span class="kw">kable</span>(<span class="kw">list</span>(installation_summary),
  <span class="dt">booktabs =</span> <span class="ot">TRUE</span>, <span class="dt">caption =</span> <span class="st">&#39;A Summary of automated app installation&#39;</span>
)
}</code></pre></div>
<p><strong>Make sure that Cytoscape is open</strong></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"> <span class="kw">cytoscapePing</span> ()</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_cytoscapeping.png" width="918" /></p>
</div>
<div id="getting-started" class="section level3">
<h3><span class="header-section-number">3.3.2</span> Getting started</h3>
<div id="confirm-that-cytoscape-is-installed-and-opened" class="section level4">
<h4><span class="header-section-number">3.3.2.1</span> Confirm that Cytoscape is installed and opened</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"> <span class="kw">cytoscapeVersionInfo</span> ()</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_cytoscapeversion.png" width="920" /></p>
</div>
<div id="browse-available-functions-commands-and-arguments" class="section level4">
<h4><span class="header-section-number">3.3.2.2</span> Browse available functions, commands and arguments</h4>
<p>Depending on what apps you have installed there is different functionality available.</p>
<p>To see all the functions available in RCy3 package</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">help</span>(<span class="dt">package=</span>RCy3)</code></pre></div>
<p>Open swagger docs for live instances of CyREST API. The CyREST API list all the functions available in a base distribution of cytoscape. The below command will launch the swagger documentation in a web browser. Functions are clustered into categories. Expanding individual categories will show all the option available. Further expanding an individual command will show detailed documentation for the function, input, outputs and allow you to try and run the function. Running the function will show the url used for the query and all returned responses.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">cyrestAPI</span>()  <span class="co"># CyREST API</span></code></pre></div>
<p>As mentioned above, there are two ways to interact with Cytoscape, through the Cyrest API or commands. To see the available commands in swagger similar to the Cyrest API.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">commandsAPI</span>()  <span class="co"># Commands API</span></code></pre></div>
<p>To get information about an individual command from the R environment you can also use the commandsHelp function. Simply specify what command you would like to get information on by adding its name to the command. For example â€œcommandsHelp(â€help stringâ€œ)â€</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">commandsHelp</span>(<span class="st">&quot;help&quot;</span>)</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_availablecommands.png" width="920" /></p>
</div>
</div>
<div id="cytoscape-basics" class="section level3">
<h3><span class="header-section-number">3.3.3</span> Cytoscape Basics</h3>
<p>Create a Cytoscape network from some basic R objects</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nodes &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">id=</span><span class="kw">c</span>(<span class="st">&quot;node 0&quot;</span>,<span class="st">&quot;node 1&quot;</span>,<span class="st">&quot;node 2&quot;</span>,<span class="st">&quot;node 3&quot;</span>),
           <span class="dt">group=</span><span class="kw">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;A&quot;</span>,<span class="st">&quot;B&quot;</span>,<span class="st">&quot;B&quot;</span>), <span class="co"># categorical strings</span>
           <span class="dt">score=</span><span class="kw">as.integer</span>(<span class="kw">c</span>(<span class="dv">20</span>,<span class="dv">10</span>,<span class="dv">15</span>,<span class="dv">5</span>)), <span class="co"># integers</span>
           <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)
edges &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">source=</span><span class="kw">c</span>(<span class="st">&quot;node 0&quot;</span>,<span class="st">&quot;node 0&quot;</span>,<span class="st">&quot;node 0&quot;</span>,<span class="st">&quot;node 2&quot;</span>),
           <span class="dt">target=</span><span class="kw">c</span>(<span class="st">&quot;node 1&quot;</span>,<span class="st">&quot;node 2&quot;</span>,<span class="st">&quot;node 3&quot;</span>,<span class="st">&quot;node 3&quot;</span>),
           <span class="dt">interaction=</span><span class="kw">c</span>(<span class="st">&quot;inhibits&quot;</span>,<span class="st">&quot;interacts&quot;</span>,<span class="st">&quot;activates&quot;</span>,<span class="st">&quot;interacts&quot;</span>),  <span class="co"># optional</span>
           <span class="dt">weight=</span><span class="kw">c</span>(<span class="fl">5.1</span>,<span class="fl">3.0</span>,<span class="fl">5.2</span>,<span class="fl">9.9</span>), <span class="co"># numeric</span>
           <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)</code></pre></div>
<table class="kable_wrapper">
<caption>
<span id="tab:unnamed-chunk-10">Table 3.1: </span>A table of example nodes and edges.
</caption>
<tbody>
<tr>
<td>
<table>
<thead>
<tr class="header">
<th align="left">id</th>
<th align="left">group</th>
<th align="right">score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">node 0</td>
<td align="left">A</td>
<td align="right">20</td>
</tr>
<tr class="even">
<td align="left">node 1</td>
<td align="left">A</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">node 2</td>
<td align="left">B</td>
<td align="right">15</td>
</tr>
<tr class="even">
<td align="left">node 3</td>
<td align="left">B</td>
<td align="right">5</td>
</tr>
</tbody>
</table>
</td>
<td>
<table>
<thead>
<tr class="header">
<th align="left">source</th>
<th align="left">target</th>
<th align="left">interaction</th>
<th align="right">weight</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">node 0</td>
<td align="left">node 1</td>
<td align="left">inhibits</td>
<td align="right">5.1</td>
</tr>
<tr class="even">
<td align="left">node 0</td>
<td align="left">node 2</td>
<td align="left">interacts</td>
<td align="right">3.0</td>
</tr>
<tr class="odd">
<td align="left">node 0</td>
<td align="left">node 3</td>
<td align="left">activates</td>
<td align="right">5.2</td>
</tr>
<tr class="even">
<td align="left">node 2</td>
<td align="left">node 3</td>
<td align="left">interacts</td>
<td align="right">9.9</td>
</tr>
</tbody>
</table>
</td>
</tr>
</tbody>
</table>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">createNetworkFromDataFrames</span>(nodes,edges, <span class="dt">title=</span><span class="st">&quot;my first network&quot;</span>, <span class="dt">collection=</span><span class="st">&quot;DataFrame Example&quot;</span>)</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_networkload_1.png" width="921" /></p>
<p>Remember. All networks we make are created in Cytoscape so get an image of the resulting network and include it in your current analysis if desired.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">initial_network_png_file_name &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(),<span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>, <span class="st">&quot;images&quot;</span>,<span class="st">&quot;initial_example_network.png&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="kw">file.exists</span>(initial_network_png_file_name)){
  <span class="co">#cytoscape hangs waiting for user response if file alreayd exists.  Remove it first</span>
  <span class="kw">file.remove</span>(initial_network_png_file_name)
  } 

<span class="co">#export the network</span>
<span class="kw">exportImage</span>(initial_network_png_file_name, <span class="dt">type =</span> <span class="st">&quot;png&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:initialnetwork"></span>
<img src="230_Isserlin_RCy3_intro/images/initial_example_network.png" alt="Example network created from dataframe" width="444" />
<p class="caption">
Figure 3.4: Example network created from dataframe
</p>
</div>
</div>
<div id="example-data-set" class="section level3">
<h3><span class="header-section-number">3.3.4</span> Example Data Set</h3>
<p>We downloaded gene expression data from the Ovarian Serous Cystadenocarcinoma project of The Cancer Genome Atlas (TCGA)<span class="citation">(International Cancer Genome et al. 2010)</span>, <a href="http://cancergenome.nih.gov" class="uri">http://cancergenome.nih.gov</a> via the Genomic Data Commons (GDC) portal<span class="citation">(Grossman et al. 2016)</span> on 2017-06-14 using TCGABiolinks R package<span class="citation">(Colaprico et al. 2016)</span>. The data includes 300 samples available as RNA-seq data, with reads mapped to a reference genome using MapSplice<span class="citation">(Wang et al. 2010)</span> and read counts per transcript determined using the RSEM method<span class="citation">(B. Li and Dewey 2011)</span>. RNA-seq data are labeled as â€˜RNA-Seq V2â€™, see details at: <a href="https://wiki.nci.nih.gov/display/TCGA/RNASeq+Version+2" class="uri">https://wiki.nci.nih.gov/display/TCGA/RNASeq+Version+2</a>). The RNA-SeqV2 data consists of raw counts similar to regular RNA-seq but RSEM (RNA-Seq by Expectation Maximization) data can be used with the edgeR method. The expression dataset of 300 tumours, with 79 classified as Immunoreactive, 72 classified as Mesenchymal, 69 classified as Differentiated, and 80 classified as Proliferative samples(class definitions were obtained from Verhaak et al.<span class="citation">(Verhaak et al. 2013)</span> Supplementary Table 1, third column). RNA-seq read counts were converted to CPM values and genes with CPM &gt; 1 in at least 50 of the samples are retained for further study (50 is the minimal sample size in the classes). The data was normalized and differential expression was calculated for each cancer class relative to the rest of the samples.</p>
<p>There are two data files: 1. Expression matrix - containing the normalized expression for each gene across all 300 samples. 1. Gene ranks - containing the p-values, FDR and foldchange values for the 4 comparisons (mesenchymal vs rest, differential vs rest, proliferative vs rest and immunoreactive vs rest)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">RNASeq_expression_matrix &lt;-<span class="st"> </span><span class="kw">read.table</span>( 
  <span class="kw">file.path</span>(<span class="kw">getwd</span>(),<span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>,<span class="st">&quot;data&quot;</span>,<span class="st">&quot;TCGA_OV_RNAseq_expression.txt&quot;</span>),  
  <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote=</span><span class="st">&quot;</span><span class="ch">\&quot;</span><span class="st">&quot;</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)

RNASeq_gene_scores &lt;-<span class="st"> </span><span class="kw">read.table</span>( 
   <span class="kw">file.path</span>(<span class="kw">getwd</span>(),<span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>,<span class="st">&quot;data&quot;</span>,<span class="st">&quot;TCGA_OV_RNAseq_All_edgeR_scores.txt&quot;</span>),  
  <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote=</span><span class="st">&quot;</span><span class="ch">\&quot;</span><span class="st">&quot;</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</code></pre></div>
</div>
<div id="finding-network-data" class="section level3">
<h3><span class="header-section-number">3.3.5</span> Finding Network Data</h3>
<p>How do I represent <em>my</em> data as a network?</p>
<p>Unfortunately, there is not a simple answer. <strong>It depends on your biological question!</strong></p>
<p>Example use cases:</p>
<ol style="list-style-type: decimal">
<li>Omics data - I have a <em>fill in the blank</em> (microarray, RNASeq, Proteomics, ATACseq, MicroRNA, GWAS â€¦) dataset. I have normalized and scored my data. How do I overlay my data on existing interaction data?</li>
<li>Coexpression data - I have a dataset that represents relationships. How do I represent it as a network.</li>
<li>Omics data - I have a <em>fill in the blank</em> (microarray, RNASeq, Proteomics, ATACseq, MicroRNA, GWAS â€¦) dataset. I have normalized and scored my data. I have run my data through a functional enrichment tool and now have a set of enriched terms associated with my dataset. How do I represent my functional enrichments as a network?</li>
</ol>
</div>
<div id="use-case-1---how-are-my-top-genes-related" class="section level3">
<h3><span class="header-section-number">3.3.6</span> Use Case 1 - How are my top genes related?</h3>
<p>Omics data - I have a <em>fill in the blank</em> (microarray, RNASeq, Proteomics, ATACseq, MicroRNA, GWAS â€¦) dataset. I have normalized and scored my data. How do I overlay my data on existing interaction data?</p>
<p>There are endless amounts of databases storing interaction data.</p>
<div class="figure"><span id="fig:ppidb"></span>
<img src="230_Isserlin_RCy3_intro/images/interaction-dbs.png" alt="Info graphic of some of the available pathway databases" width="800" />
<p class="caption">
Figure 3.5: Info graphic of some of the available pathway databases
</p>
</div>
<p>Thankfully we donâ€™t have to query each independent ally. In addition to many specialized (for example, for specific molecules, interaction type, or species) interaction databases there are also databases that collate these databases to create a broad resource that is easier to use. For example:</p>
<ul>
<li><a href="http://apps.cytoscape.org/apps/stringapp">StringApp</a> - is a protein - protein and protein- chemical database that imports data from <a href="string-db.org">String</a><span class="citation">(Szklarczyk et al. 2016)</span> (which itself includes data from multiple species, coexpression, text-mining,existing databases, and genomic context), [STITCH] into a unified, queriable database.</li>
<li><a href="https://psicquic.github.io/">PSICQUIC</a><span class="citation">(Aranda et al. 2011)</span> - a REST-ful service that is the responsibility of the database provider to set up and maintain. PSICQUIC is an additional interface that allows users to search all available databases (that support this REST-ful service). The databases are required to represent their interaction data in Proteomic Standards Initiative - molecular interaction (PSI-MI) format. To see a list of all the available data source see <a href="http://www.ebi.ac.uk/Tools/webservices/psicquic/registry/registry?action=STATUS">here</a></li>
<li><a href="http://home.ndexbio.org/index/">nDex</a><span class="citation">(Pratt et al. 2017)</span> - a network data exchange repository.</li>
<li><a href="http://genemania.org/">GeneMANIA</a><span class="citation">(Mostafavi et al. 2008)</span> - contains multiple networks (shared domains, physical interactions, pathways, predicted, co-expression, genetic interactions and co-localized network). Given a set of genes GeneMANIA selects and weights networks that optimize the connectivity between the query genes. GeneMANIA will also return additional genes that are highly related to your query set.</li>
</ul>
<p>Get a subset of genes of interest from our scored data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">top_mesenchymal_genes &lt;-<span class="st"> </span>RNASeq_gene_scores[<span class="kw">which</span>(RNASeq_gene_scores<span class="op">$</span>FDR.mesen <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>RNASeq_gene_scores<span class="op">$</span>logFC.mesen <span class="op">&gt;</span><span class="st"> </span><span class="dv">2</span>),]
<span class="kw">head</span>(top_mesenchymal_genes)
<span class="co">#&gt;       Name geneid logFC.mesen logCPM.mesen  LR.mesen PValue.mesen</span>
<span class="co">#&gt; 188   PRG4  10216    2.705305    2.6139056  95.58179     1.42e-22</span>
<span class="co">#&gt; 252  PROK1  84432    2.543381    1.3255202  68.31067     1.40e-16</span>
<span class="co">#&gt; 308  PRRX1   5396    2.077538    4.8570983 123.09925     1.33e-28</span>
<span class="co">#&gt; 434  PTGFR   5737    2.075707   -0.1960881  73.24646     1.14e-17</span>
<span class="co">#&gt; 438  PTGIS   5740    2.094198    5.8279714 165.11038     8.65e-38</span>
<span class="co">#&gt; 1214 BARX1  56033    3.267472    1.7427387 166.30064     4.76e-38</span>
<span class="co">#&gt;      FDR.mesen logFC.diff logCPM.diff   LR.diff  PValue.diff     FDR.diff</span>
<span class="co">#&gt; 188   7.34e-21 -1.2716521   2.6139056 14.107056 1.726950e-04 1.181751e-03</span>
<span class="co">#&gt; 252   4.77e-15  0.7455119   1.3255202  5.105528 2.384972e-02 6.549953e-02</span>
<span class="co">#&gt; 308   1.08e-26 -1.2367108   4.8570983 29.949104 4.440000e-08 1.250000e-06</span>
<span class="co">#&gt; 434   4.30e-16 -0.4233297  -0.1960881  2.318523 1.278413e-01 2.368387e-01</span>
<span class="co">#&gt; 438   1.20e-35 -0.4448761   5.8279714  5.696086 1.700278e-02 5.027140e-02</span>
<span class="co">#&gt; 1214  6.82e-36 -2.4411370   1.7427387 52.224346 4.950000e-13 6.970000e-11</span>
<span class="co">#&gt;      Row.names.y logFC.immuno logCPM.immuno LR.immuno PValue.immuno</span>
<span class="co">#&gt; 188   PRG4|10216   -0.4980017     2.6139056  2.651951   0.103422901</span>
<span class="co">#&gt; 252  PROK1|84432   -1.9692994     1.3255202 27.876348   0.000000129</span>
<span class="co">#&gt; 308   PRRX1|5396   -0.4914091     4.8570983  5.773502   0.016269586</span>
<span class="co">#&gt; 434   PTGFR|5737   -0.6737143    -0.1960881  6.289647   0.012144523</span>
<span class="co">#&gt; 438   PTGIS|5740   -0.6138074     5.8279714 11.708780   0.000622059</span>
<span class="co">#&gt; 1214 BARX1|56033   -0.6063633     1.7427387  4.577141   0.032401228</span>
<span class="co">#&gt;       FDR.immuno logFC.prolif logCPM.prolif  LR.prolif PValue.prolif</span>
<span class="co">#&gt; 188  0.185548905   -0.9356510     2.6139056  8.9562066  0.0027652850</span>
<span class="co">#&gt; 252  0.000001650   -1.3195933     1.3255202 13.7675841  0.0002068750</span>
<span class="co">#&gt; 308  0.042062023   -0.3494185     4.8570983  2.9943819  0.0835537930</span>
<span class="co">#&gt; 434  0.033197211   -0.9786631    -0.1960881 12.9112727  0.0003266090</span>
<span class="co">#&gt; 438  0.002779895   -1.0355143     5.8279714 31.9162051  0.0000000161</span>
<span class="co">#&gt; 1214 0.073565295   -0.2199714     1.7427387  0.6315171  0.4267993850</span>
<span class="co">#&gt;       FDR.prolif</span>
<span class="co">#&gt; 188  0.006372597</span>
<span class="co">#&gt; 252  0.000622057</span>
<span class="co">#&gt; 308  0.128528698</span>
<span class="co">#&gt; 434  0.000932503</span>
<span class="co">#&gt; 438  0.000000107</span>
<span class="co">#&gt; 1214 0.510856262</span></code></pre></div>
<p>We are going to query the String Database to get all interactions found for our set of top Mesenchymal genes.</p>
<p>Reminder: to see the parameters required by the string function or to find the right string function you can use commandsHelp.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">commandsHelp</span>(<span class="st">&quot;help string&quot;</span>)</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_availablestringcommands.png" width="922" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">commandsHelp</span>(<span class="st">&quot;help string protein query&quot;</span>)</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_availablestringarguments.png" width="921" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mesen_string_interaction_cmd &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&#39;string protein query taxonID=9606 limit=150 cutoff=0.9 query=&quot;&#39;</span>,<span class="kw">paste</span>(top_mesenchymal_genes<span class="op">$</span>Name, <span class="dt">collapse=</span><span class="st">&quot;,&quot;</span>),<span class="st">&#39;&quot;&#39;</span>,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>)
<span class="kw">commandsGET</span>(mesen_string_interaction_cmd)</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_stringnetworkload.png" width="922" /></p>
<p>Get a screenshot of the initial network</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">initial_string_network_png_file_name &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(),<span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>, <span class="st">&quot;images&quot;</span>, <span class="st">&quot;initial_string_network.png&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="kw">file.exists</span>(initial_string_network_png_file_name)){
  <span class="co">#cytoscape hangs waiting for user response if file alreayd exists.  Remove it first</span>
  response &lt;-<span class="st"> </span><span class="kw">file.remove</span>(initial_string_network_png_file_name)
} 

response &lt;-<span class="st"> </span><span class="kw">exportImage</span>(initial_string_network_png_file_name, <span class="dt">type =</span> <span class="st">&quot;png&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:initialstringnetwork"></span>
<img src="230_Isserlin_RCy3_intro/images/initial_string_network.png" alt="Initial network returned by String from our set of Mesenchymal query genes" width="444" />
<p class="caption">
Figure 3.6: Initial network returned by String from our set of Mesenchymal query genes
</p>
</div>
<p>Layout the network</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">layoutNetwork</span>(<span class="st">&#39;force-directed&#39;</span>)</code></pre></div>
<p>Check what other layout algorithms are available to try out</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getLayoutNames</span>()</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_availablelayouts.png" width="924" /> Get the parameters for a specific layout</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getLayoutPropertyNames</span>(<span class="dt">layout.name=</span><span class="st">&#39;force-directed&#39;</span>)</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_availablelayoutarguments.png" width="922" /></p>
<p>Re-layout the network using the force directed layout but specify some of the parameters</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">layoutNetwork</span>(<span class="st">&#39;force-directed defaultSpringCoefficient=0.0000008 defaultSpringLength=70&#39;</span>)</code></pre></div>
<p>Get a screenshot of the re-laid out network</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">relayout_string_network_png_file_name &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(),<span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>, <span class="st">&quot;images&quot;</span>,<span class="st">&quot;relayout_string_network.png&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="kw">file.exists</span>(relayout_string_network_png_file_name)){
  <span class="co">#cytoscape hangs waiting for user response if file alreayd exists.  Remove it first</span>
  response&lt;-<span class="st"> </span><span class="kw">file.remove</span>(relayout_string_network_png_file_name)
  } 
response &lt;-<span class="st"> </span><span class="kw">exportImage</span>(relayout_string_network_png_file_name, <span class="dt">type =</span> <span class="st">&quot;png&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:relayoutstringnetwork"></span>
<img src="230_Isserlin_RCy3_intro/images/relayout_string_network.png" alt="Initial network returned by String from our set of Mesenchymal query genes" width="444" />
<p class="caption">
Figure 3.7: Initial network returned by String from our set of Mesenchymal query genes
</p>
</div>
<p>Overlay our expression data on the String network.<br />
To do this we will be using the loadTableData function from RCy3. It is important to make sure that that your identifiers types match up. You can check what is used by String by pulling in the column names of the node attribute table.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getTableColumnNames</span>(<span class="st">&#39;node&#39;</span>)</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_availablenodecolumns.png" width="518" /></p>
<p>If you are unsure of what each column is and want to further verify the column to use you can also pull in the entire node attribute table.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">node_attribute_table_topmesen &lt;-<span class="st"> </span><span class="kw">getTableColumns</span>(<span class="dt">table=</span><span class="st">&quot;node&quot;</span>)
<span class="kw">head</span>(node_attribute_table_topmesen[,<span class="dv">3</span><span class="op">:</span><span class="dv">7</span>])</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_nodetable.png" width="926" /></p>
<p>The column â€œdisplay nameâ€ contains HGNC gene names which are also found in our Ovarian Cancer dataset.</p>
<p>To import our expression data we will match our dataset to the â€œdisplay nameâ€ node attribute.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">?loadTableData

<span class="kw">loadTableData</span>(RNASeq_gene_scores,<span class="dt">table.key.column =</span> <span class="st">&quot;display name&quot;</span>,<span class="dt">data.key.column =</span> <span class="st">&quot;Name&quot;</span>)  <span class="co">#default data.frame key is row.names</span></code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_nodeattributeload.png" width="928" /></p>
<p>Modify the Visual Style Create your own visual style to visualize your expression data on the String network.</p>
<p>Start with a default style</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">style.name =<span class="st"> &quot;MesenchymalStyle&quot;</span>
defaults.list &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">NODE_SHAPE=</span><span class="st">&quot;ellipse&quot;</span>,
                 <span class="dt">NODE_SIZE=</span><span class="dv">60</span>,
                 <span class="dt">NODE_FILL_COLOR=</span><span class="st">&quot;#AAAAAA&quot;</span>,
                 <span class="dt">EDGE_TRANSPARENCY=</span><span class="dv">120</span>)
node.label.map &lt;-<span class="st"> </span><span class="kw">mapVisualProperty</span>(<span class="st">&#39;node label&#39;</span>,<span class="st">&#39;display name&#39;</span>,<span class="st">&#39;p&#39;</span>) <span class="co"># p for passthrough; nothing else needed</span>
<span class="kw">createVisualStyle</span>(style.name, defaults.list, <span class="kw">list</span>(node.label.map))
<span class="kw">setVisualStyle</span>(<span class="dt">style.name=</span>style.name)</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_visualstyleapplied.png" width="927" /></p>
<p>Update your created style with a mapping for the Mesenchymal logFC expression. The first step is to grab the column data from Cytoscape (we can reuse the node_attribute table concept from above but we have to call the function again as we have since added our expression data) and pull out the min and max to define our data mapping range of values.</p>
<p><strong>Note</strong>: you could define the min and max based on the entire dataset or just the subset that is represented in Cytoscape currently. The two methods will give you different results. If you intend on comparing different networks created with the same dataset then it is best to calculate the min and max from the entire dataset as opposed to a subset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">min.mesen.logfc =<span class="st"> </span><span class="kw">min</span>(RNASeq_gene_scores<span class="op">$</span>logFC.mesen,<span class="dt">na.rm=</span><span class="ot">TRUE</span>)
max.mesen.logfc =<span class="st"> </span><span class="kw">max</span>(RNASeq_gene_scores<span class="op">$</span>logFC.mesen,<span class="dt">na.rm=</span><span class="ot">TRUE</span>)
data.values =<span class="st"> </span><span class="kw">c</span>(min.mesen.logfc,<span class="dv">0</span>,max.mesen.logfc)</code></pre></div>
<p>Next, we use the RColorBrewer package to help us pick good colors to pair with our data values.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(RColorBrewer)
<span class="kw">display.brewer.all</span>(<span class="kw">length</span>(data.values), <span class="dt">colorblindFriendly=</span><span class="ot">TRUE</span>, <span class="dt">type=</span><span class="st">&quot;div&quot;</span>) <span class="co"># div,qual,seq,all</span></code></pre></div>
<p><img src="230_Isserlin_RCy3_intro_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">node.colors &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="kw">length</span>(data.values), <span class="st">&quot;RdBu&quot;</span>)))</code></pre></div>
<p>Map the colors to our data value and update our visual style.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">setNodeColorMapping</span>(<span class="st">&quot;logFC.mesen&quot;</span>, data.values, node.colors, <span class="dt">style.name=</span>style.name)</code></pre></div>
<p>Remember, String includes your query proteins as well as other proteins that associate with your query proteins (including the strongest connection first). Not all of the proteins in this network are your top hits. How can we visualize which proteins are our top Mesenchymal hits?</p>
<p>Add a different border color or change the node shape for our top hits.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getNodeShapes</span>()

<span class="co">#select the Nodes of interest</span>
<span class="co">#selectNode(nodes = top_mesenchymal_genes$Name, by.col=&quot;display name&quot;)</span>
<span class="kw">setNodeShapeBypass</span>(<span class="dt">node.names =</span> top_mesenchymal_genes<span class="op">$</span>Name, <span class="dt">new.shapes =</span> <span class="st">&quot;TRIANGLE&quot;</span>)</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_availablenodeshapes.png" width="920" /></p>
<p>Change the size of the node to be correlated with the Mesenchymal p-value.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">setNodeSizeMapping</span>(<span class="dt">table.column =</span> <span class="st">&#39;LR.mesen&#39;</span>, 
                   <span class="dt">table.column.values =</span> <span class="kw">c</span>(<span class="kw">min</span>(RNASeq_gene_scores<span class="op">$</span>LR.mesen), 
                                           <span class="kw">mean</span>(RNASeq_gene_scores<span class="op">$</span>LR.mesen), 
                                           <span class="kw">max</span>(RNASeq_gene_scores<span class="op">$</span>LR.mesen)), 
                   <span class="dt">sizes =</span> <span class="kw">c</span>(<span class="dv">30</span>, <span class="dv">60</span>, <span class="dv">150</span>),<span class="dt">mapping.type =</span> <span class="st">&quot;c&quot;</span>, <span class="dt">style.name =</span> style.name)</code></pre></div>
<p>Get a screenshot of the resulting network</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mesen_string_network_png_file_name &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(),<span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>, <span class="st">&quot;images&quot;</span>,<span class="st">&quot;mesen_string_network.png&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="kw">file.exists</span>(mesen_string_network_png_file_name)){
  <span class="co">#cytoscape hangs waiting for user response if file alreayd exists.  Remove it first</span>
  response&lt;-<span class="st"> </span><span class="kw">file.remove</span>(mesen_string_network_png_file_name)
  } 
response &lt;-<span class="st"> </span><span class="kw">exportImage</span>(mesen_string_network_png_file_name, <span class="dt">type =</span> <span class="st">&quot;png&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:mesenstringnetwork"></span>
<img src="230_Isserlin_RCy3_intro/images/mesen_string_network.png" alt="Formatted String network from our set of Mesenchymal query genes.  Annotated with our expressin data" width="444" />
<p class="caption">
Figure 3.8: Formatted String network from our set of Mesenchymal query genes. Annotated with our expressin data
</p>
</div>
</div>
<div id="use-case-2---which-genes-have-similar-expression." class="section level3">
<h3><span class="header-section-number">3.3.7</span> Use Case 2 - Which genes have similar expression.</h3>
<p>Instead of querying existing resources look for correlations in your own dataset to find out which genes have similar expression. There are many tools that can analyze your data for correlation. A popular tool is Weighted Gene Correlation Network Analysis (WGCNA)<span class="citation">(Langfelder and Horvath 2008)</span> which takes expression data and calculates functional modules. As a simple example we can transform our expression dataset into a correlation matrix.</p>
<p>Using the Cytoscape App, aMatReader<span class="citation">(Settle et al. 2018)</span>, we transform our adjacency matrix into an interaction network. First we filter the correlation matrix to contain only the strongest connections (for example, only correlations greater than 0.9).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
RNASeq_expression &lt;-<span class="st"> </span>RNASeq_expression_matrix[,<span class="dv">3</span><span class="op">:</span><span class="kw">ncol</span>(RNASeq_expression_matrix)]

<span class="kw">rownames</span>(RNASeq_expression) &lt;-<span class="st"> </span>RNASeq_expression_matrix<span class="op">$</span>Name
RNAseq_correlation_matrix &lt;-<span class="st"> </span><span class="kw">cor</span>(<span class="kw">t</span>(RNASeq_expression), <span class="dt">method=</span><span class="st">&quot;pearson&quot;</span>)

<span class="co">#set the diagonal of matrix to zero - eliminate self-correlation</span>
RNAseq_correlation_matrix[ 
  <span class="kw">row</span>(RNAseq_correlation_matrix) <span class="op">==</span><span class="st"> </span><span class="kw">col</span>(RNAseq_correlation_matrix) ] &lt;-<span class="st"> </span><span class="dv">0</span>

<span class="co"># set all correlations that are less than 0.9 to zero</span>
RNAseq_correlation_matrix[<span class="kw">which</span>(RNAseq_correlation_matrix<span class="op">&lt;</span><span class="fl">0.90</span>)] &lt;-<span class="st"> </span><span class="dv">0</span>

<span class="co">#get rid of rows and columns that have no correlations with the above thresholds</span>
RNAseq_correlation_matrix &lt;-<span class="st"> </span>RNAseq_correlation_matrix[<span class="kw">which</span>(<span class="kw">rowSums</span>(RNAseq_correlation_matrix) <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>),
                          <span class="kw">which</span>(<span class="kw">colSums</span>(RNAseq_correlation_matrix) <span class="op">!=</span><span class="dv">0</span>)]

<span class="co">#write out the correlation file</span>
correlation_filename &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(), <span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>, <span class="st">&quot;data&quot;</span>, <span class="st">&quot;TCGA_OV_RNAseq_expression_correlation_matrix.txt&quot;</span>) 
<span class="kw">write.table</span>( RNAseq_correlation_matrix,  <span class="dt">file =</span> correlation_filename, <span class="dt">col.names  =</span> <span class="ot">TRUE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>)</code></pre></div>
<p>Use the CyRest call to access the aMatReader functionality.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">amat_url &lt;-<span class="st"> &quot;aMatReader/v1/import&quot;</span>
amat_params =<span class="st"> </span><span class="kw">list</span>(<span class="dt">files =</span> <span class="kw">list</span>(correlation_filename),
                   <span class="dt">delimiter =</span> <span class="st">&quot;TAB&quot;</span>,
                   <span class="dt">undirected =</span> <span class="ot">FALSE</span>,
                   <span class="dt">ignoreZeros =</span> <span class="ot">TRUE</span>,
                   <span class="dt">interactionName =</span> <span class="st">&quot;correlated with&quot;</span>,
                   <span class="dt">rowNames =</span> <span class="ot">FALSE</span>
                   )
 
response &lt;-<span class="st"> </span><span class="kw">cyrestPOST</span>(<span class="dt">operation =</span> amat_url, <span class="dt">body =</span> amat_params, <span class="dt">base.url =</span> <span class="st">&quot;http://localhost:1234&quot;</span>)

current_network_id &lt;-<span class="st"> </span>response<span class="op">$</span>data[<span class="st">&quot;suid&quot;</span>]</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#relayout network</span>
<span class="kw">layoutNetwork</span>(<span class="st">&#39;cose&#39;</span>,
              <span class="dt">network =</span> <span class="kw">as.numeric</span>(current_network_id))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">renameNetwork</span>(<span class="dt">title =</span> <span class="st">&quot;Coexpression_network_pear0_95&quot;</span>,
              <span class="dt">network =</span> <span class="kw">as.numeric</span>(current_network_id))</code></pre></div>
<p>Modify the visualization to see where each genes is predominantly expressed. Look at the 4 different p-values associated with each gene and color the nodes with the type associated with the lowest FDR.</p>
<p>Load in the scoring data. Specify the cancer type where the genes has the lowest FDR value.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nodes_in_network &lt;-<span class="st"> </span><span class="kw">rownames</span>(RNAseq_correlation_matrix)

<span class="co">#add an additional column to the gene scores table to indicate in which samples</span>
<span class="co"># the gene is significant</span>
node_class &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="dt">length =</span> <span class="kw">length</span>(nodes_in_network),<span class="dt">mode =</span> <span class="st">&quot;character&quot;</span>)
<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(nodes_in_network)){
  current_row &lt;-<span class="st"> </span><span class="kw">which</span>(RNASeq_gene_scores<span class="op">$</span>Name <span class="op">==</span><span class="st"> </span>nodes_in_network[i])
  min_pvalue &lt;-<span class="st"> </span><span class="kw">min</span>(RNASeq_gene_scores[current_row,
                                       <span class="kw">grep</span>(<span class="kw">colnames</span>(RNASeq_gene_scores), <span class="dt">pattern =</span> <span class="st">&quot;FDR&quot;</span>)])
  <span class="cf">if</span>(RNASeq_gene_scores<span class="op">$</span>FDR.mesen[current_row] <span class="op">&lt;=</span>min_pvalue){
    node_class[i] &lt;-<span class="st"> </span><span class="kw">paste</span>(node_class[i],<span class="st">&quot;mesen&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot; &quot;</span>)
  }
  <span class="cf">if</span>(RNASeq_gene_scores<span class="op">$</span>FDR.diff[current_row] <span class="op">&lt;=</span>min_pvalue){
    node_class[i] &lt;-<span class="st"> </span><span class="kw">paste</span>(node_class[i],<span class="st">&quot;diff&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot; &quot;</span>)
  }
  <span class="cf">if</span>(RNASeq_gene_scores<span class="op">$</span>FDR.prolif[current_row] <span class="op">&lt;=</span>min_pvalue){
    node_class[i] &lt;-<span class="st"> </span><span class="kw">paste</span>(node_class[i],<span class="st">&quot;prolif&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot; &quot;</span>)
  }
  <span class="cf">if</span>(RNASeq_gene_scores<span class="op">$</span>FDR.immuno[current_row] <span class="op">&lt;=</span>min_pvalue){
    node_class[i] &lt;-<span class="st"> </span><span class="kw">paste</span>(node_class[i],<span class="st">&quot;immuno&quot;</span>,<span class="dt">sep =</span> <span class="st">&quot; &quot;</span>)
  }
}
node_class &lt;-<span class="st"> </span><span class="kw">trimws</span>(node_class)
node_class_df &lt;-<span class="kw">data.frame</span>(<span class="dt">name=</span>nodes_in_network, node_class,<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)

<span class="kw">head</span>(node_class_df)
<span class="co">#&gt;     name node_class</span>
<span class="co">#&gt; 1  ABCA6      mesen</span>
<span class="co">#&gt; 2  ABCA8      mesen</span>
<span class="co">#&gt; 3   ABI3     immuno</span>
<span class="co">#&gt; 4   ACAN     prolif</span>
<span class="co">#&gt; 5  ACAP1     immuno</span>
<span class="co">#&gt; 6 ADAM12      mesen</span></code></pre></div>
<p>Map the new node attribute and the all the gene scores to the network.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">loadTableData</span>(RNASeq_gene_scores,<span class="dt">table.key.column =</span> <span class="st">&quot;name&quot;</span>,<span class="dt">data.key.column =</span> <span class="st">&quot;Name&quot;</span>)  <span class="co">#default data.frame key is row.names</span>

<span class="kw">loadTableData</span>(node_class_df,<span class="dt">table.key.column =</span> <span class="st">&quot;name&quot;</span>,<span class="dt">data.key.column =</span> <span class="st">&quot;name&quot;</span>)  <span class="co">#default data.frame key is row.names</span></code></pre></div>
<p>Create a color mapping for the different cancer types.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#create a new mapping with the different types</span>
unique_types &lt;-<span class="st"> </span><span class="kw">sort</span>(<span class="kw">unique</span>(node_class))

coul =<span class="st"> </span><span class="kw">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Set1&quot;</span>) 
 
<span class="co"># I can add more tones to this palette :</span>
coul =<span class="st"> </span><span class="kw">colorRampPalette</span>(coul)(<span class="kw">length</span>(unique_types))

<span class="kw">setNodeColorMapping</span>(<span class="dt">table.column =</span> <span class="st">&quot;node_class&quot;</span>,<span class="dt">table.column.values =</span> unique_types,
                    <span class="dt">colors =</span> coul,<span class="dt">mapping.type =</span> <span class="st">&quot;d&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">correlation_network_png_file_name &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(),<span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>, <span class="st">&quot;images&quot;</span>,<span class="st">&quot;correlation_network.png&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="kw">file.exists</span>(correlation_network_png_file_name)){
  <span class="co">#cytoscape hangs waiting for user response if file alreayd exists.  Remove it first</span>
  <span class="kw">file.remove</span>(correlation_network_png_file_name)
  } 

<span class="co">#export the network</span>
<span class="kw">exportImage</span>(correlation_network_png_file_name, <span class="dt">type =</span> <span class="st">&quot;png&quot;</span>)</code></pre></div>
<div class="figure">
<img src="230_Isserlin_RCy3_intro/images/correlation_network.png" alt="Example correlation network created using aMatReader" width="444" />
<p class="caption">
(#fig:correlation_network2)Example correlation network created using aMatReader
</p>
</div>
<p>cluster the Network</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co">#make sure it is set to the right network</span>
  <span class="kw">setCurrentNetwork</span>(<span class="dt">network =</span> <span class="kw">getNetworkName</span>(<span class="dt">suid=</span><span class="kw">as.numeric</span>(current_network_id)))

  <span class="co">#cluster the network</span>
  clustermaker_url &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;cluster mcl network=SUID:&quot;</span>,current_network_id, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)
  <span class="kw">commandsGET</span>(clustermaker_url)
  
  <span class="co">#get the clustering results</span>
  default_node_table &lt;-<span class="st"> </span><span class="kw">getTableColumns</span>(<span class="dt">table=</span> <span class="st">&quot;node&quot;</span>,<span class="dt">network =</span> <span class="kw">as.numeric</span>(current_network_id))
 
  <span class="kw">head</span>(default_node_table)</code></pre></div>
<p>Perform pathway Enrichment on one of the clusters using g:Profiler<span class="citation">(Reimand et al. 2016)</span>. g:Profiler is an online functional enrichment web service that will take your gene list and return the set of enriched pathways. For automated analysis g:Profiler has created an R library to interact with it directly from R instead of using the web page.</p>
<p>Create a function to call g:Profiler and convert the returned results into a generic enrichment map input file.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">tryCatch</span>(<span class="dt">expr =</span> { <span class="kw">library</span>(<span class="st">&quot;gProfileR&quot;</span>)}, 
         <span class="dt">error =</span> <span class="cf">function</span>(e) { <span class="kw">install.packages</span>(<span class="st">&quot;gProfileR&quot;</span>)}, <span class="dt">finally =</span> <span class="kw">library</span>(<span class="st">&quot;gProfileR&quot;</span>))

<span class="co">#function to run gprofiler using the gprofiler library</span>
<span class="co"># </span>
<span class="co"># The function takes the returned gprofiler results and formats it to the generic EM input file</span>
<span class="co">#</span>
<span class="co"># function returns a data frame in the generic EM file format.</span>
runGprofiler &lt;-<span class="st"> </span><span class="cf">function</span>(genes,<span class="dt">current_organism =</span> <span class="st">&quot;hsapiens&quot;</span>, 
                         <span class="dt">significant_only =</span> F, <span class="dt">set_size_max =</span> <span class="dv">200</span>, 
                         <span class="dt">set_size_min =</span> <span class="dv">3</span>, <span class="dt">filter_gs_size_min =</span> <span class="dv">5</span> , <span class="dt">exclude_iea =</span> F){
  
  gprofiler_results &lt;-<span class="st"> </span><span class="kw">gprofiler</span>(genes ,
                                 <span class="dt">significant=</span>significant_only,<span class="dt">ordered_query =</span> F,
                                <span class="dt">exclude_iea=</span>exclude_iea,<span class="dt">max_set_size =</span> set_size_max,
                                 <span class="dt">min_set_size =</span> set_size_min,
                                 <span class="dt">correction_method =</span> <span class="st">&quot;fdr&quot;</span>,
                                 <span class="dt">organism =</span> current_organism,
                                <span class="dt">src_filter =</span> <span class="kw">c</span>(<span class="st">&quot;GO:BP&quot;</span>,<span class="st">&quot;REAC&quot;</span>))
  
  <span class="co">#filter results</span>
  gprofiler_results &lt;-<span class="st"> </span>gprofiler_results[<span class="kw">which</span>(gprofiler_results[,<span class="st">&#39;term.size&#39;</span>] <span class="op">&gt;=</span><span class="st"> </span><span class="dv">3</span>
                                        <span class="op">&amp;</span><span class="st"> </span>gprofiler_results[,<span class="st">&#39;overlap.size&#39;</span>] <span class="op">&gt;=</span><span class="st"> </span>filter_gs_size_min ),]
  
  <span class="co"># gProfileR returns corrected p-values only.  Set p-value to corrected p-value</span>
  <span class="cf">if</span>(<span class="kw">dim</span>(gprofiler_results)[<span class="dv">1</span>] <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>){
    em_results &lt;-<span class="st"> </span><span class="kw">cbind</span>(gprofiler_results[,
                                <span class="kw">c</span>(<span class="st">&quot;term.id&quot;</span>,<span class="st">&quot;term.name&quot;</span>,<span class="st">&quot;p.value&quot;</span>,<span class="st">&quot;p.value&quot;</span>)], <span class="dv">1</span>,
                                gprofiler_results[,<span class="st">&quot;intersection&quot;</span>])
  <span class="kw">colnames</span>(em_results) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Name&quot;</span>,<span class="st">&quot;Description&quot;</span>, <span class="st">&quot;pvalue&quot;</span>,<span class="st">&quot;qvalue&quot;</span>,<span class="st">&quot;phenotype&quot;</span>,<span class="st">&quot;genes&quot;</span>)
  
  <span class="kw">return</span>(em_results)
  } <span class="cf">else</span> {
    <span class="kw">return</span>(<span class="st">&quot;no gprofiler results for supplied query&quot;</span>)
  }
}</code></pre></div>
<p>Run g:Profiler. g:Profiler will return a set of pathways and functions that are found to be enriched in our query set of genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">  current_cluster &lt;-<span class="st"> &quot;1&quot;</span>
  <span class="co">#select all the nodes in cluster 1</span>
  selectednodes &lt;-<span class="st"> </span><span class="kw">selectNodes</span>(current_cluster, <span class="dt">by.col=</span><span class="st">&quot;__mclCluster&quot;</span>)
  
  <span class="co">#create a subnetwork with cluster 1</span>
  subnetwork_suid &lt;-<span class="st"> </span><span class="kw">createSubnetwork</span>(<span class="dt">nodes=</span><span class="st">&quot;selected&quot;</span>)
  
  <span class="kw">renameNetwork</span>(<span class="st">&quot;Cluster1_Subnetwork&quot;</span>, <span class="dt">network=</span><span class="kw">as.numeric</span>(subnetwork_suid))
  
  subnetwork_node_table &lt;-<span class="st"> </span><span class="kw">getTableColumns</span>(<span class="dt">table=</span> <span class="st">&quot;node&quot;</span>,<span class="dt">network =</span> <span class="kw">as.numeric</span>(subnetwork_suid))

  em_results &lt;-<span class="st"> </span><span class="kw">runGprofiler</span>(subnetwork_node_table<span class="op">$</span>name)
  
 <span class="co">#write out the g:Profiler results</span>
 em_results_filename &lt;-<span class="kw">file.path</span>(<span class="kw">getwd</span>(),
                            <span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>, <span class="st">&quot;data&quot;</span>,<span class="kw">paste</span>(<span class="st">&quot;gprofiler_cluster&quot;</span>,current_cluster,<span class="st">&quot;enr_results.txt&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;_&quot;</span>))

  <span class="kw">write.table</span>(em_results,em_results_filename,<span class="dt">col.name=</span><span class="ot">TRUE</span>,<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,<span class="dt">row.names=</span><span class="ot">FALSE</span>,<span class="dt">quote=</span><span class="ot">FALSE</span>)
  
 
  <span class="kw">head</span>(em_results)</code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_gprofiler_emresults.png" width="514" /></p>
<p>Create an enrichment map with the returned g:Profiler results. An enrichment map is a different sort of network. Instead of nodes representing genes, nodes represent pathways or functions. Edges between these pathways or functions represent shared genes or pathway crosstalk. An enrichment map is a way to visualize your enrichment results to help reduce redundancy and uncover main themes. Pathways can also be explored in detail using the features available through the App in Cytoscape.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"> em_command =<span class="st"> </span><span class="kw">paste</span>(<span class="st">&#39;enrichmentmap build analysisType=&quot;generic&quot; &#39;</span>, 
                   <span class="st">&#39;pvalue=&#39;</span>,<span class="st">&quot;0.05&quot;</span>, <span class="st">&#39;qvalue=&#39;</span>,<span class="st">&quot;0.05&quot;</span>,
                   <span class="st">&#39;similaritycutoff=&#39;</span>,<span class="st">&quot;0.25&quot;</span>,
                   <span class="st">&#39;coeffecients=&#39;</span>,<span class="st">&quot;JACCARD&quot;</span>,
                   <span class="st">&#39;enrichmentsDataset1=&#39;</span>,em_results_filename ,
                   <span class="dt">sep=</span><span class="st">&quot; &quot;</span>)

  <span class="co">#enrichment map command will return the suid of newly created network.</span>
  em_network_suid &lt;-<span class="st"> </span><span class="kw">commandsRun</span>(em_command)
  
  <span class="kw">renameNetwork</span>(<span class="st">&quot;Cluster1_enrichmentmap&quot;</span>, <span class="dt">network=</span><span class="kw">as.numeric</span>(em_network_suid))</code></pre></div>
<p>Export image of resulting Enrichment map.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cluster1em_png_file_name &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(),<span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>, <span class="st">&quot;images&quot;</span>,<span class="st">&quot;cluster1em.png&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="kw">file.exists</span>(cluster1em_png_file_name)){
  <span class="co">#cytoscape hangs waiting for user response if file alreayd exists.  Remove it first</span>
  <span class="kw">file.remove</span>(cluster1em_png_file_name)
  } 

<span class="co">#export the network</span>
<span class="kw">exportImage</span>(cluster1em_png_file_name, <span class="dt">type =</span> <span class="st">&quot;png&quot;</span>)</code></pre></div>
<div class="figure">
<img src="230_Isserlin_RCy3_intro/images/cluster1em.png" alt="Example Enrichment Map created when running an enrichment analysis using g:Profiler with the genes that are part of cluster 1" width="444" />
<p class="caption">
(#fig:cluster1em_fig)Example Enrichment Map created when running an enrichment analysis using g:Profiler with the genes that are part of cluster 1
</p>
</div>
<p>Annotate the Enrichment map to get the general themes that are found in the enrichment results of cluster 1</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#get the column from the nodetable and node table</span>
  nodetable_colnames &lt;-<span class="st"> </span><span class="kw">getTableColumnNames</span>(<span class="dt">table=</span><span class="st">&quot;node&quot;</span>,  <span class="dt">network =</span>  <span class="kw">as.numeric</span>(em_network_suid))

  descr_attrib &lt;-<span class="st"> </span>nodetable_colnames[<span class="kw">grep</span>(nodetable_colnames, <span class="dt">pattern =</span> <span class="st">&quot;_GS_DESCR&quot;</span>)]

  <span class="co">#Autoannotate the network</span>
  autoannotate_url &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;autoannotate annotate-clusterBoosted labelColumn=&quot;</span>, descr_attrib,<span class="st">&quot; maxWords=3 &quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)
    current_name &lt;-<span class="kw">commandsGET</span>(autoannotate_url)</code></pre></div>
<p>Export image of resulting Annotated Enrichment map.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cluster1em_annot_png_file_name &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(),<span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>, <span class="st">&quot;images&quot;</span>,<span class="st">&quot;cluster1em_annot.png&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="kw">file.exists</span>(cluster1em_annot_png_file_name)){
  <span class="co">#cytoscape hangs waiting for user response if file alreayd exists.  Remove it first</span>
  <span class="kw">file.remove</span>(cluster1em_annot_png_file_name)
  } 

<span class="co">#export the network</span>
<span class="kw">exportImage</span>(cluster1em_annot_png_file_name, <span class="dt">type =</span> <span class="st">&quot;png&quot;</span>)</code></pre></div>
<div class="figure">
<img src="230_Isserlin_RCy3_intro/images/cluster1em_annot.png" alt="Example Annotated Enrichment Map created when running an enrichment analysis using g:Profiler with the genes that are part of cluster 1" width="339" />
<p class="caption">
(#fig:cluster1em_annot_fig)Example Annotated Enrichment Map created when running an enrichment analysis using g:Profiler with the genes that are part of cluster 1
</p>
</div>
<p>Dense networks small or large never look like network figures we so often see in journals. A lot of manual tweaking, reorganization and optimization is involved in getting that perfect figure ready network. The above network is what the network starts as. The below figure is what it can look like after a few minutes of manual reorganiazation. (individual clusters were selected from the auto annotate panel and separated from other clusters)</p>
<div class="figure">
<img src="230_Isserlin_RCy3_intro/images/cluster1em_annot_man.png" alt="Example Annotated Enrichment Map created when running an enrichment analysis using g:Profiler with the genes that are part of cluster 1 after manual adjusting to generate a cleaner figure" width="329" />
<p class="caption">
(#fig:cluster1em_annot_man_fig)Example Annotated Enrichment Map created when running an enrichment analysis using g:Profiler with the genes that are part of cluster 1 after manual adjusting to generate a cleaner figure
</p>
</div>
</div>
<div id="use-case-3---functional-enrichment-of-omics-set." class="section level3">
<h3><span class="header-section-number">3.3.8</span> Use Case 3 - Functional Enrichment of Omics set.</h3>
<p>Reducing our large OMICs expression set to a simple list of genes eliminates the majority of the signals present in the data. Thresholding will only highlight the strong signals neglecting the often more interesting subtle signals. In order to capitalize on the wealth of data present in the data we need to perform pathway enrichment analysis on the entire expression set. There are many tools in R or as standalone apps that perform this type of analysis.</p>
<p>To demonstrate how you can use Cytoscape and RCy3 in your enrichment analysis pipeline we will use EnrichmentBrowser package (as demonstrated in detail in the workshop <a href="https://github.com/Bioconductor/BiocWorkshops/blob/master/Geistlinger_Omics.Rmd%5D">Functional enrichment analysis of high-throughput omics data</a>) to perform pathway analysis.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="op">!</span><span class="st">&quot;EnrichmentBrowser&quot;</span> <span class="op">%in%</span><span class="st"> </span><span class="kw">installed.packages</span>()){
    <span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)
    BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;EnrichmentBrowser&quot;</span>)
}

<span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(EnrichmentBrowser))</code></pre></div>
<p>Download the latest pathway definition file from the <a href="http://download.baderlab.org/EM_Genesets/current_release/">Baderlab download site</a>. Baderlab genesets are updated on a monthly basis. Detailed information about the sources can be found <a href="http://baderlab.org/GeneSets">here</a>. Only Human, Mouse and Rat gene set files are currently available on the Baderlab downloads site. If you are working with a species other than human (and it is either rat or mouse) change the gmt_url below to correct species.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">tryCatch</span>(<span class="dt">expr =</span> { <span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(<span class="st">&quot;RCurl&quot;</span>))}, 
         <span class="dt">error =</span> <span class="cf">function</span>(e) {  <span class="kw">install.packages</span>(<span class="st">&quot;RCurl&quot;</span>)}, 
         <span class="dt">finally =</span> <span class="kw">library</span>(<span class="st">&quot;RCurl&quot;</span>))

gmt_url =<span class="st"> &quot;http://download.baderlab.org/EM_Genesets/current_release/Human/symbol/&quot;</span>


<span class="co">#list all the files on the server</span>
filenames =<span class="st"> </span><span class="kw">getURL</span>(gmt_url)
tc =<span class="st"> </span><span class="kw">textConnection</span>(filenames)
contents =<span class="st"> </span><span class="kw">suppressWarnings</span>(<span class="kw">readLines</span>(tc))
<span class="kw">close</span>(tc)

<span class="co">#get the gmt that has all the pathways and does not include terms inferred from electronic annotations(IEA)</span>
<span class="co">#start with gmt file that has pathways only</span>
rx =<span class="st"> </span><span class="kw">gregexpr</span>(<span class="st">&quot;(?&lt;=&lt;a href=</span><span class="ch">\&quot;</span><span class="st">)(.*.GOBP_AllPathways_no_GO_iea.*.)(.gmt)(?=</span><span class="ch">\&quot;</span><span class="st">&gt;)&quot;</span>,
  contents, <span class="dt">perl =</span> <span class="ot">TRUE</span>)
gmt_file =<span class="st"> </span><span class="kw">unlist</span>(<span class="kw">regmatches</span>(contents, rx))

dest_gmt_file &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(), <span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>, <span class="st">&quot;data&quot;</span>, gmt_file)

<span class="kw">download.file</span>(
    <span class="kw">paste</span>(gmt_url,gmt_file,<span class="dt">sep=</span><span class="st">&quot;&quot;</span>),
    <span class="dt">destfile=</span>dest_gmt_file
)</code></pre></div>
<p>Load in the gmt file</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">baderlab.gs &lt;-<span class="st"> </span><span class="kw">getGenesets</span>(dest_gmt_file)
baderlab.gs[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]
<span class="co">#&gt; $`THIO-MOLYBDENUM COFACTOR BIOSYNTHESIS%HUMANCYC%PWY-5963`</span>
<span class="co">#&gt; [1] &quot;MOCOS&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`PROLINE BIOSYNTHESIS I%HUMANCYC%PROSYN-PWY`</span>
<span class="co">#&gt; [1] &quot;PYCR2&quot;    &quot;ALDH18A1&quot; &quot;PYCR1&quot;    &quot;PYCRL&quot;</span></code></pre></div>
<p>Create the dataset required by EnrichmentBrowser tools</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#create the expression file - A tab separated text file containing expression values. Columns = samples/subjects; rows = features/probes/genes; NO headers, row or column names.</span>
expr &lt;-<span class="st"> </span>RNASeq_expression

sumexpr_filename &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(), <span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>,<span class="st">&quot;data&quot;</span>,<span class="st">&quot;SummarizeExperiment_expression.txt&quot;</span>) 
<span class="kw">write.table</span>( expr ,  <span class="dt">file =</span> sumexpr_filename , <span class="dt">col.names  =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>)


rowData &lt;-<span class="st"> </span>RNASeq_gene_scores[,<span class="kw">grep</span>(<span class="kw">colnames</span>(RNASeq_gene_scores), <span class="dt">pattern=</span><span class="st">&quot;mesen&quot;</span>)]
rowData &lt;-<span class="st"> </span><span class="kw">cbind</span>(RNASeq_gene_scores<span class="op">$</span>Name,rowData)
<span class="kw">colnames</span>(rowData)[<span class="dv">2</span>] &lt;-<span class="st"> &quot;FC&quot;</span>
<span class="kw">colnames</span>(rowData)[<span class="dv">6</span>] &lt;-<span class="st"> &quot;ADJ.PVAL&quot;</span>

sumexpr_rdat_filename &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(), <span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>,<span class="st">&quot;data&quot;</span>,<span class="st">&quot;SummarizeExperiment_rdat.txt&quot;</span>) 
<span class="kw">write.table</span>( rowData[,<span class="dv">1</span>] ,  <span class="dt">file =</span> sumexpr_rdat_filename , <span class="dt">col.names  =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>)

<span class="co">#load in the data classification data</span>
<span class="co"># A tab separated text file containing annotation information for the samples in either *two or three* columns. NO headers, row or column names. The number of rows/samples in this file should match the number of columns/samples of the expression matrix. The 1st column is reserved for the sample IDs; The 2nd column is reserved for a *BINARY* group assignment. Use &#39;0&#39; and &#39;1&#39; for unaffected (controls) and affected (cases) sample class</span>
classDefinitions_RNASeq &lt;-<span class="st"> </span><span class="kw">read.table</span>( 
  <span class="kw">file.path</span>(<span class="kw">getwd</span>(), <span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>,<span class="st">&quot;data&quot;</span>,<span class="st">&quot;RNASeq_classdefinitions.txt&quot;</span>), <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote=</span><span class="st">&quot;</span><span class="ch">\&quot;</span><span class="st">&quot;</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)

colData &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Sample =</span> <span class="kw">colnames</span>(RNASeq_expression),
                      <span class="dt">GROUP =</span> classDefinitions_RNASeq<span class="op">$</span>SUBTYPE, 
                      <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
<span class="kw">rownames</span>(colData) &lt;-<span class="st"> </span><span class="kw">colnames</span>(RNASeq_expression)
colData<span class="op">$</span>GROUP[<span class="kw">which</span>(colData<span class="op">$</span>GROUP <span class="op">!=</span><span class="st"> &quot;Mesenchymal&quot;</span>)] &lt;-<span class="st"> </span><span class="dv">0</span>
colData<span class="op">$</span>GROUP[<span class="kw">which</span>(colData<span class="op">$</span>GROUP <span class="op">==</span><span class="st"> &quot;Mesenchymal&quot;</span>)] &lt;-<span class="st"> </span><span class="dv">1</span>

sumexpr_cdat_filename &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(), <span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>,<span class="st">&quot;data&quot;</span>,<span class="st">&quot;SummarizeExperiment_cdat.txt&quot;</span>) 
<span class="kw">write.table</span>( colData ,  <span class="dt">file =</span> sumexpr_cdat_filename , <span class="dt">col.names  =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">quote=</span><span class="ot">FALSE</span>)

<span class="co">#create the Summarize Experiment object</span>
se_OV &lt;-<span class="st"> </span><span class="kw">readSE</span>(<span class="dt">assay.file =</span> sumexpr_filename , <span class="dt">cdat.file =</span> sumexpr_cdat_filename, <span class="dt">rdat.file =</span> sumexpr_rdat_filename)</code></pre></div>
<p>Put our precomputed p-values and fold change values into the Summarized Experiment object so we can use our rankings for the analysis</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#set the Summarized Experiment to our computed p-values and FC</span>
<span class="kw">rowData</span>(se_OV) &lt;-<span class="st"> </span>rowData</code></pre></div>
<p>Run basic Over representation analysis (ORA) using our ranked genes and our gene set file downloaded from the Baderlab genesets.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ora.all &lt;-<span class="st"> </span><span class="kw">sbea</span>(<span class="dt">method=</span><span class="st">&quot;ora&quot;</span>, <span class="dt">se=</span>se_OV, <span class="dt">gs=</span>baderlab.gs, <span class="dt">perm=</span><span class="dv">0</span>, <span class="dt">alpha=</span><span class="fl">0.05</span>)
<span class="kw">gsRanking</span>(ora.all)
<span class="co">#&gt; DataFrame with 967 rows and 4 columns</span>
<span class="co">#&gt;                                                                                            GENE.SET</span>
<span class="co">#&gt;                                                                                         &lt;character&gt;</span>
<span class="co">#&gt; 1                                                 EXTRACELLULAR MATRIX ORGANIZATION%GOBP%GO:0030198</span>
<span class="co">#&gt; 2   HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION%MSIGDB_C2%HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION</span>
<span class="co">#&gt; 3                                              EXTRACELLULAR STRUCTURE ORGANIZATION%GOBP%GO:0043062</span>
<span class="co">#&gt; 4                                        EXTRACELLULAR MATRIX ORGANIZATION%REACTOME%R-HSA-1474244.2</span>
<span class="co">#&gt; 5                                                 NABA_CORE_MATRISOME%MSIGDB_C2%NABA_CORE_MATRISOME</span>
<span class="co">#&gt; ...                                                                                             ...</span>
<span class="co">#&gt; 963                                                                 OVULATION CYCLE%GOBP%GO:0042698</span>
<span class="co">#&gt; 964                                                NEGATIVE REGULATION OF TRANSPORT%GOBP%GO:0051051</span>
<span class="co">#&gt; 965                                                  PEPTIDYL-TYROSINE MODIFICATION%GOBP%GO:0018212</span>
<span class="co">#&gt; 966                                                        NEUTROPHIL DEGRANULATION%GOBP%GO:0043312</span>
<span class="co">#&gt; 967           PROCESSING OF CAPPED INTRON-CONTAINING PRE-MRNA%REACTOME DATABASE ID RELEASE 65%72203</span>
<span class="co">#&gt;      NR.GENES NR.SIG.GENES   P.VALUE</span>
<span class="co">#&gt;     &lt;numeric&gt;    &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; 1         195          147  6.42e-21</span>
<span class="co">#&gt; 2         193          145  2.07e-20</span>
<span class="co">#&gt; 3         223          160  5.43e-19</span>
<span class="co">#&gt; 4         239          163   6.1e-16</span>
<span class="co">#&gt; 5         215          145  1.03e-13</span>
<span class="co">#&gt; ...       ...          ...       ...</span>
<span class="co">#&gt; 963        13            9    0.0487</span>
<span class="co">#&gt; 964       246          118    0.0488</span>
<span class="co">#&gt; 965        76           40     0.049</span>
<span class="co">#&gt; 966       416          194    0.0496</span>
<span class="co">#&gt; 967       213          103    0.0499</span></code></pre></div>
<p><img src="230_Isserlin_RCy3_intro/images/screenshot_ora_results.png" width="518" /></p>
<p>Take the enrichment results and create a generic enrichment map input file so we can create an Enrichment map. Description of format of the generic input file can be found <a href="http://enrichmentmap.readthedocs.io/en/latest/FileFormats.html#generic-results-files">here</a> and example generic enrichment map files can be found <a href="http://enrichmentmap.readthedocs.io/en/latest/FileFormats.html#examples-of-generic-enrichment-result-files">here</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#manually adjust p-values</span>
ora.all<span class="op">$</span>res.tbl &lt;-<span class="st"> </span><span class="kw">cbind</span>(ora.all<span class="op">$</span>res.tbl, <span class="kw">p.adjust</span>(ora.all<span class="op">$</span>res.tbl<span class="op">$</span>P.VALUE, <span class="st">&quot;BH&quot;</span>))
<span class="kw">colnames</span>(ora.all<span class="op">$</span>res.tbl)[<span class="kw">ncol</span>(ora.all<span class="op">$</span>res.tbl)] &lt;-<span class="st"> &quot;Q.VALUE&quot;</span> 

<span class="co">#create a generic enrichment map file</span>
em_results_mesen &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">name =</span> ora.all<span class="op">$</span>res.tbl<span class="op">$</span>GENE.SET,<span class="dt">descr =</span> ora.all<span class="op">$</span>res.tbl<span class="op">$</span>GENE.SET, 
                               <span class="dt">pvalue=</span>ora.all<span class="op">$</span>res.tbl<span class="op">$</span>P.VALUE, <span class="dt">qvalue=</span>ora.all<span class="op">$</span>res.tbl<span class="op">$</span>Q.VALUE, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)

<span class="co">#write out the ora results</span>
 em_results_mesen_filename &lt;-<span class="kw">file.path</span>(<span class="kw">getwd</span>(),
                            <span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>,<span class="st">&quot;data&quot;</span>,<span class="st">&quot;mesen_ora_enr_results.txt&quot;</span>)

  <span class="kw">write.table</span>(em_results_mesen,em_results_mesen_filename,<span class="dt">col.name=</span><span class="ot">TRUE</span>,<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,<span class="dt">row.names=</span><span class="ot">FALSE</span>,<span class="dt">quote=</span><span class="ot">FALSE</span>)
  
 </code></pre></div>
<p>Create an enrichment map with the returned ORA results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"> em_command =<span class="st"> </span><span class="kw">paste</span>(<span class="st">&#39;enrichmentmap build analysisType=&quot;generic&quot; &#39;</span>, <span class="st">&quot;gmtFile=&quot;</span>, dest_gmt_file,
                   <span class="st">&#39;pvalue=&#39;</span>,<span class="st">&quot;0.05&quot;</span>, <span class="st">&#39;qvalue=&#39;</span>,<span class="st">&quot;0.05&quot;</span>,
                   <span class="st">&#39;similaritycutoff=&#39;</span>,<span class="st">&quot;0.25&quot;</span>,
                   <span class="st">&#39;coeffecients=&#39;</span>,<span class="st">&quot;JACCARD&quot;</span>,
                   <span class="st">&#39;enrichmentsDataset1=&#39;</span>,em_results_mesen_filename ,
                   <span class="dt">sep=</span><span class="st">&quot; &quot;</span>)

  <span class="co">#enrichment map command will return the suid of newly created network.</span>
  em_mesen_network_suid &lt;-<span class="st"> </span><span class="kw">commandsRun</span>(em_command)
  
  <span class="kw">renameNetwork</span>(<span class="st">&quot;Mesenchymal_ORA_enrichmentmap&quot;</span>, <span class="dt">network=</span><span class="kw">as.numeric</span>(em_mesen_network_suid))</code></pre></div>
<p>Export image of resulting Enrichment map.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mesenem_png_file_name &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(),<span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>, <span class="st">&quot;images&quot;</span>,<span class="st">&quot;mesenem.png&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="kw">file.exists</span>(mesenem_png_file_name)){
  <span class="co">#cytoscape hangs waiting for user response if file alreayd exists.  Remove it first</span>
  <span class="kw">file.remove</span>(mesenem_png_file_name)
  } 

<span class="co">#export the network</span>
<span class="kw">exportImage</span>(mesenem_png_file_name, <span class="dt">type =</span> <span class="st">&quot;png&quot;</span>)</code></pre></div>
<div class="figure">
<img src="230_Isserlin_RCy3_intro/images/mesenem.png" alt="Example Enrichment Map created when running an enrichment analysis using EnrichmentBrowser ORA with the genes differential in Mesenchymal OV" width="339" />
<p class="caption">
(#fig:mesenem_fig)Example Enrichment Map created when running an enrichment analysis using EnrichmentBrowser ORA with the genes differential in Mesenchymal OV
</p>
</div>
<p>Annotate the Enrichment map to get the general themes that are found in the enrichment results. Often for very busy networks annotating the networks doesnâ€™t help to reduce the complexity but instead adds to it. To get rid of some of the pathway redundancy and density in the network create a summary of the underlying network. The summary network collapses each cluster to a summary node. Each summary node is annotated with a word tag (the top 3 words associated with the nodes of the cluster) that is computed using the Wordcloud app.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co">#get the column from the nodetable and node table</span>
  nodetable_colnames &lt;-<span class="st"> </span><span class="kw">getTableColumnNames</span>(<span class="dt">table=</span><span class="st">&quot;node&quot;</span>,  <span class="dt">network =</span>  <span class="kw">as.numeric</span>(em_mesen_network_suid))

  descr_attrib &lt;-<span class="st"> </span>nodetable_colnames[<span class="kw">grep</span>(nodetable_colnames, <span class="dt">pattern =</span> <span class="st">&quot;_GS_DESCR&quot;</span>)]

  <span class="co">#Autoannotate the network</span>
  autoannotate_url &lt;-<span class="st"> </span><span class="kw">paste</span>(<span class="st">&quot;autoannotate annotate-clusterBoosted labelColumn=&quot;</span>, descr_attrib,<span class="st">&quot; maxWords=3 &quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>)
    current_name &lt;-<span class="kw">commandsGET</span>(autoannotate_url)
    
  <span class="co">#create a summary network </span>
  <span class="kw">commandsGET</span>(<span class="st">&quot;autoannotate summary network=&#39;current&#39;&quot;</span>)
  
  <span class="co">#change the network name</span>
  summary_network_suid &lt;-<span class="st"> </span><span class="kw">getNetworkSuid</span>()
  
  <span class="kw">renameNetwork</span>(<span class="dt">title =</span> <span class="st">&quot;Mesen_ORA_summary_netowrk&quot;</span>,
              <span class="dt">network =</span> <span class="kw">as.numeric</span>(summary_network_suid))
  
  <span class="co">#get the summary node names</span>
  summary_nodes &lt;-<span class="st"> </span><span class="kw">getTableColumns</span>(<span class="dt">table=</span><span class="st">&quot;node&quot;</span>,<span class="dt">columns=</span><span class="kw">c</span>(<span class="st">&quot;name&quot;</span>))
  
  <span class="co">#clear bypass style the summary network has</span>
  <span class="kw">clearNodePropertyBypass</span>(<span class="dt">node.names =</span> summary_nodes<span class="op">$</span>name,<span class="dt">visual.property =</span> <span class="st">&quot;NODE_SIZE&quot;</span>)
  
  <span class="co">#relayout network</span>
  <span class="kw">layoutNetwork</span>(<span class="st">&#39;cose&#39;</span>,
              <span class="dt">network =</span> <span class="kw">as.numeric</span>(summary_network_suid))</code></pre></div>
<p>Export image of resulting Summarized Annotated Enrichment map.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mesenem_summary_png_file_name &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="kw">getwd</span>(),<span class="st">&quot;230_Isserlin_RCy3_intro&quot;</span>,<span class="st">&quot;images&quot;</span>,<span class="st">&quot;mesenem_summary_network.png&quot;</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="kw">file.exists</span>(mesenem_summary_png_file_name)){
  <span class="co">#cytoscape hangs waiting for user response if file alreayd exists.  Remove it first</span>
  <span class="kw">file.remove</span>(mesenem_summary_png_file_name)
  } 

<span class="co">#export the network</span>
<span class="kw">exportImage</span>(mesenem_summary_png_file_name, <span class="dt">type =</span> <span class="st">&quot;png&quot;</span>)</code></pre></div>
<div class="figure">
<img src="230_Isserlin_RCy3_intro/images/mesenem_summary_network.png" alt="Example Annotated Enrichment Map created when running an enrichment analysis using EnrichmentBrowser ORA with the genes that differential in mesenchymal OV" width="339" />
<p class="caption">
(#fig:cmesenem_annot_fig)Example Annotated Enrichment Map created when running an enrichment analysis using EnrichmentBrowser ORA with the genes that differential in mesenchymal OV
</p>
</div>
</div>
</div>
<div id="references" class="section level2">
<h2><span class="header-section-number">3.4</span> References</h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="rna-seq-data-analysis-with-deseq2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
