<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7.13 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="granges-genomic-ranges.html">
<link rel="next" href="introduction-to-bioconductor-annotation-resources.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#for-everyone"><i class="fa fa-check"></i><b>1.1</b> For Everyone</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#for-workshop-authors"><i class="fa fa-check"></i><b>1.2</b> For Workshop Authors</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#description"><i class="fa fa-check"></i><b>1.2.1</b> DESCRIPTION</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#classifying-your-workshop"><i class="fa fa-check"></i><b>1.2.2</b> Classifying your workshop</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#bookdown"><i class="fa fa-check"></i><b>1.2.3</b> Bookdown</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#deadlines-for-bioc2018"><i class="fa fa-check"></i><b>1.3</b> Deadlines for Bioc2018</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>2</b> RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-1"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>2.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>2.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="2.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>2.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="2.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>2.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="2.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>2.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>2.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>2.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="2.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>2.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="2.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>2.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>2.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="2.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>2.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="2.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>2.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="2.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>2.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis"><i class="fa fa-check"></i><b>2.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="2.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>2.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="2.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>2.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub"><i class="fa fa-check"></i><b>2.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="2.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub"><i class="fa fa-check"></i><b>2.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="2.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>2.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>2.7</b> Building reports</a><ul>
<li class="chapter" data-level="2.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>2.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="2.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>2.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>2.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="2.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>2.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="2.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>2.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="2.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>2.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="2.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>2.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="2.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>2.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="2.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>2.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>3</b> Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="3.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-1"><i class="fa fa-check"></i><b>3.1</b> Overview</a><ul>
<li class="chapter" data-level="3.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-2"><i class="fa fa-check"></i><b>3.1.1</b> Description</a></li>
<li class="chapter" data-level="3.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-1"><i class="fa fa-check"></i><b>3.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="3.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-1"><i class="fa fa-check"></i><b>3.1.3</b> Participation</a></li>
<li class="chapter" data-level="3.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="3.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-1"><i class="fa fa-check"></i><b>3.1.5</b> Time outline</a></li>
<li class="chapter" data-level="3.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started"><i class="fa fa-check"></i><b>3.2</b> Getting started</a><ul>
<li class="chapter" data-level="3.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>3.2.1</b> The data</a></li>
<li class="chapter" data-level="3.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>3.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>3.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="3.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>3.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="3.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>3.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>3.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="3.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>3.5.1</b> Normalization</a></li>
<li class="chapter" data-level="3.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>3.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>3.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="3.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>3.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="3.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>3.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><i class="fa fa-check"></i><b>4</b> Variant Functional Annotation using StatePaintR, FunciVar &amp; MotifBreakR</a><ul>
<li class="chapter" data-level="4.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#instructors-names-and-contact-information"><i class="fa fa-check"></i><b>4.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="4.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-description"><i class="fa fa-check"></i><b>4.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="4.2.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#pre-requisites-2"><i class="fa fa-check"></i><b>4.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.2.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-participation"><i class="fa fa-check"></i><b>4.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="4.2.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.2.4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#time-outline-2"><i class="fa fa-check"></i><b>4.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="4.3.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#key-concepts"><i class="fa fa-check"></i><b>4.3.1</b> Key Concepts</a></li>
<li class="chapter" data-level="4.3.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#learning-objectives"><i class="fa fa-check"></i><b>4.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>5</b> Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="5.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i><b>5.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="5.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-1"><i class="fa fa-check"></i><b>5.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="5.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-3"><i class="fa fa-check"></i><b>5.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="5.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-1"><i class="fa fa-check"></i><b>5.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>5.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="5.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-3"><i class="fa fa-check"></i><b>5.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>5.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="5.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop"><i class="fa fa-check"></i><b>5.4</b> Workshop</a></li>
<li class="chapter" data-level="5.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>5.5</b> Where does it all come from?</a></li>
<li class="chapter" data-level="5.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>5.6</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="5.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>5.7</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="5.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>5.8</b> Data types</a></li>
<li class="chapter" data-level="5.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>5.9</b> Differential expression analysis</a></li>
<li class="chapter" data-level="5.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>5.10</b> Gene sets</a></li>
<li class="chapter" data-level="5.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>5.11</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="5.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>5.12</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="5.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>5.13</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="5.14" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>5.14</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>6</b> Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="6.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructors-names-and-contact-information-2"><i class="fa fa-check"></i><b>6.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="6.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-2"><i class="fa fa-check"></i><b>6.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="6.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-4"><i class="fa fa-check"></i><b>6.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-2"><i class="fa fa-check"></i><b>6.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="6.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>6.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-4"><i class="fa fa-check"></i><b>6.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>6.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="6.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals"><i class="fa fa-check"></i><b>6.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="6.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-1"><i class="fa fa-check"></i><b>6.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="granges-genomic-ranges.html"><a href="granges-genomic-ranges.html"><i class="fa fa-check"></i><b>7</b> <em>GRanges</em>: Genomic Ranges</a><ul>
<li class="chapter" data-level="7.1" data-path="granges-genomic-ranges.html"><a href="granges-genomic-ranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>7.1</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="7.2" data-path="granges-genomic-ranges.html"><a href="granges-genomic-ranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>7.2</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="7.3" data-path="granges-genomic-ranges.html"><a href="granges-genomic-ranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>7.3</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="7.4" data-path="granges-genomic-ranges.html"><a href="granges-genomic-ranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>7.4</b> Interval set operations for <em>GRanges</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="grangeslist-groups-of-genomic-ranges.html"><a href="grangeslist-groups-of-genomic-ranges.html"><i class="fa fa-check"></i><b>8</b> <em>GRangesList</em>: Groups of Genomic Ranges</a><ul>
<li class="chapter" data-level="8.1" data-path="grangeslist-groups-of-genomic-ranges.html"><a href="grangeslist-groups-of-genomic-ranges.html#basic-grangeslist-accessors"><i class="fa fa-check"></i><b>8.1</b> Basic <em>GRangesList</em> accessors</a></li>
<li class="chapter" data-level="8.2" data-path="grangeslist-groups-of-genomic-ranges.html"><a href="grangeslist-groups-of-genomic-ranges.html#combining-grangeslist-objects"><i class="fa fa-check"></i><b>8.2</b> Combining <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="8.3" data-path="grangeslist-groups-of-genomic-ranges.html"><a href="grangeslist-groups-of-genomic-ranges.html#basic-interval-operations-for-grangeslist-objects"><i class="fa fa-check"></i><b>8.3</b> Basic interval operations for <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="8.4" data-path="grangeslist-groups-of-genomic-ranges.html"><a href="grangeslist-groups-of-genomic-ranges.html#subsetting-grangeslist-objects"><i class="fa fa-check"></i><b>8.4</b> Subsetting <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="8.5" data-path="grangeslist-groups-of-genomic-ranges.html"><a href="grangeslist-groups-of-genomic-ranges.html#looping-over-grangeslist-objects"><i class="fa fa-check"></i><b>8.5</b> Looping over <em>GRangesList</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>9</b> Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="9.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>9.1</b> Instructors</a></li>
<li class="chapter" data-level="9.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description-3"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>9.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation-3"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>9.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="9.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-1"><i class="fa fa-check"></i><b>9.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="9.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-2"><i class="fa fa-check"></i><b>9.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>9.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="9.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>9.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="9.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>9.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="9.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>9.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="9.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>9.4.4</b> Data containers</a></li>
<li class="chapter" data-level="9.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>9.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="9.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>9.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="9.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>9.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="9.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>9.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="9.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>9.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="9.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>9.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="9.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>9.4.11</b> Simple example</a></li>
<li class="chapter" data-level="9.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>9.4.12</b> Questions!</a></li>
<li class="chapter" data-level="9.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>9.4.13</b> More questions!</a></li>
<li class="chapter" data-level="9.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>9.4.14</b> Another example</a></li>
<li class="chapter" data-level="9.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>9.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="9.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>9.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="9.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>9.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="9.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>9.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="9.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>9.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="9.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>9.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="9.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>9.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="9.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>9.4.22</b> GRanges</a></li>
<li class="chapter" data-level="9.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>9.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="9.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>9.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="9.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>9.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="9.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>9.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="9.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>9.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="9.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>9.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="9.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>9.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="9.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>9.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="9.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>9.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="9.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>9.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="9.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dply-example"><i class="fa fa-check"></i><b>9.4.33</b> Organism.dply example</a></li>
<li class="chapter" data-level="9.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>9.4.34</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="9.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>9.4.35</b> BSgenome packages</a></li>
<li class="chapter" data-level="9.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>9.4.36</b> BSgenome packages</a></li>
<li class="chapter" data-level="9.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>9.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="9.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>9.4.38</b> BSgenome exercises</a></li>
<li class="chapter" data-level="9.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-1"><i class="fa fa-check"></i><b>9.4.39</b> AnnotationHub</a></li>
<li class="chapter" data-level="9.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>9.4.40</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="9.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>9.4.41</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="9.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>9.4.42</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="9.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>9.4.43</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="9.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>9.4.44</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="9.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>9.4.45</b> AnnotationHub query</a></li>
<li class="chapter" data-level="9.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>9.4.46</b> AnnotationHub query</a></li>
<li class="chapter" data-level="9.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>9.4.47</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="9.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>9.4.48</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="9.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>9.4.49</b> biomaRt</a></li>
<li class="chapter" data-level="9.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>9.4.50</b> biomaRt data sets</a></li>
<li class="chapter" data-level="9.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>9.4.51</b> biomaRt queries</a></li>
<li class="chapter" data-level="9.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>9.4.52</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="9.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>9.4.53</b> biomaRt query</a></li>
<li class="chapter" data-level="9.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>9.4.54</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>10</b> Workflow for Multi-omics Analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="10.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>10.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-4"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-5"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-4"><i class="fa fa-check"></i><b>10.2.2</b> workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>10.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-5"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-2"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-3"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>10.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="10.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summarizedexperiment-raggedexperiment"><i class="fa fa-check"></i><b>10.4.1</b> <code>SummarizedExperiment</code> / <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="10.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>10.4.2</b> <code>MultiAssayExperiment</code></a></li>
<li class="chapter" data-level="10.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>10.4.3</b> <code>RaggedExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>10.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="10.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>10.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="10.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#using-granges-objects"><i class="fa fa-check"></i><b>10.5.2</b> Using <code>GRanges</code> objects</a></li>
<li class="chapter" data-level="10.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>10.5.3</b> *Assay functions</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#qreduceassay"><i class="fa fa-check"></i><b>10.6</b> qreduceAssay</a></li>
<li class="chapter" data-level="10.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>10.7</b> Working with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="10.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>10.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="10.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>10.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="10.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>10.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="10.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>10.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="10.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>10.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>10.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="10.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>10.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="10.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>10.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="10.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>10.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#patients-with-complete-data"><i class="fa fa-check"></i><b>10.9</b> Patients with complete data</a></li>
<li class="chapter" data-level="10.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>10.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="10.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>10.11</b> Extraction</a><ul>
<li class="chapter" data-level="10.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>10.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="10.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>10.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="10.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>10.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="10.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>10.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="10.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>10.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="10.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>10.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="10.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>10.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="10.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>10.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="10.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>10.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>10.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="10.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiments"><i class="fa fa-check"></i><b>10.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiments</a></li>
<li class="chapter" data-level="10.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>10.16</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="10.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>10.16.1</b> How many samples have data for each combination of assays?</a></li>
</ul></li>
<li class="chapter" data-level="10.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#tcgautils-package"><i class="fa fa-check"></i><b>10.17</b> <code>TCGAutils</code> package</a><ul>
<li class="chapter" data-level="10.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>10.17.1</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="10.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#is-there-subtype-data-available-in-the-multiassayexperiment-obtained-from"><i class="fa fa-check"></i><b>10.17.2</b> Is there subtype data available in the <code>MultiAssayExperiment</code> obtained from</a></li>
<li class="chapter" data-level="10.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>10.17.3</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="10.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>10.17.4</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="10.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>10.17.5</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="10.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>10.17.6</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="10.17.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>10.17.7</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>11</b> Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="11.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#authors"><i class="fa fa-check"></i><b>11.1</b> Authors:</a></li>
<li class="chapter" data-level="11.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-5"><i class="fa fa-check"></i><b>11.2</b> Workshop Description:</a></li>
<li class="chapter" data-level="11.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-6"><i class="fa fa-check"></i><b>11.3</b> Pre-requisites:</a></li>
<li class="chapter" data-level="11.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-2"><i class="fa fa-check"></i><b>11.4</b> Participation:</a></li>
<li class="chapter" data-level="11.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>11.5</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="11.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>11.6</b> Workshop goals and objectives:</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>12</b> Public Data Resources and Bioconductor</a><ul>
<li class="chapter" data-level="12.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-6"><i class="fa fa-check"></i><b>12.1</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.1.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-7"><i class="fa fa-check"></i><b>12.1.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="12.1.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-5"><i class="fa fa-check"></i><b>12.1.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.1.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>12.1.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="12.1.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-6"><i class="fa fa-check"></i><b>12.1.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>12.2</b> workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-3"><i class="fa fa-check"></i><b>12.2.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-4"><i class="fa fa-check"></i><b>12.2.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-2"><i class="fa fa-check"></i><b>12.3</b> Overview</a></li>
<li class="chapter" data-level="12.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>12.4</b> GEOquery</a></li>
<li class="chapter" data-level="12.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>12.5</b> GenomicDataCommons</a></li>
<li class="chapter" data-level="12.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>12.6</b> Sequence Read Archive</a></li>
<li class="chapter" data-level="12.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>12.7</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="12.7.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata"><i class="fa fa-check"></i><b>12.7.1</b> curatedTCGAData</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#gbm-dataset-example"><i class="fa fa-check"></i><b>12.8</b> GBM dataset example</a><ul>
<li class="chapter" data-level="12.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#subtype-information"><i class="fa fa-check"></i><b>12.8.1</b> Subtype information</a></li>
<li class="chapter" data-level="12.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>12.8.2</b> TCGABiolinks</a></li>
<li class="chapter" data-level="12.8.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons-1"><i class="fa fa-check"></i><b>12.8.3</b> GenomicDataCommons</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>12.9</b> Microbiome data</a></li>
<li class="chapter" data-level="12.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>12.10</b> Pharmacogenomics</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="grangeslist-groups-of-genomic-ranges" class="section level1">
<h1><span class="header-section-number">8</span> <em>GRangesList</em>: Groups of Genomic Ranges</h1>
<p>Some important genomic features, such as spliced transcripts that are are comprised of exons, are inherently compound structures. Such a feature makes much more sense when expressed as a compound object such as a <em>GRangesList</em>. Whenever genomic features consist of multiple ranges that are grouped by a parent feature, they can be represented as a <em>GRangesList</em> object. Consider the simple example of the two transcript <code>GRangesList</code> below created using the <code>GRangesList</code> constructor.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gr1 &lt;-<span class="st"> </span><span class="kw">GRanges</span>(
    <span class="dt">seqnames =</span> <span class="st">&quot;chr2&quot;</span>, 
    <span class="dt">ranges =</span> <span class="kw">IRanges</span>(<span class="dv">103</span>, <span class="dv">106</span>),
    <span class="dt">strand =</span> <span class="st">&quot;+&quot;</span>, 
    <span class="dt">score =</span> 5L, <span class="dt">GC =</span> <span class="fl">0.45</span>)
gr2 &lt;-<span class="st"> </span><span class="kw">GRanges</span>(
    <span class="dt">seqnames =</span> <span class="kw">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr1&quot;</span>),
    <span class="dt">ranges =</span> <span class="kw">IRanges</span>(<span class="kw">c</span>(<span class="dv">107</span>, <span class="dv">113</span>), <span class="dt">width =</span> <span class="dv">3</span>),
    <span class="dt">strand =</span> <span class="kw">c</span>(<span class="st">&quot;+&quot;</span>, <span class="st">&quot;-&quot;</span>), 
    <span class="dt">score =</span> <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>, <span class="dt">GC =</span> <span class="kw">c</span>(<span class="fl">0.3</span>, <span class="fl">0.5</span>))
grl &lt;-<span class="st"> </span><span class="kw">GRangesList</span>(<span class="st">&quot;txA&quot;</span> =<span class="st"> </span>gr1, <span class="st">&quot;txB&quot;</span> =<span class="st"> </span>gr2)
grl
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames  ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1 107-109      + |     3 0.3</span>
<span class="co">#&gt;   [2]     chr1 113-115      - |     4 0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>The <code>show</code> method for a <em>GRangesList</em> object displays it as a named list of <em>GRanges</em> objects, where the names of this list are considered to be the names of the grouping feature. In the example above, the groups of individual exon ranges are represented as separate <em>GRanges</em> objects which are further organized into a list structure where each element name is a transcript name. Many other combinations of grouped and labeled <em>GRanges</em> objects are possible of course, but this example is expected to be a common arrangement.</p>
<div id="basic-grangeslist-accessors" class="section level2">
<h2><span class="header-section-number">8.1</span> Basic <em>GRangesList</em> accessors</h2>
<p>Just as with <em>GRanges</em> object, the components of the genomic coordinates within a <em>GRangesList</em> object can be extracted using simple accessor methods. Not surprisingly, the <em>GRangesList</em> objects have many of the same accessors as <em>GRanges</em> objects. The difference is that many of these methods return a list since the input is now essentially a list of <em>GRanges</em> objects. Here are a few examples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">seqnames</span>(grl)
<span class="co">#&gt; RleList of length 2</span>
<span class="co">#&gt; $txA</span>
<span class="co">#&gt; factor-Rle of length 1 with 1 run</span>
<span class="co">#&gt;   Lengths:    1</span>
<span class="co">#&gt;   Values : chr2</span>
<span class="co">#&gt; Levels(2): chr2 chr1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB</span>
<span class="co">#&gt; factor-Rle of length 2 with 1 run</span>
<span class="co">#&gt;   Lengths:    2</span>
<span class="co">#&gt;   Values : chr1</span>
<span class="co">#&gt; Levels(2): chr2 chr1</span>
<span class="kw">ranges</span>(grl)
<span class="co">#&gt; IRangesList of length 2</span>
<span class="co">#&gt; $txA</span>
<span class="co">#&gt; IRanges object with 1 range and 0 metadata columns:</span>
<span class="co">#&gt;           start       end     width</span>
<span class="co">#&gt;       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   [1]       103       106         4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB</span>
<span class="co">#&gt; IRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;           start       end     width</span>
<span class="co">#&gt;       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   [1]       107       109         3</span>
<span class="co">#&gt;   [2]       113       115         3</span>
<span class="kw">strand</span>(grl)
<span class="co">#&gt; RleList of length 2</span>
<span class="co">#&gt; $txA</span>
<span class="co">#&gt; factor-Rle of length 1 with 1 run</span>
<span class="co">#&gt;   Lengths: 1</span>
<span class="co">#&gt;   Values : +</span>
<span class="co">#&gt; Levels(3): + - *</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB</span>
<span class="co">#&gt; factor-Rle of length 2 with 2 runs</span>
<span class="co">#&gt;   Lengths: 1 1</span>
<span class="co">#&gt;   Values : + -</span>
<span class="co">#&gt; Levels(3): + - *</span></code></pre></div>
<p>The <code>length</code> and <code>names</code> methods will return the length or names of the list and the <code>seqlengths</code> method will return the set of sequence lengths.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(grl)
<span class="co">#&gt; [1] 2</span>
<span class="kw">names</span>(grl)
<span class="co">#&gt; [1] &quot;txA&quot; &quot;txB&quot;</span>
<span class="kw">seqlengths</span>(grl)
<span class="co">#&gt; chr2 chr1 </span>
<span class="co">#&gt;   NA   NA</span></code></pre></div>
<p>The <code>elementNROWS</code> method returns a list of integers corresponding to the result of calling <code>NROW</code> on each individual <em>GRanges</em> object contained by the <em>GRangesList</em>. This is a faster alternative to calling <code>lapply</code> on the <em>GRangesList</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">elementNROWS</span>(grl)
<span class="co">#&gt; txA txB </span>
<span class="co">#&gt;   1   2</span></code></pre></div>
<p><code>isEmpty</code> tests if a <em>GRangesList</em> object contains anything.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">isEmpty</span>(grl)
<span class="co">#&gt; [1] FALSE</span></code></pre></div>
<p>In the context of a <em>GRangesList</em> object, the <code>mcols</code> method performs a similar operation to what it does on a <em>GRanges</em> object. However, this metadata now refers to information at the list level instead of the level of the individual <em>GRanges</em> objects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">mcols</span>(grl) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Transcript A&quot;</span>,<span class="st">&quot;Transcript B&quot;</span>)
<span class="kw">mcols</span>(grl)
<span class="co">#&gt; DataFrame with 2 rows and 1 column</span>
<span class="co">#&gt;            value</span>
<span class="co">#&gt;      &lt;character&gt;</span>
<span class="co">#&gt; txA Transcript A</span>
<span class="co">#&gt; txB Transcript B</span></code></pre></div>
<p>Element-level metadata can be retrieved by unlisting the <code>GRangesList</code>, and extracting the metadata</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">mcols</span>(<span class="kw">unlist</span>(grl))
<span class="co">#&gt; DataFrame with 3 rows and 2 columns</span>
<span class="co">#&gt;         score        GC</span>
<span class="co">#&gt;     &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; txA         5      0.45</span>
<span class="co">#&gt; txB         3       0.3</span>
<span class="co">#&gt; txB         4       0.5</span></code></pre></div>
</div>
<div id="combining-grangeslist-objects" class="section level2">
<h2><span class="header-section-number">8.2</span> Combining <em>GRangesList</em> objects</h2>
<p><em>GRangesList</em> objects can be unlisted to combine the separate <em>GRanges</em> objects that they contain as an expanded <em>GRanges</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ul &lt;-<span class="st"> </span><span class="kw">unlist</span>(grl)
ul
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   txA     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   txB     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   txB     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Append lists using <code>append</code> or <code>c</code>.</p>
<p>A  had two <em>GRangesList</em> objects with parallel elements, and wanted to combined these element-wise into a single <em>GRangesList</em>. One solution is to use <code>pc()</code>  parallel (element-wise) <code>c()</code>. A more general solution is to concatenate the lists and then re-group by some factor, in this case the names of the elements.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">grl1 &lt;-<span class="st"> </span><span class="kw">GRangesList</span>(
    <span class="dt">gr1 =</span> <span class="kw">GRanges</span>(<span class="st">&quot;chr2&quot;</span>, <span class="kw">IRanges</span>(<span class="dv">3</span>, <span class="dv">6</span>)),
    <span class="dt">gr2 =</span> <span class="kw">GRanges</span>(<span class="st">&quot;chr1&quot;</span>, <span class="kw">IRanges</span>(<span class="kw">c</span>(<span class="dv">7</span>,<span class="dv">13</span>), <span class="dt">width =</span> <span class="dv">3</span>)))
grl2 &lt;-<span class="st"> </span><span class="kw">GRangesList</span>(
    <span class="dt">gr1 =</span> <span class="kw">GRanges</span>(<span class="st">&quot;chr2&quot;</span>, <span class="kw">IRanges</span>(<span class="dv">9</span>, <span class="dv">12</span>)),
    <span class="dt">gr2 =</span> <span class="kw">GRanges</span>(<span class="st">&quot;chr1&quot;</span>, <span class="kw">IRanges</span>(<span class="kw">c</span>(<span class="dv">25</span>,<span class="dv">38</span>), <span class="dt">width =</span> <span class="dv">3</span>)))

<span class="kw">pc</span>(grl1, grl2)
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $gr1 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr2       3-6      *</span>
<span class="co">#&gt;   [2]     chr2      9-12      *</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr2 </span>
<span class="co">#&gt; GRanges object with 4 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand</span>
<span class="co">#&gt;   [1]     chr1    7-9      *</span>
<span class="co">#&gt;   [2]     chr1  13-15      *</span>
<span class="co">#&gt;   [3]     chr1  25-27      *</span>
<span class="co">#&gt;   [4]     chr1  38-40      *</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>

grl3 &lt;-<span class="st"> </span><span class="kw">c</span>(grl1, grl2)
<span class="kw">regroup</span>(grl3, <span class="kw">names</span>(grl3))
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $gr1 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr2       3-6      *</span>
<span class="co">#&gt;   [2]     chr2      9-12      *</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr2 </span>
<span class="co">#&gt; GRanges object with 4 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand</span>
<span class="co">#&gt;   [1]     chr1    7-9      *</span>
<span class="co">#&gt;   [2]     chr1  13-15      *</span>
<span class="co">#&gt;   [3]     chr1  25-27      *</span>
<span class="co">#&gt;   [4]     chr1  38-40      *</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
</div>
<div id="basic-interval-operations-for-grangeslist-objects" class="section level2">
<h2><span class="header-section-number">8.3</span> Basic interval operations for <em>GRangesList</em> objects</h2>
<p>For interval operations, many of the same methods exist for <em>GRangesList</em> objects that exist for <em>GRanges</em> objects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">start</span>(grl)
<span class="co">#&gt; IntegerList of length 2</span>
<span class="co">#&gt; [[&quot;txA&quot;]] 103</span>
<span class="co">#&gt; [[&quot;txB&quot;]] 107 113</span>
<span class="kw">end</span>(grl)
<span class="co">#&gt; IntegerList of length 2</span>
<span class="co">#&gt; [[&quot;txA&quot;]] 106</span>
<span class="co">#&gt; [[&quot;txB&quot;]] 109 115</span>
<span class="kw">width</span>(grl)
<span class="co">#&gt; IntegerList of length 2</span>
<span class="co">#&gt; [[&quot;txA&quot;]] 4</span>
<span class="co">#&gt; [[&quot;txB&quot;]] 3 3</span></code></pre></div>
<p>These operations return a data structure representing, e.g., <em>IntegerList</em>, a list where all elements are integers; it can be convenient to use mathematical and other operations on **List* objects that work on each element, e.g.,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sum</span>(<span class="kw">width</span>(grl))  <span class="co"># sum of widths of each grl element</span>
<span class="co">#&gt; txA txB </span>
<span class="co">#&gt;   4   6</span></code></pre></div>
<p>Most of the intra-, inter- and between-range methods operate on <em>GRangesList</em> objects, e.g., to shift all the <em>GRanges</em> objects in a <em>GRangesList</em> object, or calculate the coverage. Both of these operations are also carried out across each <em>GRanges</em> list member.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">shift</span>(grl, <span class="dv">20</span>)
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   123-126      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames  ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1 127-129      + |     3 0.3</span>
<span class="co">#&gt;   [2]     chr1 133-135      - |     4 0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">coverage</span>(grl)
<span class="co">#&gt; RleList of length 2</span>
<span class="co">#&gt; $chr2</span>
<span class="co">#&gt; integer-Rle of length 106 with 2 runs</span>
<span class="co">#&gt;   Lengths: 102   4</span>
<span class="co">#&gt;   Values :   0   1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $chr1</span>
<span class="co">#&gt; integer-Rle of length 115 with 4 runs</span>
<span class="co">#&gt;   Lengths: 106   3   3   3</span>
<span class="co">#&gt;   Values :   0   1   0   1</span></code></pre></div>
</div>
<div id="subsetting-grangeslist-objects" class="section level2">
<h2><span class="header-section-number">8.4</span> Subsetting <em>GRangesList</em> objects</h2>
<p>A <em>GRangesList</em> object is behaves like a <code>list</code>: <code>[</code> returns a <em>GRangesList</em> containing a subset of the original object; <code>[[</code> or <code>\$</code> returns the <em>GRanges</em> object at that location in the list.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">grl[<span class="dv">1</span>]
grl[[<span class="dv">1</span>]]
grl[<span class="st">&quot;txA&quot;</span>]
grl<span class="op">$</span>txB</code></pre></div>
<p>In addition, subsetting a <em>GRangesList</em> also accepts a second parameter to specify which of the metadata columns you wish to select.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">grl[<span class="dv">1</span>, <span class="st">&quot;score&quot;</span>]
<span class="co">#&gt; GRangesList object of length 1:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 1 metadata column:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
grl[<span class="st">&quot;txB&quot;</span>, <span class="st">&quot;GC&quot;</span>]
<span class="co">#&gt; GRangesList object of length 1:</span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 1 metadata column:</span>
<span class="co">#&gt;       seqnames    ranges strand |        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1   107-109      + |       0.3</span>
<span class="co">#&gt;   [2]     chr1   113-115      - |       0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>The <code>head</code>, <code>tail</code>, <code>rep</code>, <code>rev</code>, and <code>window</code> methods all behave as you would expect them to for a list object. For example, the elements referred to by <code>window</code> are now list elements instead of <em>GRanges</em> elements.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rep</span>(grl[[<span class="dv">1</span>]], <span class="dt">times =</span> <span class="dv">3</span>)
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   [2]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   [3]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">rev</span>(grl)
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   [2]     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames  ranges strand | score   GC</span>
<span class="co">#&gt;   [1]     chr2 103-106      + |     5 0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">head</span>(grl, <span class="dt">n=</span><span class="dv">1</span>)
<span class="co">#&gt; GRangesList object of length 1:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">tail</span>(grl, <span class="dt">n=</span><span class="dv">1</span>)
<span class="co">#&gt; GRangesList object of length 1:</span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   [2]     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">window</span>(grl, <span class="dt">start=</span><span class="dv">1</span>, <span class="dt">end=</span><span class="dv">1</span>)
<span class="co">#&gt; GRangesList object of length 1:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
grl[<span class="kw">IRanges</span>(<span class="dt">start=</span><span class="dv">2</span>, <span class="dt">end=</span><span class="dv">2</span>)]
<span class="co">#&gt; GRangesList object of length 1:</span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   [2]     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
</div>
<div id="looping-over-grangeslist-objects" class="section level2">
<h2><span class="header-section-number">8.5</span> Looping over <em>GRangesList</em> objects</h2>
<p>For <em>GRangesList</em> objects there is also a family of <code>apply</code> methods. These include <code>lapply</code>, <code>sapply</code>, <code>mapply</code>, <code>endoapply</code>, <code>mendoapply</code>, <code>Map</code>, and <code>Reduce</code>.</p>
<p>The different looping methods defined for <em>GRangesList</em> objects are useful for returning different kinds of results. The standard <code>lapply</code> and <code>sapply</code> behave according to convention, with the <code>lapply</code> method returning a list and <code>sapply</code> returning a more simplified output.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">lapply</span>(grl, length)
<span class="co">#&gt; $txA</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB</span>
<span class="co">#&gt; [1] 2</span>
<span class="kw">sapply</span>(grl, length)
<span class="co">#&gt; txA txB </span>
<span class="co">#&gt;   1   2</span></code></pre></div>
<p>As with <em>IRanges</em> objects, there is also a multivariate version of <code>sapply</code>, called <code>mapply</code>, defined for <em>GRangesList</em> objects. And, if you dont want the results simplified, you can call the <code>Map</code> method, which does the same things as <code>mapply</code> but without simplifying the output.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">grl2 &lt;-<span class="st"> </span><span class="kw">shift</span>(grl, <span class="dv">10</span>)
<span class="kw">names</span>(grl2) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;shiftTxA&quot;</span>, <span class="st">&quot;shiftTxB&quot;</span>)

<span class="kw">mapply</span>(c, grl, grl2)
<span class="co">#&gt; $txA</span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   [2]     chr2   113-116      + |         5      0.45</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB</span>
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   [2]     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt;   [3]     chr1   117-119      + |         3       0.3</span>
<span class="co">#&gt;   [4]     chr1   123-125      - |         4       0.5</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">Map</span>(c, grl, grl2)
<span class="co">#&gt; $txA</span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   [2]     chr2   113-116      + |         5      0.45</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB</span>
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   [2]     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt;   [3]     chr1   117-119      + |         3       0.3</span>
<span class="co">#&gt;   [4]     chr1   123-125      - |         4       0.5</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Sometimes you will want to get back a modified version of the <em>GRangesList</em> that you originally passed in.</p>
<p>An endomorphism is a transformation of an object to another instance of the same class . This is achieved using the <code>endoapply</code> method, which will return the results as a <em>GRangesList</em> object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">endoapply</span>(grl, rev)
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames  ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1 113-115      - |     4 0.5</span>
<span class="co">#&gt;   [2]     chr1 107-109      + |     3 0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">mendoapply</span>(c, grl, grl2)
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   [2]     chr2   113-116      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames  ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1 107-109      + |     3 0.3</span>
<span class="co">#&gt;   [2]     chr1 113-115      - |     4 0.5</span>
<span class="co">#&gt;   [3]     chr1 117-119      + |     3 0.3</span>
<span class="co">#&gt;   [4]     chr1 123-125      - |     4 0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>The <code>Reduce</code> method will allow the <em>GRanges</em> objects to be collapsed across the whole of the <em>GRangesList</em> object. % Again, this seems like a sub-optimal example to me.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">Reduce</span>(c, grl)
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   [2]     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   [3]     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Explicit element-wise operations (<code>lapply()</code> and friends) on <em>GRangesList</em> objects with many elements can be slow. It is therefore beneficial to explore operations that work on <code>*List</code> objects directly (e.g., many of the <code>group generic' operators, see</code>?S4groupGeneric<code>, and the set and parallel set operators (e.g.,</code>union<code>,</code>punion<code>). A useful and fast strategy is to</code>unlist<code>the *GRangesList* to a *GRanges* object, operate on the *GRanges* object, then</code>relist` the result, e.g.,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gr &lt;-<span class="st"> </span><span class="kw">unlist</span>(grl)
gr<span class="op">$</span>log_score &lt;-<span class="st"> </span><span class="kw">log</span>(gr<span class="op">$</span>score)
grl &lt;-<span class="st"> </span><span class="kw">relist</span>(gr, grl)
grl
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 3 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC       log_score</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt;   txA     chr2   103-106      + |         5      0.45 1.6094379124341</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 3 metadata columns:</span>
<span class="co">#&gt;       seqnames  ranges strand | score  GC        log_score</span>
<span class="co">#&gt;   txB     chr1 107-109      + |     3 0.3 1.09861228866811</span>
<span class="co">#&gt;   txB     chr1 113-115      - |     4 0.5 1.38629436111989</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>See also <code>?extractList</code>.</p>
<p>For more information on the <code>GRangesList</code> classes be sure to consult the manual page and available methods</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">?GRangesList
<span class="kw">methods</span>(<span class="dt">class=</span><span class="st">&quot;GRangesList&quot;</span>)   <span class="co"># _partial_ list</span></code></pre></div>

<p>Interval overlapping is the process of comparing the ranges in two objects to determine if and when they overlap. As such, it is perhaps the most common operation performed on <em>GRanges</em> and <em>GRangesList</em> objects. To this end, the <em><a href="http://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></em> package provides a family of interval overlap functions. The most general of these functions is <code>findOverlaps</code>, which takes a query and a subject as inputs and returns a <em>Hits</em> object containing the index pairings for the overlapping elements.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mtch &lt;-<span class="st"> </span><span class="kw">findOverlaps</span>(gr, grl)
<span class="kw">as.matrix</span>(mtch)
<span class="co">#&gt;      queryHits subjectHits</span>
<span class="co">#&gt; [1,]         1           1</span>
<span class="co">#&gt; [2,]         2           2</span>
<span class="co">#&gt; [3,]         3           2</span></code></pre></div>
<p>As suggested in the sections discussing the nature of the <em>GRanges</em> and <em>GRangesList</em> classes, the index in the above matrix of hits for a <em>GRanges</em> object is a single range while for a <em>GRangesList</em> object it is the set of ranges that define a feature.</p>
<p>Another function in the overlaps family is <code>countOverlaps</code>, which tabulates the number of overlaps for each element in the query.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">countOverlaps</span>(gr, grl)
<span class="co">#&gt; txA txB txB </span>
<span class="co">#&gt;   1   1   1</span></code></pre></div>
<p>A third function in this family is <code>subsetByOverlaps</code>, which extracts the elements in the query that overlap at least one element in the subject.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">subsetByOverlaps</span>(gr,grl)
<span class="co">#&gt; GRanges object with 3 ranges and 3 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC        log_score</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;        &lt;numeric&gt;</span>
<span class="co">#&gt;   txA     chr2   103-106      + |         5      0.45  1.6094379124341</span>
<span class="co">#&gt;   txB     chr1   107-109      + |         3       0.3 1.09861228866811</span>
<span class="co">#&gt;   txB     chr1   113-115      - |         4       0.5 1.38629436111989</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Finally, you can use the <code>select</code> argument to get the index of the first overlapping element in the subject for each element in the query.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">findOverlaps</span>(gr, grl, <span class="dt">select=</span><span class="st">&quot;first&quot;</span>)
<span class="co">#&gt; [1] 1 2 2</span>
<span class="kw">findOverlaps</span>(grl, gr, <span class="dt">select=</span><span class="st">&quot;first&quot;</span>)
<span class="co">#&gt; [1] 1 2</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="granges-genomic-ranges.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="introduction-to-bioconductor-annotation-resources.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
