<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="fluent-genomic-data-analysis-with-plyranges.html">
<link rel="next" href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#for-everyone"><i class="fa fa-check"></i><b>1.1</b> For Everyone</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#for-workshop-authors"><i class="fa fa-check"></i><b>1.2</b> For Workshop Authors</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#description"><i class="fa fa-check"></i><b>1.2.1</b> DESCRIPTION</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#classifying-your-workshop"><i class="fa fa-check"></i><b>1.2.2</b> Classifying your workshop</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#bookdown"><i class="fa fa-check"></i><b>1.2.3</b> Bookdown</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#deadlines-for-bioc2018"><i class="fa fa-check"></i><b>1.3</b> Deadlines for Bioc2018</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html"><i class="fa fa-check"></i><b>2</b> <em>R</em> and <em>Bioconductor</em> for everyone: an introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#description-1"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="2.1.7" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-goals"><i class="fa fa-check"></i><b>2.1.7</b> Learning goals</a></li>
<li class="chapter" data-level="2.1.8" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-objectives"><i class="fa fa-check"></i><b>2.1.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-r"><i class="fa fa-check"></i><b>2.2</b> Introduction to <em>R</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#rstudio-orientation"><i class="fa fa-check"></i><b>2.2.1</b> <em>RStudio</em> orientation</a></li>
<li class="chapter" data-level="2.2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#very-basics"><i class="fa fa-check"></i><b>2.2.2</b> Very basics</a></li>
<li class="chapter" data-level="2.2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#data-input-and-manipulation"><i class="fa fa-check"></i><b>2.2.3</b> Data input and manipulation</a></li>
<li class="chapter" data-level="2.2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#writing-r-scripts"><i class="fa fa-check"></i><b>2.2.4</b> Writing <em>R</em> scripts</a></li>
<li class="chapter" data-level="2.2.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#using-packages"><i class="fa fa-check"></i><b>2.2.5</b> Using packages</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-bioconductor"><i class="fa fa-check"></i><b>2.3</b> Introduction to <em>Bioconductor</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#discovering-installing-and-learning-how-to-use-bioconductor-packages"><i class="fa fa-check"></i><b>2.3.1</b> Discovering, installing, and learning how to use <em>Bioconductor</em> packages</a></li>
<li class="chapter" data-level="2.3.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-genomic-ranges"><i class="fa fa-check"></i><b>2.3.2</b> Working with Genomic Ranges</a></li>
<li class="chapter" data-level="2.3.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-summarized-experimental-data"><i class="fa fa-check"></i><b>2.3.3</b> Working with summarized experimental data</a></li>
<li class="chapter" data-level="2.3.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#down-stream-analysis"><i class="fa fa-check"></i><b>2.3.4</b> Down-stream analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>3</b> RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview-1"><i class="fa fa-check"></i><b>3.1</b> Overview</a><ul>
<li class="chapter" data-level="3.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-2"><i class="fa fa-check"></i><b>3.1.1</b> Description</a></li>
<li class="chapter" data-level="3.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites-1"><i class="fa fa-check"></i><b>3.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="3.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation-1"><i class="fa fa-check"></i><b>3.1.3</b> Participation</a></li>
<li class="chapter" data-level="3.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="3.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline-1"><i class="fa fa-check"></i><b>3.1.5</b> Time outline</a></li>
<li class="chapter" data-level="3.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>3.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="3.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>3.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="3.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>3.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="3.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>3.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="3.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>3.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>3.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="3.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>3.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="3.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>3.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="3.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>3.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>3.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="3.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>3.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="3.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>3.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="3.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>3.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis"><i class="fa fa-check"></i><b>3.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="3.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>3.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="3.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>3.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub"><i class="fa fa-check"></i><b>3.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="3.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub"><i class="fa fa-check"></i><b>3.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="3.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>3.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>3.7</b> Building reports</a><ul>
<li class="chapter" data-level="3.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>3.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="3.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>3.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>3.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="3.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>3.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="3.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>3.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="3.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>3.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="3.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>3.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="3.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>3.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="3.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>3.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>4</b> Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-2"><i class="fa fa-check"></i><b>4.1</b> Overview</a><ul>
<li class="chapter" data-level="4.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-3"><i class="fa fa-check"></i><b>4.1.1</b> Description</a></li>
<li class="chapter" data-level="4.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-2"><i class="fa fa-check"></i><b>4.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-2"><i class="fa fa-check"></i><b>4.1.3</b> Participation</a></li>
<li class="chapter" data-level="4.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-2"><i class="fa fa-check"></i><b>4.1.5</b> Time outline</a></li>
<li class="chapter" data-level="4.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started"><i class="fa fa-check"></i><b>4.2</b> Getting started</a><ul>
<li class="chapter" data-level="4.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>4.2.1</b> The data</a></li>
<li class="chapter" data-level="4.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>4.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>4.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="4.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>4.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="4.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>4.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>4.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="4.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>4.5.1</b> Normalization</a></li>
<li class="chapter" data-level="4.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>4.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>4.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="4.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>4.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="4.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>4.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>5</b> Cytoscape Automation in R using Rcy3</a><ul>
<li class="chapter" data-level="5.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-3"><i class="fa fa-check"></i><b>5.1</b> Overview</a><ul>
<li class="chapter" data-level="5.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructors-names-and-contact-information"><i class="fa fa-check"></i><b>5.1.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="5.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description"><i class="fa fa-check"></i><b>5.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="5.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>5.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="5.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>5.1.4</b> Background:</a></li>
<li class="chapter" data-level="5.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation"><i class="fa fa-check"></i><b>5.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>5.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="5.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-3"><i class="fa fa-check"></i><b>5.1.7</b> Time outline</a></li>
<li class="chapter" data-level="5.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>5.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>5.2</b> Background</a><ul>
<li class="chapter" data-level="5.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>5.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="5.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>5.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="5.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>5.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>5.3</b> Translating biological data into Cytoscape using RCy3</a></li>
<li class="chapter" data-level="5.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>5.4</b> Set up</a><ul>
<li class="chapter" data-level="5.4.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started-1"><i class="fa fa-check"></i><b>5.4.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>5.5</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="5.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>5.6</b> Example Data Set</a></li>
<li class="chapter" data-level="5.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>5.7</b> Finding Network Data</a></li>
<li class="chapter" data-level="5.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>5.8</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="5.9" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>5.9</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="5.10" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>5.10</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
<li class="chapter" data-level="5.11" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#references"><i class="fa fa-check"></i><b>5.11</b> References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><i class="fa fa-check"></i><b>6</b> Effectively using the DelayedArray framework to support the analysis of large datasets</a><ul>
<li class="chapter" data-level="6.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-4"><i class="fa fa-check"></i><b>6.1</b> Overview</a><ul>
<li class="chapter" data-level="6.1.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#description-4"><i class="fa fa-check"></i><b>6.1.1</b> Description</a></li>
<li class="chapter" data-level="6.1.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.1.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#participation-3"><i class="fa fa-check"></i><b>6.1.3</b> Participation</a></li>
<li class="chapter" data-level="6.1.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>6.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.1.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#time-outline-4"><i class="fa fa-check"></i><b>6.1.5</b> Time outline</a></li>
<li class="chapter" data-level="6.1.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>6.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#introductory-material"><i class="fa fa-check"></i><b>6.2</b> Introductory material</a><ul>
<li class="chapter" data-level="6.2.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#discussion"><i class="fa fa-check"></i><b>6.2.1</b> Discussion</a></li>
<li class="chapter" data-level="6.2.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#why-learn-delayedarray"><i class="fa fa-check"></i><b>6.2.2</b> Why learn DelayedArray?</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-of-delayedarray-framework"><i class="fa fa-check"></i><b>6.3</b> Overview of DelayedArray framework</a><ul>
<li class="chapter" data-level="6.3.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-package"><i class="fa fa-check"></i><b>6.3.1</b> The <strong>DelayedArray</strong> package</a></li>
<li class="chapter" data-level="6.3.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#packages-that-extend-delayedarray"><i class="fa fa-check"></i><b>6.3.2</b> Packages that extend <strong>DelayedArray</strong></a></li>
<li class="chapter" data-level="6.3.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#packages-that-use-delayedarray"><i class="fa fa-check"></i><b>6.3.3</b> Packages that use <strong>DelayedArray</strong></a></li>
<li class="chapter" data-level="6.3.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-class"><i class="fa fa-check"></i><b>6.3.4</b> The <em>DelayedArray</em> class</a></li>
<li class="chapter" data-level="6.3.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fundamental-concepts"><i class="fa fa-check"></i><b>6.3.5</b> Fundamental concepts</a></li>
<li class="chapter" data-level="6.3.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#whats-out-there-already"><i class="fa fa-check"></i><b>6.3.6</b> Whatâ€™s out there already?</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#incoporating-delayedarray-into-a-package"><i class="fa fa-check"></i><b>6.4</b> Incoporating DelayedArray into a package</a><ul>
<li class="chapter" data-level="6.4.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#writing-algorithms-to-process-delayedarray-instances"><i class="fa fa-check"></i><b>6.4.1</b> Writing algorithms to process <em>DelayedArray</em> instances</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#questions-and-discussion"><i class="fa fa-check"></i><b>6.5</b> Questions and discussion</a></li>
<li class="chapter" data-level="6.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#todos"><i class="fa fa-check"></i><b>6.6</b> TODOs</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><i class="fa fa-check"></i><b>7</b> Variant Functional Annotation using StatePaintR, FunciVar &amp; MotifBreakR</a><ul>
<li class="chapter" data-level="7.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i><b>7.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="7.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-description-1"><i class="fa fa-check"></i><b>7.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="7.2.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.2.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-participation-1"><i class="fa fa-check"></i><b>7.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="7.2.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>7.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.2.4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#time-outline-5"><i class="fa fa-check"></i><b>7.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>7.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="7.3.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#key-concepts"><i class="fa fa-check"></i><b>7.3.1</b> Key Concepts</a></li>
<li class="chapter" data-level="7.3.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#learning-objectives-4"><i class="fa fa-check"></i><b>7.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>8</b> Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="8.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructors-names-and-contact-information-2"><i class="fa fa-check"></i><b>8.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="8.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-2"><i class="fa fa-check"></i><b>8.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="8.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-2"><i class="fa fa-check"></i><b>8.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="8.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>8.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-6"><i class="fa fa-check"></i><b>8.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>8.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="8.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop"><i class="fa fa-check"></i><b>8.4</b> Workshop</a></li>
<li class="chapter" data-level="8.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>8.5</b> Where does it all come from?</a></li>
<li class="chapter" data-level="8.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>8.6</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="8.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>8.7</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="8.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>8.8</b> Data types</a></li>
<li class="chapter" data-level="8.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>8.9</b> Differential expression analysis</a></li>
<li class="chapter" data-level="8.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>8.10</b> Gene sets</a></li>
<li class="chapter" data-level="8.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>8.11</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="8.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>8.12</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="8.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>8.13</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="8.14" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>8.14</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>9</b> RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="9.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>9.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-3"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-6"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-3"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-7"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>9.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="9.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals-4"><i class="fa fa-check"></i><b>9.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="9.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-5"><i class="fa fa-check"></i><b>9.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-1"><i class="fa fa-check"></i><b>9.4</b> Introduction</a></li>
<li class="chapter" data-level="9.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>9.5</b> Data packaging</a><ul>
<li class="chapter" data-level="9.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>9.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="9.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>9.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="9.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>9.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>9.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="9.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>9.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="9.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>9.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="9.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>9.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="9.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>9.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>9.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="9.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>9.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="9.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>9.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="9.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>9.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="9.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>9.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="9.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>9.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="9.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>9.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>10</b> Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="10.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructors-names-and-contact-information-3"><i class="fa fa-check"></i><b>10.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-4"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-7"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-4"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>10.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-8"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-5"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-6"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>10.4</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="10.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>10.5</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>10.6</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>10.7</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>10.8</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#grangeslist-groups-of-genomic-ranges"><i class="fa fa-check"></i><b>10.9</b> <em>GRangesList</em>: Groups of Genomic Ranges</a></li>
<li class="chapter" data-level="10.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-grangeslist-accessors"><i class="fa fa-check"></i><b>10.10</b> Basic <em>GRangesList</em> accessors</a></li>
<li class="chapter" data-level="10.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#combining-grangeslist-objects"><i class="fa fa-check"></i><b>10.11</b> Combining <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-grangeslist-objects"><i class="fa fa-check"></i><b>10.12</b> Basic interval operations for <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-grangeslist-objects"><i class="fa fa-check"></i><b>10.13</b> Subsetting <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#looping-over-grangeslist-objects"><i class="fa fa-check"></i><b>10.14</b> Looping over <em>GRangesList</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>11</b> Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="11.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructors-names-and-contact-information-4"><i class="fa fa-check"></i><b>11.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="11.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-5"><i class="fa fa-check"></i><b>11.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="11.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-8"><i class="fa fa-check"></i><b>11.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="11.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-5"><i class="fa fa-check"></i><b>11.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-9"><i class="fa fa-check"></i><b>11.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="11.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-9"><i class="fa fa-check"></i><b>11.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="11.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-6"><i class="fa fa-check"></i><b>11.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="11.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-7"><i class="fa fa-check"></i><b>11.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-1"><i class="fa fa-check"></i><b>11.4</b> Workshop</a></li>
<li class="chapter" data-level="11.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-2"><i class="fa fa-check"></i><b>11.5</b> Introduction</a><ul>
<li class="chapter" data-level="11.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>11.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="11.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>11.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="11.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>11.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup"><i class="fa fa-check"></i><b>11.6</b> Setup</a></li>
<li class="chapter" data-level="11.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>11.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="11.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>11.8</b> The Grammar</a><ul>
<li class="chapter" data-level="11.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>11.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="11.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>11.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="11.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises"><i class="fa fa-check"></i><b>11.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>11.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="11.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>11.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="11.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>11.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="11.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>11.10</b> Whatâ€™s next?</a></li>
<li class="chapter" data-level="11.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>11.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>12</b> Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="12.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>12.1</b> Instructors</a></li>
<li class="chapter" data-level="12.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description-6"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>12.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation-6"><i class="fa fa-check"></i><b>12.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-10"><i class="fa fa-check"></i><b>12.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-7"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-8"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>12.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="12.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>12.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="12.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>12.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="12.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>12.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="12.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>12.4.4</b> Data containers</a></li>
<li class="chapter" data-level="12.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>12.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="12.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>12.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="12.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>12.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="12.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>12.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="12.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>12.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="12.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>12.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="12.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>12.4.11</b> Simple example</a></li>
<li class="chapter" data-level="12.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>12.4.12</b> Questions!</a></li>
<li class="chapter" data-level="12.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>12.4.13</b> More questions!</a></li>
<li class="chapter" data-level="12.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>12.4.14</b> Another example</a></li>
<li class="chapter" data-level="12.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>12.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="12.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>12.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="12.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>12.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="12.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>12.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="12.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>12.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="12.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>12.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="12.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>12.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="12.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>12.4.22</b> GRanges</a></li>
<li class="chapter" data-level="12.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>12.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="12.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>12.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="12.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>12.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="12.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>12.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="12.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>12.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="12.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>12.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="12.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>12.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="12.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>12.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="12.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>12.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="12.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>12.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="12.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr"><i class="fa fa-check"></i><b>12.4.33</b> Organism.dplyr</a></li>
<li class="chapter" data-level="12.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-1"><i class="fa fa-check"></i><b>12.4.34</b> Organism.dplyr</a></li>
<li class="chapter" data-level="12.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-2"><i class="fa fa-check"></i><b>12.4.35</b> Organism.dplyr</a></li>
<li class="chapter" data-level="12.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>12.4.36</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="12.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>12.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="12.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>12.4.38</b> BSgenome packages</a></li>
<li class="chapter" data-level="12.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>12.4.39</b> BSgenome packages</a></li>
<li class="chapter" data-level="12.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>12.4.40</b> BSgenome exercises</a></li>
<li class="chapter" data-level="12.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-1"><i class="fa fa-check"></i><b>12.4.41</b> AnnotationHub</a></li>
<li class="chapter" data-level="12.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>12.4.42</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="12.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>12.4.43</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="12.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>12.4.44</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="12.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>12.4.45</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="12.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>12.4.46</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="12.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>12.4.47</b> AnnotationHub query</a></li>
<li class="chapter" data-level="12.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>12.4.48</b> AnnotationHub query</a></li>
<li class="chapter" data-level="12.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>12.4.49</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="12.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>12.4.50</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="12.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>12.4.51</b> biomaRt</a></li>
<li class="chapter" data-level="12.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>12.4.52</b> biomaRt data sets</a></li>
<li class="chapter" data-level="12.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>12.4.53</b> biomaRt queries</a></li>
<li class="chapter" data-level="12.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>12.4.54</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="12.4.55" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>12.4.55</b> biomaRt query</a></li>
<li class="chapter" data-level="12.4.56" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>12.4.56</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>13</b> Workflow for Multi-omics Analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>13.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-7"><i class="fa fa-check"></i><b>13.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="13.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-9"><i class="fa fa-check"></i><b>13.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-7"><i class="fa fa-check"></i><b>13.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>13.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="13.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-10"><i class="fa fa-check"></i><b>13.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="13.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-8"><i class="fa fa-check"></i><b>13.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="13.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-9"><i class="fa fa-check"></i><b>13.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>13.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="13.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>13.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="13.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>13.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="13.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>13.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>13.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="13.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>13.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="13.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>13.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="13.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>13.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>13.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="13.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>13.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="13.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>13.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="13.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>13.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="13.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>13.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="13.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>13.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="13.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>13.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>13.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="13.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>13.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="13.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>13.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="13.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>13.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>13.9</b> Complete cases</a></li>
<li class="chapter" data-level="13.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>13.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="13.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>13.11</b> Extraction</a><ul>
<li class="chapter" data-level="13.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>13.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="13.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>13.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="13.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>13.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="13.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>13.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="13.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>13.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="13.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>13.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="13.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>13.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="13.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>13.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="13.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>13.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="13.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>13.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="13.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>13.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="13.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>13.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="13.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>13.16.1</b> â€œSimplificationâ€ of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="13.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>13.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="13.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>13.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="13.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>13.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="13.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>13.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="13.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>13.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="13.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>13.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="13.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>13.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="13.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>13.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="13.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>13.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="13.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>13.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="13.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>13.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>14</b> Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="14.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>14.1</b> Instructors:</a></li>
<li class="chapter" data-level="14.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-8"><i class="fa fa-check"></i><b>14.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-10"><i class="fa fa-check"></i><b>14.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-8"><i class="fa fa-check"></i><b>14.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="14.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-11"><i class="fa fa-check"></i><b>14.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="14.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-11"><i class="fa fa-check"></i><b>14.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-11"><i class="fa fa-check"></i><b>14.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="14.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-9"><i class="fa fa-check"></i><b>14.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="14.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-10"><i class="fa fa-check"></i><b>14.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>15</b> Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="15.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>15.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="15.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-9"><i class="fa fa-check"></i><b>15.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="15.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-11"><i class="fa fa-check"></i><b>15.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="15.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-4"><i class="fa fa-check"></i><b>15.2.2</b> Participation</a></li>
<li class="chapter" data-level="15.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>15.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="15.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-12"><i class="fa fa-check"></i><b>15.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-4"><i class="fa fa-check"></i><b>15.3</b> Introduction</a></li>
<li class="chapter" data-level="15.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>15.4</b> Why maintain your package</a></li>
<li class="chapter" data-level="15.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>15.5</b> Infrastructure used for maintenance</a></li>
<li class="chapter" data-level="15.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>15.6</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="15.6.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>15.6.1</b> Build report</a></li>
<li class="chapter" data-level="15.6.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>15.6.2</b> Landing page badges</a></li>
<li class="chapter" data-level="15.6.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>15.6.3</b> Download statistics</a></li>
<li class="chapter" data-level="15.6.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>15.6.4</b> RSS feeds</a></li>
<li class="chapter" data-level="15.6.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>15.6.5</b> Support site, Bioc-devel mailing list</a></li>
<li class="chapter" data-level="15.6.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github"><i class="fa fa-check"></i><b>15.6.6</b> GitHub</a></li>
</ul></li>
<li class="chapter" data-level="15.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>15.7</b> Round up of resources available</a></li>
<li class="chapter" data-level="15.8" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#acknowledgements"><i class="fa fa-check"></i><b>15.8</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>16</b> Public Data Resources and Bioconductor</a><ul>
<li class="chapter" data-level="16.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>16.1</b> Instructor name(s) and contact information</a></li>
<li class="chapter" data-level="16.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>16.2</b> Syllabus</a><ul>
<li class="chapter" data-level="16.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-10"><i class="fa fa-check"></i><b>16.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="16.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-12"><i class="fa fa-check"></i><b>16.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="16.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-9"><i class="fa fa-check"></i><b>16.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="16.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>16.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="16.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-12"><i class="fa fa-check"></i><b>16.2.5</b> Time outline</a></li>
<li class="chapter" data-level="16.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-13"><i class="fa fa-check"></i><b>16.2.6</b> workshop goals and objectives</a></li>
<li class="chapter" data-level="16.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-10"><i class="fa fa-check"></i><b>16.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="16.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-11"><i class="fa fa-check"></i><b>16.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-5"><i class="fa fa-check"></i><b>16.3</b> Overview</a></li>
<li class="chapter" data-level="16.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>16.4</b> GEOquery</a><ul>
<li class="chapter" data-level="16.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>16.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="16.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>16.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="16.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>16.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="16.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>16.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>16.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="16.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>16.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="16.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>16.5.2</b> Basic design</a></li>
<li class="chapter" data-level="16.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>16.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="16.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>16.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="16.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>16.6.1</b> Authentication</a></li>
<li class="chapter" data-level="16.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>16.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>16.7</b> Sequence Read Archive</a></li>
<li class="chapter" data-level="16.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>16.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="16.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>16.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="16.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>16.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>16.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="16.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>16.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="16.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>16.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="16.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>16.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="16.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>16.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="16.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>16.12</b> Pharmacogenomics</a></li>
<li class="chapter" data-level="16.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>16.13</b> Bibliography</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="introduction-to-bioconductor-annotation-resources" class="section level1">
<h1><span class="header-section-number">12</span> Introduction to Bioconductor annotation resources</h1>
<div id="instructors" class="section level2">
<h2><span class="header-section-number">12.1</span> Instructors</h2>
<ul>
<li>James W. MacDonald (<a href="mailto:jmacdon@uw.edu">jmacdon@uw.edu</a>)</li>
<li>Lori Shepherd (<a href="mailto:lori.shepherd@roswellpark.org">lori.shepherd@roswellpark.org</a>)</li>
</ul>
</div>
<div id="workshop-description-6" class="section level2">
<h2><span class="header-section-number">12.2</span> Workshop Description</h2>
<p>There are various annotation packages provided by the Bioconductor project that can be used to incorporate additional information to results from high-throughput experiments. This can be as simple as mapping Ensembl IDs to corresponding HUGO gene symbols, to much more complex queries involving multiple data sources. In this workshop we will cover the various classes of annotation packages, what they contain, and how to use them efficiently.</p>
<div id="prerequisites" class="section level3">
<h3><span class="header-section-number">12.2.1</span> Prerequisites</h3>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Basic understanding of the various annotation sources (NCBI, EBI/EMBL)</li>
</ul>
<p>Useful background reading</p>
<ul>
<li>The <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/AnnotationDbi/inst/doc/IntroToAnnotationPackages.pdf">AnnotationDbi</a> vignette.</li>
<li>The <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/biomaRt/inst/doc/biomaRt.html">biomaRt</a> vignette.</li>
<li>The <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/GenomicFeatures/inst/doc/GenomicFeatures.pdf">GenomicFeatures</a> vignette.</li>
</ul>
</div>
<div id="workshop-participation-6" class="section level3">
<h3><span class="header-section-number">12.2.2</span> Workshop Participation</h3>
<p>After each type of annotation package is introduced, students will be given the opportunity to practice making their own queries.</p>
</div>
<div id="r-bioconductor-packages-used-10" class="section level3">
<h3><span class="header-section-number">12.2.3</span> <em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li>AnnotationDbi</li>
<li>AnnotationHub</li>
<li>BSgenome</li>
<li>biomaRt</li>
<li>ensembldb</li>
<li>org.Hs.eg.db</li>
<li>TxDb.Hsapiens.UCSC.hg19.knownGene</li>
<li>EnsDb.Hsapiens.v79</li>
<li>EnsDb.Mmusculus.v79</li>
<li>Homo.sapiens</li>
<li>BSgenome.Hsapiens.UCSC.hg19</li>
<li>hugene20sttranscriptcluster.db</li>
</ul>
</div>
</div>
<div id="workshop-goals-and-objectives-9" class="section level2">
<h2><span class="header-section-number">12.3</span> Workshop goals and objectives</h2>
<p>Annotating data is a complex task. For any high-throughput experiment the analyst usually starts with a set of identifiers for each thing that was measured, and in order to make the results useful to collaborators these identifiers need to be mapped to other identifiers that are either more familiar to collaborators, or that can be used for further analyses. As an example, RNA-Seq data may only have Entrez Gene IDs for each gene measured, and as part of the output you may want to include the gene symbols, which are more likely to be familiar to a Biologist.</p>
<div id="learning-goals-7" class="section level3">
<h3><span class="header-section-number">12.3.1</span> Learning goals</h3>
<ul>
<li>Understand what sort of annotation data are available</li>
<li>Understand the difference between annotation sources (NCBI and EBI/EMBL)</li>
<li>Gain familiarity with the various ways to query annotation packages</li>
<li>Get some practice making queries</li>
</ul>
</div>
<div id="learning-objectives-8" class="section level3">
<h3><span class="header-section-number">12.3.2</span> Learning objectives</h3>
<ul>
<li>Be able to use select and mapIds to map between identifiers</li>
<li>Be able to extract data from TxDb and EnsDb packages</li>
<li>Be able to make queries using biomaRt</li>
<li>Extract and utilize various data from AnnotationHub</li>
</ul>
</div>
</div>
<div id="annotation-workshop" class="section level2">
<h2><span class="header-section-number">12.4</span> Annotation Workshop</h2>
<div id="goals-for-this-workshop" class="section level3">
<h3><span class="header-section-number">12.4.1</span> Goals for this workshop</h3>
<ul>
<li><p>Learn about various annotation package types</p></li>
<li><p>Learn the basics of querying these resources</p></li>
<li><p>Discuss annotations in regard to Bioc data structures</p></li>
<li><p>Get in some practice</p></li>
</ul>
</div>
<div id="what-do-we-mean-by-annotation" class="section level3">
<h3><span class="header-section-number">12.4.2</span> What do we mean by annotation?</h3>
<p>Map a known ID to other functional or positional information</p>
<p><img src="MacDonald_Annotation/mapping.png" width="640" /></p>
</div>
<div id="specific-goal" class="section level3">
<h3><span class="header-section-number">12.4.3</span> Specific goal</h3>
<p>We have data and statistics, and we want to add other useful information</p>
<p><img src="MacDonald_Annotation/boxes.png" width="75%" /></p>
<p>The end result might be as simple as a data.frame or HTML table, or as complex as a <code>RangedSummarizedExperiment</code></p>
</div>
<div id="data-containers" class="section level3">
<h3><span class="header-section-number">12.4.4</span> Data containers</h3>
<p><img src="MacDonald_Annotation/eset.png" width="75%" /></p>
</div>
<div id="expressionset" class="section level3">
<h3><span class="header-section-number">12.4.5</span> ExpressionSet</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata/eset.Rdata&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;Bioc2018Anno&quot;</span>))
eset
<span class="co">#&gt; ExpressionSet (storageMode: lockedEnvironment)</span>
<span class="co">#&gt; assayData: 33552 features, 6 samples </span>
<span class="co">#&gt;   element names: exprs </span>
<span class="co">#&gt; protocolData: none</span>
<span class="co">#&gt; phenoData</span>
<span class="co">#&gt;   sampleNames: GSM2194079 GSM2194080 ... GSM2194084 (6 total)</span>
<span class="co">#&gt;   varLabels: title characteristics_ch1.1</span>
<span class="co">#&gt;   varMetadata: labelDescription</span>
<span class="co">#&gt; featureData</span>
<span class="co">#&gt;   featureNames: 16657436 16657440 ... 17118478 (33552 total)</span>
<span class="co">#&gt;   fvarLabels: PROBEID ENTREZID SYMBOL GENENAME</span>
<span class="co">#&gt;   fvarMetadata: labelDescription</span>
<span class="co">#&gt; experimentData: use &#39;experimentData(object)&#39;</span>
<span class="co">#&gt; Annotation: pd.hugene.2.0.st</span></code></pre></div>
</div>
<div id="expressionset-continued" class="section level3">
<h3><span class="header-section-number">12.4.6</span> ExpressionSet (continued)</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">head</span>(<span class="kw">exprs</span>(eset))
<span class="co">#&gt;          GSM2194079 GSM2194080 GSM2194081 GSM2194082 GSM2194083 GSM2194084</span>
<span class="co">#&gt; 16657436   8.505158   9.046577   8.382674   9.115481   8.715343   8.566301</span>
<span class="co">#&gt; 16657440   7.948860   8.191222   7.901911   8.459781   8.191793   8.219658</span>
<span class="co">#&gt; 16657450  10.932934  11.228553  10.948120  11.462231  11.300046  11.300886</span>
<span class="co">#&gt; 16657469   9.172462   9.344630   9.193450   9.465584   9.464020   9.135715</span>
<span class="co">#&gt; 16657473   6.222049   6.551035   6.000246   6.398798   5.892654   5.592125</span>
<span class="co">#&gt; 16657476   8.514300   8.474073   8.407196   8.811238   8.780833   8.874606</span>
<span class="kw">head</span>(<span class="kw">pData</span>(<span class="kw">phenoData</span>(eset)))
<span class="co">#&gt;                          title characteristics_ch1.1</span>
<span class="co">#&gt; GSM2194079   SW620-miR625-rep1     shRNA: miR-625-3p</span>
<span class="co">#&gt; GSM2194080   SW620-miR625-rep2     shRNA: miR-625-3p</span>
<span class="co">#&gt; GSM2194081   SW620-miR625-rep3     shRNA: miR-625-3p</span>
<span class="co">#&gt; GSM2194082 SW620-scramble-rep1       shRNA: scramble</span>
<span class="co">#&gt; GSM2194083 SW620-scramble-rep2       shRNA: scramble</span>
<span class="co">#&gt; GSM2194084 SW620-scramble-rep3       shRNA: scramble</span></code></pre></div>
</div>
<div id="expressionset-continued-1" class="section level3">
<h3><span class="header-section-number">12.4.7</span> ExpressionSet (continued)</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">head</span>(<span class="kw">pData</span>(<span class="kw">featureData</span>(eset)))
<span class="co">#&gt;           PROBEID  ENTREZID      SYMBOL</span>
<span class="co">#&gt; 16657436 16657436     84771     DDX11L2</span>
<span class="co">#&gt; 16657440 16657440 100302278   MIR1302-2</span>
<span class="co">#&gt; 16657450 16657450    402483   LINC01000</span>
<span class="co">#&gt; 16657469 16657469    140849 LINC00266-1</span>
<span class="co">#&gt; 16657473 16657473    729759      OR4F29</span>
<span class="co">#&gt; 16657476 16657476    388574   RPL23AP87</span>
<span class="co">#&gt;                                                   GENENAME</span>
<span class="co">#&gt; 16657436                     DEAD/H-box helicase 11 like 2</span>
<span class="co">#&gt; 16657440                                   microRNA 1302-2</span>
<span class="co">#&gt; 16657450       long intergenic non-protein coding RNA 1000</span>
<span class="co">#&gt; 16657469      long intergenic non-protein coding RNA 266-1</span>
<span class="co">#&gt; 16657473 olfactory receptor family 4 subfamily F member 29</span>
<span class="co">#&gt; 16657476              ribosomal protein L23a pseudogene 87</span></code></pre></div>
</div>
<div id="bioc-containers-vs-basic-structures" class="section level3">
<h3><span class="header-section-number">12.4.8</span> BioC containers vs basic structures</h3>
<div id="pros" class="section level4">
<h4><span class="header-section-number">12.4.8.1</span> Pros</h4>
<ul>
<li><p>Validity checking</p></li>
<li><p>Subsetting</p></li>
<li><p>Function dispatch</p></li>
<li><p>Automatic behaviors</p></li>
</ul>
</div>
<div id="cons" class="section level4">
<h4><span class="header-section-number">12.4.8.2</span> Cons</h4>
<ul>
<li><p>Difficult to create</p></li>
<li><p>Cumbersome to extract data by hand</p></li>
<li><p>Useful only within R</p></li>
</ul>
</div>
</div>
<div id="annotation-sources" class="section level3">
<h3><span class="header-section-number">12.4.9</span> Annotation sources</h3>
<table>
<thead>
<tr class="header">
<th align="left">Package type</th>
<th align="left">Example</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ChipDb</td>
<td align="left">hugene20sttranscriptcluster.db</td>
</tr>
<tr class="even">
<td align="left">OrgDb</td>
<td align="left">org.Hs.eg.db</td>
</tr>
<tr class="odd">
<td align="left">TxDb/EnsDb</td>
<td align="left">TxDb.Hsapiens.UCSC.hg19.knownGene; EnsDb.Hsapiens.v75</td>
</tr>
<tr class="even">
<td align="left">OrganismDb</td>
<td align="left">Homo.sapiens</td>
</tr>
<tr class="odd">
<td align="left">BSgenome</td>
<td align="left">BSgenome.Hsapiens.UCSC.hg19</td>
</tr>
<tr class="even">
<td align="left">Others</td>
<td align="left">GO.db; KEGG.db</td>
</tr>
<tr class="odd">
<td align="left">AnnotationHub</td>
<td align="left">Online resource</td>
</tr>
<tr class="even">
<td align="left">biomaRt</td>
<td align="left">Online resource</td>
</tr>
</tbody>
</table>
</div>
<div id="interacting-with-annodb-packages" class="section level3">
<h3><span class="header-section-number">12.4.10</span> Interacting with AnnoDb packages</h3>
<p>The main function is <code>select</code>:</p>
<p>select(<em>annopkg</em>, <em>keys</em>, <em>columns</em>, <em>keytype</em>)</p>
<p>Where</p>
<ul>
<li><p>annopkg is the annotation package</p></li>
<li><p>keys are the IDs that we <strong>know</strong></p></li>
<li><p>columns are the values we <strong>want</strong></p></li>
<li>keytype is the type of key used
<ul>
<li>if the keytype is the <strong>central</strong> key, it can remain unspecified</li>
</ul></li>
</ul>
</div>
<div id="simple-example" class="section level3">
<h3><span class="header-section-number">12.4.11</span> Simple example</h3>
<p>Say we have analyzed data from an Affymetrix Human Gene ST 2.0 array and want to know what the genes are. For purposes of this lab, we just select some IDs at random.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">library</span>(hugene20sttranscriptcluster.db)
<span class="kw">set.seed</span>(<span class="dv">12345</span>)
ids &lt;-<span class="st"> </span><span class="kw">featureNames</span>(eset)[<span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">25000</span>, <span class="dv">5</span>)]
ids
<span class="co">#&gt; [1] &quot;16908472&quot; &quot;16962185&quot; &quot;16920686&quot; &quot;16965513&quot; &quot;16819952&quot;</span>
<span class="kw">select</span>(hugene20sttranscriptcluster.db, ids, <span class="st">&quot;SYMBOL&quot;</span>)
<span class="co">#&gt; &#39;select()&#39; returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt;    PROBEID    SYMBOL</span>
<span class="co">#&gt; 1 16908472 LINC01494</span>
<span class="co">#&gt; 2 16962185      ALG3</span>
<span class="co">#&gt; 3 16920686      &lt;NA&gt;</span>
<span class="co">#&gt; 4 16965513      &lt;NA&gt;</span>
<span class="co">#&gt; 5 16819952      CBFB</span></code></pre></div>
</div>
<div id="questions" class="section level3">
<h3><span class="header-section-number">12.4.12</span> Questions!</h3>
<p>How do you know what the central keys are?</p>
<ul>
<li><p>If itâ€™s a ChipDb, the central key are the manufacturerâ€™s probe IDs</p></li>
<li><p>Itâ€™s sometimes in the name - org.Hs.eg.db, where â€˜egâ€™ means Entrez Gene ID</p></li>
<li><p>You can see examples using e.g., head(keys(<em>annopkg</em>)), and infer from that</p></li>
<li><p>But note that itâ€™s never necessary to know the central key, as long as you specify the keytype</p></li>
</ul>
</div>
<div id="more-questions" class="section level3">
<h3><span class="header-section-number">12.4.13</span> More questions!</h3>
<p>What keytypes or columns are available for a given annotation package?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">keytypes</span>(hugene20sttranscriptcluster.db)
<span class="co">#&gt;  [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot; </span>
<span class="co">#&gt;  [5] &quot;ENSEMBLTRANS&quot; &quot;ENTREZID&quot;     &quot;ENZYME&quot;       &quot;EVIDENCE&quot;    </span>
<span class="co">#&gt;  [9] &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;     &quot;GO&quot;           &quot;GOALL&quot;       </span>
<span class="co">#&gt; [13] &quot;IPI&quot;          &quot;MAP&quot;          &quot;OMIM&quot;         &quot;ONTOLOGY&quot;    </span>
<span class="co">#&gt; [17] &quot;ONTOLOGYALL&quot;  &quot;PATH&quot;         &quot;PFAM&quot;         &quot;PMID&quot;        </span>
<span class="co">#&gt; [21] &quot;PROBEID&quot;      &quot;PROSITE&quot;      &quot;REFSEQ&quot;       &quot;SYMBOL&quot;      </span>
<span class="co">#&gt; [25] &quot;UCSCKG&quot;       &quot;UNIGENE&quot;      &quot;UNIPROT&quot;</span>
<span class="kw">columns</span>(hugene20sttranscriptcluster.db)
<span class="co">#&gt;  [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot; </span>
<span class="co">#&gt;  [5] &quot;ENSEMBLTRANS&quot; &quot;ENTREZID&quot;     &quot;ENZYME&quot;       &quot;EVIDENCE&quot;    </span>
<span class="co">#&gt;  [9] &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;     &quot;GO&quot;           &quot;GOALL&quot;       </span>
<span class="co">#&gt; [13] &quot;IPI&quot;          &quot;MAP&quot;          &quot;OMIM&quot;         &quot;ONTOLOGY&quot;    </span>
<span class="co">#&gt; [17] &quot;ONTOLOGYALL&quot;  &quot;PATH&quot;         &quot;PFAM&quot;         &quot;PMID&quot;        </span>
<span class="co">#&gt; [21] &quot;PROBEID&quot;      &quot;PROSITE&quot;      &quot;REFSEQ&quot;       &quot;SYMBOL&quot;      </span>
<span class="co">#&gt; [25] &quot;UCSCKG&quot;       &quot;UNIGENE&quot;      &quot;UNIPROT&quot;</span></code></pre></div>
</div>
<div id="another-example" class="section level3">
<h3><span class="header-section-number">12.4.14</span> Another example</h3>
<p>There is one issue with <code>select</code> however.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
ids &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&#39;16737401&#39;</span>,<span class="st">&#39;16657436&#39;</span> ,<span class="st">&#39;16678303&#39;</span>)
<span class="kw">select</span>(hugene20sttranscriptcluster.db, ids, <span class="kw">c</span>(<span class="st">&quot;SYMBOL&quot;</span>,<span class="st">&quot;MAP&quot;</span>))
<span class="co">#&gt; &#39;select()&#39; returned 1:many mapping between keys and columns</span>
<span class="co">#&gt;     PROBEID       SYMBOL     MAP</span>
<span class="co">#&gt; 1  16737401        TRAF6   11p12</span>
<span class="co">#&gt; 2  16657436      DDX11L1 1p36.33</span>
<span class="co">#&gt; 3  16657436 LOC102725121 1p36.33</span>
<span class="co">#&gt; 4  16657436      DDX11L2  2q14.1</span>
<span class="co">#&gt; 5  16657436      DDX11L9 15q26.3</span>
<span class="co">#&gt; 6  16657436     DDX11L10 16p13.3</span>
<span class="co">#&gt; 7  16657436      DDX11L5  9p24.3</span>
<span class="co">#&gt; 8  16657436     DDX11L16    Xq28</span>
<span class="co">#&gt; 9  16657436     DDX11L16    Yq12</span>
<span class="co">#&gt; 10 16678303         ARF1 1q42.13</span></code></pre></div>
</div>
<div id="the-mapids-function" class="section level3">
<h3><span class="header-section-number">12.4.15</span> The <code>mapIds</code> function</h3>
<p>An alternative to <code>select</code> is <code>mapIds</code>, which gives control of duplicates</p>
<ul>
<li><p>Same arguments as <code>select</code> with slight differences</p>
<ul>
<li><p>The columns argument can only specify one column</p></li>
<li><p>The keytype argument <strong>must</strong> be specified</p></li>
<li><p>An additional argument, multiVals used to control duplicates</p></li>
</ul></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">mapIds</span>(hugene20sttranscriptcluster.db, ids, <span class="st">&quot;SYMBOL&quot;</span>, <span class="st">&quot;PROBEID&quot;</span>)
<span class="co">#&gt; &#39;select()&#39; returned 1:many mapping between keys and columns</span>
<span class="co">#&gt;  16737401  16657436  16678303 </span>
<span class="co">#&gt;   &quot;TRAF6&quot; &quot;DDX11L1&quot;    &quot;ARF1&quot;</span></code></pre></div>
</div>
<div id="choices-for-multivals" class="section level3">
<h3><span class="header-section-number">12.4.16</span> Choices for multiVals</h3>
<p>Default is <code>first</code>, where we just choose the first of the duplicates. Other choices are <code>list</code>, <code>CharacterList</code>, <code>filter</code>, <code>asNA</code> or a user-specified function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">mapIds</span>(hugene20sttranscriptcluster.db, ids, <span class="st">&quot;SYMBOL&quot;</span>, <span class="st">&quot;PROBEID&quot;</span>, <span class="dt">multiVals =</span> <span class="st">&quot;list&quot;</span>)
<span class="co">#&gt; &#39;select()&#39; returned 1:many mapping between keys and columns</span>
<span class="co">#&gt; $`16737401`</span>
<span class="co">#&gt; [1] &quot;TRAF6&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`16657436`</span>
<span class="co">#&gt; [1] &quot;DDX11L1&quot;      &quot;LOC102725121&quot; &quot;DDX11L2&quot;      &quot;DDX11L9&quot;     </span>
<span class="co">#&gt; [5] &quot;DDX11L10&quot;     &quot;DDX11L5&quot;      &quot;DDX11L16&quot;    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`16678303`</span>
<span class="co">#&gt; [1] &quot;ARF1&quot;</span></code></pre></div>
</div>
<div id="choices-for-multivals-continued" class="section level3">
<h3><span class="header-section-number">12.4.17</span> Choices for multiVals (continued)</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">mapIds</span>(hugene20sttranscriptcluster.db, ids, <span class="st">&quot;SYMBOL&quot;</span>, <span class="st">&quot;PROBEID&quot;</span>, <span class="dt">multiVals =</span> <span class="st">&quot;CharacterList&quot;</span>)
<span class="co">#&gt; &#39;select()&#39; returned 1:many mapping between keys and columns</span>
<span class="co">#&gt; CharacterList of length 3</span>
<span class="co">#&gt; [[&quot;16737401&quot;]] TRAF6</span>
<span class="co">#&gt; [[&quot;16657436&quot;]] DDX11L1 LOC102725121 DDX11L2 DDX11L9 DDX11L10 DDX11L5 DDX11L16</span>
<span class="co">#&gt; [[&quot;16678303&quot;]] ARF1</span>
<span class="kw">mapIds</span>(hugene20sttranscriptcluster.db, ids, <span class="st">&quot;SYMBOL&quot;</span>, <span class="st">&quot;PROBEID&quot;</span>, <span class="dt">multiVals =</span> <span class="st">&quot;filter&quot;</span>)
<span class="co">#&gt; &#39;select()&#39; returned 1:many mapping between keys and columns</span>
<span class="co">#&gt; 16737401 16678303 </span>
<span class="co">#&gt;  &quot;TRAF6&quot;   &quot;ARF1&quot;</span>
<span class="kw">mapIds</span>(hugene20sttranscriptcluster.db, ids, <span class="st">&quot;SYMBOL&quot;</span>, <span class="st">&quot;PROBEID&quot;</span>, <span class="dt">multiVals =</span> <span class="st">&quot;asNA&quot;</span>)
<span class="co">#&gt; &#39;select()&#39; returned 1:many mapping between keys and columns</span>
<span class="co">#&gt; 16737401 16657436 16678303 </span>
<span class="co">#&gt;  &quot;TRAF6&quot;       NA   &quot;ARF1&quot;</span></code></pre></div>
</div>
<div id="chipdborgdb-questions" class="section level3">
<h3><span class="header-section-number">12.4.18</span> ChipDb/OrgDb questions</h3>
<p>Using either the hugene20sttranscriptcluster.db or org.Hs.eg.db package,</p>
<ul>
<li><p>What gene symbol corresponds to Entrez Gene ID 1000?</p></li>
<li><p>What is the Ensembl Gene ID for PPARG?</p></li>
<li><p>What is the UniProt ID for GAPDH?</p></li>
<li><p>How many of the probesets from the ExpressionSet (eset) we loaded map to a single gene? How many donâ€™t map to a gene at all?</p></li>
</ul>
</div>
<div id="txdb-packages" class="section level3">
<h3><span class="header-section-number">12.4.19</span> TxDb packages</h3>
<p>TxDb packages contain positional information; the contents can be inferred by the package name</p>
<p><strong>TxDb.Species.Source.Build.Table</strong></p>
<ul>
<li><p>TxDb.Hsapiens.UCSC.hg19.knownGene</p>
<ul>
<li><p><em>Homo sapiens</em></p></li>
<li><p>UCSC genome browser</p></li>
<li><p>hg19 (their version of GRCh37)</p></li>
<li><p>knownGene table</p></li>
</ul></li>
</ul>
<p>TxDb.Dmelanogaster.UCSC.dm3.ensGene TxDb.Athaliana.BioMart.plantsmart22</p>
</div>
<div id="ensdb-packages" class="section level3">
<h3><span class="header-section-number">12.4.20</span> EnsDb packages</h3>
<p>EnsDb packages are similar to TxDb packages, but based on Ensembl mappings</p>
<p>EnsDb.Hsapiens.v79 EnsDb.Mmusculus.v79 EnsDb.Rnorvegicus.v79</p>
</div>
<div id="transcript-packages" class="section level3">
<h3><span class="header-section-number">12.4.21</span> Transcript packages</h3>
<p>As with ChipDb and OrgDb packages, <code>select</code> and <code>mapIds</code> can be used to make queries</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">select</span>(TxDb.Hsapiens.UCSC.hg19.knownGene, <span class="kw">c</span>(<span class="st">&quot;1&quot;</span>,<span class="st">&quot;10&quot;</span>),
       <span class="kw">c</span>(<span class="st">&quot;TXNAME&quot;</span>,<span class="st">&quot;TXCHROM&quot;</span>,<span class="st">&quot;TXSTART&quot;</span>,<span class="st">&quot;TXEND&quot;</span>), <span class="st">&quot;GENEID&quot;</span>)
<span class="co">#&gt; &#39;select()&#39; returned 1:many mapping between keys and columns</span>
<span class="co">#&gt;   GENEID     TXNAME TXCHROM  TXSTART    TXEND</span>
<span class="co">#&gt; 1      1 uc002qsd.4   chr19 58858172 58864865</span>
<span class="co">#&gt; 2      1 uc002qsf.2   chr19 58859832 58874214</span>
<span class="co">#&gt; 3     10 uc003wyw.1    chr8 18248755 18258723</span>
<span class="kw">select</span>(EnsDb.Hsapiens.v79, <span class="kw">c</span>(<span class="st">&quot;1&quot;</span>, <span class="st">&quot;10&quot;</span>),
       <span class="kw">c</span>(<span class="st">&quot;GENEID&quot;</span>,<span class="st">&quot;GENENAME&quot;</span>,<span class="st">&quot;SEQNAME&quot;</span>,<span class="st">&quot;GENESEQSTART&quot;</span>,<span class="st">&quot;GENESEQEND&quot;</span>), <span class="st">&quot;ENTREZID&quot;</span>)
<span class="co">#&gt;   ENTREZID          GENEID GENENAME SEQNAME GENESEQSTART GENESEQEND</span>
<span class="co">#&gt; 1        1 ENSG00000121410     A1BG      19     58345178   58353499</span>
<span class="co">#&gt; 2       10 ENSG00000156006     NAT2       8     18391245   18401218</span></code></pre></div>
<p>But this is not how one normally uses themâ€¦</p>
</div>
<div id="granges" class="section level3">
<h3><span class="header-section-number">12.4.22</span> GRanges</h3>
<p>The normal use case for transcript packages is to extract positional information into a <code>GRanges</code> or <code>GRangesList</code> object. An example is the genomic position of all genes:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
gns &lt;-<span class="st"> </span><span class="kw">genes</span>(TxDb.Hsapiens.UCSC.hg19.knownGene)
gns
<span class="co">#&gt; GRanges object with 23056 ranges and 1 metadata column:</span>
<span class="co">#&gt;         seqnames              ranges strand |     gene_id</span>
<span class="co">#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;       1    chr19   58858172-58874214      - |           1</span>
<span class="co">#&gt;      10     chr8   18248755-18258723      + |          10</span>
<span class="co">#&gt;     100    chr20   43248163-43280376      - |         100</span>
<span class="co">#&gt;    1000    chr18   25530930-25757445      - |        1000</span>
<span class="co">#&gt;   10000     chr1 243651535-244006886      - |       10000</span>
<span class="co">#&gt;     ...      ...                 ...    ... .         ...</span>
<span class="co">#&gt;    9991     chr9 114979995-115095944      - |        9991</span>
<span class="co">#&gt;    9992    chr21   35736323-35743440      + |        9992</span>
<span class="co">#&gt;    9993    chr22   19023795-19109967      - |        9993</span>
<span class="co">#&gt;    9994     chr6   90539619-90584155      + |        9994</span>
<span class="co">#&gt;    9997    chr22   50961997-50964905      - |        9997</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 93 sequences (1 circular) from hg19 genome</span></code></pre></div>
</div>
<div id="grangeslist" class="section level3">
<h3><span class="header-section-number">12.4.23</span> GRangesList</h3>
<p>Or the genomic position of all transcripts <strong>by</strong> gene:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
txs &lt;-<span class="st"> </span><span class="kw">transcriptsBy</span>(TxDb.Hsapiens.UCSC.hg19.knownGene)
txs
<span class="co">#&gt; GRangesList object of length 23459:</span>
<span class="co">#&gt; $1 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">#&gt;          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt;   [1]    chr19 58858172-58864865      - |     70455  uc002qsd.4</span>
<span class="co">#&gt;   [2]    chr19 58859832-58874214      - |     70456  uc002qsf.2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $10 </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames            ranges strand | tx_id    tx_name</span>
<span class="co">#&gt;   [1]     chr8 18248755-18258723      + | 31944 uc003wyw.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $100 </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames            ranges strand | tx_id    tx_name</span>
<span class="co">#&gt;   [1]    chr20 43248163-43280376      - | 72132 uc002xmj.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ...</span>
<span class="co">#&gt; &lt;23456 more elements&gt;</span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 93 sequences (1 circular) from hg19 genome</span></code></pre></div>
</div>
<div id="other-accessors" class="section level3">
<h3><span class="header-section-number">12.4.24</span> Other accessors</h3>
<ul>
<li><p>Positional information can be extracted for <code>transcripts</code>, <code>genes</code>, coding sequences (<code>cds</code>), <code>promoters</code> and <code>exons</code>.</p></li>
<li><p>Positional information can be extracted for most of the above, grouped by a second element. For example, our <code>transcriptsBy</code> call was all transcripts, grouped by gene.</p></li>
<li><p>More detail on these *Ranges objects is beyond the scope of this workshop, but why we want them is not.</p></li>
</ul>
</div>
<div id="why-ranges-objects" class="section level3">
<h3><span class="header-section-number">12.4.25</span> Why *Ranges objects</h3>
<p>The main rationale for *Ranges objects is to allow us to easily select and subset data based on genomic position information. This is really powerful!</p>
<p><code>GRanges</code> and <code>GRangesLists</code> act like data.frames and lists, and can be subsetted using the <code>[</code> function. As a really artificial example:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
txs[txs <span class="op">%over%</span><span class="st"> </span>gns[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>,]]
<span class="co">#&gt; GRangesList object of length 3:</span>
<span class="co">#&gt; $1 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">#&gt;          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt;   [1]    chr19 58858172-58864865      - |     70455  uc002qsd.4</span>
<span class="co">#&gt;   [2]    chr19 58859832-58874214      - |     70456  uc002qsf.2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $10 </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames            ranges strand | tx_id    tx_name</span>
<span class="co">#&gt;   [1]     chr8 18248755-18258723      + | 31944 uc003wyw.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $162968 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames            ranges strand | tx_id    tx_name</span>
<span class="co">#&gt;   [1]    chr19 58865723-58874214      - | 70457 uc002qsh.2</span>
<span class="co">#&gt;   [2]    chr19 58865723-58874214      - | 70458 uc002qsi.2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 93 sequences (1 circular) from hg19 genome</span></code></pre></div>
</div>
<div id="ranges-use-cases" class="section level3">
<h3><span class="header-section-number">12.4.26</span> *Ranges use cases</h3>
<ul>
<li><p>Gene expression changes near differentially methylated CpG islands</p></li>
<li><p>Closest genes to a set of interesting SNPs</p></li>
<li><p>Genes near DNAseI hypersensitivity clusters</p></li>
<li><p>Number of CpGs measured over Gene X by Chip Y</p></li>
</ul>
</div>
<div id="summarizedexperiment-objects" class="section level3">
<h3><span class="header-section-number">12.4.27</span> SummarizedExperiment objects</h3>
<p>SummarizedExperiment objects are like ExpressionSets, but the row-wise annotations are GRanges, so you can subset by genomic locations:</p>
<p><img src="MacDonald_Annotation/subset.png" width="75%" /></p>
<p>SummarizedExperiment objects are popular objects for representing expression data and other rectangular data (feature x sample data). Incoming packages are now strongly recommended to use this class representation instead of ExpressionSet.</p>
</div>
<div id="txdb-exercises" class="section level3">
<h3><span class="header-section-number">12.4.28</span> TxDb exercises</h3>
<ul>
<li><p>How many transcripts does PPARG have, according to UCSC?</p></li>
<li><p>Does Ensembl agree?</p></li>
<li>How many genes are between 2858473 and 3271812 on chr2 in the hg19 genome?
<ul>
<li>Hint: you make a <code>GRanges</code> like this - <code>GRanges(&quot;chr2&quot;, IRanges(2858473,3271812))</code></li>
</ul></li>
</ul>
</div>
<div id="organismdb-packages" class="section level3">
<h3><span class="header-section-number">12.4.29</span> OrganismDb packages</h3>
<p>OrganismDb packages are meta-packages that contain an OrgDb, a TxDb, and a GO.db package and allow cross-queries between those packages.</p>
<p>All previous accessors work; <code>select</code>, <code>mapIds</code>, <code>transcripts</code>, etc.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">library</span>(Homo.sapiens)
Homo.sapiens
<span class="co">#&gt; OrganismDb Object:</span>
<span class="co">#&gt; # Includes GODb Object:  GO.db </span>
<span class="co">#&gt; # With data about:  Gene Ontology </span>
<span class="co">#&gt; # Includes OrgDb Object:  org.Hs.eg.db </span>
<span class="co">#&gt; # Gene data about:  Homo sapiens </span>
<span class="co">#&gt; # Taxonomy Id:  9606 </span>
<span class="co">#&gt; # Includes TxDb Object:  TxDb.Hsapiens.UCSC.hg19.knownGene </span>
<span class="co">#&gt; # Transcriptome data about:  Homo sapiens </span>
<span class="co">#&gt; # Based on genome:  hg19 </span>
<span class="co">#&gt; # The OrgDb gene id ENTREZID is mapped to the TxDb gene id GENEID .</span></code></pre></div>
</div>
<div id="organismdb-packages-1" class="section level3">
<h3><span class="header-section-number">12.4.30</span> OrganismDb packages</h3>
<ul>
<li><p>Updateable - can change TxDb object</p></li>
<li><p>columns and keytypes span all underlying objects</p></li>
<li><p>Calls to TxDb accessors include a â€˜columnsâ€™ argument</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">head</span>(<span class="kw">genes</span>(Homo.sapiens, <span class="dt">columns =</span> <span class="kw">c</span>(<span class="st">&quot;ENTREZID&quot;</span>,<span class="st">&quot;ALIAS&quot;</span>,<span class="st">&quot;UNIPROT&quot;</span>)),<span class="dv">4</span>)
<span class="co">#&gt; &#39;select()&#39; returned 1:many mapping between keys and columns</span>
<span class="co">#&gt; GRanges object with 4 ranges and 3 metadata columns:</span>
<span class="co">#&gt;        seqnames            ranges strand |                 ALIAS</span>
<span class="co">#&gt;           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |       &lt;CharacterList&gt;</span>
<span class="co">#&gt;      1    chr19 58858172-58874214      - |       A1B,ABG,GAB,...</span>
<span class="co">#&gt;     10     chr8 18248755-18258723      + |   AAC2,NAT-2,PNAT,...</span>
<span class="co">#&gt;    100    chr20 43248163-43280376      - |                   ADA</span>
<span class="co">#&gt;   1000    chr18 25530930-25757445      - | CD325,CDHN,CDw325,...</span>
<span class="co">#&gt;                         UNIPROT     ENTREZID</span>
<span class="co">#&gt;                 &lt;CharacterList&gt; &lt;FactorList&gt;</span>
<span class="co">#&gt;      1            P04217,V9HWD8            1</span>
<span class="co">#&gt;     10            A4Z6T7,P11245           10</span>
<span class="co">#&gt;    100 A0A0S2Z381,P00813,F5GWI4          100</span>
<span class="co">#&gt;   1000        P19022,A0A024RC42         1000</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 93 sequences (1 circular) from hg19 genome</span></code></pre></div>
</div>
<div id="organismdb-exercises" class="section level3">
<h3><span class="header-section-number">12.4.31</span> OrganismDb exercises</h3>
<ul>
<li><p>Get all the GO terms for BRCA1</p></li>
<li><p>What gene does the UCSC transcript ID uc002fai.3 map to?</p></li>
<li><p>How many other transcripts does that gene have?</p></li>
<li><p>Get all the transcripts from the hg19 genome build, along with their Ensembl gene ID, UCSC transcript ID and gene symbol</p></li>
</ul>
</div>
<div id="organism.dplyr-package" class="section level3">
<h3><span class="header-section-number">12.4.32</span> Organism.dplyr package</h3>
<ul>
<li><p>Combines the data from TxDb and Org.Db associated packages into local database.</p></li>
<li>Allows functions from both <em>org.*</em> and <em>TxDb.*</em>
<ul>
<li><code>keytypes()</code>, <code>select()</code>, â€¦</li>
<li><code>exons()</code>, <code>promoters()</code>, â€¦</li>
</ul></li>
<li><p>Allows for filtering and display of combined TxDb and Org.Db information through <code>dplyr</code> functions.</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(Organism.dplyr)
<span class="co">#&gt; Loading required package: dplyr</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;dplyr&#39;</span>
<span class="co">#&gt; The following objects are masked from &#39;package:stats&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from &#39;package:base&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="co">#&gt; Loading required package: AnnotationFilter</span>

<span class="co"># src = src_organism(&quot;TxDb.Hsapiens.UCSC.hg19.knownGene&quot;)</span>
src &lt;-<span class="st"> </span><span class="kw">src_organism</span>(<span class="dt">dbpath =</span> <span class="kw">hg38light</span>())
src
<span class="co">#&gt; src:  sqlite 3.22.0 [/usr/local/lib/R/site-library/Organism.dplyr/extdata/light.hg38.knownGene.sqlite]</span>
<span class="co">#&gt; tbls: id, id_accession, id_go, id_go_all, id_omim_pm, id_protein,</span>
<span class="co">#&gt;   id_transcript, ranges_cds, ranges_exon, ranges_gene, ranges_tx</span></code></pre></div>
</div>
<div id="organism.dplyr" class="section level3">
<h3><span class="header-section-number">12.4.33</span> Organism.dplyr</h3>
<p>Get promoters from a TxDb object (we use a small version)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">options</span>(<span class="dt">width =</span> <span class="dv">120</span>)
<span class="kw">promoters</span>(src)
<span class="co">#&gt; &lt;SQL&gt;</span>
<span class="co">#&gt; SELECT *</span>
<span class="co">#&gt; FROM `ranges_tx`</span>
<span class="co">#&gt; GRanges object with 88 ranges and 2 metadata columns:</span>
<span class="co">#&gt;                      seqnames              ranges strand |     tx_id     tx_name</span>
<span class="co">#&gt;                         &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt;   uc001hzz.2             chr1 243843037-243845236      - |     15880  uc001hzz.2</span>
<span class="co">#&gt;   uc021plu.1             chr1 243843385-243845584      - |     15881  uc021plu.1</span>
<span class="co">#&gt;   uc001iab.3             chr1 243843083-243845282      - |     15882  uc001iab.3</span>
<span class="co">#&gt;   uc057qvr.1             chr1 243849929-243852128      - |     15883  uc057qvr.1</span>
<span class="co">#&gt;   uc057qvt.1             chr1 243614947-243617146      - |     15884  uc057qvt.1</span>
<span class="co">#&gt;          ...              ...                 ...    ... .       ...         ...</span>
<span class="co">#&gt;   uc064xqh.1 chrUn_GL000220v1       110025-112224      + |    197741  uc064xqh.1</span>
<span class="co">#&gt;   uc064xqi.1 chrUn_GL000220v1       112151-114350      + |    197742  uc064xqi.1</span>
<span class="co">#&gt;   uc064xqj.1 chrUn_GL000220v1       115428-117627      + |    197743  uc064xqj.1</span>
<span class="co">#&gt;   uc064xqk.1 chrUn_GL000220v1       116197-118396      + |    197744  uc064xqk.1</span>
<span class="co">#&gt;   uc033dnj.2 chrUn_GL000220v1       153997-156196      + |    197750  uc033dnj.2</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
</div>
<div id="organism.dplyr-1" class="section level3">
<h3><span class="header-section-number">12.4.34</span> Organism.dplyr</h3>
<p>Extract a table from the underlying database</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">tbl</span>(src, <span class="st">&quot;id&quot;</span>)
<span class="co">#&gt; # Source:   table&lt;id&gt; [?? x 6]</span>
<span class="co">#&gt; # Database: sqlite 3.22.0 []</span>
<span class="co">#&gt;    entrez map      ensembl         symbol genename               alias   </span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;    &lt;chr&gt;           &lt;chr&gt;  &lt;chr&gt;                  &lt;chr&gt;   </span>
<span class="co">#&gt;  1 1      19q13.4  ENSG00000121410 A1BG   alpha-1-B glycoprotein A1B     </span>
<span class="co">#&gt;  2 1      19q13.4  ENSG00000121410 A1BG   alpha-1-B glycoprotein ABG     </span>
<span class="co">#&gt;  3 1      19q13.4  ENSG00000121410 A1BG   alpha-1-B glycoprotein GAB     </span>
<span class="co">#&gt;  4 1      19q13.4  ENSG00000121410 A1BG   alpha-1-B glycoprotein HYST2477</span>
<span class="co">#&gt;  5 1      19q13.4  ENSG00000121410 A1BG   alpha-1-B glycoprotein A1BG    </span>
<span class="co">#&gt;  6 10     8p22     ENSG00000156006 NAT2   N-acetyltransferase 2  AAC2    </span>
<span class="co">#&gt;  7 10     8p22     ENSG00000156006 NAT2   N-acetyltransferase 2  NAT-2   </span>
<span class="co">#&gt;  8 10     8p22     ENSG00000156006 NAT2   N-acetyltransferase 2  PNAT    </span>
<span class="co">#&gt;  9 10     8p22     ENSG00000156006 NAT2   N-acetyltransferase 2  NAT2    </span>
<span class="co">#&gt; 10 100    20q13.12 ENSG00000196839 ADA    adenosine deaminase    ADA     </span>
<span class="co">#&gt; # ... with more rows</span></code></pre></div>
</div>
<div id="organism.dplyr-2" class="section level3">
<h3><span class="header-section-number">12.4.35</span> Organism.dplyr</h3>
<p>Make a complex query between tables in the underlying database</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">inner_join</span>(<span class="kw">tbl</span>(src, <span class="st">&quot;id&quot;</span>), <span class="kw">tbl</span>(src, <span class="st">&quot;ranges_gene&quot;</span>)) <span class="op">%&gt;%</span>
<span class="st">            </span><span class="kw">filter</span>(symbol <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;ADA&quot;</span>, <span class="st">&quot;NAT2&quot;</span>)) <span class="op">%&gt;%</span>
<span class="st">            </span>dplyr<span class="op">::</span><span class="kw">select</span>(gene_chrom, gene_start, gene_end,
            gene_strand, symbol, alias, map)
<span class="co">#&gt; Joining, by = &quot;entrez&quot;</span>
<span class="co">#&gt; # Source:   lazy query [?? x 7]</span>
<span class="co">#&gt; # Database: sqlite 3.22.0 []</span>
<span class="co">#&gt;   gene_chrom gene_start gene_end gene_strand symbol alias map     </span>
<span class="co">#&gt;   &lt;chr&gt;           &lt;int&gt;    &lt;int&gt; &lt;chr&gt;       &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;   </span>
<span class="co">#&gt; 1 chr8         18391245 18401218 +           NAT2   AAC2  8p22    </span>
<span class="co">#&gt; 2 chr8         18391245 18401218 +           NAT2   NAT-2 8p22    </span>
<span class="co">#&gt; 3 chr8         18391245 18401218 +           NAT2   PNAT  8p22    </span>
<span class="co">#&gt; 4 chr8         18391245 18401218 +           NAT2   NAT2  8p22    </span>
<span class="co">#&gt; 5 chr20        44619522 44651742 -           ADA    ADA   20q13.12</span></code></pre></div>
</div>
<div id="organism.dplyr-exercises" class="section level3">
<h3><span class="header-section-number">12.4.36</span> Organism.dplyr exercises</h3>
<ul>
<li><p>How many supported organisms are implemented in Organism.dplyr?</p></li>
<li><p>Display the ensembl Id and genename description for symbol â€œNAT2â€.</p></li>
<li><p>Show all the alias for â€œNAT2â€ in the database.</p></li>
<li><p>Display Gene ontology (GO) information for gene symbol â€œNAT2â€.</p></li>
</ul>
</div>
<div id="bsgenome-packages" class="section level3">
<h3><span class="header-section-number">12.4.37</span> BSgenome packages</h3>
<p>BSgenome packages contain sequence information for a given species/build. There are many such packages - you can get a listing using <code>available.genomes</code></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">library</span>(BSgenome)
<span class="kw">head</span>(<span class="kw">available.genomes</span>())
<span class="co">#&gt; [1] &quot;BSgenome.Alyrata.JGI.v1&quot;                 &quot;BSgenome.Amellifera.BeeBase.assembly4&quot;  </span>
<span class="co">#&gt; [3] &quot;BSgenome.Amellifera.UCSC.apiMel2&quot;        &quot;BSgenome.Amellifera.UCSC.apiMel2.masked&quot;</span>
<span class="co">#&gt; [5] &quot;BSgenome.Athaliana.TAIR.04232008&quot;        &quot;BSgenome.Athaliana.TAIR.TAIR9&quot;</span></code></pre></div>
</div>
<div id="bsgenome-packages-1" class="section level3">
<h3><span class="header-section-number">12.4.38</span> BSgenome packages</h3>
<p>We can load and inspect a BSgenome package</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">library</span>(BSgenome.Hsapiens.UCSC.hg19)
Hsapiens
<span class="co">#&gt; Human genome:</span>
<span class="co">#&gt; # organism: Homo sapiens (Human)</span>
<span class="co">#&gt; # provider: UCSC</span>
<span class="co">#&gt; # provider version: hg19</span>
<span class="co">#&gt; # release date: Feb. 2009</span>
<span class="co">#&gt; # release name: Genome Reference Consortium GRCh37</span>
<span class="co">#&gt; # 93 sequences:</span>
<span class="co">#&gt; #   chr1                  chr2                  chr3                  chr4                  chr5                 </span>
<span class="co">#&gt; #   chr6                  chr7                  chr8                  chr9                  chr10                </span>
<span class="co">#&gt; #   chr11                 chr12                 chr13                 chr14                 chr15                </span>
<span class="co">#&gt; #   chr16                 chr17                 chr18                 chr19                 chr20                </span>
<span class="co">#&gt; #   chr21                 chr22                 chrX                  chrY                  chrM                 </span>
<span class="co">#&gt; #   ...                   ...                   ...                   ...                   ...                  </span>
<span class="co">#&gt; #   chrUn_gl000227        chrUn_gl000228        chrUn_gl000229        chrUn_gl000230        chrUn_gl000231       </span>
<span class="co">#&gt; #   chrUn_gl000232        chrUn_gl000233        chrUn_gl000234        chrUn_gl000235        chrUn_gl000236       </span>
<span class="co">#&gt; #   chrUn_gl000237        chrUn_gl000238        chrUn_gl000239        chrUn_gl000240        chrUn_gl000241       </span>
<span class="co">#&gt; #   chrUn_gl000242        chrUn_gl000243        chrUn_gl000244        chrUn_gl000245        chrUn_gl000246       </span>
<span class="co">#&gt; #   chrUn_gl000247        chrUn_gl000248        chrUn_gl000249                                                   </span>
<span class="co">#&gt; # (use &#39;seqnames()&#39; to see all the sequence names, use the &#39;$&#39; or &#39;[[&#39; operator to access a given sequence)</span></code></pre></div>
</div>
<div id="bsgenome-packages-2" class="section level3">
<h3><span class="header-section-number">12.4.39</span> BSgenome packages</h3>
<p>The main accessor is <code>getSeq</code>, and you can get data by sequence (e.g., entire chromosome or unplaced scaffold), or by passing in a GRanges object, to get just a region.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">getSeq</span>(Hsapiens, <span class="st">&quot;chr1&quot;</span>)
<span class="co">#&gt;   249250621-letter &quot;DNAString&quot; instance</span>
<span class="co">#&gt; seq: NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span>
<span class="kw">getSeq</span>(Hsapiens, gns[<span class="st">&quot;5467&quot;</span>,])
<span class="co">#&gt;   A DNAStringSet instance of length 1</span>
<span class="co">#&gt;     width seq                                                                                       names               </span>
<span class="co">#&gt; [1] 85634 GCGGAGCGTGTGACGCTGCGGCCGCCGCGGACCTGGGGATTAA...ACTTTAAATAAATGGGAATTAAATATTTAAGAGCTGACTGGAA 5467</span></code></pre></div>
<p>The Biostrings package contains most of the code for dealing with these <code>*StringSet</code> objects - please see the Biostrings vignettes and help pages for more information.</p>
</div>
<div id="bsgenome-exercises" class="section level3">
<h3><span class="header-section-number">12.4.40</span> BSgenome exercises</h3>
<ul>
<li>Get the sequences for all transcripts of the TP53 gene</li>
</ul>
</div>
<div id="annotationhub-1" class="section level3">
<h3><span class="header-section-number">12.4.41</span> AnnotationHub</h3>
<p>AnnotationHub is a package that allows us to query and download many different annotation objects, without having to explicitly install them.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">library</span>(AnnotationHub)
hub &lt;-<span class="st"> </span><span class="kw">AnnotationHub</span>()
<span class="co">#&gt; snapshotDate(): 2018-06-27</span>
hub
<span class="co">#&gt; AnnotationHub with 44925 records</span>
<span class="co">#&gt; # snapshotDate(): 2018-06-27 </span>
<span class="co">#&gt; # $dataprovider: BroadInstitute, Ensembl, UCSC, ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/, Haemcode, Inparanoid8, Pazar,...</span>
<span class="co">#&gt; # $species: Homo sapiens, Mus musculus, Drosophila melanogaster, Bos taurus, Pan troglodytes, Rattus norvegicus, Dan...</span>
<span class="co">#&gt; # $rdataclass: GRanges, BigWigFile, FaFile, TwoBitFile, Rle, OrgDb, ChainFile, EnsDb, Inparanoid8Db, TxDb</span>
<span class="co">#&gt; # additional mcols(): taxonomyid, genome, description, coordinate_1_based, maintainer, rdatadateadded,</span>
<span class="co">#&gt; #   preparerclass, tags, rdatapath, sourceurl, sourcetype </span>
<span class="co">#&gt; # retrieve records with, e.g., &#39;object[[&quot;AH2&quot;]]&#39; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;             title                                                   </span>
<span class="co">#&gt;   AH2     | Ailuropoda_melanoleuca.ailMel1.69.dna.toplevel.fa       </span>
<span class="co">#&gt;   AH3     | Ailuropoda_melanoleuca.ailMel1.69.dna_rm.toplevel.fa    </span>
<span class="co">#&gt;   AH4     | Ailuropoda_melanoleuca.ailMel1.69.dna_sm.toplevel.fa    </span>
<span class="co">#&gt;   AH5     | Ailuropoda_melanoleuca.ailMel1.69.ncrna.fa              </span>
<span class="co">#&gt;   AH6     | Ailuropoda_melanoleuca.ailMel1.69.pep.all.fa            </span>
<span class="co">#&gt;   ...       ...                                                     </span>
<span class="co">#&gt;   AH63655 | phastCons46wayPrimates.UCSC.hg19.chrX.rds               </span>
<span class="co">#&gt;   AH63656 | phastCons46wayPrimates.UCSC.hg19.chrY.rds               </span>
<span class="co">#&gt;   AH63657 | Alternative Splicing Annotation for Homo sapiens (Human)</span>
<span class="co">#&gt;   AH63658 | Allele data from the IPD IMGT/HLA database              </span>
<span class="co">#&gt;   AH63659 | Allele data from the IPD KIR database</span></code></pre></div>
</div>
<div id="querying-annotationhub-1" class="section level3">
<h3><span class="header-section-number">12.4.42</span> Querying AnnotationHub</h3>
<p>Finding the â€˜rightâ€™ resource on AnnotationHub is like using Google - a well posed query is necessary to find what you are after. Useful queries are based on</p>
<ul>
<li><p>Data provider</p></li>
<li><p>Data class</p></li>
<li><p>Species</p></li>
<li><p>Data source</p></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">names</span>(<span class="kw">mcols</span>(hub))
<span class="co">#&gt;  [1] &quot;title&quot;              &quot;dataprovider&quot;       &quot;species&quot;            &quot;taxonomyid&quot;         &quot;genome&quot;            </span>
<span class="co">#&gt;  [6] &quot;description&quot;        &quot;coordinate_1_based&quot; &quot;maintainer&quot;         &quot;rdatadateadded&quot;     &quot;preparerclass&quot;     </span>
<span class="co">#&gt; [11] &quot;tags&quot;               &quot;rdataclass&quot;         &quot;rdatapath&quot;          &quot;sourceurl&quot;          &quot;sourcetype&quot;</span></code></pre></div>
</div>
<div id="annotationhub-data-providers" class="section level3">
<h3><span class="header-section-number">12.4.43</span> AnnotationHub Data providers</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">unique</span>(hub<span class="op">$</span>dataprovider)
<span class="co">#&gt;  [1] &quot;Ensembl&quot;                               &quot;UCSC&quot;                                 </span>
<span class="co">#&gt;  [3] &quot;RefNet&quot;                                &quot;Inparanoid8&quot;                          </span>
<span class="co">#&gt;  [5] &quot;NHLBI&quot;                                 &quot;ChEA&quot;                                 </span>
<span class="co">#&gt;  [7] &quot;Pazar&quot;                                 &quot;NIH Pathway Interaction Database&quot;     </span>
<span class="co">#&gt;  [9] &quot;Haemcode&quot;                              &quot;BroadInstitute&quot;                       </span>
<span class="co">#&gt; [11] &quot;PRIDE&quot;                                 &quot;Gencode&quot;                              </span>
<span class="co">#&gt; [13] &quot;CRIBI&quot;                                 &quot;Genoscope&quot;                            </span>
<span class="co">#&gt; [15] &quot;MISO, VAST-TOOLS, UCSC&quot;                &quot;UWashington&quot;                          </span>
<span class="co">#&gt; [17] &quot;Stanford&quot;                              &quot;dbSNP&quot;                                </span>
<span class="co">#&gt; [19] &quot;BioMart&quot;                               &quot;GeneOntology&quot;                         </span>
<span class="co">#&gt; [21] &quot;KEGG&quot;                                  &quot;URGI&quot;                                 </span>
<span class="co">#&gt; [23] &quot;ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/&quot; &quot;EMBL-EBI&quot;</span></code></pre></div>
</div>
<div id="annotationhub-data-classes" class="section level3">
<h3><span class="header-section-number">12.4.44</span> AnnotationHub Data classes</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">unique</span>(hub<span class="op">$</span>rdataclass)
<span class="co">#&gt;  [1] &quot;FaFile&quot;                            &quot;GRanges&quot;                           &quot;data.frame&quot;                       </span>
<span class="co">#&gt;  [4] &quot;Inparanoid8Db&quot;                     &quot;TwoBitFile&quot;                        &quot;ChainFile&quot;                        </span>
<span class="co">#&gt;  [7] &quot;SQLiteConnection&quot;                  &quot;biopax&quot;                            &quot;BigWigFile&quot;                       </span>
<span class="co">#&gt; [10] &quot;AAStringSet&quot;                       &quot;MSnSet&quot;                            &quot;mzRpwiz&quot;                          </span>
<span class="co">#&gt; [13] &quot;mzRident&quot;                          &quot;list&quot;                              &quot;TxDb&quot;                             </span>
<span class="co">#&gt; [16] &quot;Rle&quot;                               &quot;EnsDb&quot;                             &quot;VcfFile&quot;                          </span>
<span class="co">#&gt; [19] &quot;igraph&quot;                            &quot;OrgDb&quot;                             &quot;data.frame, DNAStringSet, GRanges&quot;</span></code></pre></div>
</div>
<div id="annotationhub-species" class="section level3">
<h3><span class="header-section-number">12.4.45</span> AnnotationHub Species</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">head</span>(<span class="kw">unique</span>(hub<span class="op">$</span>species))
<span class="co">#&gt; [1] &quot;Ailuropoda melanoleuca&quot; &quot;Anolis carolinensis&quot;    &quot;Bos taurus&quot;             &quot;Caenorhabditis elegans&quot;</span>
<span class="co">#&gt; [5] &quot;Callithrix jacchus&quot;     &quot;Canis familiaris&quot;</span>
<span class="kw">length</span>(<span class="kw">unique</span>(hub<span class="op">$</span>species))
<span class="co">#&gt; [1] 1971</span></code></pre></div>
</div>
<div id="annotationhub-data-sources" class="section level3">
<h3><span class="header-section-number">12.4.46</span> AnnotationHub Data sources</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">unique</span>(hub<span class="op">$</span>sourcetype)
<span class="co">#&gt;  [1] &quot;FASTA&quot;        &quot;UCSC track&quot;   &quot;GTF&quot;          &quot;TSV&quot;          &quot;Inparanoid&quot;   &quot;TwoBit&quot;       &quot;Chain&quot;       </span>
<span class="co">#&gt;  [8] &quot;GRASP&quot;        &quot;Zip&quot;          &quot;CSV&quot;          &quot;BioPax&quot;       &quot;BioPaxLevel2&quot; &quot;RData&quot;        &quot;BED&quot;         </span>
<span class="co">#&gt; [15] &quot;BigWig&quot;       &quot;tab&quot;          &quot;mzTab&quot;        &quot;mzML&quot;         &quot;mzid&quot;         &quot;GFF&quot;          &quot;ensembl&quot;     </span>
<span class="co">#&gt; [22] &quot;VCF&quot;          &quot;NCBI/ensembl&quot; &quot;NCBI/UniProt&quot;</span></code></pre></div>
</div>
<div id="annotationhub-query" class="section level3">
<h3><span class="header-section-number">12.4.47</span> AnnotationHub query</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
qry &lt;-<span class="st"> </span><span class="kw">query</span>(hub, <span class="kw">c</span>(<span class="st">&quot;granges&quot;</span>,<span class="st">&quot;homo sapiens&quot;</span>,<span class="st">&quot;ensembl&quot;</span>))
qry
<span class="co">#&gt; AnnotationHub with 56 records</span>
<span class="co">#&gt; # snapshotDate(): 2018-06-27 </span>
<span class="co">#&gt; # $dataprovider: Ensembl, UCSC</span>
<span class="co">#&gt; # $species: Homo sapiens</span>
<span class="co">#&gt; # $rdataclass: GRanges</span>
<span class="co">#&gt; # additional mcols(): taxonomyid, genome, description, coordinate_1_based, maintainer, rdatadateadded,</span>
<span class="co">#&gt; #   preparerclass, tags, rdatapath, sourceurl, sourcetype </span>
<span class="co">#&gt; # retrieve records with, e.g., &#39;object[[&quot;AH5046&quot;]]&#39; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;             title                                          </span>
<span class="co">#&gt;   AH5046  | Ensembl Genes                                  </span>
<span class="co">#&gt;   AH5160  | Ensembl Genes                                  </span>
<span class="co">#&gt;   AH5311  | Ensembl Genes                                  </span>
<span class="co">#&gt;   AH5434  | Ensembl Genes                                  </span>
<span class="co">#&gt;   AH5435  | Ensembl EST Genes                              </span>
<span class="co">#&gt;   ...       ...                                            </span>
<span class="co">#&gt;   AH60085 | Homo_sapiens.GRCh38.91.gtf                     </span>
<span class="co">#&gt;   AH61125 | Homo_sapiens.GRCh38.92.abinitio.gtf            </span>
<span class="co">#&gt;   AH61126 | Homo_sapiens.GRCh38.92.chr.gtf                 </span>
<span class="co">#&gt;   AH61127 | Homo_sapiens.GRCh38.92.chr_patch_hapl_scaff.gtf</span>
<span class="co">#&gt;   AH61128 | Homo_sapiens.GRCh38.92.gtf</span></code></pre></div>
</div>
<div id="annotationhub-query-1" class="section level3">
<h3><span class="header-section-number">12.4.48</span> AnnotationHub query</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
qry<span class="op">$</span>sourceurl
<span class="co">#&gt;  [1] &quot;rtracklayer://hgdownload.cse.ucsc.edu/goldenpath/hg19/database/ensGene&quot;                                  </span>
<span class="co">#&gt;  [2] &quot;rtracklayer://hgdownload.cse.ucsc.edu/goldenpath/hg18/database/ensGene&quot;                                  </span>
<span class="co">#&gt;  [3] &quot;rtracklayer://hgdownload.cse.ucsc.edu/goldenpath/hg17/database/ensGene&quot;                                  </span>
<span class="co">#&gt;  [4] &quot;rtracklayer://hgdownload.cse.ucsc.edu/goldenpath/hg16/database/ensGene&quot;                                  </span>
<span class="co">#&gt;  [5] &quot;rtracklayer://hgdownload.cse.ucsc.edu/goldenpath/hg16/database/ensEstGene&quot;                               </span>
<span class="co">#&gt;  [6] &quot;ftp://ftp.ensembl.org/pub/release-70/gtf/homo_sapiens/Homo_sapiens.GRCh37.70.gtf.gz&quot;                     </span>
<span class="co">#&gt;  [7] &quot;ftp://ftp.ensembl.org/pub/release-69/gtf/homo_sapiens/Homo_sapiens.GRCh37.69.gtf.gz&quot;                     </span>
<span class="co">#&gt;  [8] &quot;ftp://ftp.ensembl.org/pub/release-71/gtf/homo_sapiens/Homo_sapiens.GRCh37.71.gtf.gz&quot;                     </span>
<span class="co">#&gt;  [9] &quot;ftp://ftp.ensembl.org/pub/release-72/gtf/homo_sapiens/Homo_sapiens.GRCh37.72.gtf.gz&quot;                     </span>
<span class="co">#&gt; [10] &quot;ftp://ftp.ensembl.org/pub/release-73/gtf/homo_sapiens/Homo_sapiens.GRCh37.73.gtf.gz&quot;                     </span>
<span class="co">#&gt; [11] &quot;ftp://ftp.ensembl.org/pub/release-74/gtf/homo_sapiens/Homo_sapiens.GRCh37.74.gtf.gz&quot;                     </span>
<span class="co">#&gt; [12] &quot;ftp://ftp.ensembl.org/pub/release-75/gtf/homo_sapiens/Homo_sapiens.GRCh37.75.gtf.gz&quot;                     </span>
<span class="co">#&gt; [13] &quot;ftp://ftp.ensembl.org/pub/release-78/gtf/homo_sapiens/Homo_sapiens.GRCh38.78.gtf.gz&quot;                     </span>
<span class="co">#&gt; [14] &quot;ftp://ftp.ensembl.org/pub/release-76/gtf/homo_sapiens/Homo_sapiens.GRCh38.76.gtf.gz&quot;                     </span>
<span class="co">#&gt; [15] &quot;ftp://ftp.ensembl.org/pub/release-79/gtf/homo_sapiens/Homo_sapiens.GRCh38.79.gtf.gz&quot;                     </span>
<span class="co">#&gt; [16] &quot;ftp://ftp.ensembl.org/pub/release-77/gtf/homo_sapiens/Homo_sapiens.GRCh38.77.gtf.gz&quot;                     </span>
<span class="co">#&gt; [17] &quot;ftp://ftp.ensembl.org/pub/release-80/gtf/homo_sapiens/Homo_sapiens.GRCh38.80.gtf.gz&quot;                     </span>
<span class="co">#&gt; [18] &quot;ftp://ftp.ensembl.org/pub/release-81/gtf/homo_sapiens/Homo_sapiens.GRCh38.81.gtf.gz&quot;                     </span>
<span class="co">#&gt; [19] &quot;ftp://ftp.ensembl.org/pub/release-82/gtf/homo_sapiens/Homo_sapiens.GRCh38.82.gtf.gz&quot;                     </span>
<span class="co">#&gt; [20] &quot;ftp://ftp.ensembl.org/pub/release-83/gtf/homo_sapiens/Homo_sapiens.GRCh38.83.gtf.gz&quot;                     </span>
<span class="co">#&gt; [21] &quot;ftp://ftp.ensembl.org/pub/release-84/gtf/homo_sapiens/Homo_sapiens.GRCh38.84.abinitio.gtf.gz&quot;            </span>
<span class="co">#&gt; [22] &quot;ftp://ftp.ensembl.org/pub/release-84/gtf/homo_sapiens/Homo_sapiens.GRCh38.84.chr.gtf.gz&quot;                 </span>
<span class="co">#&gt; [23] &quot;ftp://ftp.ensembl.org/pub/release-84/gtf/homo_sapiens/Homo_sapiens.GRCh38.84.chr_patch_hapl_scaff.gtf.gz&quot;</span>
<span class="co">#&gt; [24] &quot;ftp://ftp.ensembl.org/pub/release-84/gtf/homo_sapiens/Homo_sapiens.GRCh38.84.gtf.gz&quot;                     </span>
<span class="co">#&gt; [25] &quot;ftp://ftp.ensembl.org/pub/release-85/gtf/homo_sapiens/Homo_sapiens.GRCh38.85.abinitio.gtf.gz&quot;            </span>
<span class="co">#&gt; [26] &quot;ftp://ftp.ensembl.org/pub/release-85/gtf/homo_sapiens/Homo_sapiens.GRCh38.85.chr.gtf.gz&quot;                 </span>
<span class="co">#&gt; [27] &quot;ftp://ftp.ensembl.org/pub/release-85/gtf/homo_sapiens/Homo_sapiens.GRCh38.85.chr_patch_hapl_scaff.gtf.gz&quot;</span>
<span class="co">#&gt; [28] &quot;ftp://ftp.ensembl.org/pub/release-85/gtf/homo_sapiens/Homo_sapiens.GRCh38.85.gtf.gz&quot;                     </span>
<span class="co">#&gt; [29] &quot;ftp://ftp.ensembl.org/pub/release-86/gtf/homo_sapiens/Homo_sapiens.GRCh38.86.abinitio.gtf.gz&quot;            </span>
<span class="co">#&gt; [30] &quot;ftp://ftp.ensembl.org/pub/release-86/gtf/homo_sapiens/Homo_sapiens.GRCh38.86.chr.gtf.gz&quot;                 </span>
<span class="co">#&gt; [31] &quot;ftp://ftp.ensembl.org/pub/release-86/gtf/homo_sapiens/Homo_sapiens.GRCh38.86.chr_patch_hapl_scaff.gtf.gz&quot;</span>
<span class="co">#&gt; [32] &quot;ftp://ftp.ensembl.org/pub/release-86/gtf/homo_sapiens/Homo_sapiens.GRCh38.86.gtf.gz&quot;                     </span>
<span class="co">#&gt; [33] &quot;ftp://ftp.ensembl.org/pub/release-87/gtf/homo_sapiens/Homo_sapiens.GRCh38.87.abinitio.gtf.gz&quot;            </span>
<span class="co">#&gt; [34] &quot;ftp://ftp.ensembl.org/pub/release-87/gtf/homo_sapiens/Homo_sapiens.GRCh38.87.chr.gtf.gz&quot;                 </span>
<span class="co">#&gt; [35] &quot;ftp://ftp.ensembl.org/pub/release-87/gtf/homo_sapiens/Homo_sapiens.GRCh38.87.chr_patch_hapl_scaff.gtf.gz&quot;</span>
<span class="co">#&gt; [36] &quot;ftp://ftp.ensembl.org/pub/release-87/gtf/homo_sapiens/Homo_sapiens.GRCh38.87.gtf.gz&quot;                     </span>
<span class="co">#&gt; [37] &quot;ftp://ftp.ensembl.org/pub/release-88/gtf/homo_sapiens/Homo_sapiens.GRCh38.88.abinitio.gtf.gz&quot;            </span>
<span class="co">#&gt; [38] &quot;ftp://ftp.ensembl.org/pub/release-88/gtf/homo_sapiens/Homo_sapiens.GRCh38.88.chr.gtf.gz&quot;                 </span>
<span class="co">#&gt; [39] &quot;ftp://ftp.ensembl.org/pub/release-88/gtf/homo_sapiens/Homo_sapiens.GRCh38.88.chr_patch_hapl_scaff.gtf.gz&quot;</span>
<span class="co">#&gt; [40] &quot;ftp://ftp.ensembl.org/pub/release-88/gtf/homo_sapiens/Homo_sapiens.GRCh38.88.gtf.gz&quot;                     </span>
<span class="co">#&gt; [41] &quot;ftp://ftp.ensembl.org/pub/release-89/gtf/homo_sapiens/Homo_sapiens.GRCh38.89.abinitio.gtf.gz&quot;            </span>
<span class="co">#&gt; [42] &quot;ftp://ftp.ensembl.org/pub/release-89/gtf/homo_sapiens/Homo_sapiens.GRCh38.89.chr.gtf.gz&quot;                 </span>
<span class="co">#&gt; [43] &quot;ftp://ftp.ensembl.org/pub/release-89/gtf/homo_sapiens/Homo_sapiens.GRCh38.89.chr_patch_hapl_scaff.gtf.gz&quot;</span>
<span class="co">#&gt; [44] &quot;ftp://ftp.ensembl.org/pub/release-89/gtf/homo_sapiens/Homo_sapiens.GRCh38.89.gtf.gz&quot;                     </span>
<span class="co">#&gt; [45] &quot;ftp://ftp.ensembl.org/pub/release-90/gtf/homo_sapiens/Homo_sapiens.GRCh38.90.abinitio.gtf.gz&quot;            </span>
<span class="co">#&gt; [46] &quot;ftp://ftp.ensembl.org/pub/release-90/gtf/homo_sapiens/Homo_sapiens.GRCh38.90.chr.gtf.gz&quot;                 </span>
<span class="co">#&gt; [47] &quot;ftp://ftp.ensembl.org/pub/release-90/gtf/homo_sapiens/Homo_sapiens.GRCh38.90.chr_patch_hapl_scaff.gtf.gz&quot;</span>
<span class="co">#&gt; [48] &quot;ftp://ftp.ensembl.org/pub/release-90/gtf/homo_sapiens/Homo_sapiens.GRCh38.90.gtf.gz&quot;                     </span>
<span class="co">#&gt; [49] &quot;ftp://ftp.ensembl.org/pub/release-91/gtf/homo_sapiens/Homo_sapiens.GRCh38.91.abinitio.gtf.gz&quot;            </span>
<span class="co">#&gt; [50] &quot;ftp://ftp.ensembl.org/pub/release-91/gtf/homo_sapiens/Homo_sapiens.GRCh38.91.chr.gtf.gz&quot;                 </span>
<span class="co">#&gt; [51] &quot;ftp://ftp.ensembl.org/pub/release-91/gtf/homo_sapiens/Homo_sapiens.GRCh38.91.chr_patch_hapl_scaff.gtf.gz&quot;</span>
<span class="co">#&gt; [52] &quot;ftp://ftp.ensembl.org/pub/release-91/gtf/homo_sapiens/Homo_sapiens.GRCh38.91.gtf.gz&quot;                     </span>
<span class="co">#&gt; [53] &quot;ftp://ftp.ensembl.org/pub/release-92/gtf/homo_sapiens/Homo_sapiens.GRCh38.92.abinitio.gtf.gz&quot;            </span>
<span class="co">#&gt; [54] &quot;ftp://ftp.ensembl.org/pub/release-92/gtf/homo_sapiens/Homo_sapiens.GRCh38.92.chr.gtf.gz&quot;                 </span>
<span class="co">#&gt; [55] &quot;ftp://ftp.ensembl.org/pub/release-92/gtf/homo_sapiens/Homo_sapiens.GRCh38.92.chr_patch_hapl_scaff.gtf.gz&quot;</span>
<span class="co">#&gt; [56] &quot;ftp://ftp.ensembl.org/pub/release-92/gtf/homo_sapiens/Homo_sapiens.GRCh38.92.gtf.gz&quot;</span></code></pre></div>
</div>
<div id="selecting-annotationhub-resource" class="section level3">
<h3><span class="header-section-number">12.4.49</span> Selecting AnnotationHub resource</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
whatIwant &lt;-<span class="st"> </span>qry[[<span class="st">&quot;AH50377&quot;</span>]]</code></pre></div>
<p>We can use these data as they are, or convert to a TxDb format:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
GRCh38TxDb &lt;-<span class="st"> </span><span class="kw">makeTxDbFromGRanges</span>(whatIwant)
GRCh38TxDb
<span class="co">#&gt; TxDb object:</span>
<span class="co">#&gt; # Db type: TxDb</span>
<span class="co">#&gt; # Supporting package: GenomicFeatures</span>
<span class="co">#&gt; # Genome: GRCh38</span>
<span class="co">#&gt; # transcript_nrow: 199184</span>
<span class="co">#&gt; # exon_nrow: 675836</span>
<span class="co">#&gt; # cds_nrow: 270225</span>
<span class="co">#&gt; # Db created by: GenomicFeatures package from Bioconductor</span>
<span class="co">#&gt; # Creation time: 2018-07-18 18:42:38 +0000 (Wed, 18 Jul 2018)</span>
<span class="co">#&gt; # GenomicFeatures version at creation time: 1.33.0</span>
<span class="co">#&gt; # RSQLite version at creation time: 2.1.1</span>
<span class="co">#&gt; # DBSCHEMAVERSION: 1.2</span></code></pre></div>
</div>
<div id="annotationhub-exercises" class="section level3">
<h3><span class="header-section-number">12.4.50</span> AnnotationHub exercises</h3>
<ul>
<li><p>How many resources are on AnnotationHub for Atlantic salmon (Salmo salar)?</p></li>
<li><p>Get the most recent Ensembl build for domesticated dog (Canis familiaris) and make a TxDb</p></li>
</ul>
</div>
<div id="biomart" class="section level3">
<h3><span class="header-section-number">12.4.51</span> biomaRt</h3>
<p>The biomaRt package allows queries to an Ensembl Biomart server. We can see the choices of servers that we can use:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw">library</span>(biomaRt)
<span class="kw">listMarts</span>()
<span class="co">#&gt;                biomart               version</span>
<span class="co">#&gt; 1 ENSEMBL_MART_ENSEMBL      Ensembl Genes 93</span>
<span class="co">#&gt; 2   ENSEMBL_MART_MOUSE      Mouse strains 93</span>
<span class="co">#&gt; 3     ENSEMBL_MART_SNP  Ensembl Variation 93</span>
<span class="co">#&gt; 4 ENSEMBL_MART_FUNCGEN Ensembl Regulation 93</span></code></pre></div>
</div>
<div id="biomart-data-sets" class="section level3">
<h3><span class="header-section-number">12.4.52</span> biomaRt data sets</h3>
<p>And we can then check for the available data sets on a particular server.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
mart &lt;-<span class="st"> </span><span class="kw">useMart</span>(<span class="st">&quot;ENSEMBL_MART_ENSEMBL&quot;</span>)
<span class="kw">head</span>(<span class="kw">listDatasets</span>(mart))
<span class="co">#&gt;                       dataset                        description      version</span>
<span class="co">#&gt; 1  acarolinensis_gene_ensembl     Anole lizard genes (AnoCar2.0)    AnoCar2.0</span>
<span class="co">#&gt; 2   amelanoleuca_gene_ensembl              Panda genes (ailMel1)      ailMel1</span>
<span class="co">#&gt; 3     amexicanus_gene_ensembl        Cave fish genes (AstMex102)    AstMex102</span>
<span class="co">#&gt; 4     anancymaae_gene_ensembl Ma&#39;s night monkey genes (Anan_2.0)     Anan_2.0</span>
<span class="co">#&gt; 5 aplatyrhynchos_gene_ensembl          Duck genes (BGI_duck_1.0) BGI_duck_1.0</span>
<span class="co">#&gt; 6        btaurus_gene_ensembl                 Cow genes (UMD3.1)       UMD3.1</span></code></pre></div>
</div>
<div id="biomart-queries" class="section level3">
<h3><span class="header-section-number">12.4.53</span> biomaRt queries</h3>
<p>After setting up a <code>mart</code> object pointing to the server and data set that we care about, we can make queries. We first set up the <code>mart</code> object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
mart &lt;-<span class="st"> </span><span class="kw">useMart</span>(<span class="st">&quot;ENSEMBL_MART_ENSEMBL&quot;</span>,<span class="st">&quot;hsapiens_gene_ensembl&quot;</span>)</code></pre></div>
<p>Queries are of the form</p>
<p>getBM(<em>attributes</em>, <em>filters</em>, <em>values</em>, <em>mart</em>)</p>
<p>where</p>
<ul>
<li><p>attributes are the things we <strong>want</strong></p></li>
<li><p>filters are the <em>types of</em> IDs we <strong>have</strong></p></li>
<li><p>values are the IDs we <strong>have</strong></p></li>
<li><p>mart is the <code>mart</code> object we set up</p></li>
</ul>
</div>
<div id="biomart-attributes-and-filters" class="section level3">
<h3><span class="header-section-number">12.4.54</span> biomaRt attributes and filters</h3>
<p>Both attributes and filters have rather inscrutable names, but a listing can be accessed using</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
atrib &lt;-<span class="st"> </span><span class="kw">listAttributes</span>(mart)
filts &lt;-<span class="st"> </span><span class="kw">listFilters</span>(mart)
<span class="kw">head</span>(atrib)
<span class="co">#&gt;                            name                  description         page</span>
<span class="co">#&gt; 1               ensembl_gene_id               Gene stable ID feature_page</span>
<span class="co">#&gt; 2       ensembl_gene_id_version       Gene stable ID version feature_page</span>
<span class="co">#&gt; 3         ensembl_transcript_id         Transcript stable ID feature_page</span>
<span class="co">#&gt; 4 ensembl_transcript_id_version Transcript stable ID version feature_page</span>
<span class="co">#&gt; 5            ensembl_peptide_id            Protein stable ID feature_page</span>
<span class="co">#&gt; 6    ensembl_peptide_id_version    Protein stable ID version feature_page</span>
<span class="kw">head</span>(filts)
<span class="co">#&gt;              name              description</span>
<span class="co">#&gt; 1 chromosome_name Chromosome/scaffold name</span>
<span class="co">#&gt; 2           start                    Start</span>
<span class="co">#&gt; 3             end                      End</span>
<span class="co">#&gt; 4      band_start               Band Start</span>
<span class="co">#&gt; 5        band_end                 Band End</span>
<span class="co">#&gt; 6    marker_start             Marker Start</span></code></pre></div>
</div>
<div id="biomart-query" class="section level3">
<h3><span class="header-section-number">12.4.55</span> biomaRt query</h3>
<p>A simple example query</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
afyids &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;1000_at&quot;</span>,<span class="st">&quot;1001_at&quot;</span>,<span class="st">&quot;1002_f_at&quot;</span>,<span class="st">&quot;1007_s_at&quot;</span>)
<span class="kw">getBM</span>(<span class="kw">c</span>(<span class="st">&quot;affy_hg_u95av2&quot;</span>, <span class="st">&quot;hgnc_symbol&quot;</span>), <span class="kw">c</span>(<span class="st">&quot;affy_hg_u95av2&quot;</span>), afyids, mart)
<span class="co">#&gt;   affy_hg_u95av2 hgnc_symbol</span>
<span class="co">#&gt; 1        1000_at       MAPK3</span>
<span class="co">#&gt; 2      1007_s_at        DDR1</span>
<span class="co">#&gt; 3      1002_f_at            </span>
<span class="co">#&gt; 4      1002_f_at     CYP2C19</span>
<span class="co">#&gt; 5        1001_at        TIE1</span></code></pre></div>
</div>
<div id="biomart-exercises" class="section level3">
<h3><span class="header-section-number">12.4.56</span> biomaRt exercises</h3>
<ul>
<li><p>Get the Ensembl gene IDs and HUGO symbol for Entrez Gene IDs 672, 5468 and 7157</p></li>
<li><p>What do you get if you query for the â€˜gene_exonâ€™ for GAPDH?</p></li>
</ul>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="fluent-genomic-data-analysis-with-plyranges.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
