<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="solving-common-bioinformatic-challenges-using-genomicranges.html">
<link rel="next" href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-122837239-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-122837239-1');
</script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-workshops"><i class="fa fa-check"></i><b>1.1</b> The Workshops</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html"><i class="fa fa-check"></i><b>2</b> 100: <em>R</em> and <em>Bioconductor</em> for everyone: an introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#description"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="2.1.7" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-goals"><i class="fa fa-check"></i><b>2.1.7</b> Learning goals</a></li>
<li class="chapter" data-level="2.1.8" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-objectives"><i class="fa fa-check"></i><b>2.1.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-r"><i class="fa fa-check"></i><b>2.2</b> Introduction to <em>R</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#rstudio-orientation"><i class="fa fa-check"></i><b>2.2.1</b> <em>RStudio</em> orientation</a></li>
<li class="chapter" data-level="2.2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#very-basics"><i class="fa fa-check"></i><b>2.2.2</b> Very basics</a></li>
<li class="chapter" data-level="2.2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#data-input-and-manipulation"><i class="fa fa-check"></i><b>2.2.3</b> Data input and manipulation</a></li>
<li class="chapter" data-level="2.2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#writing-r-scripts"><i class="fa fa-check"></i><b>2.2.4</b> Writing <em>R</em> scripts</a></li>
<li class="chapter" data-level="2.2.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#using-packages"><i class="fa fa-check"></i><b>2.2.5</b> Using packages</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-bioconductor"><i class="fa fa-check"></i><b>2.3</b> Introduction to <em>Bioconductor</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#discovering-installing-and-learning-how-to-use-bioconductor-packages"><i class="fa fa-check"></i><b>2.3.1</b> Discovering, installing, and learning how to use <em>Bioconductor</em> packages</a></li>
<li class="chapter" data-level="2.3.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-genomic-ranges"><i class="fa fa-check"></i><b>2.3.2</b> Working with Genomic Ranges</a></li>
<li class="chapter" data-level="2.3.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-summarized-experimental-data"><i class="fa fa-check"></i><b>2.3.3</b> Working with summarized experimental data</a></li>
<li class="chapter" data-level="2.3.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#down-stream-analysis"><i class="fa fa-check"></i><b>2.3.4</b> Down-stream analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>3</b> 101: Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>3.1</b> Instructors</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description"><i class="fa fa-check"></i><b>3.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="3.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>3.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="3.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation"><i class="fa fa-check"></i><b>3.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="3.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="3.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-1"><i class="fa fa-check"></i><b>3.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="3.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-1"><i class="fa fa-check"></i><b>3.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>3.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="3.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>3.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="3.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>3.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="3.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>3.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="3.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>3.4.4</b> Data containers</a></li>
<li class="chapter" data-level="3.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>3.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="3.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>3.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="3.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>3.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="3.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>3.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="3.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>3.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="3.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>3.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="3.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>3.4.11</b> Simple example</a></li>
<li class="chapter" data-level="3.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>3.4.12</b> Questions!</a></li>
<li class="chapter" data-level="3.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>3.4.13</b> More questions!</a></li>
<li class="chapter" data-level="3.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>3.4.14</b> Another example</a></li>
<li class="chapter" data-level="3.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>3.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="3.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>3.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="3.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>3.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="3.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>3.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="3.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>3.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="3.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>3.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="3.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>3.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="3.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>3.4.22</b> GRanges</a></li>
<li class="chapter" data-level="3.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>3.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="3.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>3.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="3.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>3.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="3.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>3.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="3.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>3.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="3.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>3.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="3.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>3.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="3.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>3.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="3.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>3.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="3.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>3.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="3.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr"><i class="fa fa-check"></i><b>3.4.33</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-1"><i class="fa fa-check"></i><b>3.4.34</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-2"><i class="fa fa-check"></i><b>3.4.35</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>3.4.36</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="3.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>3.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>3.4.38</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>3.4.39</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>3.4.40</b> BSgenome exercises</a></li>
<li class="chapter" data-level="3.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub"><i class="fa fa-check"></i><b>3.4.41</b> AnnotationHub</a></li>
<li class="chapter" data-level="3.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub"><i class="fa fa-check"></i><b>3.4.42</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="3.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>3.4.43</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="3.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>3.4.44</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="3.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>3.4.45</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="3.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>3.4.46</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="3.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>3.4.47</b> AnnotationHub query</a></li>
<li class="chapter" data-level="3.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>3.4.48</b> AnnotationHub query</a></li>
<li class="chapter" data-level="3.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>3.4.49</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="3.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>3.4.50</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="3.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>3.4.51</b> biomaRt</a></li>
<li class="chapter" data-level="3.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>3.4.52</b> biomaRt data sets</a></li>
<li class="chapter" data-level="3.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>3.4.53</b> biomaRt queries</a></li>
<li class="chapter" data-level="3.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>3.4.54</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="3.4.55" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>3.4.55</b> biomaRt query</a></li>
<li class="chapter" data-level="3.4.56" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>3.4.56</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>4</b> 102: Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>4.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-1"><i class="fa fa-check"></i><b>4.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="4.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-1"><i class="fa fa-check"></i><b>4.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-1"><i class="fa fa-check"></i><b>4.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="4.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-1"><i class="fa fa-check"></i><b>4.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="4.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-2"><i class="fa fa-check"></i><b>4.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="4.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-2"><i class="fa fa-check"></i><b>4.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#introduction-1"><i class="fa fa-check"></i><b>4.4</b> Introduction</a><ul>
<li class="chapter" data-level="4.4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#what-is-the-ranges-infrastructure"><i class="fa fa-check"></i><b>4.4.1</b> What is the Ranges infrastructure?</a></li>
<li class="chapter" data-level="4.4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#why-use-the-ranges-infrastructure"><i class="fa fa-check"></i><b>4.4.2</b> Why use the Ranges infrastructure?</a></li>
<li class="chapter" data-level="4.4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>4.4.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#setup"><i class="fa fa-check"></i><b>4.5</b> Setup</a></li>
<li class="chapter" data-level="4.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>4.6</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="4.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#constructing-a-granges-object-from-data.frame"><i class="fa fa-check"></i><b>4.7</b> Constructing a <em>GRanges</em> object from data.frame</a></li>
<li class="chapter" data-level="4.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#loading-a-granges-object-from-a-standard-file-format"><i class="fa fa-check"></i><b>4.8</b> Loading a <em>GRanges</em> object from a standard file format</a></li>
<li class="chapter" data-level="4.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-manipulation-of-granges-objects"><i class="fa fa-check"></i><b>4.9</b> Basic manipulation of <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>4.10</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>4.11</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#aggregating-granges-objects"><i class="fa fa-check"></i><b>4.12</b> Aggregating <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>4.13</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>4.14</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.15" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#finding-overlaps-between-granges-objects"><i class="fa fa-check"></i><b>4.15</b> Finding overlaps between <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.16" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises"><i class="fa fa-check"></i><b>4.16</b> Exercises</a></li>
<li class="chapter" data-level="4.17" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>4.17</b> Example: exploring BigWig files from AnnotationHub</a><ul>
<li class="chapter" data-level="4.17.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#extracting-data-from-annotationhub"><i class="fa fa-check"></i><b>4.17.1</b> Extracting data from AnnotationHub</a></li>
<li class="chapter" data-level="4.17.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#reading-bigwig-files"><i class="fa fa-check"></i><b>4.17.2</b> Reading BigWig files</a></li>
<li class="chapter" data-level="4.17.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-1"><i class="fa fa-check"></i><b>4.17.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4.18" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>4.18</b> Worked example: coverage analysis of BAM files</a><ul>
<li class="chapter" data-level="4.18.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-2"><i class="fa fa-check"></i><b>4.18.1</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4.19" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#conclusions"><i class="fa fa-check"></i><b>4.19</b> Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>5</b> 103: Public data resources and Bioconductor</a><ul>
<li class="chapter" data-level="5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>5.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>5.2</b> Syllabus</a><ul>
<li class="chapter" data-level="5.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-2"><i class="fa fa-check"></i><b>5.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="5.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-2"><i class="fa fa-check"></i><b>5.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="5.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-2"><i class="fa fa-check"></i><b>5.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>5.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="5.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-2"><i class="fa fa-check"></i><b>5.2.5</b> Time outline</a></li>
<li class="chapter" data-level="5.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>5.2.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="5.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-3"><i class="fa fa-check"></i><b>5.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="5.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-3"><i class="fa fa-check"></i><b>5.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-1"><i class="fa fa-check"></i><b>5.3</b> Overview</a></li>
<li class="chapter" data-level="5.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>5.4</b> GEOquery</a><ul>
<li class="chapter" data-level="5.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>5.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="5.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>5.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="5.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>5.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="5.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>5.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>5.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="5.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>5.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="5.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>5.5.2</b> Basic design</a></li>
<li class="chapter" data-level="5.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>5.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>5.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="5.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>5.6.1</b> Authentication</a></li>
<li class="chapter" data-level="5.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>5.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>5.7</b> Sequence Read Archive</a><ul>
<li class="chapter" data-level="5.7.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage-1"><i class="fa fa-check"></i><b>5.7.1</b> Usage</a></li>
<li class="chapter" data-level="5.7.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#query-syntax"><i class="fa fa-check"></i><b>5.7.2</b> Query syntax</a></li>
<li class="chapter" data-level="5.7.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#using-the-raw-api-without-rbioconductor"><i class="fa fa-check"></i><b>5.7.3</b> Using the raw API without R/Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>5.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="5.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>5.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="5.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>5.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="5.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>5.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="5.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>5.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="5.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>5.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="5.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>5.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="5.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>5.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>5.12</b> Pharmacogenomics</a><ul>
<li class="chapter" data-level="5.12.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started"><i class="fa fa-check"></i><b>5.12.1</b> Getting started</a></li>
<li class="chapter" data-level="5.12.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-pharmacogx-datasets-pharmacosets"><i class="fa fa-check"></i><b>5.12.2</b> Overview of PharmacoGx datasets (PharmacoSets)</a></li>
<li class="chapter" data-level="5.12.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#exploring-drug-sensitivity-datasets"><i class="fa fa-check"></i><b>5.12.3</b> Exploring drug sensitivity datasets</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>5.13</b> Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>6</b> 200: RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>6.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-3"><i class="fa fa-check"></i><b>6.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="6.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-3"><i class="fa fa-check"></i><b>6.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="6.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>6.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-3"><i class="fa fa-check"></i><b>6.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>6.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="6.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals-4"><i class="fa fa-check"></i><b>6.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="6.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-4"><i class="fa fa-check"></i><b>6.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-2"><i class="fa fa-check"></i><b>6.4</b> Introduction</a></li>
<li class="chapter" data-level="6.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>6.5</b> Data packaging</a><ul>
<li class="chapter" data-level="6.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>6.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="6.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>6.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="6.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>6.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>6.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="6.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>6.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="6.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>6.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="6.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>6.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="6.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>6.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis"><i class="fa fa-check"></i><b>6.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="6.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>6.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="6.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>6.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="6.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>6.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="6.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>6.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="6.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>6.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="6.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>6.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>7</b> 201: RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview-2"><i class="fa fa-check"></i><b>7.1</b> Overview</a><ul>
<li class="chapter" data-level="7.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-1"><i class="fa fa-check"></i><b>7.1.1</b> Description</a></li>
<li class="chapter" data-level="7.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation-1"><i class="fa fa-check"></i><b>7.1.3</b> Participation</a></li>
<li class="chapter" data-level="7.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>7.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline-4"><i class="fa fa-check"></i><b>7.1.5</b> Time outline</a></li>
<li class="chapter" data-level="7.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>7.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>7.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="7.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>7.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="7.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>7.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="7.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>7.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="7.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>7.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>7.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="7.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>7.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="7.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>7.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="7.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>7.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>7.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="7.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>7.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="7.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>7.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="7.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>7.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>7.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="7.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>7.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="7.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>7.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub-1"><i class="fa fa-check"></i><b>7.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="7.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>7.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="7.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>7.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>7.7</b> Building reports</a><ul>
<li class="chapter" data-level="7.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>7.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="7.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>7.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>7.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="7.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>7.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="7.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>7.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="7.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>7.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="7.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>7.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="7.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>7.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="7.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>7.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>8</b> 202: Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="8.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-3"><i class="fa fa-check"></i><b>8.1</b> Overview</a><ul>
<li class="chapter" data-level="8.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-2"><i class="fa fa-check"></i><b>8.1.1</b> Description</a></li>
<li class="chapter" data-level="8.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-2"><i class="fa fa-check"></i><b>8.1.3</b> Participation</a></li>
<li class="chapter" data-level="8.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>8.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-5"><i class="fa fa-check"></i><b>8.1.5</b> Time outline</a></li>
<li class="chapter" data-level="8.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>8.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started-1"><i class="fa fa-check"></i><b>8.2</b> Getting started</a><ul>
<li class="chapter" data-level="8.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>8.2.1</b> The data</a></li>
<li class="chapter" data-level="8.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>8.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>8.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="8.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>8.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="8.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>8.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>8.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="8.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>8.5.1</b> Normalization</a></li>
<li class="chapter" data-level="8.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>8.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>8.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="8.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>8.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="8.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>8.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>9</b> 210: Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="9.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>9.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-4"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-6"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-4"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-6"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>9.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="9.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>9.4</b> Where does it all come from?</a></li>
<li class="chapter" data-level="9.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.5</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="9.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>9.6</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="9.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>9.7</b> Data types</a></li>
<li class="chapter" data-level="9.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>9.8</b> Differential expression analysis</a></li>
<li class="chapter" data-level="9.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>9.9</b> Gene sets</a></li>
<li class="chapter" data-level="9.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>9.10</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="9.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>9.11</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="9.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.12</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="9.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>9.13</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>10</b> 220: Workflow for multi-omics analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="10.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information-2"><i class="fa fa-check"></i><b>10.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-5"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-7"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-5"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>10.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-7"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-5"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-5"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>10.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="10.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="10.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>10.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="10.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>10.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>10.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="10.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>10.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="10.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>10.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="10.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>10.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>10.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="10.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>10.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="10.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>10.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="10.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>10.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="10.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>10.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="10.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>10.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>10.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="10.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>10.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="10.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>10.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="10.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>10.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>10.9</b> Complete cases</a></li>
<li class="chapter" data-level="10.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>10.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="10.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>10.11</b> Extraction</a><ul>
<li class="chapter" data-level="10.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>10.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="10.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>10.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="10.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>10.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="10.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>10.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="10.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>10.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="10.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>10.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="10.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>10.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="10.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>10.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="10.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>10.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>10.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="10.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>10.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="10.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>10.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="10.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>10.16.1</b> Simplification of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="10.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>10.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="10.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>10.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="10.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>10.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="10.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>10.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="10.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>10.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="10.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>10.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="10.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>10.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="10.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>10.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="10.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>10.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="10.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>10.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="10.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>10.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>11</b> 230: Cytoscape automation in R using Rcy3</a><ul>
<li class="chapter" data-level="11.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-4"><i class="fa fa-check"></i><b>11.1</b> Overview</a><ul>
<li class="chapter" data-level="11.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructor-names-and-contact-information-3"><i class="fa fa-check"></i><b>11.1.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="11.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description-6"><i class="fa fa-check"></i><b>11.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="11.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>11.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="11.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>11.1.4</b> Background:</a></li>
<li class="chapter" data-level="11.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation-6"><i class="fa fa-check"></i><b>11.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>11.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="11.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-8"><i class="fa fa-check"></i><b>11.1.7</b> Time outline</a></li>
<li class="chapter" data-level="11.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>11.2</b> Background</a><ul>
<li class="chapter" data-level="11.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>11.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="11.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>11.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="11.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>11.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>11.3</b> Translating biological data into Cytoscape using RCy3</a></li>
<li class="chapter" data-level="11.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>11.4</b> Set up</a><ul>
<li class="chapter" data-level="11.4.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started-2"><i class="fa fa-check"></i><b>11.4.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>11.5</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="11.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>11.6</b> Example Data Set</a></li>
<li class="chapter" data-level="11.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>11.7</b> Finding Network Data</a></li>
<li class="chapter" data-level="11.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>11.8</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="11.9" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>11.9</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="11.10" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>11.10</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>12</b> 240: Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="12.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructor-names-and-contact-information-4"><i class="fa fa-check"></i><b>12.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="12.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-7"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-8"><i class="fa fa-check"></i><b>12.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-7"><i class="fa fa-check"></i><b>12.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>12.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="12.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-9"><i class="fa fa-check"></i><b>12.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-6"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-6"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop"><i class="fa fa-check"></i><b>12.4</b> Workshop</a></li>
<li class="chapter" data-level="12.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-3"><i class="fa fa-check"></i><b>12.5</b> Introduction</a><ul>
<li class="chapter" data-level="12.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>12.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="12.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>12.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="12.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for-1"><i class="fa fa-check"></i><b>12.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup-1"><i class="fa fa-check"></i><b>12.6</b> Setup</a></li>
<li class="chapter" data-level="12.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>12.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="12.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>12.8</b> The Grammar</a><ul>
<li class="chapter" data-level="12.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>12.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="12.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>12.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="12.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises-3"><i class="fa fa-check"></i><b>12.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>12.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="12.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>12.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="12.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files-1"><i class="fa fa-check"></i><b>12.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="12.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>12.10</b> Whats next?</a></li>
<li class="chapter" data-level="12.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>12.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html"><i class="fa fa-check"></i><b>13</b> 250: Working with genomic data in R with the DECIPHER package</a><ul>
<li class="chapter" data-level="13.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#overview-5"><i class="fa fa-check"></i><b>13.1</b> Overview</a><ul>
<li class="chapter" data-level="13.1.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-description-8"><i class="fa fa-check"></i><b>13.1.1</b> Workshop Description</a></li>
<li class="chapter" data-level="13.1.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#pre-requisites-9"><i class="fa fa-check"></i><b>13.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.1.3" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-participation-8"><i class="fa fa-check"></i><b>13.1.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.1.4" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#r-bioconductor-packages-used-9"><i class="fa fa-check"></i><b>13.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="13.1.5" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#time-outline-10"><i class="fa fa-check"></i><b>13.1.5</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.2</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="13.2.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-goals-7"><i class="fa fa-check"></i><b>13.2.1</b> Learning goals</a></li>
<li class="chapter" data-level="13.2.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-objectives-7"><i class="fa fa-check"></i><b>13.2.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>14</b> 260: Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="14.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>14.1</b> Instructors:</a></li>
<li class="chapter" data-level="14.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-9"><i class="fa fa-check"></i><b>14.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-10"><i class="fa fa-check"></i><b>14.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-9"><i class="fa fa-check"></i><b>14.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="14.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-10"><i class="fa fa-check"></i><b>14.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="14.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-11"><i class="fa fa-check"></i><b>14.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-11"><i class="fa fa-check"></i><b>14.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="14.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-8"><i class="fa fa-check"></i><b>14.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="14.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-8"><i class="fa fa-check"></i><b>14.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#abstract"><i class="fa fa-check"></i><b>14.4</b> Abstract</a></li>
<li class="chapter" data-level="14.5" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#introduction-5"><i class="fa fa-check"></i><b>14.5</b> Introduction</a><ul>
<li class="chapter" data-level="14.5.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#downloading-pharmacoset-objects"><i class="fa fa-check"></i><b>14.5.1</b> Downloading PharmacoSet objects</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#reproducibility"><i class="fa fa-check"></i><b>14.6</b> Reproducibility</a><ul>
<li class="chapter" data-level="14.6.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#plotting-drug-dose-response-data"><i class="fa fa-check"></i><b>14.6.1</b> Plotting Drug-Dose Response Data</a></li>
<li class="chapter" data-level="14.6.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pharmacological-profiles"><i class="fa fa-check"></i><b>14.6.2</b> Pharmacological profiles</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#replication"><i class="fa fa-check"></i><b>14.7</b> Replication</a><ul>
<li class="chapter" data-level="14.7.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#consistency-of-pharmacological-profiles"><i class="fa fa-check"></i><b>14.7.1</b> Consistency of pharmacological profiles</a></li>
<li class="chapter" data-level="14.7.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#consistency-assessment-improved-by-modified-concordance-index"><i class="fa fa-check"></i><b>14.7.2</b> consistency assessment improved by Modified Concordance Index</a></li>
<li class="chapter" data-level="14.7.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#known-biomarkers"><i class="fa fa-check"></i><b>14.7.3</b> Known Biomarkers</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#machine-learning-and-biomarker-discovery"><i class="fa fa-check"></i><b>14.8</b> Machine Learning and Biomarker Discovery</a><ul>
<li class="chapter" data-level="14.8.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#bonus-using-the-connectivity-map-for-drug-repurposing"><i class="fa fa-check"></i><b>14.8.1</b> Bonus: Using the Connectivity Map for drug repurposing</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#session-info"><i class="fa fa-check"></i><b>14.9</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><i class="fa fa-check"></i><b>15</b> 500: Effectively using the DelayedArray framework to support the analysis of large datasets</a><ul>
<li class="chapter" data-level="15.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-6"><i class="fa fa-check"></i><b>15.1</b> Overview</a><ul>
<li class="chapter" data-level="15.1.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#description-3"><i class="fa fa-check"></i><b>15.1.1</b> Description</a></li>
<li class="chapter" data-level="15.1.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#pre-requisites-11"><i class="fa fa-check"></i><b>15.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="15.1.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#participation-3"><i class="fa fa-check"></i><b>15.1.3</b> Participation</a></li>
<li class="chapter" data-level="15.1.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#r-bioconductor-packages-used-11"><i class="fa fa-check"></i><b>15.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="15.1.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#time-outline-12"><i class="fa fa-check"></i><b>15.1.5</b> Time outline</a></li>
<li class="chapter" data-level="15.1.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-goals-and-objectives-12"><i class="fa fa-check"></i><b>15.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#introductory-material"><i class="fa fa-check"></i><b>15.2</b> Introductory material</a><ul>
<li class="chapter" data-level="15.2.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#discussion"><i class="fa fa-check"></i><b>15.2.1</b> Discussion</a></li>
<li class="chapter" data-level="15.2.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#why-learn-delayedarray"><i class="fa fa-check"></i><b>15.2.2</b> Why learn DelayedArray?</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-of-delayedarray-framework"><i class="fa fa-check"></i><b>15.3</b> Overview of DelayedArray framework</a><ul>
<li class="chapter" data-level="15.3.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-package"><i class="fa fa-check"></i><b>15.3.1</b> The <strong>DelayedArray</strong> package</a></li>
<li class="chapter" data-level="15.3.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-class"><i class="fa fa-check"></i><b>15.3.2</b> The <em>DelayedArray</em> class</a></li>
<li class="chapter" data-level="15.3.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#operating-on-delayedarray-objects"><i class="fa fa-check"></i><b>15.3.3</b> Operating on <em>DelayedArray</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#whats-out-there-already"><i class="fa fa-check"></i><b>15.4</b> Whats out there already?</a></li>
<li class="chapter" data-level="15.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#incoporating-delayedarray-into-a-package"><i class="fa fa-check"></i><b>15.5</b> Incoporating DelayedArray into a package</a><ul>
<li class="chapter" data-level="15.5.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#writing-algorithms-to-process-delayedarray-instances"><i class="fa fa-check"></i><b>15.5.1</b> Writing algorithms to process <em>DelayedArray</em> instances</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#questions-and-discussion"><i class="fa fa-check"></i><b>15.6</b> Questions and discussion</a></li>
<li class="chapter" data-level="15.7" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#todos"><i class="fa fa-check"></i><b>15.7</b> TODOs</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>16</b> 510: Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="16.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-2"><i class="fa fa-check"></i><b>16.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="16.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-10"><i class="fa fa-check"></i><b>16.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="16.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-12"><i class="fa fa-check"></i><b>16.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="16.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-4"><i class="fa fa-check"></i><b>16.2.2</b> Participation</a></li>
<li class="chapter" data-level="16.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>16.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="16.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-13"><i class="fa fa-check"></i><b>16.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-6"><i class="fa fa-check"></i><b>16.3</b> Introduction</a><ul>
<li class="chapter" data-level="16.3.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>16.3.1</b> Why maintain your package</a></li>
<li class="chapter" data-level="16.3.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>16.3.2</b> Infrastructure used for maintenance</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>16.4</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="16.4.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>16.4.1</b> Build report</a></li>
<li class="chapter" data-level="16.4.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#version-numbering"><i class="fa fa-check"></i><b>16.4.2</b> Version Numbering</a></li>
<li class="chapter" data-level="16.4.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>16.4.3</b> Landing page badges</a></li>
<li class="chapter" data-level="16.4.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>16.4.4</b> Download statistics</a></li>
<li class="chapter" data-level="16.4.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>16.4.5</b> RSS feeds</a></li>
<li class="chapter" data-level="16.4.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>16.4.6</b> Support site, Bioc-devel mailing list</a></li>
<li class="chapter" data-level="16.4.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github"><i class="fa fa-check"></i><b>16.4.7</b> GitHub</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>16.5</b> Round up of resources available</a></li>
<li class="chapter" data-level="16.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#acknowledgements"><i class="fa fa-check"></i><b>16.6</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="bibliography-1.html"><a href="bibliography-1.html"><i class="fa fa-check"></i><b>17</b> Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="public-data-resources-and-bioconductor" class="section level1">
<h1><span class="header-section-number">5</span> 103: Public data resources and Bioconductor</h1>
<div id="instructor-names-and-contact-information" class="section level2">
<h2><span class="header-section-number">5.1</span> Instructor names and contact information</h2>
<ul>
<li>Levi Waldron &lt;levi.waldron at sph.cuny.edu&gt; (City University of New York, New York, NY, USA)</li>
<li>Benjamin Haibe-Kains &lt;benjamin.haibe.kains at utoronto.ca&gt; (Princess Margaret Cancer Center, Toronto, Canada)</li>
<li>Sean Davis <sdavis2 at mail.nih.gov> (Center for Cancer Research, National Cancer Institute, National Institutes of Health, Bethesda, MD, USA)</li>
</ul>
</div>
<div id="syllabus" class="section level2">
<h2><span class="header-section-number">5.2</span> Syllabus</h2>
<div id="workshop-description-2" class="section level3">
<h3><span class="header-section-number">5.2.1</span> Workshop Description</h3>
<p>The goal of this workshop is to introduce Bioconductor packages for finding, accessing, and using large-scale public data resources including the Gene Expression Omnibus <a href="https://www.ncbi.nlm.nih.gov/geo">GEO</a>, Sequence Read Archive <a href="https://www.ncbi.nlm.nih.gov/sra">SRA</a>, the Genomic Data Commons <a href="https://portal.gdc.cancer.gov/">GDC</a>, and Bioconductor-hosted curated data resources for metagenomics, pharmacogenomics <a href="http://pharmacodb.ca/">PharmacoDB</a>, and The Cancer Genome Atlas.</p>
</div>
<div id="pre-requisites-2" class="section level3">
<h3><span class="header-section-number">5.2.2</span> Pre-requisites</h3>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Familiarity with the ExpressionSet and SummarizedExperiment classes</li>
<li>Basic familiarity with omics technologies such as microarray and NGS sequencing</li>
</ul>
<p>Interested students can prepare by reviewing vignettes of the packages listed in R/Bioconductor packages used to gain background on aspects of interest to them.</p>
<p>Some more general background on these resources is published in <span class="citation">Kannan et al. (2016)</span>.</p>
</div>
<div id="workshop-participation-2" class="section level3">
<h3><span class="header-section-number">5.2.3</span> Workshop Participation</h3>
<p>Each component will include runnable examples of typical usage that students are encouraged to run during demonstration of that component.</p>
</div>
<div id="rbioconductor-packages-used" class="section level3">
<h3><span class="header-section-number">5.2.4</span> R/Bioconductor packages used</h3>
<ul>
<li><em><a href="http://bioconductor.org/packages/GEOquery">GEOquery</a></em>: Access to the NCBI Gene Expression Omnibus (GEO), a public repository of gene expression (primarily microarray) data.</li>
<li><em><a href="http://bioconductor.org/packages/GenomicDataCommons">GenomicDataCommons</a></em>: Access to the NIH / NCI Genomic Data Commons RESTful service.</li>
<li><em><a href="https://github.com/seandavi/SRAdbV2">SRAdbV2</a></em>: A compilation of metadata from the NCBI Sequence Read Archive, the largest public repository of sequencing data from the next generation of sequencing platforms, and tools</li>
<li><em><a href="http://bioconductor.org/packages/curatedTCGAData">curatedTCGAData</a></em>: Curated data from The Cancer Genome Atlas (TCGA) as MultiAssayExperiment Objects</li>
<li><em><a href="http://bioconductor.org/packages/curatedMetagenomicData">curatedMetagenomicData</a></em>: Curated metagenomic data of the human microbiome</li>
<li><em><a href="http://bioconductor.org/packages/HMP16SData">HMP16SData</a></em>: Curated metagenomic data of the human microbiome</li>
<li><em><a href="http://bioconductor.org/packages/PharmacoGx">PharmacoGx</a></em>: Curated large-scale preclinical pharmacogenomic data and basic analysis tools</li>
</ul>
</div>
<div id="time-outline-2" class="section level3">
<h3><span class="header-section-number">5.2.5</span> Time outline</h3>
<p>This is a 1h45m workshop.</p>
<table>
<thead>
<tr class="header">
<th>Activity</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Overview</td>
<td>10m</td>
</tr>
<tr class="even">
<td>GEOquery</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>GenomicDataCommons</td>
<td>20m</td>
</tr>
<tr class="even">
<td>Sequence Read Archive</td>
<td>20m</td>
</tr>
<tr class="odd">
<td>curatedTCGAData</td>
<td>10m</td>
</tr>
<tr class="even">
<td>curatedMetagenomicData and HMP16SData</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>PharmacoGx</td>
<td>20m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives-3" class="section level3">
<h3><span class="header-section-number">5.2.6</span> Workshop goals and objectives</h3>
<p>Bioconductor provides access to significant amounts of publicly available experimental data. This workshop introduces students to Bioconductor interfaces to the NCBIs Gene Expression Omnibus, Genomic Data Commons, Sequence Read Archive and PharmacoDB. It additionally introduces curated resources providing The Cancer Genome Atlas, the Human Microbiome Project and other microbiome studies, and major pharmacogenomic studies, as native Bioconductor objects ready for analysis and comparison to in-house datasets.</p>
</div>
<div id="learning-goals-3" class="section level3">
<h3><span class="header-section-number">5.2.7</span> Learning goals</h3>
<ul>
<li>search NCBI resources for publicly available omics data</li>
<li>quickly use data from the TCGA and the Human Microbiome Project</li>
</ul>
</div>
<div id="learning-objectives-3" class="section level3">
<h3><span class="header-section-number">5.2.8</span> Learning objectives</h3>
<ul>
<li>find and download processed microarray and RNA-seq datasets from the Gene Expression Omnibus</li>
<li>find and download omics data from the Genomic Data Commons and Sequence Read Archive</li>
<li>download and manipulate data from The Cancer Genome Atlas and Human Microbiome Project</li>
<li>download and explore pharmacogenomics data</li>
</ul>
</div>
</div>
<div id="overview-1" class="section level2">
<h2><span class="header-section-number">5.3</span> Overview</h2>
<p>Before proceeding, ensure that the following packages are installed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">required_pkgs =<span class="st"> </span><span class="kw">c</span>(
  <span class="st">&quot;TCGAbiolinks&quot;</span>, 
  <span class="st">&quot;GEOquery&quot;</span>, 
  <span class="st">&quot;GenomicDataCommons&quot;</span>,
  <span class="st">&quot;limma&quot;</span>,
  <span class="st">&quot;curatedTCGAData&quot;</span>,
  <span class="st">&quot;recount&quot;</span>,
  <span class="st">&quot;curatedMetagenomicData&quot;</span>,
  <span class="st">&quot;phyloseq&quot;</span>,
  <span class="st">&quot;HMP16SData&quot;</span>,
  <span class="st">&quot;caTools&quot;</span>,
  <span class="st">&quot;piano&quot;</span>,
  <span class="st">&quot;isa&quot;</span>,
  <span class="st">&quot;VennDiagram&quot;</span>,
  <span class="st">&quot;downloader&quot;</span>,
    <span class="st">&quot;gdata&quot;</span>,
    <span class="st">&quot;AnnotationDbi&quot;</span>,
    <span class="st">&quot;hgu133a.db&quot;</span>,
  <span class="st">&quot;PharmacoGx&quot;</span>)
BiocManager<span class="op">::</span><span class="kw">install</span>(required_pkgs)</code></pre></div>
</div>
<div id="geoquery" class="section level2">
<h2><span class="header-section-number">5.4</span> GEOquery</h2>
<p><span class="citation">(Davis and Meltzer 2007)</span></p>
<p>The NCBI Gene Expression Omnibus (GEO) serves as a public repository for a wide range of high-throughput experimental data. These data include single and dual channel microarray-based experiments measuring mRNA, genomic DNA, and protein abundance, as well as non-array techniques such as serial analysis of gene expression (SAGE), mass spectrometry proteomic data, and high-throughput sequencing data. The <em><a href="http://bioconductor.org/packages/GEOquery">GEOquery</a></em> package <span class="citation">(Davis and Meltzer 2007)</span> forms a bridge between this public repository and the analysis capabilities in Bioconductor.</p>
<div id="overview-of-geo" class="section level3">
<h3><span class="header-section-number">5.4.1</span> Overview of GEO</h3>
<p>At the most basic level of organization of GEO, there are four basic entity types. The first three (Sample, Platform, and Series) are supplied by users; the fourth, the dataset, is compiled and curated by GEO staff from the user-submitted data. See <a href="https://www.ncbi.nlm.nih.gov/geo/">the GEO home page</a> for more information.</p>
<div id="platforms" class="section level4">
<h4><span class="header-section-number">5.4.1.1</span> Platforms</h4>
<p>A Platform record describes the list of elements on the array (e.g., cDNAs, oligonucleotide probesets, ORFs, antibodies) or the list of elements that may be detected and quantified in that experiment (e.g., SAGE tags, peptides). Each Platform record is assigned a unique and stable GEO accession number (GPLxxx). A Platform may reference many Samples that have been submitted by multiple submitters.</p>
</div>
<div id="samples" class="section level4">
<h4><span class="header-section-number">5.4.1.2</span> Samples</h4>
<p>A Sample record describes the conditions under which an individual Sample was handled, the manipulations it underwent, and the abundance measurement of each element derived from it. Each Sample record is assigned a unique and stable GEO accession number (GSMxxx). A Sample entity must reference only one Platform and may be included in multiple Series.</p>
</div>
<div id="series" class="section level4">
<h4><span class="header-section-number">5.4.1.3</span> Series</h4>
<p>A Series record defines a set of related Samples considered to be part of a group, how the Samples are related, and if and how they are ordered. A Series provides a focal point and description of the experiment as a whole. Series records may also contain tables describing extracted data, summary conclusions, or analyses. Each Series record is assigned a unique and stable GEO accession number (GSExxx). Series records are available in a couple of formats which are handled by GEOquery independently. The smaller and new GSEMatrix files are quite fast to parse; a simple flag is used by GEOquery to choose to use GSEMatrix files (see below).</p>
</div>
<div id="datasets" class="section level4">
<h4><span class="header-section-number">5.4.1.4</span> Datasets</h4>
<p>GEO DataSets (GDSxxx) are curated sets of GEO Sample data. There are hundreds of GEO datasets available, but GEO discontinued creating GDS records several years ago. We mention them here for completeness only.</p>
</div>
</div>
<div id="getting-started-using-geoquery" class="section level3">
<h3><span class="header-section-number">5.4.2</span> Getting Started using GEOquery</h3>
<p>Getting data from GEO is really quite easy. There is only one command that is needed, <code>getGEO</code>. This one function interprets its input to determine how to get the data from GEO and then parse the data into useful R data structures.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(GEOquery)</code></pre></div>
<p>With the library loaded, we are free to access any GEO accession.</p>
</div>
<div id="use-case-mds-plot-of-cancer-data" class="section level3">
<h3><span class="header-section-number">5.4.3</span> Use case: MDS plot of cancer data</h3>
<p>The data we are going to access are from <a href="https://doi.org/10.1158/1055-9965.EPI-17-0461">this paper</a>.</p>
<blockquote>
<p>Background: The tumor microenvironment is an important factor in cancer immunotherapy response. To further understand how a tumor affects the local immune system, we analyzed immune gene expression differences between matching normal and tumor tissue.Methods: We analyzed public and new gene expression data from solid cancers and isolated immune cell populations. We also determined the correlation between CD8, FoxP3 IHC, and our gene signatures.Results: We observed that regulatory T cells (Tregs) were one of the main drivers of immune gene expression differences between normal and tumor tissue. A tumor-specific CD8 signature was slightly lower in tumor tissue compared with normal of most (12 of 16) cancers, whereas a Treg signature was higher in tumor tissue of all cancers except liver. Clustering by Treg signature found two groups in colorectal cancer datasets. The high Treg cluster had more samples that were consensus molecular subtype 1/4, right-sided, and microsatellite-instable, compared with the low Treg cluster. Finally, we found that the correlation between signature and IHC was low in our small dataset, but samples in the high Treg cluster had significantly more CD8+ and FoxP3+ cells compared with the low Treg cluster.Conclusions: Treg gene expression is highly indicative of the overall tumor immune environment.Impact: In comparison with the consensus molecular subtype and microsatellite status, the Treg signature identifies more colorectal tumors with high immune activation that may benefit from cancer immunotherapy.</p>
</blockquote>
<p>In this little exercise, we will:</p>
<ol style="list-style-type: decimal">
<li>Access public omics data using the GEOquery package</li>
<li>Convert the public omics data to a <code>SummarizedExperiment</code> object.</li>
<li>Perform a simple unsupervised analysis to visualize these public data.</li>
</ol>
<p>Use the <a href="https://bioconductor.org/packages/GEOquery">GEOquery</a> package to fetch data about <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE103512">GSE103512</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gse =<span class="st"> </span><span class="kw">getGEO</span>(<span class="st">&quot;GSE103512&quot;</span>)[[<span class="dv">1</span>]]</code></pre></div>
<p>Note that <code>getGEO</code>, when used to retrieve <em>GSE</em> records, returns a list. The members of the list each represent one <em>GEO Platform</em>, since each <em>GSE</em> record can contain multiple related datasets (eg., gene expression and DNA methylation). In this case, the list is of length one, but it is still necessary to grab the first elment.</p>
<p>The first stepa detailis to convert from the older Bioconductor data structure (GEOquery was written in 2007), the <code>ExpressionSet</code>, to the newer <code>SummarizedExperiment</code>. One line suffices.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(SummarizedExperiment)
se =<span class="st"> </span><span class="kw">as</span>(gse, <span class="st">&quot;SummarizedExperiment&quot;</span>)</code></pre></div>
<p>Examine two variables of interest, cancer type and tumor/normal status.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">with</span>(<span class="kw">colData</span>(se),<span class="kw">table</span>(<span class="st">`</span><span class="dt">cancer.type.ch1</span><span class="st">`</span>,<span class="st">`</span><span class="dt">normal.ch1</span><span class="st">`</span>))
<span class="co">#&gt;                normal.ch1</span>
<span class="co">#&gt; cancer.type.ch1 no yes</span>
<span class="co">#&gt;           BC    65  10</span>
<span class="co">#&gt;           CRC   57  12</span>
<span class="co">#&gt;           NSCLC 60   9</span>
<span class="co">#&gt;           PCA   60   7</span></code></pre></div>
<p>Filter gene expression by variance to find most informative genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sds =<span class="st"> </span><span class="kw">apply</span>(<span class="kw">assay</span>(se, <span class="st">&#39;exprs&#39;</span>),<span class="dv">1</span>,sd)
dat =<span class="st"> </span><span class="kw">assay</span>(se, <span class="st">&#39;exprs&#39;</span>)[<span class="kw">order</span>(sds,<span class="dt">decreasing =</span> <span class="ot">TRUE</span>)[<span class="dv">1</span><span class="op">:</span><span class="dv">500</span>],]</code></pre></div>
<p>Perform <a href="https://en.wikipedia.org/wiki/Multidimensional_scaling">multidimensional scaling</a> and prepare for plotting. We will be using ggplot2, so we need to make a data.frame before plotting.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mdsvals =<span class="st"> </span><span class="kw">cmdscale</span>(<span class="kw">dist</span>(<span class="kw">t</span>(dat)))
mdsvals =<span class="st"> </span><span class="kw">as.data.frame</span>(mdsvals)
mdsvals<span class="op">$</span>Type=<span class="kw">factor</span>(<span class="kw">colData</span>(se)[,<span class="st">&#39;cancer.type.ch1&#39;</span>])
mdsvals<span class="op">$</span>Normal =<span class="st"> </span><span class="kw">factor</span>(<span class="kw">colData</span>(se)[,<span class="st">&#39;normal.ch1&#39;</span>])
<span class="kw">head</span>(mdsvals)
<span class="co">#&gt;                   V1        V2 Type Normal</span>
<span class="co">#&gt; GSM2772660  8.531331 -18.57115   BC     no</span>
<span class="co">#&gt; GSM2772661  8.991591 -13.63764   BC     no</span>
<span class="co">#&gt; GSM2772662 10.788973 -13.48403   BC     no</span>
<span class="co">#&gt; GSM2772663  3.127105 -19.13529   BC     no</span>
<span class="co">#&gt; GSM2772664 13.056599 -13.88711   BC     no</span>
<span class="co">#&gt; GSM2772665  7.903717 -13.24731   BC     no</span></code></pre></div>
<p>And do the plot.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ggplot2)
<span class="kw">ggplot</span>(mdsvals, <span class="kw">aes</span>(<span class="dt">x=</span>V1,<span class="dt">y=</span>V2,<span class="dt">shape=</span>Normal,<span class="dt">color=</span>Type)) <span class="op">+</span><span class="st"> </span>
<span class="st">    </span><span class="kw">geom_point</span>( <span class="dt">alpha=</span><span class="fl">0.6</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">text=</span><span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">18</span>))</code></pre></div>
<p><img src="103_Waldron_PublicData_files/figure-html/mdsplot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="accessing-raw-data-from-geo" class="section level3">
<h3><span class="header-section-number">5.4.4</span> Accessing Raw Data from GEO</h3>
<p>NCBI GEO accepts (but has not always required) raw data such as .CEL files, .CDF files, images, etc. It is also not uncommon for some RNA-seq or other sequencing datasets to supply <em>only</em> raw data (with accompanying sample information, of course), necessitating Sometimes, it is useful to get quick access to such data. A single function, <code>getGEOSuppFiles</code>, can take as an argument a GEO accession and will download all the raw data associate with that accession. By default, the function will create a directory in the current working directory to store the raw data for the chosen GEO accession.</p>
</div>
</div>
<div id="genomicdatacommons" class="section level2">
<h2><span class="header-section-number">5.5</span> GenomicDataCommons</h2>
<p>From the <a href="https://gdc.nci.nih.gov/about-gdc">Genomic Data Commons (GDC) website</a>:</p>
<blockquote>
<p>The National Cancer Institutes (NCIs) Genomic Data Commons (GDC) is a data sharing platform that promotes precision medicine in oncology. It is not just a database or a tool; it is an expandable knowledge network supporting the import and standardization of genomic and clinical data from cancer research programs. The GDC contains NCI-generated data from some of the largest and most comprehensive cancer genomic datasets, including The Cancer Genome Atlas (TCGA) and Therapeutically Applicable Research to Generate Effective Therapies (TARGET). For the first time, these datasets have been harmonized using a common set of bioinformatics pipelines, so that the data can be directly compared. As a growing knowledge system for cancer, the GDC also enables researchers to submit data, and harmonizes these data for import into the GDC. As more researchers add clinical and genomic data to the GDC, it will become an even more powerful tool for making discoveries about the molecular basis of cancer that may lead to better care for patients.</p>
</blockquote>
<p>The <a href="https://gdc.cancer.gov/developers/gdc-data-model/gdc-data-model-components">data model for the GDC is complex</a>, but it worth a quick overview and a graphical representation is included here.</p>
<div class="figure">
<img src="Waldron_PublicData/all_nodes_040318.png" alt="The data model is encoded as a so-called property graph. Nodes represent entities such as Projects, Cases, Diagnoses, Files (various kinds), and Annotations. The relationships between these entities are maintained as edges. Both nodes and edges may have Properties that supply instance details." />
<p class="caption">The data model is encoded as a so-called property graph. Nodes represent entities such as Projects, Cases, Diagnoses, Files (various kinds), and Annotations. The relationships between these entities are maintained as edges. Both nodes and edges may have Properties that supply instance details.</p>
</div>
<p>The GDC API exposes these nodes and edges in a somewhat simplified set of <a href="https://en.wikipedia.org/wiki/Representational_state_transfer">RESTful</a> endpoints.</p>
<div id="quickstart" class="section level3">
<h3><span class="header-section-number">5.5.1</span> Quickstart</h3>
<p>This quickstart section is just meant to show basic functionality. More details of functionality are included further on in this vignette and in function-specific help.</p>
<p>To report bugs or problems, either <a href="https://github.com/Bioconductor/GenomicDataCommons/issues">submit a new issue</a> or submit a <code>bug.report(package='GenomicDataCommons')</code> from within R (which will redirect you to the new issue on GitHub).</p>
<div id="installation" class="section level4">
<h4><span class="header-section-number">5.5.1.1</span> Installation</h4>
<p>Installation of the <em><a href="http://bioconductor.org/packages/GenomicDataCommons">GenomicDataCommons</a></em> package is identical to installation of other Bioconductor packages.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">&#39;BiocManager&#39;</span>)
BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&#39;GenomicDataCommons&#39;</span>)</code></pre></div>
<p>After installation, load the library in order to use it.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(GenomicDataCommons)</code></pre></div>
</div>
<div id="check-connectivity-and-status" class="section level4">
<h4><span class="header-section-number">5.5.1.2</span> Check connectivity and status</h4>
<p>The <em><a href="http://bioconductor.org/packages/GenomicDataCommons">GenomicDataCommons</a></em> package relies on having network connectivity. In addition, the NCI GDC API must also be operational and not under maintenance. Checking <code>status</code> can be used to check this connectivity and functionality.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">GenomicDataCommons<span class="op">::</span><span class="kw">status</span>()
<span class="co">#&gt; $commit</span>
<span class="co">#&gt; [1] &quot;e9e20d6f97f2bf6dd3b3261e36ead57c56a4c7cc&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $data_release</span>
<span class="co">#&gt; [1] &quot;Data Release 12.0 - June 13, 2018&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $status</span>
<span class="co">#&gt; [1] &quot;OK&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $tag</span>
<span class="co">#&gt; [1] &quot;1.14.1&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $version</span>
<span class="co">#&gt; [1] 1</span></code></pre></div>
</div>
<div id="find-data" class="section level4">
<h4><span class="header-section-number">5.5.1.3</span> Find data</h4>
<p>The following code builds a <code>manifest</code> that can be used to guide the download of raw data. Here, filtering finds gene expression files quantified as raw counts using <code>HTSeq</code> from ovarian cancer patients.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ge_manifest =<span class="st"> </span><span class="kw">files</span>() <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">filter</span>( <span class="op">~</span><span class="st"> </span>cases.project.project_id <span class="op">==</span><span class="st"> &#39;TCGA-OV&#39;</span> <span class="op">&amp;</span>
<span class="st">                </span>type <span class="op">==</span><span class="st"> &#39;gene_expression&#39;</span> <span class="op">&amp;</span>
<span class="st">                </span>analysis.workflow_type <span class="op">==</span><span class="st"> &#39;HTSeq - Counts&#39;</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">manifest</span>()</code></pre></div>
</div>
<div id="download-data" class="section level4">
<h4><span class="header-section-number">5.5.1.4</span> Download data</h4>
<p>After the 379 gene expression files specified in the query above. Using multiple processes to do the download very significantly speeds up the transfer in many cases. On a standard 1Gb connection, the following completes in about 30 seconds. The first time the data are downloaded, R will ask to create a cache directory (see <code>?gdc_cache</code> for details of setting and interacting with the cache). Resulting downloaded files will be stored in the cache directory. Future access to the same files will be directly from the cache, alleviating multiple downloads.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fnames =<span class="st"> </span><span class="kw">lapply</span>(ge_manifest<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>],gdcdata)</code></pre></div>
<p>If the download had included controlled-access data, the download above would have needed to include a <code>token</code>. Details are available in <a href="public-data-resources-and-bioconductor.html#authentication">the authentication section below</a>.</p>
</div>
<div id="metadata-queries" class="section level4">
<h4><span class="header-section-number">5.5.1.5</span> Metadata queries</h4>
<p>The <em><a href="http://bioconductor.org/packages/GenomicDataCommons">GenomicDataCommons</a></em> can access the significant clinical, demographic, biospecimen, and annotation information contained in the NCI GDC.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">expands =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;diagnoses&quot;</span>,<span class="st">&quot;annotations&quot;</span>,
             <span class="st">&quot;demographic&quot;</span>,<span class="st">&quot;exposures&quot;</span>)
projResults =<span class="st"> </span><span class="kw">projects</span>() <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">results</span>(<span class="dt">size=</span><span class="dv">10</span>)
<span class="kw">str</span>(projResults,<span class="dt">list.len=</span><span class="dv">5</span>)
<span class="co">#&gt; List of 8</span>
<span class="co">#&gt;  $ dbgap_accession_number: chr [1:10] &quot;phs001179&quot; &quot;phs000470&quot; NA NA ...</span>
<span class="co">#&gt;  $ disease_type          :List of 10</span>
<span class="co">#&gt;   ..$ FM-AD    : chr [1:23] &quot;Germ Cell Neoplasms&quot; &quot;Acinar Cell Neoplasms&quot; &quot;Miscellaneous Tumors&quot; &quot;Thymic Epithelial Neoplasms&quot; ...</span>
<span class="co">#&gt;   ..$ TARGET-RT: chr &quot;Rhabdoid Tumor&quot;</span>
<span class="co">#&gt;   ..$ TCGA-UCS : chr &quot;Uterine Carcinosarcoma&quot;</span>
<span class="co">#&gt;   ..$ TCGA-LUSC: chr &quot;Lung Squamous Cell Carcinoma&quot;</span>
<span class="co">#&gt;   ..$ TCGA-BRCA: chr &quot;Breast Invasive Carcinoma&quot;</span>
<span class="co">#&gt;   .. [list output truncated]</span>
<span class="co">#&gt;  $ released              : logi [1:10] TRUE TRUE TRUE TRUE TRUE TRUE ...</span>
<span class="co">#&gt;  $ state                 : chr [1:10] &quot;open&quot; &quot;open&quot; &quot;open&quot; &quot;open&quot; ...</span>
<span class="co">#&gt;  $ primary_site          :List of 10</span>
<span class="co">#&gt;   ..$ FM-AD    : chr [1:42] &quot;Kidney&quot; &quot;Testis&quot; &quot;Unknown&quot; &quot;Other and unspecified parts of biliary tract&quot; ...</span>
<span class="co">#&gt;   ..$ TARGET-RT: chr &quot;Kidney&quot;</span>
<span class="co">#&gt;   ..$ TCGA-UCS : chr &quot;Uterus&quot;</span>
<span class="co">#&gt;   ..$ TCGA-LUSC: chr &quot;Lung&quot;</span>
<span class="co">#&gt;   ..$ TCGA-BRCA: chr &quot;Breast&quot;</span>
<span class="co">#&gt;   .. [list output truncated]</span>
<span class="co">#&gt;   [list output truncated]</span>
<span class="co">#&gt;  - attr(*, &quot;row.names&quot;)= int [1:10] 1 2 3 4 5 6 7 8 9 10</span>
<span class="co">#&gt;  - attr(*, &quot;class&quot;)= chr [1:3] &quot;GDCprojectsResults&quot; &quot;GDCResults&quot; &quot;list&quot;</span>
<span class="kw">names</span>(projResults)
<span class="co">#&gt; [1] &quot;dbgap_accession_number&quot; &quot;disease_type&quot;          </span>
<span class="co">#&gt; [3] &quot;released&quot;               &quot;state&quot;                 </span>
<span class="co">#&gt; [5] &quot;primary_site&quot;           &quot;project_id&quot;            </span>
<span class="co">#&gt; [7] &quot;id&quot;                     &quot;name&quot;</span>
<span class="co"># or listviewer::jsonedit(clinResults)</span></code></pre></div>
</div>
</div>
<div id="basic-design" class="section level3">
<h3><span class="header-section-number">5.5.2</span> Basic design</h3>
<p>This package design is meant to have some similarities to the hadleyverse approach of dplyr. Roughly, the functionality for finding and accessing files and metadata can be divided into:</p>
<ol style="list-style-type: decimal">
<li>Simple query constructors based on GDC API endpoints.</li>
<li>A set of verbs that when applied, adjust filtering, field selection, and faceting (fields for aggregation) and result in a new query object (an endomorphism)</li>
<li>A set of verbs that take a query and return results from the GDC</li>
</ol>
<p>In addition, there are exhiliary functions for asking the GDC API for information about available and default fields, slicing BAM files, and downloading actual data files. Here is an overview of functionality<a href="#fn3" class="footnoteRef" id="fnref3"><sup>3</sup></a>.</p>
<ul>
<li>Creating a query
<ul>
<li><code>projects()</code></li>
<li><code>cases()</code></li>
<li><code>files()</code></li>
<li><code>annotations()</code></li>
</ul></li>
<li>Manipulating a query
<ul>
<li><code>filter()</code></li>
<li><code>facet()</code></li>
<li><code>select()</code></li>
</ul></li>
<li>Introspection on the GDC API fields
<ul>
<li><code>mapping()</code></li>
<li><code>available_fields()</code></li>
<li><code>default_fields()</code></li>
<li><code>grep_fields()</code></li>
<li><code>field_picker()</code></li>
<li><code>available_values()</code></li>
<li><code>available_expand()</code></li>
</ul></li>
<li>Executing an API call to retrieve query results
<ul>
<li><code>results()</code></li>
<li><code>count()</code></li>
<li><code>response()</code></li>
</ul></li>
<li>Raw data file downloads
<ul>
<li><code>gdcdata()</code></li>
<li><code>transfer()</code></li>
<li><code>gdc_client()</code></li>
</ul></li>
<li>Summarizing and aggregating field values (faceting)
<ul>
<li><code>aggregations()</code></li>
</ul></li>
<li>Authentication
<ul>
<li><code>gdc_token()</code></li>
</ul></li>
<li>BAM file slicing
<ul>
<li><code>slicing()</code></li>
</ul></li>
</ul>
</div>
<div id="usage" class="section level3">
<h3><span class="header-section-number">5.5.3</span> Usage</h3>
<p>There are two main classes of operations when working with the NCI GDC.</p>
<ol style="list-style-type: decimal">
<li><a href="public-data-resources-and-bioconductor.html#querying-metadata">Querying metadata and finding data files</a> (e.g., finding all gene expression quantifications data files for all colon cancer patients).</li>
<li><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download">Transferring raw or processed data</a> from the GDC to another computer (e.g., downloading raw or processed data)</li>
</ol>
<p>Both classes of operation are reviewed in detail in the following sections.</p>
</div>
</div>
<div id="querying-metadata" class="section level2">
<h2><span class="header-section-number">5.6</span> Querying metadata</h2>
<p>Vast amounts of metadata about cases (patients, basically), files, projects, and so-called annotations are available via the NCI GDC API. Typically, one will want to query metadata to either focus in on a set of files for download or transfer <em>or</em> to perform so-called aggregations (pivot-tables, facets, similar to the R <code>table()</code> functionality).</p>
<p>Querying metadata starts with <a href="public-data-resources-and-bioconductor.html#creating-a-query">creating a blank query</a>. One will often then want to <a href="public-data-resources-and-bioconductor.html#filtering"><code>filter</code></a> the query to limit results prior to <a href="public-data-resources-and-bioconductor.html#retrieving-results">retrieving results</a>. The GenomicDataCommons package has <a href="public-data-resources-and-bioconductor.html#fields-and-values">helper functions for listing fields</a> that are available for filtering.</p>
<p>In addition to fetching results, the GDC API allows <a href="public-data-resources-and-bioconductor.html#facets-and-aggregation">faceting, or aggregating,</a>, useful for compiling reports, generating dashboards, or building user interfaces to GDC data (see GDC web query interface for a non-R-based example).</p>
<div id="creating-a-query" class="section level4">
<h4><span class="header-section-number">5.6.0.1</span> Creating a query</h4>
<p>The <em><a href="http://bioconductor.org/packages/GenomicDataCommons">GenomicDataCommons</a></em> package accesses the <em>same</em> API as the <em>GDC</em> website. Therefore, a useful approach, particularly for beginning users is to examine the filters available on the <a href="https://portal.gdc.cancer.gov/repository">GDC repository pages</a> to find appropriate filtering criteria. From there, converting those checkboxes to a GenomicDataCommons <code>query()</code> is relatively straightforward. Note that only a small subset of the <code>available_fields()</code> are available by default on the website.</p>
<div class="figure">
<img src="Waldron_PublicData/gdc_filtering.png" alt="A screenshot of an example query of the GDC repository portal." />
<p class="caption">A screenshot of an example query of the GDC repository portal.</p>
</div>
<p>A query of the GDC starts its life in R. Queries follow the four metadata endpoints available at the GDC. In particular, there are four convenience functions that each create <code>GDCQuery</code> objects (actually, specific subclasses of <code>GDCQuery</code>):</p>
<ul>
<li><code>projects()</code></li>
<li><code>cases()</code></li>
<li><code>files()</code></li>
<li><code>annotations()</code></li>
</ul>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pquery =<span class="st"> </span><span class="kw">projects</span>()</code></pre></div>
<p>The <code>pquery</code> object is now an object of (S3) class, <code>GDCQuery</code> (and <code>gdc_projects</code> and <code>list</code>). The object contains the following elements:</p>
<ul>
<li>fields: This is a character vector of the fields that will be returned when we <a href="public-data-resources-and-bioconductor.html#retrieving-results">retrieve data</a>. If no fields are specified to, for example, the <code>projects()</code> function, the default fields from the GDC are used (see <code>default_fields()</code>)</li>
<li>filters: This will contain results after calling the <a href="public-data-resources-and-bioconductor.html#filtering"><code>filter()</code> method</a> and will be used to filter results on <a href="public-data-resources-and-bioconductor.html#retrieving-results">retrieval</a>.</li>
<li>facets: A character vector of field names that will be used for <a href="public-data-resources-and-bioconductor.html#facets-and-aggregation">aggregating data</a> in a call to <code>aggregations()</code>.</li>
<li>archive: One of either default or <a href="https://gdc-portal.nci.nih.gov/legacy-archive/">legacy</a>.</li>
<li>token: A character(1) token from the GDC. See <a href="public-data-resources-and-bioconductor.html#authentication">the authentication section</a> for details, but note that, in general, the token is not necessary for metadata query and retrieval, only for actual data download.</li>
</ul>
<p>Looking at the actual object (get used to using <code>str()</code>!), note that the query contains no results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(pquery)
<span class="co">#&gt; List of 5</span>
<span class="co">#&gt;  $ fields : chr [1:16] &quot;awg_review&quot; &quot;dbgap_accession_number&quot; &quot;disease_type&quot; &quot;in_review&quot; ...</span>
<span class="co">#&gt;  $ filters: NULL</span>
<span class="co">#&gt;  $ facets : NULL</span>
<span class="co">#&gt;  $ legacy : logi FALSE</span>
<span class="co">#&gt;  $ expand : NULL</span>
<span class="co">#&gt;  - attr(*, &quot;class&quot;)= chr [1:3] &quot;gdc_projects&quot; &quot;GDCQuery&quot; &quot;list&quot;</span></code></pre></div>
</div>
<div id="retrieving-results" class="section level4">
<h4><span class="header-section-number">5.6.0.2</span> Retrieving results</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#size-and-from">[ GDC pagination documentation ]</a></p>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#sort">[ GDC sorting documentation ]</a></p>
<p>With a query object available, the next step is to retrieve results from the GDC. The GenomicDataCommons package. The most basic type of results we can get is a simple <code>count()</code> of records available that satisfy the filter criteria. Note that we have not set any filters, so a <code>count()</code> here will represent all the project records publicly available at the GDC in the default archive&quot;</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pcount =<span class="st"> </span><span class="kw">count</span>(pquery)
<span class="co"># or</span>
pcount =<span class="st"> </span>pquery <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>()
pcount
<span class="co">#&gt; [1] 40</span></code></pre></div>
<p>The <code>results()</code> method will fetch actual results.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">presults =<span class="st"> </span>pquery <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">results</span>()</code></pre></div>
<p>These results are returned from the GDC in <a href="http://www.json.org/">JSON</a> format and converted into a (potentially nested) list in R. The <code>str()</code> method is useful for taking a quick glimpse of the data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">str</span>(presults)
<span class="co">#&gt; List of 8</span>
<span class="co">#&gt;  $ dbgap_accession_number: chr [1:10] &quot;phs001179&quot; &quot;phs000470&quot; NA NA ...</span>
<span class="co">#&gt;  $ disease_type          :List of 10</span>
<span class="co">#&gt;   ..$ FM-AD    : chr [1:23] &quot;Germ Cell Neoplasms&quot; &quot;Acinar Cell Neoplasms&quot; &quot;Miscellaneous Tumors&quot; &quot;Thymic Epithelial Neoplasms&quot; ...</span>
<span class="co">#&gt;   ..$ TARGET-RT: chr &quot;Rhabdoid Tumor&quot;</span>
<span class="co">#&gt;   ..$ TCGA-UCS : chr &quot;Uterine Carcinosarcoma&quot;</span>
<span class="co">#&gt;   ..$ TCGA-LUSC: chr &quot;Lung Squamous Cell Carcinoma&quot;</span>
<span class="co">#&gt;   ..$ TCGA-BRCA: chr &quot;Breast Invasive Carcinoma&quot;</span>
<span class="co">#&gt;   ..$ TCGA-SKCM: chr &quot;Skin Cutaneous Melanoma&quot;</span>
<span class="co">#&gt;   ..$ TARGET-OS: chr &quot;Osteosarcoma&quot;</span>
<span class="co">#&gt;   ..$ TCGA-THYM: chr &quot;Thymoma&quot;</span>
<span class="co">#&gt;   ..$ TARGET-WT: chr &quot;High-Risk Wilms Tumor&quot;</span>
<span class="co">#&gt;   ..$ TCGA-ESCA: chr &quot;Esophageal Carcinoma&quot;</span>
<span class="co">#&gt;  $ released              : logi [1:10] TRUE TRUE TRUE TRUE TRUE TRUE ...</span>
<span class="co">#&gt;  $ state                 : chr [1:10] &quot;open&quot; &quot;open&quot; &quot;open&quot; &quot;open&quot; ...</span>
<span class="co">#&gt;  $ primary_site          :List of 10</span>
<span class="co">#&gt;   ..$ FM-AD    : chr [1:42] &quot;Kidney&quot; &quot;Testis&quot; &quot;Unknown&quot; &quot;Other and unspecified parts of biliary tract&quot; ...</span>
<span class="co">#&gt;   ..$ TARGET-RT: chr &quot;Kidney&quot;</span>
<span class="co">#&gt;   ..$ TCGA-UCS : chr &quot;Uterus&quot;</span>
<span class="co">#&gt;   ..$ TCGA-LUSC: chr &quot;Lung&quot;</span>
<span class="co">#&gt;   ..$ TCGA-BRCA: chr &quot;Breast&quot;</span>
<span class="co">#&gt;   ..$ TCGA-SKCM: chr &quot;Skin&quot;</span>
<span class="co">#&gt;   ..$ TARGET-OS: chr &quot;Bone&quot;</span>
<span class="co">#&gt;   ..$ TCGA-THYM: chr &quot;Thymus&quot;</span>
<span class="co">#&gt;   ..$ TARGET-WT: chr &quot;Kidney&quot;</span>
<span class="co">#&gt;   ..$ TCGA-ESCA: chr &quot;Esophagus&quot;</span>
<span class="co">#&gt;  $ project_id            : chr [1:10] &quot;FM-AD&quot; &quot;TARGET-RT&quot; &quot;TCGA-UCS&quot; &quot;TCGA-LUSC&quot; ...</span>
<span class="co">#&gt;  $ id                    : chr [1:10] &quot;FM-AD&quot; &quot;TARGET-RT&quot; &quot;TCGA-UCS&quot; &quot;TCGA-LUSC&quot; ...</span>
<span class="co">#&gt;  $ name                  : chr [1:10] &quot;Foundation Medicine Adult Cancer Clinical Dataset (FM-AD)&quot; &quot;Rhabdoid Tumor&quot; &quot;Uterine Carcinosarcoma&quot; &quot;Lung Squamous Cell Carcinoma&quot; ...</span>
<span class="co">#&gt;  - attr(*, &quot;row.names&quot;)= int [1:10] 1 2 3 4 5 6 7 8 9 10</span>
<span class="co">#&gt;  - attr(*, &quot;class&quot;)= chr [1:3] &quot;GDCprojectsResults&quot; &quot;GDCResults&quot; &quot;list&quot;</span></code></pre></div>
<p>A default of only 10 records are returned. We can use the <code>size</code> and <code>from</code> arguments to <code>results()</code> to either page through results or to change the number of results. Finally, there is a convenience method, <code>results_all()</code> that will simply fetch all the available results given a query. Note that <code>results_all()</code> may take a long time and return HUGE result sets if not used carefully. Use of a combination of <code>count()</code> and <code>results()</code> to get a sense of the expected data size is probably warranted before calling <code>results_all()</code></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(<span class="kw">ids</span>(presults))
<span class="co">#&gt; [1] 10</span>
presults =<span class="st"> </span>pquery <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">results_all</span>()
<span class="kw">length</span>(<span class="kw">ids</span>(presults))
<span class="co">#&gt; [1] 40</span>
<span class="co"># includes all records</span>
<span class="kw">length</span>(<span class="kw">ids</span>(presults)) <span class="op">==</span><span class="st"> </span><span class="kw">count</span>(pquery)
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<p>Extracting subsets of results or manipulating the results into a more conventional R data structure is not easily generalizable. However, the <a href="https://github.com/hadley/purrr">purrr</a>, <a href="https://renkun.me/rlist/">rlist</a>, and <a href="https://cran.r-project.org/web/packages/data.tree/vignettes/data.tree.html">data.tree</a> packages are all potentially of interest for manipulating complex, nested list structures. For viewing the results in an interactive viewer, consider the <a href="https://github.com/timelyportfolio/listviewer">listviewer</a> package.</p>
</div>
<div id="fields-and-values" class="section level4">
<h4><span class="header-section-number">5.6.0.3</span> Fields and Values</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#fields">[ GDC <code>fields</code> documentation ]</a></p>
<p>Central to querying and retrieving data from the GDC is the ability to specify which fields to return, filtering by fields and values, and faceting or aggregating. The GenomicDataCommons package includes two simple functions, <code>available_fields()</code> and <code>default_fields()</code>. Each can operate on a character(1) endpoint name (cases, files, annotations, or projects) or a <code>GDCQuery</code> object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">default_fields</span>(<span class="st">&#39;files&#39;</span>)
<span class="co">#&gt;  [1] &quot;access&quot;                &quot;acl&quot;                  </span>
<span class="co">#&gt;  [3] &quot;batch_id&quot;              &quot;created_datetime&quot;     </span>
<span class="co">#&gt;  [5] &quot;data_category&quot;         &quot;data_format&quot;          </span>
<span class="co">#&gt;  [7] &quot;data_type&quot;             &quot;error_type&quot;           </span>
<span class="co">#&gt;  [9] &quot;experimental_strategy&quot; &quot;file_autocomplete&quot;    </span>
<span class="co">#&gt; [11] &quot;file_id&quot;               &quot;file_name&quot;            </span>
<span class="co">#&gt; [13] &quot;file_size&quot;             &quot;file_state&quot;           </span>
<span class="co">#&gt; [15] &quot;imaging_date&quot;          &quot;magnification&quot;        </span>
<span class="co">#&gt; [17] &quot;md5sum&quot;                &quot;origin&quot;               </span>
<span class="co">#&gt; [19] &quot;platform&quot;              &quot;read_pair_number&quot;     </span>
<span class="co">#&gt; [21] &quot;revision&quot;              &quot;state&quot;                </span>
<span class="co">#&gt; [23] &quot;state_comment&quot;         &quot;submitter_id&quot;         </span>
<span class="co">#&gt; [25] &quot;tags&quot;                  &quot;type&quot;                 </span>
<span class="co">#&gt; [27] &quot;updated_datetime&quot;</span>
<span class="co"># The number of fields available for files endpoint</span>
<span class="kw">length</span>(<span class="kw">available_fields</span>(<span class="st">&#39;files&#39;</span>))
<span class="co">#&gt; [1] 703</span>
<span class="co"># The first few fields available for files endpoint</span>
<span class="kw">head</span>(<span class="kw">available_fields</span>(<span class="st">&#39;files&#39;</span>))
<span class="co">#&gt; [1] &quot;access&quot;                    &quot;acl&quot;                      </span>
<span class="co">#&gt; [3] &quot;analysis.analysis_id&quot;      &quot;analysis.analysis_type&quot;   </span>
<span class="co">#&gt; [5] &quot;analysis.batch_id&quot;         &quot;analysis.created_datetime&quot;</span></code></pre></div>
<p>The fields to be returned by a query can be specified following a similar paradigm to that of the dplyr package. The <code>select()</code> function is a verb that resets the fields slot of a <code>GDCQuery</code>; note that this is not quite analogous to the dplyr <code>select()</code> verb that limits from already-present fields. We <em>completely replace</em> the fields when using <code>select()</code> on a <code>GDCQuery</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Default fields here</span>
qcases =<span class="st"> </span><span class="kw">cases</span>()
qcases<span class="op">$</span>fields
<span class="co">#&gt;  [1] &quot;aliquot_ids&quot;              &quot;analyte_ids&quot;             </span>
<span class="co">#&gt;  [3] &quot;batch_id&quot;                 &quot;case_autocomplete&quot;       </span>
<span class="co">#&gt;  [5] &quot;case_id&quot;                  &quot;created_datetime&quot;        </span>
<span class="co">#&gt;  [7] &quot;days_to_index&quot;            &quot;days_to_lost_to_followup&quot;</span>
<span class="co">#&gt;  [9] &quot;disease_type&quot;             &quot;index_date&quot;              </span>
<span class="co">#&gt; [11] &quot;lost_to_followup&quot;         &quot;portion_ids&quot;             </span>
<span class="co">#&gt; [13] &quot;primary_site&quot;             &quot;sample_ids&quot;              </span>
<span class="co">#&gt; [15] &quot;slide_ids&quot;                &quot;state&quot;                   </span>
<span class="co">#&gt; [17] &quot;submitter_aliquot_ids&quot;    &quot;submitter_analyte_ids&quot;   </span>
<span class="co">#&gt; [19] &quot;submitter_id&quot;             &quot;submitter_portion_ids&quot;   </span>
<span class="co">#&gt; [21] &quot;submitter_sample_ids&quot;     &quot;submitter_slide_ids&quot;     </span>
<span class="co">#&gt; [23] &quot;updated_datetime&quot;</span>
<span class="co"># set up query to use ALL available fields</span>
<span class="co"># Note that checking of fields is done by select()</span>
qcases =<span class="st"> </span><span class="kw">cases</span>() <span class="op">%&gt;%</span><span class="st"> </span>GenomicDataCommons<span class="op">::</span><span class="kw">select</span>(<span class="kw">available_fields</span>(<span class="st">&#39;cases&#39;</span>))
<span class="kw">head</span>(qcases<span class="op">$</span>fields)
<span class="co">#&gt; [1] &quot;case_id&quot;                   &quot;aliquot_ids&quot;              </span>
<span class="co">#&gt; [3] &quot;analyte_ids&quot;               &quot;annotations.annotation_id&quot;</span>
<span class="co">#&gt; [5] &quot;annotations.batch_id&quot;      &quot;annotations.case_id&quot;</span></code></pre></div>
<p>Finding fields of interest is such a common operation that the GenomicDataCommons includes the <code>grep_fields()</code> function and the <code>field_picker()</code> widget. See the appropriate help pages for details.</p>
</div>
<div id="facets-and-aggregation" class="section level4">
<h4><span class="header-section-number">5.6.0.4</span> Facets and aggregation</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#facets">[ GDC <code>facet</code> documentation ]</a></p>
<p>The GDC API offers a feature known as aggregation or faceting. By specifying one or more fields (of appropriate type), the GDC can return to us a count of the number of records matching each potential value. This is similar to the R <code>table</code> method. Multiple fields can be returned at once, but the GDC API does not have a cross-tabulation feature; all aggregations are only on one field at a time. Results of <code>aggregation()</code> calls come back as a list of data.frames (actually, tibbles).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># total number of files of a specific type</span>
res =<span class="st"> </span><span class="kw">files</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">facet</span>(<span class="kw">c</span>(<span class="st">&#39;type&#39;</span>,<span class="st">&#39;data_type&#39;</span>)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">aggregations</span>()
res<span class="op">$</span>type
<span class="co">#&gt;                            key doc_count</span>
<span class="co">#&gt; 1      simple_somatic_mutation     64015</span>
<span class="co">#&gt; 2   annotated_somatic_mutation     63580</span>
<span class="co">#&gt; 3                aligned_reads     45985</span>
<span class="co">#&gt; 4          copy_number_segment     44752</span>
<span class="co">#&gt; 5              gene_expression     34713</span>
<span class="co">#&gt; 6                  slide_image     30036</span>
<span class="co">#&gt; 7       biospecimen_supplement     25151</span>
<span class="co">#&gt; 8             mirna_expression     22976</span>
<span class="co">#&gt; 9          clinical_supplement     12496</span>
<span class="co">#&gt; 10      methylation_beta_value     12359</span>
<span class="co">#&gt; 11 aggregated_somatic_mutation       186</span>
<span class="co">#&gt; 12     masked_somatic_mutation       132</span></code></pre></div>
<p>Using <code>aggregations()</code> is an also easy way to learn the contents of individual fields and forms the basis for faceted search pages.</p>
</div>
<div id="filtering" class="section level4">
<h4><span class="header-section-number">5.6.0.5</span> Filtering</h4>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#filters-specifying-the-query">[ GDC <code>filtering</code> documentation ]</a></p>
<p>The GenomicDataCommons package uses a form of non-standard evaluation to specify R-like queries that are then translated into an R list. That R list is, upon calling a method that fetches results from the GDC API, translated into the appropriate JSON string. The R expression uses the formula interface as suggested by Hadley Wickham in his <a href="https://cran.r-project.org/web/packages/dplyr/vignettes/nse.html">vignette on non-standard evaluation</a></p>
<blockquote>
<p>Its best to use a formula because a formula captures both the expression to evaluate and the environment where the evaluation occurs. This is important if the expression is a mixture of variables in a data frame and objects in the local environment [for example].</p>
</blockquote>
<p>For the user, these details will not be too important except to note that a filter expression must begin with a ~.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">qfiles =<span class="st"> </span><span class="kw">files</span>()
qfiles <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>() <span class="co"># all files</span>
<span class="co">#&gt; [1] 356381</span></code></pre></div>
<p>To limit the file type, we can refer back to the <a href="public-data-resources-and-bioconductor.html#facets-and-aggregation">section on faceting</a> to see the possible values for the file field type. For example, to filter file results to only gene_expression files, we simply specify a filter.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">qfiles =<span class="st"> </span><span class="kw">files</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">~</span><span class="st"> </span>type <span class="op">==</span><span class="st"> &#39;gene_expression&#39;</span>)
<span class="co"># here is what the filter looks like after translation</span>
<span class="kw">str</span>(<span class="kw">get_filter</span>(qfiles))
<span class="co">#&gt; List of 2</span>
<span class="co">#&gt;  $ op     : &#39;scalar&#39; chr &quot;=&quot;</span>
<span class="co">#&gt;  $ content:List of 2</span>
<span class="co">#&gt;   ..$ field: chr &quot;type&quot;</span>
<span class="co">#&gt;   ..$ value: chr &quot;gene_expression&quot;</span></code></pre></div>
<p>What if we want to create a filter based on the project (TCGA-OVCA, for example)? Well, we have a couple of possible ways to discover available fields. The first is based on base R functionality and some intuition.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">grep</span>(<span class="st">&#39;pro&#39;</span>,<span class="kw">available_fields</span>(<span class="st">&#39;files&#39;</span>),<span class="dt">value=</span><span class="ot">TRUE</span>)
<span class="co">#&gt;  [1] &quot;cases.diagnoses.progression_free_survival&quot;               </span>
<span class="co">#&gt;  [2] &quot;cases.diagnoses.progression_free_survival_event&quot;         </span>
<span class="co">#&gt;  [3] &quot;cases.diagnoses.progression_or_recurrence&quot;               </span>
<span class="co">#&gt;  [4] &quot;cases.project.awg_review&quot;                                </span>
<span class="co">#&gt;  [5] &quot;cases.project.dbgap_accession_number&quot;                    </span>
<span class="co">#&gt;  [6] &quot;cases.project.disease_type&quot;                              </span>
<span class="co">#&gt;  [7] &quot;cases.project.in_review&quot;                                 </span>
<span class="co">#&gt;  [8] &quot;cases.project.intended_release_date&quot;                     </span>
<span class="co">#&gt;  [9] &quot;cases.project.is_legacy&quot;                                 </span>
<span class="co">#&gt; [10] &quot;cases.project.name&quot;                                      </span>
<span class="co">#&gt; [11] &quot;cases.project.primary_site&quot;                              </span>
<span class="co">#&gt; [12] &quot;cases.project.program.dbgap_accession_number&quot;            </span>
<span class="co">#&gt; [13] &quot;cases.project.program.name&quot;                              </span>
<span class="co">#&gt; [14] &quot;cases.project.program.program_id&quot;                        </span>
<span class="co">#&gt; [15] &quot;cases.project.project_id&quot;                                </span>
<span class="co">#&gt; [16] &quot;cases.project.releasable&quot;                                </span>
<span class="co">#&gt; [17] &quot;cases.project.release_requested&quot;                         </span>
<span class="co">#&gt; [18] &quot;cases.project.released&quot;                                  </span>
<span class="co">#&gt; [19] &quot;cases.project.request_submission&quot;                        </span>
<span class="co">#&gt; [20] &quot;cases.project.state&quot;                                     </span>
<span class="co">#&gt; [21] &quot;cases.project.submission_enabled&quot;                        </span>
<span class="co">#&gt; [22] &quot;cases.samples.days_to_sample_procurement&quot;                </span>
<span class="co">#&gt; [23] &quot;cases.samples.method_of_sample_procurement&quot;              </span>
<span class="co">#&gt; [24] &quot;cases.samples.portions.slides.number_proliferating_cells&quot;</span>
<span class="co">#&gt; [25] &quot;cases.tissue_source_site.project&quot;</span></code></pre></div>
<p>Interestingly, the project information is nested inside the case. We dont need to know that detail other than to know that we now have a few potential guesses for where our information might be in the files records. We need to know where because we need to construct the appropriate filter.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">files</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">facet</span>(<span class="st">&#39;cases.project.project_id&#39;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">aggregations</span>()
<span class="co">#&gt; $cases.project.project_id</span>
<span class="co">#&gt;            key doc_count</span>
<span class="co">#&gt; 1        FM-AD     36134</span>
<span class="co">#&gt; 2    TCGA-BRCA     31511</span>
<span class="co">#&gt; 3    TCGA-LUAD     17051</span>
<span class="co">#&gt; 4    TCGA-UCEC     16130</span>
<span class="co">#&gt; 5    TCGA-HNSC     15266</span>
<span class="co">#&gt; 6      TCGA-OV     15057</span>
<span class="co">#&gt; 7    TCGA-THCA     14420</span>
<span class="co">#&gt; 8    TCGA-LUSC     15323</span>
<span class="co">#&gt; 9     TCGA-LGG     14723</span>
<span class="co">#&gt; 10   TCGA-KIRC     15082</span>
<span class="co">#&gt; 11   TCGA-PRAD     14287</span>
<span class="co">#&gt; 12   TCGA-COAD     14270</span>
<span class="co">#&gt; 13    TCGA-GBM     11973</span>
<span class="co">#&gt; 14   TCGA-SKCM     12724</span>
<span class="co">#&gt; 15   TCGA-STAD     12845</span>
<span class="co">#&gt; 16   TCGA-BLCA     11710</span>
<span class="co">#&gt; 17   TCGA-LIHC     10814</span>
<span class="co">#&gt; 18   TCGA-CESC      8593</span>
<span class="co">#&gt; 19   TCGA-KIRP      8506</span>
<span class="co">#&gt; 20   TCGA-SARC      7493</span>
<span class="co">#&gt; 21   TCGA-PAAD      5306</span>
<span class="co">#&gt; 22   TCGA-ESCA      5270</span>
<span class="co">#&gt; 23   TCGA-PCPG      5032</span>
<span class="co">#&gt; 24   TCGA-READ      4918</span>
<span class="co">#&gt; 25   TCGA-TGCT      4217</span>
<span class="co">#&gt; 26   TCGA-THYM      3444</span>
<span class="co">#&gt; 27   TCGA-LAML      3960</span>
<span class="co">#&gt; 28  TARGET-NBL      2795</span>
<span class="co">#&gt; 29    TCGA-ACC      2546</span>
<span class="co">#&gt; 30   TCGA-KICH      2324</span>
<span class="co">#&gt; 31   TCGA-MESO      2330</span>
<span class="co">#&gt; 32  TARGET-AML      2170</span>
<span class="co">#&gt; 33    TCGA-UVM      2179</span>
<span class="co">#&gt; 34    TCGA-UCS      1658</span>
<span class="co">#&gt; 35   TARGET-WT      1406</span>
<span class="co">#&gt; 36   TCGA-DLBC      1330</span>
<span class="co">#&gt; 37   TCGA-CHOL      1348</span>
<span class="co">#&gt; 38   TARGET-OS        47</span>
<span class="co">#&gt; 39   TARGET-RT       174</span>
<span class="co">#&gt; 40 TARGET-CCSK        15</span></code></pre></div>
<p>We note that <code>cases.project.project_id</code> looks like it is a good fit. We also note that <code>TCGA-OV</code> is the correct project_id, not <code>TCGA-OVCA</code>. Note that <em>unlike with dplyr and friends, the <code>filter()</code> method here <strong>replaces</strong> the filter and does not build on any previous filters</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">qfiles =<span class="st"> </span><span class="kw">files</span>() <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw">filter</span>( <span class="op">~</span><span class="st"> </span>cases.project.project_id <span class="op">==</span><span class="st"> &#39;TCGA-OV&#39;</span> <span class="op">&amp;</span><span class="st"> </span>type <span class="op">==</span><span class="st"> &#39;gene_expression&#39;</span>)
<span class="kw">str</span>(<span class="kw">get_filter</span>(qfiles))
<span class="co">#&gt; List of 2</span>
<span class="co">#&gt;  $ op     : &#39;scalar&#39; chr &quot;and&quot;</span>
<span class="co">#&gt;  $ content:List of 2</span>
<span class="co">#&gt;   ..$ :List of 2</span>
<span class="co">#&gt;   .. ..$ op     : &#39;scalar&#39; chr &quot;=&quot;</span>
<span class="co">#&gt;   .. ..$ content:List of 2</span>
<span class="co">#&gt;   .. .. ..$ field: chr &quot;cases.project.project_id&quot;</span>
<span class="co">#&gt;   .. .. ..$ value: chr &quot;TCGA-OV&quot;</span>
<span class="co">#&gt;   ..$ :List of 2</span>
<span class="co">#&gt;   .. ..$ op     : &#39;scalar&#39; chr &quot;=&quot;</span>
<span class="co">#&gt;   .. ..$ content:List of 2</span>
<span class="co">#&gt;   .. .. ..$ field: chr &quot;type&quot;</span>
<span class="co">#&gt;   .. .. ..$ value: chr &quot;gene_expression&quot;</span>
qfiles <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>()
<span class="co">#&gt; [1] 1137</span></code></pre></div>
<p>Asking for a <code>count()</code> of results given these new filter criteria gives <code>r qfiles %&gt;% count()</code> results. Generating a manifest for bulk downloads is as simple as asking for the manifest from the current query.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">manifest_df =<span class="st"> </span>qfiles <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">manifest</span>()
<span class="kw">head</span>(manifest_df)
<span class="co">#&gt; # A tibble: 6 x 5</span>
<span class="co">#&gt;   id                filename                  md5               size state</span>
<span class="co">#&gt;   &lt;chr&gt;             &lt;chr&gt;                     &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;</span>
<span class="co">#&gt; 1 567ced20-00cf-46 b2552f6f-dd15-410f-a621- 9af0d993c40aec 258324 live </span>
<span class="co">#&gt; 2 05692746-1770-47 701b8c71-6c05-4e5b-ac10- 8e9816f4d9b871 526537 live </span>
<span class="co">#&gt; 3 e2d47640-8565-43 b2552f6f-dd15-410f-a621- e05190ed65c8a8 543367 live </span>
<span class="co">#&gt; 4 bc6dab72-dc5a-4c a1c4f19e-079e-47e7-8939- 110d8cda0ccdf6 253059 live </span>
<span class="co">#&gt; 5 0a176c20-f3f3-4b 01eac123-1e21-440d-9495- b40921f17128a9 540592 live </span>
<span class="co">#&gt; 6 2ae73487-7acf-42 12c8b289-b9d0-4697-b3a6- 4d3c2b951d94f0 549437 live</span></code></pre></div>
<p>Note that we might still not be quite there. Looking at filenames, there are suspiciously named files that might include FPKM, FPKM-UQ, or counts. Another round of <code>grep</code> and <code>available_fields</code>, looking for type turned up that the field analysis.workflow_type has the appropriate filter criteria.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">qfiles =<span class="st"> </span><span class="kw">files</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>( <span class="op">~</span><span class="st"> </span>cases.project.project_id <span class="op">==</span><span class="st"> &#39;TCGA-OV&#39;</span> <span class="op">&amp;</span>
<span class="st">                            </span>type <span class="op">==</span><span class="st"> &#39;gene_expression&#39;</span> <span class="op">&amp;</span>
<span class="st">                            </span>analysis.workflow_type <span class="op">==</span><span class="st"> &#39;HTSeq - Counts&#39;</span>)
manifest_df =<span class="st"> </span>qfiles <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">manifest</span>()
<span class="kw">nrow</span>(manifest_df)
<span class="co">#&gt; [1] 379</span></code></pre></div>
<p>The GDC Data Transfer Tool can be used (from R, <code>transfer()</code> or from the command-line) to orchestrate high-performance, restartable transfers of all the files in the manifest. See <a href="bulk-downloads">the bulk downloads section</a> for details.</p>
</div>
<div id="authentication" class="section level3">
<h3><span class="header-section-number">5.6.1</span> Authentication</h3>
<p><a href="https://docs.gdc.cancer.gov/API/Users_Guide/Search_and_Retrieval/#facets">[ GDC authentication documentation ]</a></p>
<p>The GDC offers both controlled-access and open data. As of this writing, only data stored as files is controlled-access; that is, metadata accessible via the GDC is all open data and some files are open and some are controlled-access. Controlled-access data are only available after <a href="https://gdc.cancer.gov/access-data/obtaining-access-controlled-data">going through the process of obtaining access.</a></p>
<p>After controlled-access to one or more datasets has been granted, logging into the GDC web portal will allow you to <a href="https://docs.gdc.cancer.gov/Data_Portal/Users_Guide/Authentication/#gdc-authentication-tokens">access a GDC authentication token</a>, which can be downloaded and then used to access available controlled-access data via the GenomicDataCommons package.</p>
<p>The GenomicDataCommons uses authentication tokens only for downloading data (see <code>transfer</code> and <code>gdcdata</code> documentation). The package includes a helper function, <code>gdc_token</code>, that looks for the token to be stored in one of three ways (resolved in this order):</p>
<ol style="list-style-type: decimal">
<li>As a string stored in the environment variable, <code>GDC_TOKEN</code></li>
<li>As a file, stored in the file named by the environment variable, <code>GDC_TOKEN_FILE</code></li>
<li>In a file in the user home directory, called <code>.gdc_token</code></li>
</ol>
<p>As a concrete example:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">token =<span class="st"> </span><span class="kw">gdc_token</span>()
<span class="kw">transfer</span>(...,<span class="dt">token=</span>token)
<span class="co"># or</span>
<span class="kw">transfer</span>(...,<span class="dt">token=</span><span class="kw">get_token</span>())</code></pre></div>
</div>
<div id="datafile-access-and-download" class="section level3">
<h3><span class="header-section-number">5.6.2</span> Datafile access and download</h3>
<p>The <code>gdcdata</code> function takes a character vector of one or more file ids. A simple way of producing such a vector is to produce a <code>manifest</code> data frame and then pass in the first column, which will contain file ids.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fnames =<span class="st"> </span><span class="kw">gdcdata</span>(manifest_df<span class="op">$</span>id[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>],<span class="dt">progress=</span><span class="ot">FALSE</span>)</code></pre></div>
<p>Note that for controlled-access data, a GDC <a href="public-data-resources-and-bioconductor.html#authentication">authentication token</a> is required. Using the <code>BiocParallel</code> package may be useful for downloading in parallel, particularly for large numbers of smallish files.</p>
<p>The bulk download functionality is only efficient (as of v1.2.0 of the GDC Data Transfer Tool) for relatively large files, so use this approach only when transferring BAM files or larger VCF files, for example. Otherwise, consider using the approach shown above, perhaps in parallel.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fnames =<span class="st"> </span><span class="kw">gdcdata</span>(manifest_df<span class="op">$</span>id[<span class="dv">3</span><span class="op">:</span><span class="dv">10</span>], <span class="dt">access_method =</span> <span class="st">&#39;client&#39;</span>)</code></pre></div>
</div>
</div>
<div id="sequence-read-archive" class="section level2">
<h2><span class="header-section-number">5.7</span> Sequence Read Archive</h2>
<p>The SRAdbV2 package is currently available from <a href="https://github.com/seandavi/SRAdbV2">GitHub</a> and is under active development. Either the <em><a href="https://CRAN.R-project.org/package=devtools">devtools</a></em> package or the <em><a href="https://CRAN.R-project.org/package=BiocManager">BiocManager</a></em> package can be used for easy installation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">&#39;BiocManager&#39;</span>)
BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&#39;seandavi/SRAdbV2&#39;</span>)</code></pre></div>
<div id="usage-1" class="section level3">
<h3><span class="header-section-number">5.7.1</span> Usage</h3>
<div id="loading-the-library" class="section level4">
<h4><span class="header-section-number">5.7.1.1</span> Loading the library</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(SRAdbV2)
<span class="co">#&gt; Loading required package: R6</span></code></pre></div>
</div>
<div id="the-omicidx" class="section level4">
<h4><span class="header-section-number">5.7.1.2</span> The Omicidx</h4>
<p>The entrypoint for using the SRAdbV2 system is the <code>Omicidx</code>, an <a href="https://cran.r-project.org/web/packages/R6/vignettes/Introduction.html">R6 class</a>. To start, create a new instance of the class.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">oidx =<span class="st"> </span>Omicidx<span class="op">$</span><span class="kw">new</span>()</code></pre></div>
<p>Typing <code>oidx$</code> and then <code>TAB</code> will give possible completions. Note the search completion.</p>
</div>
<div id="queries" class="section level4">
<h4><span class="header-section-number">5.7.1.3</span> Queries</h4>
<p>Once an instance of <code>Omicidx</code> is created (here we will call the instance <code>oidx</code>), search capabilities are available via <code>oidx$search()</code>. The one interesting parameter is the <code>q</code> parameter. This parameter takes a string formatted as a Lucene query string. See below for <a href="public-data-resources-and-bioconductor.html#query-syntax">Query syntax</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">query=<span class="kw">paste</span>(
  <span class="kw">paste0</span>(<span class="st">&#39;sample_taxon_id:&#39;</span>, <span class="dv">10116</span>),
  <span class="st">&#39;AND experiment_library_strategy:&quot;rna seq&quot;&#39;</span>,
  <span class="st">&#39;AND experiment_library_source:transcriptomic&#39;</span>,
  <span class="st">&#39;AND experiment_platform:illumina&#39;</span>)
z =<span class="st"> </span>oidx<span class="op">$</span><span class="kw">search</span>(<span class="dt">q=</span>query,<span class="dt">entity=</span><span class="st">&#39;full&#39;</span>,<span class="dt">size=</span>100L)</code></pre></div>
<p>The <code>entity</code> parameter is one of the SRA entity types available via the API. The <code>size</code> parameter is the number of records that will be returned in each chunk.</p>
</div>
<div id="fetching-results" class="section level4">
<h4><span class="header-section-number">5.7.1.4</span> Fetching results</h4>
<p>Because result sets can be large, we have a special method that allows us to scroll through the results or to simply get them <em>en bloc</em>. The first step for result retrieval, then, is to get a <code>Scroller</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">s =<span class="st"> </span>z<span class="op">$</span><span class="kw">scroll</span>()
s
<span class="co">#&gt; &lt;Scroller&gt;</span>
<span class="co">#&gt;   Public:</span>
<span class="co">#&gt;     clone: function (deep = FALSE) </span>
<span class="co">#&gt;     collate: function (limit = Inf) </span>
<span class="co">#&gt;     count: active binding</span>
<span class="co">#&gt;     fetched: active binding</span>
<span class="co">#&gt;     has_next: function () </span>
<span class="co">#&gt;     initialize: function (search, progress = interactive()) </span>
<span class="co">#&gt;     reset: function () </span>
<span class="co">#&gt;     yield: function () </span>
<span class="co">#&gt;   Private:</span>
<span class="co">#&gt;     .count: NULL</span>
<span class="co">#&gt;     .fetched: 0</span>
<span class="co">#&gt;     .last: FALSE</span>
<span class="co">#&gt;     progress: FALSE</span>
<span class="co">#&gt;     scroll: 1m</span>
<span class="co">#&gt;     scroll_id: NULL</span>
<span class="co">#&gt;     search: Searcher, R6</span></code></pre></div>
<p>Methods such as <code>s$count</code> allow introspection into the available number of results, in this case, 8886 records.</p>
<p>The <code>Scroller</code> provides two different approaches to accessing the resulting data.</p>
<div id="collating-entire-result-sets" class="section level5">
<h5><span class="header-section-number">5.7.1.4.1</span> Collating entire result sets</h5>
<p>The first approach to getting results of a query back into R is the most convenient, but for large result sets, the entire dataset is loaded into memory and may take significant time if network connections are slow.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># for VERY large result sets, this may take</span>
<span class="co"># quite a bit of time and/or memory. An</span>
<span class="co"># alternative is to use s$chunk() to retrieve</span>
<span class="co"># one batch of records at a time and process</span>
<span class="co"># incrementally.</span>
res =<span class="st"> </span>s<span class="op">$</span><span class="kw">collate</span>(<span class="dt">limit =</span> <span class="dv">1000</span>)
<span class="kw">head</span>(res)
<span class="co">#&gt; # A tibble: 6 x 85</span>
<span class="co">#&gt;   experiment_Insdc experiment_LastMet experiment_LastUpd</span>
<span class="co">#&gt;   &lt;lgl&gt;            &lt;dttm&gt;              &lt;dttm&gt;             </span>
<span class="co">#&gt; 1 TRUE             2018-04-10 11:28:04 2018-04-10 11:52:20</span>
<span class="co">#&gt; 2 TRUE             2018-05-06 06:12:59 2018-05-06 06:42:38</span>
<span class="co">#&gt; 3 TRUE             2013-08-28 07:50:02 2014-04-13 06:08:21</span>
<span class="co">#&gt; 4 TRUE             2018-04-10 11:28:00 2018-04-10 11:34:38</span>
<span class="co">#&gt; 5 TRUE             2016-12-07 17:16:11 2017-06-21 23:05:24</span>
<span class="co">#&gt; 6 TRUE             2016-10-21 15:03:02 2016-10-22 13:55:27</span>
<span class="co">#&gt; # ... with 82 more variables: experiment_Published &lt;dttm&gt;,</span>
<span class="co">#&gt; #   experiment_Received &lt;dttm&gt;, experiment_Status &lt;chr&gt;,</span>
<span class="co">#&gt; #   experiment_accession &lt;chr&gt;, experiment_alias &lt;chr&gt;,</span>
<span class="co">#&gt; #   experiment_center_name &lt;chr&gt;, experiment_identifiers &lt;list&gt;,</span>
<span class="co">#&gt; #   experiment_instrument_model &lt;chr&gt;,</span>
<span class="co">#&gt; #   experiment_library_construction_protocol &lt;chr&gt;,</span>
<span class="co">#&gt; #   experiment_library_layout &lt;chr&gt;, experiment_library_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   experiment_library_selection &lt;chr&gt;, experiment_library_source &lt;chr&gt;,</span>
<span class="co">#&gt; #   experiment_library_strategy &lt;chr&gt;, experiment_platform &lt;chr&gt;,</span>
<span class="co">#&gt; #   experiment_title &lt;chr&gt;, run_FileDate &lt;dbl&gt;, run_FileMd5 &lt;chr&gt;,</span>
<span class="co">#&gt; #   run_FileSize &lt;int&gt;, run_Insdc &lt;lgl&gt;, run_LastMetaUpdate &lt;dttm&gt;,</span>
<span class="co">#&gt; #   run_LastUpdate &lt;dttm&gt;, run_Published &lt;dttm&gt;, run_Received &lt;dttm&gt;,</span>
<span class="co">#&gt; #   run_Status &lt;chr&gt;, run_accession &lt;chr&gt;, run_alias &lt;chr&gt;,</span>
<span class="co">#&gt; #   run_attributes &lt;list&gt;, run_bases &lt;dbl&gt;, run_center_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   run_identifiers &lt;list&gt;, run_nreads &lt;int&gt;, run_reads &lt;list&gt;,</span>
<span class="co">#&gt; #   run_spot_length &lt;int&gt;, run_spots &lt;int&gt;, sample_BioSample &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_Insdc &lt;lgl&gt;, sample_LastMetaUpdate &lt;dttm&gt;,</span>
<span class="co">#&gt; #   sample_LastUpdate &lt;dttm&gt;, sample_Published &lt;dttm&gt;,</span>
<span class="co">#&gt; #   sample_Received &lt;dttm&gt;, sample_Status &lt;chr&gt;, sample_accession &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_alias &lt;chr&gt;, sample_attributes &lt;list&gt;,</span>
<span class="co">#&gt; #   sample_center_name &lt;chr&gt;, sample_identifiers &lt;list&gt;,</span>
<span class="co">#&gt; #   sample_organism &lt;chr&gt;, sample_taxon_id &lt;int&gt;, sample_title &lt;chr&gt;,</span>
<span class="co">#&gt; #   study_BioProject &lt;chr&gt;, study_Insdc &lt;lgl&gt;,</span>
<span class="co">#&gt; #   study_LastMetaUpdate &lt;dttm&gt;, study_LastUpdate &lt;dttm&gt;,</span>
<span class="co">#&gt; #   study_Published &lt;dttm&gt;, study_Received &lt;dttm&gt;, study_Status &lt;chr&gt;,</span>
<span class="co">#&gt; #   study_abstract &lt;chr&gt;, study_accession &lt;chr&gt;, study_alias &lt;chr&gt;,</span>
<span class="co">#&gt; #   study_attributes &lt;list&gt;, study_center_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   study_description &lt;chr&gt;, study_identifiers &lt;list&gt;, study_title &lt;chr&gt;,</span>
<span class="co">#&gt; #   study_type &lt;chr&gt;, experiment_attributes &lt;list&gt;,</span>
<span class="co">#&gt; #   experiment_broker_name &lt;chr&gt;, experiment_design &lt;chr&gt;,</span>
<span class="co">#&gt; #   run_broker_name &lt;chr&gt;, run_center &lt;chr&gt;, sample_broker_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   sample_description &lt;chr&gt;, study_broker_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   run_file_addons &lt;list&gt;, experiment_library_layout_length &lt;dbl&gt;,</span>
<span class="co">#&gt; #   experiment_library_layout_sdev &lt;chr&gt;, sample_xrefs &lt;list&gt;,</span>
<span class="co">#&gt; #   experiment_xrefs &lt;list&gt;, sample_GEO &lt;chr&gt;, study_GEO &lt;chr&gt;,</span>
<span class="co">#&gt; #   study_xrefs &lt;list&gt;</span></code></pre></div>
<p>Note that the scroller now reports that it has fetched (<code>s$fetched</code>) 1000 records.</p>
<p>To reuse a <code>Scroller</code>, we must reset it first.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">s<span class="op">$</span><span class="kw">reset</span>()
s
<span class="co">#&gt; &lt;Scroller&gt;</span>
<span class="co">#&gt;   Public:</span>
<span class="co">#&gt;     clone: function (deep = FALSE) </span>
<span class="co">#&gt;     collate: function (limit = Inf) </span>
<span class="co">#&gt;     count: active binding</span>
<span class="co">#&gt;     fetched: active binding</span>
<span class="co">#&gt;     has_next: function () </span>
<span class="co">#&gt;     initialize: function (search, progress = interactive()) </span>
<span class="co">#&gt;     reset: function () </span>
<span class="co">#&gt;     yield: function () </span>
<span class="co">#&gt;   Private:</span>
<span class="co">#&gt;     .count: 8886</span>
<span class="co">#&gt;     .fetched: 0</span>
<span class="co">#&gt;     .last: FALSE</span>
<span class="co">#&gt;     progress: FALSE</span>
<span class="co">#&gt;     scroll: 1m</span>
<span class="co">#&gt;     scroll_id: NULL</span>
<span class="co">#&gt;     search: Searcher, R6</span></code></pre></div>
</div>
<div id="yielding-chunks" class="section level5">
<h5><span class="header-section-number">5.7.1.4.2</span> Yielding chunks</h5>
<p>The second approach is to iterate through results using the <code>yield</code> method. This approach allows the user to perform processing on chunks of data as they arrive in R.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">j =<span class="st"> </span><span class="dv">0</span>
## fetch only 500 records, but
## `yield` will return NULL
## after ALL records have been fetched
<span class="cf">while</span>(s<span class="op">$</span>fetched <span class="op">&lt;</span><span class="st"> </span><span class="dv">500</span>) {
    res =<span class="st"> </span>s<span class="op">$</span><span class="kw">yield</span>()
    <span class="co"># do something interesting with `res` here if you like</span>
    j =<span class="st"> </span>j <span class="op">+</span><span class="st"> </span><span class="dv">1</span>
    <span class="kw">message</span>(<span class="kw">sprintf</span>(<span class="st">&#39;total of %d fetched records, loop iteration # %d&#39;</span>, s<span class="op">$</span>fetched, j))
}
<span class="co">#&gt; total of 100 fetched records, loop iteration # 1</span>
<span class="co">#&gt; total of 200 fetched records, loop iteration # 2</span>
<span class="co">#&gt; total of 300 fetched records, loop iteration # 3</span>
<span class="co">#&gt; total of 400 fetched records, loop iteration # 4</span>
<span class="co">#&gt; total of 500 fetched records, loop iteration # 5</span></code></pre></div>
<p>The <code>Scroller</code> also has a <code>has_next()</code> method that will report <code>TRUE</code> if the result set has not been fully fetched. Using the <code>reset()</code> method will move the cursor back to the beginning of the result set.</p>
</div>
</div>
</div>
<div id="query-syntax" class="section level3">
<h3><span class="header-section-number">5.7.2</span> Query syntax</h3>
<div id="terms" class="section level4">
<h4><span class="header-section-number">5.7.2.1</span> Terms</h4>
<p>A query is broken up into terms and operators. There are two types of terms: Single Terms and Phrases. A Single Term is a single word such as test or hello. A Phrase is a group of words surrounded by double quotes such as hello dolly. Multiple terms can be combined together with Boolean operators to form a more complex query (see below).</p>
</div>
<div id="fields" class="section level4">
<h4><span class="header-section-number">5.7.2.2</span> Fields</h4>
<p>Queries support fielded data. When performing a search you can either specify a field, or use the default field. The field names and default field is implementation specific. You can search any field by typing the field name followed by a colon : and then the term you are looking for. As an example, lets assume a Lucene index contains two fields, title and abstract. If you want to find the document entitled The Right Way which contains the text dont go this way in the abstract, you can enter:</p>
<pre><code>title:&quot;The Right Way&quot; AND abstract:go</code></pre>
<p>or</p>
<p>Note: The field is only valid for the term that it directly precedes, so the query</p>
<pre><code>title:Do it right</code></pre>
<p>will only find Do in the title field. It will find it and right in any other fields.</p>
</div>
<div id="wildcard-searches" class="section level4">
<h4><span class="header-section-number">5.7.2.3</span> Wildcard Searches</h4>
<p>Lucene supports single and multiple character wildcard searches within single terms (not within phrase queries). To perform a single character wildcard search use the ? symbol. To perform a multiple character wildcard search use the *&quot; symbol. The single character wildcard search looks for terms that match that with the single character replaced. For example, to search for text or test you can use the search:</p>
<pre><code>te?t</code></pre>
<p>Multiple character wildcard searches looks for 0 or more characters. For example, to search for test, tests or tester, you can use the search:</p>
<pre><code>test*</code></pre>
<p>You can also use the wildcard searches in the middle of a term.</p>
<pre><code>te*t</code></pre>
<p>Note: You cannot use a * or ? symbol as the first character of a search.</p>
</div>
<div id="fuzzy-searches" class="section level4">
<h4><span class="header-section-number">5.7.2.4</span> Fuzzy Searches</h4>
<p>Lucene supports fuzzy searches based on the Levenshtein Distance, or Edit Distance algorithm. To do a fuzzy search use the tilde, ~, symbol at the end of a Single word Term. For example to search for a term similar in spelling to roam use the fuzzy search:</p>
<pre><code>roam~</code></pre>
<p>This search will find terms like foam and roams.</p>
<p>Starting with Lucene 1.9 an additional (optional) parameter can specify the required similarity. The value is between 0 and 1, with a value closer to 1 only terms with a higher similarity will be matched. For example:</p>
<pre><code>roam~0.8</code></pre>
<p>The default that is used if the parameter is not given is 0.5.</p>
</div>
<div id="proximity-searches" class="section level4">
<h4><span class="header-section-number">5.7.2.5</span> Proximity Searches</h4>
<p>Lucene supports finding words are a within a specific distance away. To do a proximity search use the tilde, ~, symbol at the end of a Phrase. For example to search for a apache and jakarta within 10 words of each other in a document use the search:</p>
<pre><code>&quot;jakarta apache&quot;~10</code></pre>
</div>
<div id="range-searches" class="section level4">
<h4><span class="header-section-number">5.7.2.6</span> Range Searches</h4>
<p>Range Queries allow one to match documents whose field(s) values are between the lower and upper bound specified by the Range Query. Range Queries can be inclusive or exclusive of the upper and lower bounds. Sorting is done lexicographically.</p>
<pre><code>mod_date:[20020101 TO 20030101]</code></pre>
<p>This will find documents whose mod_date fields have values between 20020101 and 20030101, inclusive. Note that Range Queries are not reserved for date fields. You could also use range queries with non-date fields:</p>
<pre><code>title:{Aida TO Carmen}</code></pre>
<p>This will find all documents whose titles are between Aida and Carmen, but not including Aida and Carmen. Inclusive range queries are denoted by square brackets. Exclusive range queries are denoted by curly brackets.</p>
</div>
<div id="boolean-operators" class="section level4">
<h4><span class="header-section-number">5.7.2.7</span> Boolean Operators</h4>
<p>Boolean operators allow terms to be combined through logic operators. Lucene supports AND, +, OR, NOT and - as Boolean operators(Note: Boolean operators must be ALL CAPS).</p>
<div id="or" class="section level5">
<h5><span class="header-section-number">5.7.2.7.1</span> OR</h5>
<p>The OR operator is the default conjunction operator. This means that if there is no Boolean operator between two terms, the OR operator is used. The OR operator links two terms and finds a matching document if either of the terms exist in a document. This is equivalent to a union using sets. The symbol || can be used in place of the word OR.</p>
<p>To search for documents that contain either jakarta apache or just jakarta use the query:</p>
<pre><code>&quot;jakarta apache&quot; jakarta</code></pre>
<p>or</p>
<pre><code>&quot;jakarta apache&quot; OR jakarta</code></pre>
<p>The AND operator matches documents where both terms exist anywhere in the text of a single document. This is equivalent to an intersection using sets. The symbol &amp;&amp; can be used in place of the word AND. To search for documents that contain jakarta apache and Apache Lucene use the query:</p>
<p>jakarta apache AND Apache Lucene</p>
</div>
<div id="section" class="section level5">
<h5><span class="header-section-number">5.7.2.7.2</span> +</h5>
<p>The + or required operator requires that the term after the + symbol exist somewhere in a the field of a single document. To search for documents that must contain jakarta and may contain lucene use the query:</p>
<pre><code>+jakarta lucene</code></pre>
</div>
<div id="not" class="section level5">
<h5><span class="header-section-number">5.7.2.7.3</span> NOT</h5>
<p>The NOT operator excludes documents that contain the term after NOT. This is equivalent to a difference using sets. The symbol ! can be used in place of the word NOT. To search for documents that contain jakarta apache but not Apache Lucene use the query:</p>
<pre><code>&quot;jakarta apache&quot; NOT &quot;Apache Lucene&quot;</code></pre>
<p>Note: The NOT operator cannot be used with just one term. For example, the following search will return no results:</p>
<pre><code>NOT &quot;jakarta apache&quot;</code></pre>
</div>
<div id="section-1" class="section level5">
<h5><span class="header-section-number">5.7.2.7.4</span> -</h5>
<p>The - or prohibit operator excludes documents that contain the term after the - symbol. To search for documents that contain jakarta apache but not Apache Lucene use the query:</p>
<pre><code>&quot;jakarta apache&quot; -&quot;Apache Lucene&quot;</code></pre>
</div>
</div>
<div id="grouping" class="section level4">
<h4><span class="header-section-number">5.7.2.8</span> Grouping</h4>
<p>Lucene supports using parentheses to group clauses to form sub queries. This can be very useful if you want to control the boolean logic for a query. To search for either jakarta or apache and website use the query:</p>
<pre><code>(jakarta OR apache) AND website</code></pre>
<p>This eliminates any confusion and makes sure you that website must exist and either term jakarta or apache may exist.</p>
<p>Lucene supports using parentheses to group multiple clauses to a single field.</p>
<p>To search for a title that contains both the word return and the phrase pink panther use the query:</p>
<pre><code>title:(+return +&quot;pink panther&quot;)</code></pre>
</div>
<div id="escaping-special-characters" class="section level4">
<h4><span class="header-section-number">5.7.2.9</span> Escaping Special Characters</h4>
<p>Lucene supports escaping special characters that are part of the query syntax. The current list special characters are</p>
<pre><code>+ - &amp;&amp; || ! ( ) { } [ ] ^ &quot; ~ * ? : \</code></pre>
<p>To escape these character use the before the character. For example to search for (1+1):2 use the query:</p>
<pre><code>\(1\+1\)\:2</code></pre>
</div>
</div>
<div id="using-the-raw-api-without-rbioconductor" class="section level3">
<h3><span class="header-section-number">5.7.3</span> Using the raw API without R/Bioconductor</h3>
<p>The SRAdbV2 is a client to a high-performance web-based API. As such, the web API is perfectly usable from either a simple test page, accessible here:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sra_browse_API</span>()</code></pre></div>
<p>The web-based API provides a useful interface for experiment with queries. It also returns URLs associated with the example queries, facilitating querying with other tools like <code>curl</code> or <code>wget</code></p>
<p>The API is described using the OpenAPI standard, also known as <a href="https://swagger.io/">Swagger</a>. Tooling exists to quickly scaffold clients in any language (basically) based on the json available here:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sra_get_swagger_json_url</span>()
<span class="co">#&gt; [1] &quot;https://api-omicidx.cancerdatasci.org/sra/1.0/swagger.json&quot;</span></code></pre></div>
</div>
</div>
<div id="accessing-the-cancer-genome-atlas-tcga" class="section level2">
<h2><span class="header-section-number">5.8</span> Accessing The Cancer Genome Atlas (TCGA)</h2>
<p>We summarize two approaches to accessing TCGA data: 1. <em><a href="http://bioconductor.org/packages/TCGAbiolinks">TCGAbiolinks</a></em>: a. data access through <em><a href="http://bioconductor.org/packages/GenomicDataCommons">GenomicDataCommons</a></em> b. provides data both from the legacy Firehose pipeline used by the TCGA publications (alignments based on hg18 and hg19 builds<a href="#fn4" class="footnoteRef" id="fnref4"><sup>4</sup></a>), and the GDC harmonized GRCh38 pipeline<a href="#fn5" class="footnoteRef" id="fnref5"><sup>5</sup></a>. c. downloads files from the Genomic Data Commons, and provides conversion to <code>(Ranged)SummarizedExperiment</code> where possible 2. <em><a href="http://bioconductor.org/packages/curatedTCGAData">curatedTCGAData</a></em>: a. data access through <em><a href="http://bioconductor.org/packages/ExperimentHub">ExperimentHub</a></em> b. provides data from the legacy Firehose pipeline<a href="#fn6" class="footnoteRef" id="fnref6"><sup>6</sup></a> c. provides individual assays as <code>(Ranged)SummarizedExperiment</code> and <code>RaggedExperiment</code>, integrates multiple assays within and across cancer types using <code>MultiAssayExperiment</code></p>
<div id="tcgabiolinks" class="section level3">
<h3><span class="header-section-number">5.8.1</span> TCGAbiolinks</h3>
<p>We demonstrate here generating a <code>RangedSummarizedExperiment</code> for RNA-seq data from adrenocortical carcinoma (ACC). For additional information and options, see the TCGAbiolinks vignettes<a href="#fn7" class="footnoteRef" id="fnref7"><sup>7</sup></a>.</p>
<p>Load packages:</p>
<p>Search for matching data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(TCGAbiolinks)
<span class="kw">library</span>(SummarizedExperiment)
query &lt;-<span class="st"> </span><span class="kw">GDCquery</span>(<span class="dt">project =</span> <span class="st">&quot;TCGA-ACC&quot;</span>,
                           <span class="dt">data.category =</span> <span class="st">&quot;Gene expression&quot;</span>,
                           <span class="dt">data.type =</span> <span class="st">&quot;Gene expression quantification&quot;</span>,
                           <span class="dt">platform =</span> <span class="st">&quot;Illumina HiSeq&quot;</span>, 
                           <span class="dt">file.type  =</span> <span class="st">&quot;normalized_results&quot;</span>,
                           <span class="dt">experimental.strategy =</span> <span class="st">&quot;RNA-Seq&quot;</span>,
                           <span class="dt">legacy =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>Download data and convert it to <code>RangedSummarizedExperiment</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gdcdir &lt;-<span class="st"> </span><span class="kw">file.path</span>(<span class="st">&quot;Waldron_PublicData&quot;</span>, <span class="st">&quot;GDCdata&quot;</span>)
<span class="kw">GDCdownload</span>(query, <span class="dt">method =</span> <span class="st">&quot;api&quot;</span>, <span class="dt">files.per.chunk =</span> <span class="dv">10</span>,
            <span class="dt">directory =</span> gdcdir)
ACCse &lt;-<span class="st"> </span><span class="kw">GDCprepare</span>(query, <span class="dt">directory =</span> gdcdir)
ACCse</code></pre></div>
</div>
<div id="curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects" class="section level3">
<h3><span class="header-section-number">5.8.2</span> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</h3>
<p><em><a href="http://bioconductor.org/packages/curatedTCGAData">curatedTCGAData</a></em> does not interface with the Genomic Data Commons, but downloads data from Bioconductors <em><a href="http://bioconductor.org/packages/ExperimentHub">ExperimentHub</a></em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(curatedTCGAData)
<span class="kw">library</span>(MultiAssayExperiment)</code></pre></div>
<p>By default, the <code>curatedTCGAData()</code> function will only show available datasets, and not download anything. The arguments are shown here only for demonstration, the same result is obtained with no arguments:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">curatedTCGAData</span>(<span class="dt">diseaseCode =</span> <span class="st">&quot;*&quot;</span>, <span class="dt">assays =</span> <span class="st">&quot;*&quot;</span>)
<span class="co">#&gt; Please see the list below for available cohorts and assays</span>
<span class="co">#&gt; Available Cancer codes:</span>
<span class="co">#&gt;  ACC BLCA BRCA CESC CHOL COAD DLBC ESCA GBM HNSC KICH</span>
<span class="co">#&gt;  KIRC KIRP LAML LGG LIHC LUAD LUSC MESO OV PAAD PCPG</span>
<span class="co">#&gt;  PRAD READ SARC SKCM STAD TGCT THCA THYM UCEC UCS UVM </span>
<span class="co">#&gt; Available Data Types:</span>
<span class="co">#&gt;  CNACGH CNASeq CNASNP CNVSNP GISTICA GISTICT</span>
<span class="co">#&gt;  Methylation miRNAArray miRNASeqGene mRNAArray</span>
<span class="co">#&gt;  Mutation RNASeq2GeneNorm RNASeqGene RPPAArray</span></code></pre></div>
<p>Check potential files to be downloaded for adrenocortical carcinoma (ACC):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">curatedTCGAData</span>(<span class="dt">diseaseCode =</span> <span class="st">&quot;ACC&quot;</span>)
<span class="co">#&gt;                                  ACC_CNASNP </span>
<span class="co">#&gt;                   &quot;ACC_CNASNP-20160128.rda&quot; </span>
<span class="co">#&gt;                                  ACC_CNVSNP </span>
<span class="co">#&gt;                   &quot;ACC_CNVSNP-20160128.rda&quot; </span>
<span class="co">#&gt;                        ACC_GISTIC_AllByGene </span>
<span class="co">#&gt;         &quot;ACC_GISTIC_AllByGene-20160128.rda&quot; </span>
<span class="co">#&gt;                ACC_GISTIC_ThresholdedByGene </span>
<span class="co">#&gt; &quot;ACC_GISTIC_ThresholdedByGene-20160128.rda&quot; </span>
<span class="co">#&gt;                             ACC_Methylation </span>
<span class="co">#&gt;              &quot;ACC_Methylation-20160128.rda&quot; </span>
<span class="co">#&gt;                            ACC_miRNASeqGene </span>
<span class="co">#&gt;             &quot;ACC_miRNASeqGene-20160128.rda&quot; </span>
<span class="co">#&gt;                                ACC_Mutation </span>
<span class="co">#&gt;                 &quot;ACC_Mutation-20160128.rda&quot; </span>
<span class="co">#&gt;                         ACC_RNASeq2GeneNorm </span>
<span class="co">#&gt;          &quot;ACC_RNASeq2GeneNorm-20160128.rda&quot; </span>
<span class="co">#&gt;                               ACC_RPPAArray </span>
<span class="co">#&gt;                &quot;ACC_RPPAArray-20160128.rda&quot;</span></code></pre></div>
<p>Actually download the reverse phase protein array (RPPA) and RNA-seq data for ACC</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ACCmae &lt;-<span class="st"> </span><span class="kw">curatedTCGAData</span>(<span class="st">&quot;ACC&quot;</span>, <span class="kw">c</span>(<span class="st">&quot;RPPAArray&quot;</span>, <span class="st">&quot;RNASeq2GeneNorm&quot;</span>), 
                          <span class="dt">dry.run=</span><span class="ot">FALSE</span>)
ACCmae
<span class="co">#&gt; A MultiAssayExperiment object of 2 listed</span>
<span class="co">#&gt;  experiments with user-defined names and respective classes. </span>
<span class="co">#&gt;  Containing an ExperimentList class object of length 2: </span>
<span class="co">#&gt;  [1] ACC_RNASeq2GeneNorm-20160128: SummarizedExperiment with 20501 rows and 79 columns </span>
<span class="co">#&gt;  [2] ACC_RPPAArray-20160128: SummarizedExperiment with 192 rows and 46 columns </span>
<span class="co">#&gt; Features: </span>
<span class="co">#&gt;  experiments() - obtain the ExperimentList instance </span>
<span class="co">#&gt;  colData() - the primary/phenotype DataFrame </span>
<span class="co">#&gt;  sampleMap() - the sample availability DataFrame </span>
<span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment </span>
<span class="co">#&gt;  *Format() - convert into a long or wide DataFrame </span>
<span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></code></pre></div>
<p><strong>Note</strong>. Data will be downloaded the first time the above command is run; subsequent times it will be loaded from local cache.</p>
<p>This object contains 822 columns of clinical, pathological, specimen, and subtypes data in its <code>colData</code>, merged from all available data levels (1-4) of the Firehose pipeline:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(<span class="kw">colData</span>(ACCmae))
<span class="co">#&gt; [1]  79 822</span>
<span class="kw">head</span>(<span class="kw">colnames</span>(<span class="kw">colData</span>(ACCmae)))
<span class="co">#&gt; [1] &quot;patientID&quot;             &quot;years_to_birth&quot;        &quot;vital_status&quot;         </span>
<span class="co">#&gt; [4] &quot;days_to_death&quot;         &quot;days_to_last_followup&quot; &quot;tumor_tissue_site&quot;</span></code></pre></div>
<p>See the <em><a href="http://bioconductor.org/packages/MultiAssayExperiment">MultiAssayExperiment</a></em> vignette <span class="citation">(Ramos et al. 2017)</span> and the <em>Workflow for Multi-omics Analysis with MultiAssayExperiment</em> workshop for details on using this object.</p>
<div id="subtype-information" class="section level4">
<h4><span class="header-section-number">5.8.2.1</span> Subtype information</h4>
<p>Some cancer datasets contain associated subtype information within the clinical datasets provided. This subtype information is included in the metadata of <code>colData</code> of the <code>MultiAssayExperiment</code> object. To obtain these variable names, run the <code>metadata</code> function on the <code>colData</code> of the object such as:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw">metadata</span>(<span class="kw">colData</span>(ACCmae))[[<span class="st">&quot;subtypes&quot;</span>]])
<span class="co">#&gt;         ACC_annotations   ACC_subtype</span>
<span class="co">#&gt; 1            Patient_ID        SAMPLE</span>
<span class="co">#&gt; 2 histological_subtypes     Histology</span>
<span class="co">#&gt; 3         mrna_subtypes       C1A/C1B</span>
<span class="co">#&gt; 4         mrna_subtypes       mRNA_K4</span>
<span class="co">#&gt; 5                  cimp    MethyLevel</span>
<span class="co">#&gt; 6     microrna_subtypes miRNA cluster</span></code></pre></div>
</div>
</div>
</div>
<div id="recount-reproducible-rna-seq-analysis-using-recount2" class="section level2">
<h2><span class="header-section-number">5.9</span> recount: Reproducible RNA-seq Analysis Using recount2</h2>
<p>The <em><a href="http://bioconductor.org/packages/recount">recount</a></em><span class="citation">(Collado-Torres et al. 2017)</span> package provides uniformly processed <code>RangedSummarizedExperiment</code> objects at the gene, exon, or exon-exon junctions level, the raw counts, the phenotype metadata used, the urls to sample coverage bigWig files and mean coverage bigWig file, for every study available. The <code>RangedSummarizedExperiment</code> objects can be used for differential expression analysis. These are also accessible through a web interface.<a href="#fn8" class="footnoteRef" id="fnref8"><sup>8</sup></a></p>
<pre><code>#&gt; No methods found in package &#39;IRanges&#39; for request: &#39;subset&#39; when loading &#39;derfinder&#39;</code></pre>
<p><em><a href="http://bioconductor.org/packages/recount">recount</a></em> provides a search function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(recount)
project_info &lt;-<span class="st"> </span><span class="kw">abstract_search</span>(<span class="st">&#39;GSE32465&#39;</span>)</code></pre></div>
<p>It is not an ExperimentHub package, so downloading and serializing is slightly more involved in involves two steps: first, download the gene-level RangedSummarizedExperiment data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">download_study</span>(project_info<span class="op">$</span>project)
<span class="co">#&gt; 2018-07-23 20:44:16 downloading file rse_gene.Rdata to SRP009615</span></code></pre></div>
<p>followed by loading the data</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">load</span>(<span class="kw">file.path</span>(project_info<span class="op">$</span>project, <span class="st">&#39;rse_gene.Rdata&#39;</span>))</code></pre></div>
</div>
<div id="curateddata-packages-for-standardized-cancer-transcriptomes" class="section level2">
<h2><span class="header-section-number">5.10</span> curated*Data packages for standardized cancer transcriptomes</h2>
<p>There are focused databases of cancer microarray data for several cancer types, which can be useful for researchers of those cancer types or for methodological development: * <em><a href="http://bioconductor.org/packages/curatedOvarianData">curatedOvarianData</a></em><span class="citation">(Ganzfried et al. 2013)</span>: Clinically Annotated Data for the Ovarian Cancer Transcriptome (data available with additional options through the <em><a href="http://bioconductor.org/packages/MetaGxOvarian">MetaGxOvarian</a></em> package). * <em><a href="http://bioconductor.org/packages/curatedBladderData">curatedBladderData</a></em>: Clinically Annotated Data for the Bladder Cancer Transcriptome * <em><a href="http://bioconductor.org/packages/curatedCRCData">curatedCRCData</a></em>: Clinically Annotated Data for the Colorectal Cancer Transcriptome</p>
<p>These provide data from the Gene Expression Omnibus and other sources, but use a formally vocabulary for clinicopathological data and use a common pipeline for preprocessing of microarray data (for Affymetrix, other for other platforms the processed data are provided as processed by original study authors), merging probesets, and mapping to gene symbols. The pipeline is described by <span class="citation">Ganzfried et al. (2013)</span>.</p>
</div>
<div id="microbiome-data" class="section level2">
<h2><span class="header-section-number">5.11</span> Microbiome data</h2>
<p>Bioconductor provides curated resources of microbiome data. Most microbiome data are generated either by targeted amplicon sequencing (usually of variable regions of the 16S ribosomal RNA gene) or by metagenomic shotgun sequencing (MGX). These two approaches are analyzed by different sequence analysis tools, but downstream statistical and ecological analysis can involve any of the following types of data: * taxonomic abundance at different levels of the taxonomic hierarchy * phylogenetic distances and the phylogenetic tree of life * metabolic potential of the microbiome * abundance of microbial genes and gene families</p>
<p>A review of types and properties of microbiome data is provided by <span class="citation">(Morgan and Huttenhower 2012)</span>.</p>
<div id="curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub" class="section level3">
<h3><span class="header-section-number">5.11.1</span> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</h3>
<p><code>curatedMetagenomicData</code><span class="citation">(Pasolli et al. 2017)</span> provides 6 types of processed data for &gt;30 publicly available whole-metagenome shotgun sequencing datasets (obtained from the Sequence Read Archive):</p>
<ol style="list-style-type: decimal">
<li>Species-level taxonomic profiles, expressed as relative abundance from kingdom to strain level</li>
<li>Presence of unique, clade-specific markers</li>
<li>Abundance of unique, clade-specific markers</li>
<li>Abundance of gene families</li>
<li>Metabolic pathway coverage</li>
<li>Metabolic pathway abundance</li>
</ol>
<p>Types 1-3 are generated by <a href="http://huttenhower.sph.harvard.edu/metaphlan2">MetaPhlAn2</a>; 4-6 are generated by <a href="http://huttenhower.sph.harvard.edu/humann2">HUMAnN2</a>.</p>
<p>Currently, <code>curatedMetagenomicData</code> provides:</p>
<ul>
<li>6386 samples from 31 datasets, primarily of the human gut but including body sites profiled in the Human Microbiome Project</li>
<li>Processed data from whole-metagenome shotgun metagenomics, with manually-curated metadata, as integrated and documented Bioconductor ExpressionSet objects</li>
<li>~80 fields of specimen metadata from original papers, supplementary files, and websites, with manual curation to standardize annotations</li>
<li>Processing of data through the <a href="http://huttenhower.sph.harvard.edu/metaphlan2">MetaPhlAn2</a> pipeline for taxonomic abundance, and <a href="http://huttenhower.sph.harvard.edu/humann2">HUMAnN2</a> pipeline for metabolic analysis</li>
<li>These represent ~100TB of raw sequencing data, but the processed data provided are much smaller.</li>
</ul>
<p>These datasets are documented in the <a href="https://bioconductor.org/packages/devel/data/experiment/manuals/curatedMetagenomicData/man/curatedMetagenomicData.pdf">reference manual</a>.</p>
<p>This is an <em><a href="http://bioconductor.org/packages/ExperimentHub">ExperimentHub</a></em> package, and its main workhorse function is <code>curatedMetagenomicData()</code>:</p>
<p>The manually curated metadata for all available samples are provided in a single table <code>combined_metadata</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(curatedMetagenomicData)
?combined_metadata
<span class="kw">View</span>(<span class="kw">data.frame</span>(combined_metadata))</code></pre></div>
<p>The main function provides a <code>list</code> of <code>ExpressionSet</code> objects:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">oral &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;BritoIL_2016.metaphlan_bugs_list.oralcavity&quot;</span>,
          <span class="st">&quot;Castro-NallarE_2015.metaphlan_bugs_list.oralcavity&quot;</span>)
esl &lt;-<span class="st"> </span><span class="kw">curatedMetagenomicData</span>(oral, <span class="dt">dryrun =</span> <span class="ot">FALSE</span>)
<span class="co">#&gt; Working on BritoIL_2016.metaphlan_bugs_list.oralcavity</span>
<span class="co">#&gt; snapshotDate(): 2018-07-17</span>
<span class="co">#&gt; see ?curatedMetagenomicData and browseVignettes(&#39;curatedMetagenomicData&#39;) for documentation</span>
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.ExperimentHub/1179&#39;</span>
<span class="co">#&gt; Working on Castro-NallarE_2015.metaphlan_bugs_list.oralcavity</span>
<span class="co">#&gt; snapshotDate(): 2018-07-17</span>
<span class="co">#&gt; see ?curatedMetagenomicData and browseVignettes(&#39;curatedMetagenomicData&#39;) for documentation</span>
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.ExperimentHub/391&#39;</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">esl
<span class="co">#&gt; List of length 2</span>
<span class="co">#&gt; names(2): BritoIL_2016.metaphlan_bugs_list.oralcavity ...</span></code></pre></div>
<p>These <code>ExpressionSet</code> objects can also be converted to <em><a href="http://bioconductor.org/packages/phyloseq">phyloseq</a></em> object for ecological analysis and differential abundance analysis using the <em><a href="http://bioconductor.org/packages/DESeq2">DESeq2</a></em> package, using the <code>ExpressionSet2phyloseq()</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ExpressionSet2phyloseq</span>( esl[[<span class="dv">1</span>]], <span class="dt">phylogenetictree =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; Loading required namespace: phyloseq</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 535 taxa and 140 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 140 samples by 17 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 535 taxa by 8 taxonomic ranks ]</span>
<span class="co">#&gt; phy_tree()    Phylogenetic Tree: [ 535 tips and 534 internal nodes ]</span></code></pre></div>
<p>See the documentation of <em><a href="http://bioconductor.org/packages/phyloseq">phyloseq</a></em> for more on ecological and differential abundance analysis of the microbiome.</p>
</div>
<div id="hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project" class="section level3">
<h3><span class="header-section-number">5.11.2</span> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(HMP16SData))</code></pre></div>
<p><em><a href="http://bioconductor.org/packages/HMP16SData">HMP16SData</a></em><span class="citation">(Schiffer et al. 2018)</span> is a Bioconductor ExperimentData package of the Human Microbiome Project (HMP) 16S rRNA sequencing data. Taxonomic count data files are provided as downloaded from the HMP Data Analysis and Coordination Center from its QIIME pipeline. Processed data is provided as <code>SummarizedExperiment</code> class objects via <em><a href="http://bioconductor.org/packages/ExperimentHub">ExperimentHub</a></em>. Like other ExperimentHub-based packages, a convenience function does downloading, automatic local caching, and serializing of a Bioconductor data class. This returns taxonomic counts from the V1-3 variable region of the 16S rRNA gene, along with the unrestricted participant data and phylogenetic tree.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">V13</span>()
<span class="co">#&gt; snapshotDate(): 2018-07-17</span>
<span class="co">#&gt; see ?HMP16SData and browseVignettes(&#39;HMP16SData&#39;) for documentation</span>
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.ExperimentHub/1117&#39;</span>
<span class="co">#&gt; class: SummarizedExperiment </span>
<span class="co">#&gt; dim: 43140 2898 </span>
<span class="co">#&gt; metadata(2): experimentData phylogeneticTree</span>
<span class="co">#&gt; assays(1): 16SrRNA</span>
<span class="co">#&gt; rownames(43140): OTU_97.1 OTU_97.10 ... OTU_97.9997 OTU_97.9999</span>
<span class="co">#&gt; rowData names(7): CONSENSUS_LINEAGE SUPERKINGDOM ... FAMILY GENUS</span>
<span class="co">#&gt; colnames(2898): 700013549 700014386 ... 700114963 700114965</span>
<span class="co">#&gt; colData names(7): RSID VISITNO ... HMP_BODY_SUBSITE SRS_SAMPLE_ID</span></code></pre></div>
<p>This can also be converted to <em><a href="http://bioconductor.org/packages/phyloseq">phyloseq</a></em> for ecological and differential abundance analysis; see the <em><a href="http://bioconductor.org/packages/HMP16SData">HMP16SData</a></em> vignette for details.</p>
</div>
</div>
<div id="pharmacogenomics" class="section level2">
<h2><span class="header-section-number">5.12</span> Pharmacogenomics</h2>
<p>Pharmacogenomics holds great promise for the development of biomarkers of drug response and the design of new therapeutic options, which are key challenges in precision medicine. However, such data are scattered and lack standards for efficient access and analysis, consequently preventing the realization of the full potential of pharmacogenomics. To address these issues, we implemented <em><a href="http://bioconductor.org/packages/PharmacoGx">PharmacoGx</a></em>, an easy-to-use, open source package for integrative analysis of multiple pharmacogenomic datasets. ~PharmacoGx` provides a unified framework for downloading and analyzing large pharmacogenomic datasets which are extensively curated to ensure maximum overlap and consistency.</p>
<p>Examples of <code>PharmacoGx</code> usage in biomedical research can be found in the following publications: * <a href="https://academic.oup.com/bioinformatics/article/32/8/1244/1744214/PharmacoGx-an-R-package-for-analysis-of-large">Smirnov et al. PharmacoGx: an R package for analysis of large pharmacogenomic datasets.&quot; Bioinformatics (2015): 1244-1246</a>. * <a href="https://f1000research.com/articles/5-825/v1">Safikhani et al., Assessment of pharmacogenomic agreement, F1000 Research (2017)</a> * <a href="https://f1000research.com/articles/5-2333/v3">Safikhani et al., Revisiting inconsistency in large pharmacogenomic studies, F1000 Research (2017)</a> * <a href="https://doi.org/10.1093/jamia/ocx062">Yao et al., Tissue specificity of in vitro drug sensitivity, JAMIA (2017)</a> * <a href="https://www.nature.com/articles/s41467-017-01153-8">Safikhani et al., Gene isoforms as expression-based biomarkers predictive of drug response in vitro, Nature Communications (2017)</a> * <a href="http://cancerres.aacrjournals.org/content/77/11/3057.long">El-Hachem et al., Integrative Cancer Pharmacogenomics to Infer Large-Scale Drug Taxonomy, Cancer Research (2017)</a></p>
<div id="getting-started" class="section level3">
<h3><span class="header-section-number">5.12.1</span> Getting started</h3>
<p>Let us first load the <code>PharmacoGx</code> library.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(PharmacoGx)</code></pre></div>
<p>We can now access large-scale preclinical pharmacogenomic datasets that have been fully curated for ease of use.</p>
</div>
<div id="overview-of-pharmacogx-datasets-pharmacosets" class="section level3">
<h3><span class="header-section-number">5.12.2</span> Overview of PharmacoGx datasets (PharmacoSets)</h3>
<p>To efficiently store and analyze large pharmacogenomic datasets, we developed the <code>PharmacoSet</code> class (also referred to as <code>PSet</code>), which acts as a data container storing pharmacological and molecular data along with experimental metadata (detailed structure provided in Supplementary materials). This class enables efficient implementation of curated annotations for cell lines, drug compounds and molecular features, which facilitates comparisons between different datasets stored as <code>PharmacoSet</code> objects.</p>
<p>We have made the PharmacoSet objects of the curated datasets available for download using functions provided in the package. A table of available PharmacoSet objects can be obtained by using the <em>availablePSets</em> function. Any of the PharmacoSets in the table can then be downloaded by calling <em>downloadPSet</em>, which saves the datasets into a directory of the users choice, and returns the data into the R session.</p>
<p><a href="Waldron_PublicData/PharmacoSet_structure.png">Structure of the PharmacoSet class</a></p>
<p>To get a list of all the available <code>PharmacoSets</code> in PharmacoGx, we can use the availablePSets` function, which returns a table providing key information for each dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">availablePSets</span>(<span class="dt">saveDir=</span><span class="kw">file.path</span>(<span class="st">&quot;.&quot;</span>, <span class="st">&quot;Waldron_PublicData&quot;</span>))
<span class="co">#&gt;                       PSet.Name Dataset.Type Available.Molecular.Profiles</span>
<span class="co">#&gt; CCLE_2013             CCLE_2013  sensitivity                 rna/mutation</span>
<span class="co">#&gt; CCLE                       CCLE  sensitivity      rna/rnaseq/mutation/cnv</span>
<span class="co">#&gt; GDSC_2013             GDSC_2013  sensitivity                 rna/mutation</span>
<span class="co">#&gt; GDSC                       GDSC  sensitivity rna/rna2/mutation/fusion/cnv</span>
<span class="co">#&gt; GDSC1000               GDSC1000  sensitivity                          rna</span>
<span class="co">#&gt; gCSI                       gCSI  sensitivity                   rnaseq/cnv</span>
<span class="co">#&gt; FIMM                       FIMM  sensitivity                             </span>
<span class="co">#&gt; CTRPv2                   CTRPv2  sensitivity                             </span>
<span class="co">#&gt; CMAP                       CMAP perturbation                          rna</span>
<span class="co">#&gt; L1000_compounds L1000_compounds perturbation                          rna</span>
<span class="co">#&gt; L1000_genetic     L1000_genetic perturbation                          rna</span>
<span class="co">#&gt;                             Date.Updated</span>
<span class="co">#&gt; CCLE_2013       Tue Sep 15 18:50:07 2015</span>
<span class="co">#&gt; CCLE            Thu Dec 10 18:17:14 2015</span>
<span class="co">#&gt; GDSC_2013       Mon Oct  5 16:07:54 2015</span>
<span class="co">#&gt; GDSC            Wed Dec 30 10:44:21 2015</span>
<span class="co">#&gt; GDSC1000        Thu Aug 25 11:13:00 2016</span>
<span class="co">#&gt; gCSI            Mon Jun 13 18:50:12 2016</span>
<span class="co">#&gt; FIMM             Mon Oct 3 17:14:00 2016</span>
<span class="co">#&gt; CTRPv2          Thu Aug 25 11:15:00 2016</span>
<span class="co">#&gt; CMAP            Mon Sep 21 02:38:45 2015</span>
<span class="co">#&gt; L1000_compounds Mon Jan 25 12:51:00 2016</span>
<span class="co">#&gt; L1000_genetic   Mon Jan 25 12:51:00 2016</span>
<span class="co">#&gt;                                                                                                  URL</span>
<span class="co">#&gt; CCLE_2013       https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CCLE_Nature2013.RData</span>
<span class="co">#&gt; CCLE                       https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CCLE.RData</span>
<span class="co">#&gt; GDSC_2013        https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CGP_Nature2013.RData</span>
<span class="co">#&gt; GDSC                       https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/GDSC.RData</span>
<span class="co">#&gt; GDSC1000               https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/GDSC1000.RData</span>
<span class="co">#&gt; gCSI                       https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/gCSI.RData</span>
<span class="co">#&gt; FIMM                       https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/FIMM.RData</span>
<span class="co">#&gt; CTRPv2                   https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CTRPv2.RData</span>
<span class="co">#&gt; CMAP                       https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CMAP.RData</span>
<span class="co">#&gt; L1000_compounds https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/L1000_compounds.RData</span>
<span class="co">#&gt; L1000_genetic     https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/L1000_genetic.RData</span></code></pre></div>
<p>There are currently 11 datasets available in PharmacoGx, including 8 sensitivity datasets and 3 perturbation datasets (see below).</p>
<div id="drug-sensitivity-datasets" class="section level4">
<h4><span class="header-section-number">5.12.2.1</span> Drug Sensitivity Datasets</h4>
<p>Drug sensitivity datasets refer to pharmacogenomic data where cancer cells are molecularly profiled at baseline (before drug treatment), and the effect of drug treatment on cell viability is measured using a pharmacological assay (e.g., Cell Titer-Glo). These datasets can be used for biomarker discovery by correlating the molecular features of cancer cells to their response to drugs of interest.</p>
<p><a href="Waldron_PublicData/Drug_sensitivity.png">Schematic view of the drug sensitivity datasets.</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">psets &lt;-<span class="st"> </span><span class="kw">availablePSets</span>(<span class="dt">saveDir=</span><span class="kw">file.path</span>(<span class="st">&quot;.&quot;</span>, <span class="st">&quot;Waldron_PublicData&quot;</span>))
psets[psets[ , <span class="st">&quot;Dataset.Type&quot;</span>] <span class="op">==</span><span class="st"> &quot;sensitivity&quot;</span>, ]
<span class="co">#&gt;           PSet.Name Dataset.Type Available.Molecular.Profiles</span>
<span class="co">#&gt; CCLE_2013 CCLE_2013  sensitivity                 rna/mutation</span>
<span class="co">#&gt; CCLE           CCLE  sensitivity      rna/rnaseq/mutation/cnv</span>
<span class="co">#&gt; GDSC_2013 GDSC_2013  sensitivity                 rna/mutation</span>
<span class="co">#&gt; GDSC           GDSC  sensitivity rna/rna2/mutation/fusion/cnv</span>
<span class="co">#&gt; GDSC1000   GDSC1000  sensitivity                          rna</span>
<span class="co">#&gt; gCSI           gCSI  sensitivity                   rnaseq/cnv</span>
<span class="co">#&gt; FIMM           FIMM  sensitivity                             </span>
<span class="co">#&gt; CTRPv2       CTRPv2  sensitivity                             </span>
<span class="co">#&gt;                       Date.Updated</span>
<span class="co">#&gt; CCLE_2013 Tue Sep 15 18:50:07 2015</span>
<span class="co">#&gt; CCLE      Thu Dec 10 18:17:14 2015</span>
<span class="co">#&gt; GDSC_2013 Mon Oct  5 16:07:54 2015</span>
<span class="co">#&gt; GDSC      Wed Dec 30 10:44:21 2015</span>
<span class="co">#&gt; GDSC1000  Thu Aug 25 11:13:00 2016</span>
<span class="co">#&gt; gCSI      Mon Jun 13 18:50:12 2016</span>
<span class="co">#&gt; FIMM       Mon Oct 3 17:14:00 2016</span>
<span class="co">#&gt; CTRPv2    Thu Aug 25 11:15:00 2016</span>
<span class="co">#&gt;                                                                                            URL</span>
<span class="co">#&gt; CCLE_2013 https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CCLE_Nature2013.RData</span>
<span class="co">#&gt; CCLE                 https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CCLE.RData</span>
<span class="co">#&gt; GDSC_2013  https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CGP_Nature2013.RData</span>
<span class="co">#&gt; GDSC                 https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/GDSC.RData</span>
<span class="co">#&gt; GDSC1000         https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/GDSC1000.RData</span>
<span class="co">#&gt; gCSI                 https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/gCSI.RData</span>
<span class="co">#&gt; FIMM                 https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/FIMM.RData</span>
<span class="co">#&gt; CTRPv2             https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CTRPv2.RData</span></code></pre></div>
<p>Notably, the Genomics of Drug Sensitivity in Cancer <a href="https://www.cancerrxgene.org/">GDSC</a>and the Cancer Cell Line Encyclopedia <a href="https://portals.broadinstitute.org/ccle">CCLE</a> are large drug sensitivity datasets published in seminal studies in Nature, <a href="https://www.nature.com/articles/nature11005">Garnett et al., https://www.nature.com/articles/nature11005, Nature (2012)</a> and <a href="https://www.nature.com/articles/nature11003">Barretina et al., The Cancer Cell Line Encyclopedia enables predictive modelling of anticancer drug sensitivity, Nature (2012)</a>, respectively.</p>
</div>
<div id="drug-perturbation-datasets" class="section level4">
<h4><span class="header-section-number">5.12.2.2</span> Drug Perturbation Datasets</h4>
<p>Drug perturbation datasets refer to pharmacogenomic data where gene expression profiles are measured before and after short-term (e.g., 6h) drug treatment to identify genes that are up- and down-regulated due to the drug treatment. These datasets can be to classify drug (drug taxonomy), infer their mechanism of action, or find drugs with similar effects (drug repurposing).</p>
<p><a href="Waldron_PublicData/Drug_perturbation.png">Schematic view of drug perturbation datasets</a></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">psets &lt;-<span class="st"> </span><span class="kw">availablePSets</span>(<span class="dt">saveDir=</span><span class="kw">file.path</span>(<span class="st">&quot;.&quot;</span>, <span class="st">&quot;Waldron_PublicData&quot;</span>))
psets[psets[ , <span class="st">&quot;Dataset.Type&quot;</span>] <span class="op">==</span><span class="st"> &quot;perturbation&quot;</span>, ]
<span class="co">#&gt;                       PSet.Name Dataset.Type Available.Molecular.Profiles</span>
<span class="co">#&gt; CMAP                       CMAP perturbation                          rna</span>
<span class="co">#&gt; L1000_compounds L1000_compounds perturbation                          rna</span>
<span class="co">#&gt; L1000_genetic     L1000_genetic perturbation                          rna</span>
<span class="co">#&gt;                             Date.Updated</span>
<span class="co">#&gt; CMAP            Mon Sep 21 02:38:45 2015</span>
<span class="co">#&gt; L1000_compounds Mon Jan 25 12:51:00 2016</span>
<span class="co">#&gt; L1000_genetic   Mon Jan 25 12:51:00 2016</span>
<span class="co">#&gt;                                                                                                  URL</span>
<span class="co">#&gt; CMAP                       https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/CMAP.RData</span>
<span class="co">#&gt; L1000_compounds https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/L1000_compounds.RData</span>
<span class="co">#&gt; L1000_genetic     https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/L1000_genetic.RData</span></code></pre></div>
<p>Large drug perturbation data have been generated within the Connectivity Map Project <a href="https://clue.io/cmap">CAMP</a>, with CMAPv2 and CMAPv3 available from <code>PharmacoGx</code>, published in <a href="http://science.sciencemag.org/content/313/5795/1929">Lamb et al., The Connectivity Map: Using Gene-Expression Signatures to Connect Small Molecules, Genes, and Disease, Science (2006)</a> and <a href="https://www.cell.com/cell/abstract/S0092-8674(17)31309-0">Subramanian et al., A Next Generation Connectivity Map: L1000 Platform and the First 1,000,000 Profiles, Cell (2017)</a>, respectively.</p>
</div>
</div>
<div id="exploring-drug-sensitivity-datasets" class="section level3">
<h3><span class="header-section-number">5.12.3</span> Exploring drug sensitivity datasets</h3>
<p>The <a href="https://bioconductor.github.io/BiocWorkshops/biomarker-discovery-from-large-pharmacogenomics-datasets.html">Biomarker discovery from large pharmacogenomics datasets</a> workshop demonstrates analyses of <em><a href="http://bioconductor.org/packages/PharmacoGx">PharmacoGx</a></em> data.</p>
</div>
</div>
<div id="bibliography" class="section level2">
<h2><span class="header-section-number">5.13</span> Bibliography</h2>

</div>
</div>
<div class="footnotes">
<hr />
<ol start="3">
<li id="fn3"><p>See individual function and methods documentation for specific details.<a href="public-data-resources-and-bioconductor.html#fnref3"></a></p></li>
<li id="fn4"><p><a href="https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-Q%C2%A0Whatreferencegenomebuildareyouusing">https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-Q%C2%A0Whatreferencegenomebuildareyouusing</a><a href="public-data-resources-and-bioconductor.html#fnref4"></a></p></li>
<li id="fn5"><p><a href="https://gdc.cancer.gov/about-data/data-harmonization-and-generation/genomic-data-harmonization-0" class="uri">https://gdc.cancer.gov/about-data/data-harmonization-and-generation/genomic-data-harmonization-0</a><a href="public-data-resources-and-bioconductor.html#fnref5"></a></p></li>
<li id="fn6"><p><a href="https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-Q%C2%A0Whatreferencegenomebuildareyouusing">https://confluence.broadinstitute.org/display/GDAC/FAQ#FAQ-Q%C2%A0Whatreferencegenomebuildareyouusing</a><a href="public-data-resources-and-bioconductor.html#fnref6"></a></p></li>
<li id="fn7"><p><a href="https://bioconductor.org/packages/release/bioc/vignettes/TCGAbiolinks/inst/doc/download_prepare.html" class="uri">https://bioconductor.org/packages/release/bioc/vignettes/TCGAbiolinks/inst/doc/download_prepare.html</a><a href="public-data-resources-and-bioconductor.html#fnref7"></a></p></li>
<li id="fn8"><p><a href="https://jhubiostatistics.shinyapps.io/recount/" class="uri">https://jhubiostatistics.shinyapps.io/recount/</a><a href="public-data-resources-and-bioconductor.html#fnref8"></a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="solving-common-bioinformatic-challenges-using-genomicranges.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
