<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7.14 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html">
<link rel="next" href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-workshops"><i class="fa fa-check"></i><b>1.1</b> The Workshops</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html"><i class="fa fa-check"></i><b>2</b> 100: <em>R</em> and <em>Bioconductor</em> for everyone: an introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#description"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="2.1.7" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-goals"><i class="fa fa-check"></i><b>2.1.7</b> Learning goals</a></li>
<li class="chapter" data-level="2.1.8" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-objectives"><i class="fa fa-check"></i><b>2.1.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-r"><i class="fa fa-check"></i><b>2.2</b> Introduction to <em>R</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#rstudio-orientation"><i class="fa fa-check"></i><b>2.2.1</b> <em>RStudio</em> orientation</a></li>
<li class="chapter" data-level="2.2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#very-basics"><i class="fa fa-check"></i><b>2.2.2</b> Very basics</a></li>
<li class="chapter" data-level="2.2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#data-input-and-manipulation"><i class="fa fa-check"></i><b>2.2.3</b> Data input and manipulation</a></li>
<li class="chapter" data-level="2.2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#writing-r-scripts"><i class="fa fa-check"></i><b>2.2.4</b> Writing <em>R</em> scripts</a></li>
<li class="chapter" data-level="2.2.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#using-packages"><i class="fa fa-check"></i><b>2.2.5</b> Using packages</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-bioconductor"><i class="fa fa-check"></i><b>2.3</b> Introduction to <em>Bioconductor</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#discovering-installing-and-learning-how-to-use-bioconductor-packages"><i class="fa fa-check"></i><b>2.3.1</b> Discovering, installing, and learning how to use <em>Bioconductor</em> packages</a></li>
<li class="chapter" data-level="2.3.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-genomic-ranges"><i class="fa fa-check"></i><b>2.3.2</b> Working with Genomic Ranges</a></li>
<li class="chapter" data-level="2.3.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-summarized-experimental-data"><i class="fa fa-check"></i><b>2.3.3</b> Working with summarized experimental data</a></li>
<li class="chapter" data-level="2.3.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#down-stream-analysis"><i class="fa fa-check"></i><b>2.3.4</b> Down-stream analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>3</b> 101: Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>3.1</b> Instructors</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description"><i class="fa fa-check"></i><b>3.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="3.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>3.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="3.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation"><i class="fa fa-check"></i><b>3.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="3.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="3.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-1"><i class="fa fa-check"></i><b>3.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="3.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-1"><i class="fa fa-check"></i><b>3.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>3.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="3.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>3.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="3.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>3.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="3.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>3.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="3.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>3.4.4</b> Data containers</a></li>
<li class="chapter" data-level="3.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>3.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="3.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>3.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="3.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>3.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="3.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>3.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="3.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>3.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="3.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>3.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="3.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>3.4.11</b> Simple example</a></li>
<li class="chapter" data-level="3.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>3.4.12</b> Questions!</a></li>
<li class="chapter" data-level="3.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>3.4.13</b> More questions!</a></li>
<li class="chapter" data-level="3.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>3.4.14</b> Another example</a></li>
<li class="chapter" data-level="3.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>3.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="3.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>3.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="3.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>3.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="3.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>3.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="3.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>3.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="3.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>3.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="3.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>3.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="3.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>3.4.22</b> GRanges</a></li>
<li class="chapter" data-level="3.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>3.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="3.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>3.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="3.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>3.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="3.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>3.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="3.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>3.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="3.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>3.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="3.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>3.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="3.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>3.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="3.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>3.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="3.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>3.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="3.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr"><i class="fa fa-check"></i><b>3.4.33</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-1"><i class="fa fa-check"></i><b>3.4.34</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-2"><i class="fa fa-check"></i><b>3.4.35</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>3.4.36</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="3.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>3.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>3.4.38</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>3.4.39</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>3.4.40</b> BSgenome exercises</a></li>
<li class="chapter" data-level="3.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub"><i class="fa fa-check"></i><b>3.4.41</b> AnnotationHub</a></li>
<li class="chapter" data-level="3.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub"><i class="fa fa-check"></i><b>3.4.42</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="3.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>3.4.43</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="3.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>3.4.44</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="3.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>3.4.45</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="3.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>3.4.46</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="3.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>3.4.47</b> AnnotationHub query</a></li>
<li class="chapter" data-level="3.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>3.4.48</b> AnnotationHub query</a></li>
<li class="chapter" data-level="3.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>3.4.49</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="3.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>3.4.50</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="3.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>3.4.51</b> biomaRt</a></li>
<li class="chapter" data-level="3.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>3.4.52</b> biomaRt data sets</a></li>
<li class="chapter" data-level="3.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>3.4.53</b> biomaRt queries</a></li>
<li class="chapter" data-level="3.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>3.4.54</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="3.4.55" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>3.4.55</b> biomaRt query</a></li>
<li class="chapter" data-level="3.4.56" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>3.4.56</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>4</b> 102: Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructors-names-and-contact-information"><i class="fa fa-check"></i><b>4.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-1"><i class="fa fa-check"></i><b>4.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="4.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-1"><i class="fa fa-check"></i><b>4.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-1"><i class="fa fa-check"></i><b>4.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="4.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-1"><i class="fa fa-check"></i><b>4.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="4.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-2"><i class="fa fa-check"></i><b>4.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="4.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-2"><i class="fa fa-check"></i><b>4.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#introduction-1"><i class="fa fa-check"></i><b>4.4</b> Introduction</a><ul>
<li class="chapter" data-level="4.4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#what-is-the-ranges-infrastructure"><i class="fa fa-check"></i><b>4.4.1</b> What is the Ranges infrastructure?</a></li>
<li class="chapter" data-level="4.4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#why-use-the-ranges-infrastructure"><i class="fa fa-check"></i><b>4.4.2</b> Why use the Ranges infrastructure?</a></li>
<li class="chapter" data-level="4.4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>4.4.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#setup"><i class="fa fa-check"></i><b>4.5</b> Setup</a></li>
<li class="chapter" data-level="4.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>4.6</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="4.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#constructing-a-granges-object-from-data.frame"><i class="fa fa-check"></i><b>4.7</b> Constructing a <em>GRanges</em> object from data.frame</a></li>
<li class="chapter" data-level="4.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#loading-a-granges-object-from-a-standard-file-format"><i class="fa fa-check"></i><b>4.8</b> Loading a <em>GRanges</em> object from a standard file format</a></li>
<li class="chapter" data-level="4.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-manipulation-of-granges-objects"><i class="fa fa-check"></i><b>4.9</b> Basic manipulation of <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>4.10</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>4.11</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#aggregating-granges-objects"><i class="fa fa-check"></i><b>4.12</b> Aggregating <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>4.13</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>4.14</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.15" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#finding-overlaps-between-granges-objects"><i class="fa fa-check"></i><b>4.15</b> Finding overlaps between <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.16" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises"><i class="fa fa-check"></i><b>4.16</b> Exercises</a></li>
<li class="chapter" data-level="4.17" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>4.17</b> Example: exploring BigWig files from AnnotationHub</a><ul>
<li class="chapter" data-level="4.17.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#extracting-data-from-annotationhub"><i class="fa fa-check"></i><b>4.17.1</b> Extracting data from AnnotationHub</a></li>
<li class="chapter" data-level="4.17.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#reading-bigwig-files"><i class="fa fa-check"></i><b>4.17.2</b> Reading BigWig files</a></li>
<li class="chapter" data-level="4.17.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-1"><i class="fa fa-check"></i><b>4.17.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4.18" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>4.18</b> Worked example: coverage analysis of BAM files</a><ul>
<li class="chapter" data-level="4.18.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-2"><i class="fa fa-check"></i><b>4.18.1</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4.19" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#conclusions"><i class="fa fa-check"></i><b>4.19</b> Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>5</b> 103: Public data resources and Bioconductor</a><ul>
<li class="chapter" data-level="5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>5.1</b> Instructor name(s) and contact information</a></li>
<li class="chapter" data-level="5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>5.2</b> Syllabus</a><ul>
<li class="chapter" data-level="5.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-2"><i class="fa fa-check"></i><b>5.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="5.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-2"><i class="fa fa-check"></i><b>5.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="5.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-2"><i class="fa fa-check"></i><b>5.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>5.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="5.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-2"><i class="fa fa-check"></i><b>5.2.5</b> Time outline</a></li>
<li class="chapter" data-level="5.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>5.2.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="5.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-3"><i class="fa fa-check"></i><b>5.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="5.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-3"><i class="fa fa-check"></i><b>5.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-1"><i class="fa fa-check"></i><b>5.3</b> Overview</a></li>
<li class="chapter" data-level="5.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>5.4</b> GEOquery</a><ul>
<li class="chapter" data-level="5.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>5.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="5.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>5.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="5.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>5.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="5.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>5.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>5.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="5.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>5.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="5.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>5.5.2</b> Basic design</a></li>
<li class="chapter" data-level="5.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>5.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>5.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="5.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>5.6.1</b> Authentication</a></li>
<li class="chapter" data-level="5.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>5.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>5.7</b> Sequence Read Archive</a><ul>
<li class="chapter" data-level="5.7.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage-1"><i class="fa fa-check"></i><b>5.7.1</b> Usage</a></li>
<li class="chapter" data-level="5.7.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#query-syntax"><i class="fa fa-check"></i><b>5.7.2</b> Query syntax</a></li>
<li class="chapter" data-level="5.7.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#using-the-raw-api-without-rbioconductor"><i class="fa fa-check"></i><b>5.7.3</b> Using the raw API without R/Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>5.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="5.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>5.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="5.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>5.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="5.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>5.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="5.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>5.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="5.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>5.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="5.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>5.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="5.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>5.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>5.12</b> Pharmacogenomics</a><ul>
<li class="chapter" data-level="5.12.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started"><i class="fa fa-check"></i><b>5.12.1</b> Getting started</a></li>
<li class="chapter" data-level="5.12.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-pharmacogx-datasets-pharmacosets"><i class="fa fa-check"></i><b>5.12.2</b> Overview of PharmacoGx datasets (PharmacoSets)</a></li>
<li class="chapter" data-level="5.12.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#exploring-a-drug-sensitivity-dataset"><i class="fa fa-check"></i><b>5.12.3</b> Exploring a drug sensitivity dataset</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>5.13</b> Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>6</b> 200: RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>6.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-3"><i class="fa fa-check"></i><b>6.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="6.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-3"><i class="fa fa-check"></i><b>6.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="6.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>6.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-3"><i class="fa fa-check"></i><b>6.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>6.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="6.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals-4"><i class="fa fa-check"></i><b>6.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="6.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-4"><i class="fa fa-check"></i><b>6.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-2"><i class="fa fa-check"></i><b>6.4</b> Introduction</a></li>
<li class="chapter" data-level="6.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>6.5</b> Data packaging</a><ul>
<li class="chapter" data-level="6.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>6.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="6.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>6.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="6.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>6.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>6.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="6.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>6.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="6.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>6.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="6.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>6.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="6.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>6.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis"><i class="fa fa-check"></i><b>6.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="6.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>6.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="6.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>6.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="6.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>6.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="6.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>6.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="6.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>6.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="6.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>6.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>7</b> 201: RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview-2"><i class="fa fa-check"></i><b>7.1</b> Overview</a><ul>
<li class="chapter" data-level="7.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-1"><i class="fa fa-check"></i><b>7.1.1</b> Description</a></li>
<li class="chapter" data-level="7.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation-1"><i class="fa fa-check"></i><b>7.1.3</b> Participation</a></li>
<li class="chapter" data-level="7.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>7.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline-4"><i class="fa fa-check"></i><b>7.1.5</b> Time outline</a></li>
<li class="chapter" data-level="7.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>7.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>7.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="7.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>7.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="7.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>7.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="7.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>7.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="7.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>7.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>7.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="7.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>7.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="7.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>7.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="7.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>7.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>7.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="7.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>7.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="7.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>7.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="7.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>7.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>7.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="7.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>7.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="7.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>7.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub-1"><i class="fa fa-check"></i><b>7.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="7.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>7.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="7.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>7.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>7.7</b> Building reports</a><ul>
<li class="chapter" data-level="7.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>7.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="7.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>7.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>7.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="7.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>7.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="7.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>7.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="7.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>7.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="7.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>7.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="7.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>7.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="7.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>7.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>8</b> 202: Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="8.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-3"><i class="fa fa-check"></i><b>8.1</b> Overview</a><ul>
<li class="chapter" data-level="8.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-2"><i class="fa fa-check"></i><b>8.1.1</b> Description</a></li>
<li class="chapter" data-level="8.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-2"><i class="fa fa-check"></i><b>8.1.3</b> Participation</a></li>
<li class="chapter" data-level="8.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>8.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-5"><i class="fa fa-check"></i><b>8.1.5</b> Time outline</a></li>
<li class="chapter" data-level="8.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>8.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started-1"><i class="fa fa-check"></i><b>8.2</b> Getting started</a><ul>
<li class="chapter" data-level="8.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>8.2.1</b> The data</a></li>
<li class="chapter" data-level="8.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>8.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>8.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="8.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>8.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="8.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>8.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>8.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="8.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>8.5.1</b> Normalization</a></li>
<li class="chapter" data-level="8.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>8.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>8.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="8.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>8.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="8.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>8.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>9</b> 210: Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="9.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>9.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-4"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-6"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-4"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-6"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>9.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="9.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop"><i class="fa fa-check"></i><b>9.4</b> Workshop</a></li>
<li class="chapter" data-level="9.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>9.5</b> Where does it all come from?</a></li>
<li class="chapter" data-level="9.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.6</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="9.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>9.7</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="9.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>9.8</b> Data types</a></li>
<li class="chapter" data-level="9.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>9.9</b> Differential expression analysis</a></li>
<li class="chapter" data-level="9.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>9.10</b> Gene sets</a></li>
<li class="chapter" data-level="9.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>9.11</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="9.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>9.12</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="9.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.13</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="9.14" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>9.14</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>10</b> 220: Workflow for multi-omics analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="10.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information-2"><i class="fa fa-check"></i><b>10.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-5"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-7"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-5"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>10.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-7"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-5"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-5"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>10.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="10.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="10.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>10.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="10.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>10.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>10.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="10.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>10.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="10.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>10.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="10.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>10.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>10.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="10.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>10.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="10.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>10.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="10.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>10.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="10.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>10.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="10.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>10.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>10.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="10.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>10.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="10.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>10.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="10.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>10.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>10.9</b> Complete cases</a></li>
<li class="chapter" data-level="10.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>10.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="10.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>10.11</b> Extraction</a><ul>
<li class="chapter" data-level="10.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>10.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="10.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>10.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="10.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>10.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="10.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>10.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="10.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>10.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="10.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>10.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="10.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>10.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="10.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>10.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="10.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>10.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>10.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="10.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>10.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="10.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>10.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="10.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>10.16.1</b> Simplification of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="10.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>10.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="10.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>10.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="10.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>10.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="10.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>10.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="10.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>10.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="10.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>10.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="10.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>10.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="10.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>10.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="10.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>10.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="10.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>10.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="10.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>10.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>11</b> 230: Cytoscape automation in R using Rcy3</a><ul>
<li class="chapter" data-level="11.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-4"><i class="fa fa-check"></i><b>11.1</b> Overview</a><ul>
<li class="chapter" data-level="11.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i><b>11.1.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="11.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description-6"><i class="fa fa-check"></i><b>11.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="11.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>11.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="11.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>11.1.4</b> Background:</a></li>
<li class="chapter" data-level="11.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation-6"><i class="fa fa-check"></i><b>11.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>11.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="11.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-8"><i class="fa fa-check"></i><b>11.1.7</b> Time outline</a></li>
<li class="chapter" data-level="11.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>11.2</b> Background</a><ul>
<li class="chapter" data-level="11.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>11.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="11.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>11.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="11.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>11.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>11.3</b> Translating biological data into Cytoscape using RCy3</a></li>
<li class="chapter" data-level="11.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>11.4</b> Set up</a><ul>
<li class="chapter" data-level="11.4.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started-2"><i class="fa fa-check"></i><b>11.4.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>11.5</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="11.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>11.6</b> Example Data Set</a></li>
<li class="chapter" data-level="11.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>11.7</b> Finding Network Data</a></li>
<li class="chapter" data-level="11.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>11.8</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="11.9" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>11.9</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="11.10" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>11.10</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>12</b> 240: Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="12.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructors-names-and-contact-information-2"><i class="fa fa-check"></i><b>12.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="12.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-7"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-8"><i class="fa fa-check"></i><b>12.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-7"><i class="fa fa-check"></i><b>12.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>12.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="12.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-9"><i class="fa fa-check"></i><b>12.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-6"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-6"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-1"><i class="fa fa-check"></i><b>12.4</b> Workshop</a></li>
<li class="chapter" data-level="12.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-3"><i class="fa fa-check"></i><b>12.5</b> Introduction</a><ul>
<li class="chapter" data-level="12.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>12.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="12.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>12.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="12.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for-1"><i class="fa fa-check"></i><b>12.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup-1"><i class="fa fa-check"></i><b>12.6</b> Setup</a></li>
<li class="chapter" data-level="12.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>12.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="12.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>12.8</b> The Grammar</a><ul>
<li class="chapter" data-level="12.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>12.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="12.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>12.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="12.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises-3"><i class="fa fa-check"></i><b>12.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>12.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="12.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>12.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="12.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files-1"><i class="fa fa-check"></i><b>12.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="12.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>12.10</b> Whats next?</a></li>
<li class="chapter" data-level="12.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>12.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html"><i class="fa fa-check"></i><b>13</b> 250: Working with genomic data in R with the DECIPHER package</a><ul>
<li class="chapter" data-level="13.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#overview-5"><i class="fa fa-check"></i><b>13.1</b> Overview</a><ul>
<li class="chapter" data-level="13.1.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-description-8"><i class="fa fa-check"></i><b>13.1.1</b> Workshop Description</a></li>
<li class="chapter" data-level="13.1.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#pre-requisites-9"><i class="fa fa-check"></i><b>13.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.1.3" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-participation-8"><i class="fa fa-check"></i><b>13.1.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.1.4" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#r-bioconductor-packages-used-9"><i class="fa fa-check"></i><b>13.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="13.1.5" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#time-outline-10"><i class="fa fa-check"></i><b>13.1.5</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.2</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="13.2.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-goals-7"><i class="fa fa-check"></i><b>13.2.1</b> Learning goals</a></li>
<li class="chapter" data-level="13.2.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-objectives-7"><i class="fa fa-check"></i><b>13.2.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>14</b> 260: Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="14.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>14.1</b> Instructors:</a></li>
<li class="chapter" data-level="14.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-9"><i class="fa fa-check"></i><b>14.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-10"><i class="fa fa-check"></i><b>14.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-9"><i class="fa fa-check"></i><b>14.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="14.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-10"><i class="fa fa-check"></i><b>14.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="14.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-11"><i class="fa fa-check"></i><b>14.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-11"><i class="fa fa-check"></i><b>14.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="14.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-8"><i class="fa fa-check"></i><b>14.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="14.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-8"><i class="fa fa-check"></i><b>14.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#abstract"><i class="fa fa-check"></i><b>14.4</b> Abstract</a></li>
<li class="chapter" data-level="14.5" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#introduction-5"><i class="fa fa-check"></i><b>14.5</b> Introduction</a><ul>
<li class="chapter" data-level="14.5.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#downloading-pharmacoset-objects"><i class="fa fa-check"></i><b>14.5.1</b> Downloading PharmacoSet objects</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#reproducibility"><i class="fa fa-check"></i><b>14.6</b> Reproducibility</a><ul>
<li class="chapter" data-level="14.6.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#plotting-drug-dose-response-data"><i class="fa fa-check"></i><b>14.6.1</b> Plotting Drug-Dose Response Data</a></li>
<li class="chapter" data-level="14.6.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pharmacological-profiles"><i class="fa fa-check"></i><b>14.6.2</b> Pharmacological profiles</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#replication"><i class="fa fa-check"></i><b>14.7</b> Replication</a><ul>
<li class="chapter" data-level="14.7.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#consistency-of-pharmacological-profiles"><i class="fa fa-check"></i><b>14.7.1</b> Consistency of pharmacological profiles</a></li>
<li class="chapter" data-level="14.7.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#consistency-assessment-improved-by-modified-concordance-index"><i class="fa fa-check"></i><b>14.7.2</b> consistency assessment improved by Modified Concordance Index</a></li>
<li class="chapter" data-level="14.7.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#known-biomarkers"><i class="fa fa-check"></i><b>14.7.3</b> Known Biomarkers</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#machine-learning-and-biomarker-discovery"><i class="fa fa-check"></i><b>14.8</b> Machine Learning and Biomarker Discovery</a><ul>
<li class="chapter" data-level="14.8.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#bonus-using-the-connectivity-map-for-drug-repurposing"><i class="fa fa-check"></i><b>14.8.1</b> Bonus: Using the Connectivity Map for drug repurposing</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#session-info"><i class="fa fa-check"></i><b>14.9</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><i class="fa fa-check"></i><b>15</b> 500: Effectively using the DelayedArray framework to support the analysis of large datasets</a><ul>
<li class="chapter" data-level="15.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-6"><i class="fa fa-check"></i><b>15.1</b> Overview</a><ul>
<li class="chapter" data-level="15.1.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#description-3"><i class="fa fa-check"></i><b>15.1.1</b> Description</a></li>
<li class="chapter" data-level="15.1.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#pre-requisites-11"><i class="fa fa-check"></i><b>15.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="15.1.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#participation-3"><i class="fa fa-check"></i><b>15.1.3</b> Participation</a></li>
<li class="chapter" data-level="15.1.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#r-bioconductor-packages-used-11"><i class="fa fa-check"></i><b>15.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="15.1.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#time-outline-12"><i class="fa fa-check"></i><b>15.1.5</b> Time outline</a></li>
<li class="chapter" data-level="15.1.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-goals-and-objectives-12"><i class="fa fa-check"></i><b>15.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#introductory-material"><i class="fa fa-check"></i><b>15.2</b> Introductory material</a><ul>
<li class="chapter" data-level="15.2.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#discussion"><i class="fa fa-check"></i><b>15.2.1</b> Discussion</a></li>
<li class="chapter" data-level="15.2.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#why-learn-delayedarray"><i class="fa fa-check"></i><b>15.2.2</b> Why learn DelayedArray?</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-of-delayedarray-framework"><i class="fa fa-check"></i><b>15.3</b> Overview of DelayedArray framework</a><ul>
<li class="chapter" data-level="15.3.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-package"><i class="fa fa-check"></i><b>15.3.1</b> The <strong>DelayedArray</strong> package</a></li>
<li class="chapter" data-level="15.3.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-class"><i class="fa fa-check"></i><b>15.3.2</b> The <em>DelayedArray</em> class</a></li>
<li class="chapter" data-level="15.3.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#operating-on-delayedarray-objects"><i class="fa fa-check"></i><b>15.3.3</b> Operating on <em>DelayedArray</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#whats-out-there-already"><i class="fa fa-check"></i><b>15.4</b> Whats out there already?</a></li>
<li class="chapter" data-level="15.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#incoporating-delayedarray-into-a-package"><i class="fa fa-check"></i><b>15.5</b> Incoporating DelayedArray into a package</a><ul>
<li class="chapter" data-level="15.5.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#writing-algorithms-to-process-delayedarray-instances"><i class="fa fa-check"></i><b>15.5.1</b> Writing algorithms to process <em>DelayedArray</em> instances</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#questions-and-discussion"><i class="fa fa-check"></i><b>15.6</b> Questions and discussion</a></li>
<li class="chapter" data-level="15.7" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#todos"><i class="fa fa-check"></i><b>15.7</b> TODOs</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>16</b> 510: Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="16.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>16.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="16.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-10"><i class="fa fa-check"></i><b>16.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="16.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-12"><i class="fa fa-check"></i><b>16.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="16.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-4"><i class="fa fa-check"></i><b>16.2.2</b> Participation</a></li>
<li class="chapter" data-level="16.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>16.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="16.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-13"><i class="fa fa-check"></i><b>16.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-6"><i class="fa fa-check"></i><b>16.3</b> Introduction</a></li>
<li class="chapter" data-level="16.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>16.4</b> Why maintain your package</a></li>
<li class="chapter" data-level="16.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>16.5</b> Infrastructure used for maintenance</a></li>
<li class="chapter" data-level="16.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>16.6</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="16.6.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>16.6.1</b> Build report</a></li>
<li class="chapter" data-level="16.6.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>16.6.2</b> Landing page badges</a></li>
<li class="chapter" data-level="16.6.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>16.6.3</b> Download statistics</a></li>
<li class="chapter" data-level="16.6.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>16.6.4</b> RSS feeds</a></li>
<li class="chapter" data-level="16.6.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>16.6.5</b> Support site, Bioc-devel mailing list</a></li>
<li class="chapter" data-level="16.6.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github"><i class="fa fa-check"></i><b>16.6.6</b> GitHub</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>16.7</b> Round up of resources available</a></li>
<li class="chapter" data-level="16.8" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#acknowledgements"><i class="fa fa-check"></i><b>16.8</b> Acknowledgements</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rna-seq-data-analysis-with-deseq2" class="section level1">
<h1><span class="header-section-number">7</span> 201: RNA-seq data analysis with DESeq2</h1>
<p>Authors: Michael I. Love<a href="#fn12" class="footnoteRef" id="fnref12"><sup>12</sup></a>, Simon Anders<a href="#fn13" class="footnoteRef" id="fnref13"><sup>13</sup></a>, Wolfgang Huber<a href="#fn14" class="footnoteRef" id="fnref14"><sup>14</sup></a> Last modified: 25 June, 2018.</p>
<!-- 
  bookdown::render_book("201_Love_DESeq2.Rmd", "bookdown::gitbook") 
-->
<div id="overview-2" class="section level2">
<h2><span class="header-section-number">7.1</span> Overview</h2>
<div id="description-1" class="section level3">
<h3><span class="header-section-number">7.1.1</span> Description</h3>
<p>In this workshop, we will give a quick overview of the most useful functions in the DESeq2 package, and a basic RNA-seq analysis. We will cover: how to quantify transcript expression from FASTQ files using Salmon, import quantification from Salmon with tximport and tximeta, generate plots for quality control and exploratory data analysis EDA (also using MultiQC), perform differential expression (DE) (also using apeglm), overlap with other experimental data (using AnnotationHub), and build reports (using ReportingTools and Glimma). We will give a short example of integration of DESeq2 with the zinbwave package for single-cell RNA-seq differential expression. The workshop is designed to be a lab with plenty of time for questions throughout the lab.</p>
</div>
<div id="pre-requisites-4" class="section level3">
<h3><span class="header-section-number">7.1.2</span> Pre-requisites</h3>
<ul>
<li>Basic knowledge of R syntax</li>
</ul>
<p>Non-essential background reading:</p>
<ul>
<li>DESeq2 paper: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4302049/" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4302049/</a></li>
<li>tximport paper: <a href="https://f1000research.com/articles/4-1521/v2" class="uri">https://f1000research.com/articles/4-1521/v2</a></li>
<li>apeglm paper: <a href="https://www.biorxiv.org/content/early/2018/04/17/303255" class="uri">https://www.biorxiv.org/content/early/2018/04/17/303255</a></li>
</ul>
</div>
<div id="participation-1" class="section level3">
<h3><span class="header-section-number">7.1.3</span> Participation</h3>
<p>Students will participate by following along an Rmarkdown document, and asking questions throughout the workshop.</p>
</div>
<div id="r-bioconductor-packages-used-4" class="section level3">
<h3><span class="header-section-number">7.1.4</span> <em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li>DESeq2</li>
<li>tximport</li>
<li>apeglm</li>
<li>AnnotationHub</li>
<li>ReportingTools</li>
<li>Glimma</li>
<li>splatter</li>
<li>zinbwave</li>
</ul>
</div>
<div id="time-outline-4" class="section level3">
<h3><span class="header-section-number">7.1.5</span> Time outline</h3>
<table>
<thead>
<tr class="header">
<th align="left">Activity</th>
<th align="left">Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Overview of packages</td>
<td align="left">20m</td>
</tr>
<tr class="even">
<td align="left">Quantification and import</td>
<td align="left">20m</td>
</tr>
<tr class="odd">
<td align="left">EDA and DE</td>
<td align="left">20m</td>
</tr>
<tr class="even">
<td align="left">Downstream analysis &amp; reports</td>
<td align="left">20m</td>
</tr>
<tr class="odd">
<td align="left">ZINB-WaVE integration</td>
<td align="left">20m</td>
</tr>
<tr class="even">
<td align="left">Additional questions</td>
<td align="left">20m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives-5" class="section level3">
<h3><span class="header-section-number">7.1.6</span> Workshop goals and objectives</h3>
<p>Learning goals</p>
<ul>
<li>Visually assess quality of RNA-seq data</li>
<li>Perform basic differential analysis of RNA-seq data</li>
<li>Compare RNA-seq results with other experimental data</li>
</ul>
<p>Learning objectives</p>
<ul>
<li>Quantify transcript expression from FASTQ files</li>
<li>Import quantification into R/Bioconductor</li>
<li>Perform quality control and exploratory data analysis</li>
<li>Perform differential expression</li>
<li>Overlap with other experimental data</li>
<li>Build dynamic reports</li>
<li>Integrate DESeq2 and zinbwave for single-cell RNA-seq data</li>
</ul>
</div>
</div>
<div id="preparing-data-for-deseq2" class="section level2">
<h2><span class="header-section-number">7.2</span> Preparing data for <em>DESeq2</em></h2>
<div id="experimental-data" class="section level3">
<h3><span class="header-section-number">7.2.1</span> Experimental data</h3>
<p>The data used in this workflow is stored in the <em>airway</em> package that summarizes an RNA-seq experiment wherein airway smooth muscle cells were treated with dexamethasone, a synthetic glucocorticoid steroid with anti-inflammatory effects <span class="citation">(Himes et al. 2014)</span>. Glucocorticoids are used, for example, by people with asthma to reduce inflammation of the airways. In the experiment, four primary human airway smooth muscle cell lines were treated with 1 micromolar dexamethasone for 18 hours. For each of the four cell lines, we have a treated and an untreated sample. For more description of the experiment see the <a href="http://www.ncbi.nlm.nih.gov/pubmed/24926665">PubMed entry 24926665</a> and for raw data see the <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52778">GEO entry GSE52778</a>.</p>
<p>We will show how to import RNA-seq quantification data using an alternative dataset (the <em>tximportData</em> package which is used in the <em>tximport</em> vignette). Afterward we will load counts for the <em>airway</em> dataset, which were counted using <em>summarizeOverlaps</em> from the <em>GenomicAlignments</em> package. As described below, we recommend the <em>tximport</em> pipeline for producing count matrices, but we do not yet have a Bioconductor package containing the necessary quantification files for the <em>airway</em> dataset.</p>
</div>
<div id="modeling-count-data" class="section level3">
<h3><span class="header-section-number">7.2.2</span> Modeling count data</h3>
<p>As input, the count-based statistical methods, such as <em>DESeq2</em> <span class="citation">(Love, Huber, and Anders 2014)</span>, <em>edgeR</em> <span class="citation">(Robinson, McCarthy, and Smyth 2009)</span>, <em>limma</em> with the voom method <span class="citation">(Law et al. 2014)</span>, <em>DSS</em> <span class="citation">(Wu, Wang, and Wu 2013)</span>, <em>EBSeq</em> <span class="citation">(Leng et al. 2013)</span> and <em>baySeq</em> <span class="citation">(Hardcastle and Kelly 2010)</span>, expect input data as obtained, e.g., from RNA-seq or another high-throughput sequencing experiment, in the form of a matrix of counts. The value in the <em>i</em>-th row and the <em>j</em>-th column of the matrix tells how many reads (or fragments, for paired-end RNA-seq) have been assigned to gene <em>i</em> in sample <em>j</em>. Analogously, for other types of assays, the rows of the matrix might correspond e.g., to binding regions (with ChIP-Seq), species of bacteria (with metagenomic datasets), or peptide sequences (with quantitative mass spectrometry).</p>
<p>The values in the matrix should be counts of sequencing reads/fragments. This is important for the statistical models used by <em>DESeq2</em> and <em>edgeR</em> to hold, as only counts allow assessing the measurement precision correctly. It is important to not provide counts that were pre-normalized for sequencing depth (also called library size), as the statistical model is most powerful when applied to un-normalized counts and is designed to account for library size differences internally.</p>
</div>
<div id="transcript-abundances" class="section level3">
<h3><span class="header-section-number">7.2.3</span> Transcript abundances</h3>
<p>In this workflow, we will show how to use transcript abundances as quantified by the <a href="https://combine-lab.github.io/salmon/">Salmon</a> <span class="citation">(Patro et al. 2017)</span> software package. <em>Salmon</em> and other methods, such as <a href="http://www.cs.cmu.edu/~ckingsf/software/sailfish/">Sailfish</a> <span class="citation">(Patro, Mount, and Kingsford 2014)</span>, <a href="https://pachterlab.github.io/kallisto/">kallisto</a> <span class="citation">(Bray et al. 2016)</span>, or <a href="http://deweylab.github.io/RSEM/">RSEM</a> <span class="citation">(Bo Li and Dewey 2011)</span>, estimate the relative abundances of all (known, annotated) transcripts without aligning reads. Because estimating the abundance of the transcripts involves an inference step, the counts are <em>estimated</em>. Most methods either use a statistical framework called Estimation-Maximization or Bayesian techniques to estimate the abundances and counts. Following quantification, we will use the <em>tximport</em> <span class="citation">(Soneson, Love, and Robinson 2015)</span> package for assembling estimated count and offset matrices for use with Bioconductor differential gene expression packages.</p>
<p>The advantages of using the transcript abundance quantifiers in conjunction with <em>tximport</em> to produce gene-level count matrices and normalizing offsets, are:</p>
<ol style="list-style-type: decimal">
<li>this approach corrects for any potential changes in gene length across samples (e.g.from differential isoform usage) <span class="citation">(Trapnell et al. 2013)</span></li>
<li>some of these methods are substantially faster and require less memory and less disk usage compared to alignment-based methods</li>
<li>it is possible to avoid discarding those fragments that can align to multiple genes with homologous sequence <span class="citation">(Robert and Watson 2015)</span>.</li>
</ol>
<p>Note that transcript abundance quantifiers skip the generation of large files which store read alignments (SAM or BAM files), instead producing smaller files which store estimated abundances, counts and effective lengths per transcript. For more details, see the manuscript describing this approach <span class="citation">(Soneson, Love, and Robinson 2015)</span> and the <em>tximport</em> package vignette for software details.</p>
<p>A full tutorial on how to use the <em>Salmon</em> software for quantifying transcript abundance can be found <a href="https://combine-lab.github.io/salmon/getting_started/">here</a>.</p>
</div>
<div id="salmon-quantification" class="section level3">
<h3><span class="header-section-number">7.2.4</span> <em>Salmon</em> quantification</h3>
<p>We begin by providing <em>Salmon</em> with the sequence of all of the reference transcripts, which we will call the <em>reference transcriptome</em>. We recommend to use the GENCODE human transcripts, which can be downloaded from the <a href="https://www.gencodegenes.org/">GENCODE website</a>. On the command line, creating the transcriptome index looks like:</p>
<pre><code>salmon index -i gencode.v99_salmon_0.10.0 -t gencode.v99.transcripts.fa.gz</code></pre>
<p>The <code>0.10.0</code> refers to the version of <em>Salmon</em> that was used, and is useful to put into the index name.</p>
<p>To quantify an individual sample, <code>sample_01</code>, the following command can be used:</p>
<pre><code>salmon quant -i gencode.v99_salmon_0.10.0 -p 6 --libType A \
  --gcBias --biasSpeedSamp 5 \
  -1 sample_01_1.fastq.gz -2 sample_01_2.fastq.gz \
  -o sample_01</code></pre>
<p>In simple English, this command says to quantify a sample using this transcriptome index, with 6 threads, using automatic <a href="http://salmon.readthedocs.io/en/latest/library_type.html">library type</a> detection, using GC bias correction (the bias speed part is now longer needed with current versions of <em>Salmon</em>), here are the first and second read, and use this output directory. The output directory will be created if it doesnt exist, though if earlier parts of the path do not exist, it will give an error. A single sample of human RNA-seq usually takes ~5 minutes with the GC bias correction.</p>
<p>Rather than writing the above command on the command line multiple times for each sample, it is possible to loop over files using a bash loop, or more advanced workflow management systems such as Snakemake <span class="citation">(Kster and Rahmann 2012)</span> or Nextflow <span class="citation">(Di Tommaso et al. 2017)</span>.</p>
</div>
</div>
<div id="importing-into-r-with-tximport" class="section level2">
<h2><span class="header-section-number">7.3</span> Importing into R with <em>tximport</em></h2>
<div id="specifying-file-locations" class="section level3">
<h3><span class="header-section-number">7.3.1</span> Specifying file locations</h3>
<p>Following quantification, we can use <em>tximport</em> to import the data into R and perform statistical analysis using Bioconductor packages. Normally, we would simply point <em>tximport</em> to the <code>quant.sf</code> files on our machine. However, because we are distributing these files as part of an R package, we have to do some extra steps, to figure out where the R package, and so the files, are located on <em>your</em> machine.</p>
<p>We will show how to import <em>Salmon</em> quantification files using the data in the <em>tximportData</em> package. The quantified samples are six samples from the <a href="http://www.geuvadis.org/web/geuvadis">GEUVADIS Project</a> <span class="citation">(Lappalainen et al. 2013)</span>. The output directories from the above <em>Salmon</em> quantification calls has been stored in the <code>extdata</code> directory of the <em>tximportData</em> package. The R function <em>system.file</em> can be used to find out where on your computer the files from a package have been installed. Here we ask for the full path to the <code>extdata</code> directory, where R packages store external data, that is part of the <em>tximportData</em> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;tximportData&quot;</span>)
dir &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="dt">package=</span><span class="st">&quot;tximportData&quot;</span>)
<span class="kw">list.files</span>(dir)
<span class="co">#&gt;  [1] &quot;cufflinks&quot;               &quot;kallisto&quot;               </span>
<span class="co">#&gt;  [3] &quot;kallisto_boot&quot;           &quot;rsem&quot;                   </span>
<span class="co">#&gt;  [5] &quot;sailfish&quot;                &quot;salmon&quot;                 </span>
<span class="co">#&gt;  [7] &quot;salmon_dm&quot;               &quot;salmon_gibbs&quot;           </span>
<span class="co">#&gt;  [9] &quot;samples_extended.txt&quot;    &quot;samples.txt&quot;            </span>
<span class="co">#&gt; [11] &quot;tx2gene.csv&quot;             &quot;tx2gene.gencode.v27.csv&quot;</span></code></pre></div>
<p>The <em>Salmon</em> quantification directories are in the <code>salmon</code> directory.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">list.files</span>(<span class="kw">file.path</span>(dir,<span class="st">&quot;salmon&quot;</span>))
<span class="co">#&gt; [1] &quot;ERR188021&quot; &quot;ERR188088&quot; &quot;ERR188288&quot; &quot;ERR188297&quot; &quot;ERR188329&quot; &quot;ERR188356&quot;</span></code></pre></div>
<p>The identifiers used here are the <em>ERR</em> identifiers from the <a href="https://www.ebi.ac.uk/ena">European Nucleotide Archive</a>. We need to create a named vector pointing to the quantification files. We will create a vector of filenames first by reading in a table that contains the sample IDs, and then combining this with <code>dir</code> and <code>&quot;quant.sf.gz&quot;</code>. (We gzipped the quantification files to make the data package smaller, this is not a problem for R functions that we use to import the files.)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samples &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="kw">file.path</span>(dir,<span class="st">&quot;samples.txt&quot;</span>), <span class="dt">header=</span><span class="ot">TRUE</span>)
samples
<span class="co">#&gt;   pop center                assay    sample experiment       run</span>
<span class="co">#&gt; 1 TSI  UNIGE NA20503.1.M_111124_5 ERS185497  ERX163094 ERR188297</span>
<span class="co">#&gt; 2 TSI  UNIGE NA20504.1.M_111124_7 ERS185242  ERX162972 ERR188088</span>
<span class="co">#&gt; 3 TSI  UNIGE NA20505.1.M_111124_6 ERS185048  ERX163009 ERR188329</span>
<span class="co">#&gt; 4 TSI  UNIGE NA20507.1.M_111124_7 ERS185412  ERX163158 ERR188288</span>
<span class="co">#&gt; 5 TSI  UNIGE NA20508.1.M_111124_2 ERS185362  ERX163159 ERR188021</span>
<span class="co">#&gt; 6 TSI  UNIGE NA20514.1.M_111124_4 ERS185217  ERX163062 ERR188356</span>
files &lt;-<span class="st"> </span><span class="kw">file.path</span>(dir, <span class="st">&quot;salmon&quot;</span>, samples<span class="op">$</span>run, <span class="st">&quot;quant.sf.gz&quot;</span>)
<span class="kw">names</span>(files) &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&quot;sample&quot;</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">6</span>)
<span class="kw">all</span>(<span class="kw">file.exists</span>(files))
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
</div>
<div id="mapping-transcripts-to-genes" class="section level3">
<h3><span class="header-section-number">7.3.2</span> Mapping transcripts to genes</h3>
<p>Transcripts need to be associated with gene IDs for gene-level summarization. We therefore will construct a <em>data.frame</em> called <code>tx2gene</code> with two columns: 1) transcript ID and 2) gene ID. The column names do not matter but this column order must be used. The transcript ID must be the same one used in the abundance files. This can most easily be accomplished by downloading the GTF file at the same time that the transcriptome FASTA is downloaded, and generating <code>tx2gene</code> from the GTF file using Bioconductors <em>TxDb</em> infrastructure.</p>
<p>Generating a <em>TxDb</em> from a GTF file can be easily accomplished with the <em>makeTxDbFromGFF</em> function. This step requires a few minutes of waiting, and a large file. We therefore skip this step, but show the code that is used to create the <code>tx2gene</code> table, assuming the correct <em>TxDb</em> object has been created.</p>
<p>Creating the <code>tx2gene</code> <em>data.frame</em> can be accomplished by calling the <em>select</em> function from the <em>AnnotationDbi</em> package on a <em>TxDb</em> object. The following code could be used to construct such a table:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;TxDb.Hsapiens.UCSC.hg38.knownGene&quot;</span>)
txdb &lt;-<span class="st"> </span>TxDb.Hsapiens.UCSC.hg38.knownGene
k &lt;-<span class="st"> </span><span class="kw">keys</span>(txdb, <span class="dt">keytype=</span><span class="st">&quot;TXNAME&quot;</span>)
tx2gene &lt;-<span class="st"> </span><span class="kw">select</span>(txdb, k, <span class="st">&quot;GENEID&quot;</span>, <span class="st">&quot;TXNAME&quot;</span>)
<span class="co">#&gt; &#39;select()&#39; returned 1:1 mapping between keys and columns</span></code></pre></div>
<p>In this case, weve used the Gencode v27 CHR transcripts to build our <em>Salmon</em> index, and we used <code>makeTxDbFromGFF</code> and code similar to the chunk above to build the <code>tx2gene</code> table. We then read in a pre-constructed <code>tx2gene</code> table:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;readr&quot;</span>)
tx2gene &lt;-<span class="st"> </span><span class="kw">read_csv</span>(<span class="kw">file.path</span>(dir, <span class="st">&quot;tx2gene.gencode.v27.csv&quot;</span>))
<span class="co">#&gt; Parsed with column specification:</span>
<span class="co">#&gt; cols(</span>
<span class="co">#&gt;   TXNAME = col_character(),</span>
<span class="co">#&gt;   GENEID = col_character()</span>
<span class="co">#&gt; )</span>
<span class="kw">head</span>(tx2gene)
<span class="co">#&gt; # A tibble: 6 x 2</span>
<span class="co">#&gt;   TXNAME            GENEID           </span>
<span class="co">#&gt;   &lt;chr&gt;             &lt;chr&gt;            </span>
<span class="co">#&gt; 1 ENST00000456328.2 ENSG00000223972.5</span>
<span class="co">#&gt; 2 ENST00000450305.2 ENSG00000223972.5</span>
<span class="co">#&gt; 3 ENST00000473358.1 ENSG00000243485.5</span>
<span class="co">#&gt; 4 ENST00000469289.1 ENSG00000243485.5</span>
<span class="co">#&gt; 5 ENST00000607096.1 ENSG00000284332.1</span>
<span class="co">#&gt; 6 ENST00000606857.1 ENSG00000268020.3</span></code></pre></div>
</div>
<div id="tximport-command" class="section level3">
<h3><span class="header-section-number">7.3.3</span> <em>tximport</em> command</h3>
<p>Finally the following line of code imports <em>Salmon</em> transcript quantifications into R, collapsing to the gene level using the information in <code>tx2gene</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;tximport&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;jsonlite&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;readr&quot;</span>)
txi &lt;-<span class="st"> </span><span class="kw">tximport</span>(files, <span class="dt">type=</span><span class="st">&quot;salmon&quot;</span>, <span class="dt">tx2gene=</span>tx2gene)
<span class="co">#&gt; reading in files with read_tsv</span>
<span class="co">#&gt; 1 2 3 4 5 6 </span>
<span class="co">#&gt; summarizing abundance</span>
<span class="co">#&gt; summarizing counts</span>
<span class="co">#&gt; summarizing length</span></code></pre></div>
<p>The <code>txi</code> object is simply a list of matrices (and one character vector):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(txi)
<span class="co">#&gt; [1] &quot;abundance&quot;           &quot;counts&quot;              &quot;length&quot;             </span>
<span class="co">#&gt; [4] &quot;countsFromAbundance&quot;</span>
txi<span class="op">$</span>counts[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]
<span class="co">#&gt;                       sample1  sample2    sample3</span>
<span class="co">#&gt; ENSG00000000003.14    2.58012    2.000   27.09648</span>
<span class="co">#&gt; ENSG00000000005.5     0.00000    0.000    0.00000</span>
<span class="co">#&gt; ENSG00000000419.12 1056.99960 1337.997 1452.99497</span>
txi<span class="op">$</span>length[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]
<span class="co">#&gt;                      sample1  sample2   sample3</span>
<span class="co">#&gt; ENSG00000000003.14 3621.7000 735.4220 2201.6223</span>
<span class="co">#&gt; ENSG00000000005.5   195.6667 195.6667  195.6667</span>
<span class="co">#&gt; ENSG00000000419.12  871.5077 905.0540  845.7278</span>
txi<span class="op">$</span>abundance[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]
<span class="co">#&gt;                       sample1   sample2   sample3</span>
<span class="co">#&gt; ENSG00000000003.14  0.0354884  0.119404  0.411491</span>
<span class="co">#&gt; ENSG00000000005.5   0.0000000  0.000000  0.000000</span>
<span class="co">#&gt; ENSG00000000419.12 60.4173800 64.909276 57.441353</span>
txi<span class="op">$</span>countsFromAbundance
<span class="co">#&gt; [1] &quot;no&quot;</span></code></pre></div>
<p>If we were continuing with the GEUVADIS samples, we would then create a <em>DESeqDataSet</em> with the following line of code. Because there are no differences among the samples (same population and same sequencing batch), we specify a <em>design formula</em> of ~1, meaning we can only fit an intercept term  so we cannot perform differential expression analysis with these samples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;DESeq2&quot;</span>)
dds &lt;-<span class="st"> </span><span class="kw">DESeqDataSetFromTximport</span>(txi, samples, <span class="op">~</span><span class="dv">1</span>)
<span class="co">#&gt; using counts and average transcript lengths from tximport</span>
dds<span class="op">$</span>center
<span class="co">#&gt; [1] UNIGE UNIGE UNIGE UNIGE UNIGE UNIGE</span>
<span class="co">#&gt; Levels: UNIGE</span>
dds<span class="op">$</span>pop
<span class="co">#&gt; [1] TSI TSI TSI TSI TSI TSI</span>
<span class="co">#&gt; Levels: TSI</span></code></pre></div>
</div>
</div>
<div id="exploratory-data-analysis" class="section level2">
<h2><span class="header-section-number">7.4</span> Exploratory data analysis</h2>
<div id="simple-eda" class="section level3">
<h3><span class="header-section-number">7.4.1</span> Simple EDA</h3>
<p>We will now switch over to the <em>airway</em> experiment, counts of which are already prepared in a <em>SummarizedExperiment</em> object. In this case, the object that we load is the output of the <em>summarizeOverlaps</em> function in the <em>GenomicAlignments</em> package, and the exact code used to produce this object can be seen by typing <code>vignette(&quot;airway&quot;)</code> into the R session, to pull up the <em>airway</em> software vignette. There are multiple ways to produce a count table and import it into <em>DESeq2</em>, and these are summarized in this section of the <a href="https://bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html#deseq2-import-functions">RNA-seq gene-level workflow</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;airway&quot;</span>)
<span class="kw">data</span>(<span class="st">&quot;airway&quot;</span>)</code></pre></div>
<p>We want to specify that <code>untrt</code> is the reference level for the dex variable:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">airway<span class="op">$</span>dex &lt;-<span class="st"> </span><span class="kw">relevel</span>(airway<span class="op">$</span>dex, <span class="st">&quot;untrt&quot;</span>)
airway<span class="op">$</span>dex
<span class="co">#&gt; [1] untrt trt   untrt trt   untrt trt   untrt trt  </span>
<span class="co">#&gt; Levels: untrt trt</span></code></pre></div>
<p>We can quickly check the millions of fragments that uniquely aligned to the genes (the second argument of <em>round</em> tells how many decimal points to keep).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">round</span>( <span class="kw">colSums</span>(<span class="kw">assay</span>(airway)) <span class="op">/</span><span class="st"> </span><span class="fl">1e6</span>, <span class="dv">1</span> )
<span class="co">#&gt; SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 </span>
<span class="co">#&gt;       20.6       18.8       25.3       15.2       24.4       30.8 </span>
<span class="co">#&gt; SRR1039520 SRR1039521 </span>
<span class="co">#&gt;       19.1       21.2</span></code></pre></div>
<p>We can inspect the information about the samples, by pulling out the <code>colData</code> slot of the <em>SummarizedExperiment</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">colData</span>(airway)
<span class="co">#&gt; DataFrame with 8 rows and 9 columns</span>
<span class="co">#&gt;            SampleName     cell      dex    albut        Run avgLength</span>
<span class="co">#&gt;              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;</span>
<span class="co">#&gt; SRR1039508 GSM1275862   N61311    untrt    untrt SRR1039508       126</span>
<span class="co">#&gt; SRR1039509 GSM1275863   N61311      trt    untrt SRR1039509       126</span>
<span class="co">#&gt; SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126</span>
<span class="co">#&gt; SRR1039513 GSM1275867  N052611      trt    untrt SRR1039513        87</span>
<span class="co">#&gt; SRR1039516 GSM1275870  N080611    untrt    untrt SRR1039516       120</span>
<span class="co">#&gt; SRR1039517 GSM1275871  N080611      trt    untrt SRR1039517       126</span>
<span class="co">#&gt; SRR1039520 GSM1275874  N061011    untrt    untrt SRR1039520       101</span>
<span class="co">#&gt; SRR1039521 GSM1275875  N061011      trt    untrt SRR1039521        98</span>
<span class="co">#&gt;            Experiment    Sample    BioSample</span>
<span class="co">#&gt;              &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;</span>
<span class="co">#&gt; SRR1039508  SRX384345 SRS508568 SAMN02422669</span>
<span class="co">#&gt; SRR1039509  SRX384346 SRS508567 SAMN02422675</span>
<span class="co">#&gt; SRR1039512  SRX384349 SRS508571 SAMN02422678</span>
<span class="co">#&gt; SRR1039513  SRX384350 SRS508572 SAMN02422670</span>
<span class="co">#&gt; SRR1039516  SRX384353 SRS508575 SAMN02422682</span>
<span class="co">#&gt; SRR1039517  SRX384354 SRS508576 SAMN02422673</span>
<span class="co">#&gt; SRR1039520  SRX384357 SRS508579 SAMN02422683</span>
<span class="co">#&gt; SRR1039521  SRX384358 SRS508580 SAMN02422677</span>
<span class="kw">table</span>(airway<span class="op">$</span>cell)
<span class="co">#&gt; </span>
<span class="co">#&gt; N052611 N061011 N080611  N61311 </span>
<span class="co">#&gt;       2       2       2       2</span>
<span class="kw">table</span>(airway<span class="op">$</span>dex)
<span class="co">#&gt; </span>
<span class="co">#&gt; untrt   trt </span>
<span class="co">#&gt;     4     4</span></code></pre></div>
<p>If we had not already loaded <em>DESeq2</em>, we would do this, and then create a <em>DESeqDataSet</em>. We want to control for the cell line, while testing for differences across dexamethasone treatment, so we use a design of <code>~ cell + dex</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;DESeq2&quot;</span>)
dds &lt;-<span class="st"> </span><span class="kw">DESeqDataSet</span>(airway, <span class="dt">design =</span> <span class="op">~</span><span class="st"> </span>cell <span class="op">+</span><span class="st"> </span>dex)</code></pre></div>
<p>We will perform a minimal filtering to reduce the size of the dataset. We do not need to retain genes if they do not have a count of 5 or more for 4 or more samples as these genes will have no statistical power to detect differences, and no information to compute distances between samples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">keep &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">counts</span>(dds) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">5</span>) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">4</span>
<span class="kw">table</span>(keep)
<span class="co">#&gt; keep</span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt; 46070 18032</span>
dds &lt;-<span class="st"> </span>dds[keep,]</code></pre></div>
<p>Some very basic exploratory analysis is to examine a boxplot of the counts for each sample. We will take the logarithm so that large counts do not dominate the boxplot:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(<span class="kw">log10</span>(<span class="kw">counts</span>(dds)<span class="op">+</span><span class="dv">1</span>))</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<p>The main function in <em>DESeq2</em> involves computation of <em>size factors</em> which normalize for differences in sequencing depth among samples. We can also compute these size factors manually, so that the <em>normalized counts</em> are available for plotting:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dds &lt;-<span class="st"> </span><span class="kw">estimateSizeFactors</span>(dds)
<span class="kw">boxplot</span>(<span class="kw">log10</span>(<span class="kw">counts</span>(dds,<span class="dt">normalized=</span><span class="ot">TRUE</span>)<span class="op">+</span><span class="dv">1</span>))</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
</div>
<div id="data-transformation-for-eda" class="section level3">
<h3><span class="header-section-number">7.4.2</span> Data transformation for EDA</h3>
<p>Taking the logarithm of counts plus a pseudocount of 1 is a common transformation, but it tends to inflate the sampling variance of low counts such that it is even larger than biological variation across groups of samples. In <em>DESeq2</em> we therefore provide transformations which produce log-scale data such that the systematic trends have been removed. Our recommended transformation is the variance-stabilizing transformation, or VST, and it can be called with the <em>vst</em> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">vsd &lt;-<span class="st"> </span><span class="kw">vst</span>(dds)
<span class="kw">class</span>(vsd)
<span class="co">#&gt; [1] &quot;DESeqTransform&quot;</span>
<span class="co">#&gt; attr(,&quot;package&quot;)</span>
<span class="co">#&gt; [1] &quot;DESeq2&quot;</span></code></pre></div>
<p>This function does not return a <em>DESeqDataSet</em>, because it does not return counts, but instead continuous values (on the log2 scale). We can access the transformed data with <em>assay</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">assay</span>(vsd)[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]
<span class="co">#&gt;                 SRR1039508 SRR1039509 SRR1039512</span>
<span class="co">#&gt; ENSG00000000003   9.456925   9.074623   9.608160</span>
<span class="co">#&gt; ENSG00000000419   8.952752   9.262092   9.145782</span>
<span class="co">#&gt; ENSG00000000457   8.193711   8.098664   8.032656</span></code></pre></div>
</div>
<div id="principal-components-plot" class="section level3">
<h3><span class="header-section-number">7.4.3</span> Principal components plot</h3>
<p>The VST data is appropriate for calculating distances between samples or for performing PCA. More information about PCA and distance calculation can be found in the <a href="https://bioconductor.org/packages/release/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html#deseq2-import-functions">RNA-seq gene-level workflow</a>. In short, PCA plots allow us to visualize the most dominant axes of variation in our data, which is useful for both quality control, and to get a sense of how large the inter-sample differences are across and within conditions. Here we see that PC1 (the primary axis of variation in the data) separates the treated and untreated samples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotPCA</span>(vsd, <span class="st">&quot;dex&quot;</span>)</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<p>With some additional <em>ggplot2</em> code, we can also indicate which samples belong to which cell line:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;ggplot2&quot;</span>)
pcaData &lt;-<span class="st"> </span><span class="kw">plotPCA</span>(vsd, <span class="dt">intgroup =</span> <span class="kw">c</span>( <span class="st">&quot;dex&quot;</span>, <span class="st">&quot;cell&quot;</span>), <span class="dt">returnData =</span> <span class="ot">TRUE</span>)
percentVar &lt;-<span class="st"> </span><span class="kw">round</span>(<span class="dv">100</span> <span class="op">*</span><span class="st"> </span><span class="kw">attr</span>(pcaData, <span class="st">&quot;percentVar&quot;</span>))
<span class="kw">ggplot</span>(pcaData, <span class="kw">aes</span>(<span class="dt">x =</span> PC1, <span class="dt">y =</span> PC2, <span class="dt">color =</span> dex, <span class="dt">shape =</span> cell)) <span class="op">+</span>
<span class="st">  </span><span class="kw">geom_point</span>(<span class="dt">size =</span><span class="dv">3</span>) <span class="op">+</span>
<span class="st">  </span><span class="kw">xlab</span>(<span class="kw">paste0</span>(<span class="st">&quot;PC1: &quot;</span>, percentVar[<span class="dv">1</span>], <span class="st">&quot;% variance&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">ylab</span>(<span class="kw">paste0</span>(<span class="st">&quot;PC2: &quot;</span>, percentVar[<span class="dv">2</span>], <span class="st">&quot;% variance&quot;</span>)) <span class="op">+</span>
<span class="st">  </span><span class="kw">coord_fixed</span>()</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<p>Note that we do not recommend working with the transformed data for the primary differential expression analysis. Instead we will use the original counts and a <em>generalized linear model</em> (GLM) which takes into account the expected variance from either low or high counts. For statistical details, please refer to the <em>DESeq2</em> methods paper <span class="citation">(Love, Huber, and Anders 2014)</span>.</p>
</div>
</div>
<div id="differential-expression-analysis-1" class="section level2">
<h2><span class="header-section-number">7.5</span> Differential expression analysis</h2>
<div id="standard-de-steps" class="section level3">
<h3><span class="header-section-number">7.5.1</span> Standard DE steps</h3>
<p>Differential expression analysis in <em>DESeq2</em> is performed by calling the following two functions:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds)
<span class="co">#&gt; using pre-existing size factors</span>
<span class="co">#&gt; estimating dispersions</span>
<span class="co">#&gt; gene-wise dispersion estimates</span>
<span class="co">#&gt; mean-dispersion relationship</span>
<span class="co">#&gt; final dispersion estimates</span>
<span class="co">#&gt; fitting model and testing</span>
res &lt;-<span class="st"> </span><span class="kw">results</span>(dds)</code></pre></div>
<p>The results table <code>res</code> contains the results for each gene (in the same order as in the <em>DESeqDataSet</em>). If we want to see the top genes, we can order it like so:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(res[<span class="kw">order</span>(res<span class="op">$</span>pvalue),])
<span class="co">#&gt; log2 fold change (MLE): dex trt vs untrt </span>
<span class="co">#&gt; Wald test p-value: dex trt vs untrt </span>
<span class="co">#&gt; DataFrame with 6 rows and 6 columns</span>
<span class="co">#&gt;                         baseMean   log2FoldChange             lfcSE</span>
<span class="co">#&gt;                        &lt;numeric&gt;        &lt;numeric&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt; ENSG00000152583 997.522193389904 4.57431721934304 0.183934560913345</span>
<span class="co">#&gt; ENSG00000165995 495.289924523775 3.29060376160918 0.132397954562143</span>
<span class="co">#&gt; ENSG00000120129 3409.85238036378  2.9472509434796 0.122471899795457</span>
<span class="co">#&gt; ENSG00000101347  12707.320121355  3.7664043884956 0.156934450302937</span>
<span class="co">#&gt; ENSG00000189221 2342.17328482568 3.35311264853201 0.142537730294312</span>
<span class="co">#&gt; ENSG00000211445 12292.1234547129 3.72983474166149 0.167361554596787</span>
<span class="co">#&gt;                             stat                pvalue</span>
<span class="co">#&gt;                        &lt;numeric&gt;             &lt;numeric&gt;</span>
<span class="co">#&gt; ENSG00000152583 24.8692643548272 1.60060784368496e-136</span>
<span class="co">#&gt; ENSG00000165995 24.8538866970538 2.34741049897627e-136</span>
<span class="co">#&gt; ENSG00000120129 24.0647115656888 5.85597945778505e-128</span>
<span class="co">#&gt; ENSG00000101347 23.9998571456117 2.79035105617505e-127</span>
<span class="co">#&gt; ENSG00000189221 23.5243864316382 2.29647414481068e-122</span>
<span class="co">#&gt; ENSG00000211445 22.2860904384375 5.04142736936512e-110</span>
<span class="co">#&gt;                                  padj</span>
<span class="co">#&gt;                             &lt;numeric&gt;</span>
<span class="co">#&gt; ENSG00000152583 2.03426593841284e-132</span>
<span class="co">#&gt; ENSG00000165995 2.03426593841284e-132</span>
<span class="co">#&gt; ENSG00000120129 3.38319453207768e-124</span>
<span class="co">#&gt; ENSG00000101347 1.20905911264065e-123</span>
<span class="co">#&gt; ENSG00000189221 7.96049797557172e-119</span>
<span class="co">#&gt; ENSG00000211445  1.4563003194306e-106</span></code></pre></div>
<p>We can plot the counts for the top gene using <code>plotCounts</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotCounts</span>(dds, <span class="kw">which.min</span>(res<span class="op">$</span>pvalue), <span class="st">&quot;dex&quot;</span>)</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<p>We can examine all the log2 fold changes (LFC) due to dexamethasone treatment over the mean of counts using <code>plotMA</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotMA</span>(res, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">5</span>,<span class="dv">5</span>))</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
<p>Note that there are many large LFC which are not significant (grey points) on the left side of the MA-plot above. These obtain a large LFC because of the imprecision of log counts. For more informative visualization and more accurate ranking of genes by effect size (the log fold change may sometimes be referred to as an <em>effect size</em>), we recommend to use <em>DESeq2</em>s functionality for shrinking LFCs. Our most recent methodological development is the <em>apeglm</em> shrinkage estimator, which is available in <em>DESeq2</em>s <em>lfcShrink</em> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;apeglm&quot;</span>)
<span class="kw">resultsNames</span>(dds)
<span class="co">#&gt; [1] &quot;Intercept&quot;               &quot;cell_N061011_vs_N052611&quot;</span>
<span class="co">#&gt; [3] &quot;cell_N080611_vs_N052611&quot; &quot;cell_N61311_vs_N052611&quot; </span>
<span class="co">#&gt; [5] &quot;dex_trt_vs_untrt&quot;</span>
res2 &lt;-<span class="st"> </span><span class="kw">lfcShrink</span>(dds, <span class="dt">coef=</span><span class="st">&quot;dex_trt_vs_untrt&quot;</span>, <span class="dt">type=</span><span class="st">&quot;apeglm&quot;</span>)
<span class="co">#&gt; using &#39;apeglm&#39; for LFC shrinkage. If used in published research, please cite:</span>
<span class="co">#&gt;     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for</span>
<span class="co">#&gt;     sequence count data: removing the noise and preserving large differences.</span>
<span class="co">#&gt;     bioRxiv. https://doi.org/10.1101/303255</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">plotMA</span>(res, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">3</span>,<span class="dv">3</span>), <span class="dt">main=</span><span class="st">&quot;No shrinkage&quot;</span>)
<span class="kw">plotMA</span>(res2, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">3</span>,<span class="dv">3</span>), <span class="dt">main=</span><span class="st">&quot;apeglm&quot;</span>)</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
</div>
<div id="minimum-effect-size" class="section level3">
<h3><span class="header-section-number">7.5.2</span> Minimum effect size</h3>
<p>If we dont want to report as significant genes with small LFC, we can specify a minimum <em>biologically meaningful</em> effect size, by choosing an LFC and testing against this. We can either perform such a threshold test using the unshrunken LFCs or the LFCs provided by <em>lfcShrink</em> using the <em>apeglm</em> method:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res.lfc &lt;-<span class="st"> </span><span class="kw">results</span>(dds, <span class="dt">lfcThreshold=</span><span class="dv">1</span>)
res.lfc2 &lt;-<span class="st"> </span><span class="kw">lfcShrink</span>(dds, <span class="dt">coef=</span><span class="st">&quot;dex_trt_vs_untrt&quot;</span>, <span class="dt">type=</span><span class="st">&quot;apeglm&quot;</span>,
                      <span class="dt">lfcThreshold=</span><span class="dv">1</span>)
<span class="co">#&gt; using &#39;apeglm&#39; for LFC shrinkage. If used in published research, please cite:</span>
<span class="co">#&gt;     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for</span>
<span class="co">#&gt;     sequence count data: removing the noise and preserving large differences.</span>
<span class="co">#&gt;     bioRxiv. https://doi.org/10.1101/303255</span>
<span class="co">#&gt; computing FSOS &#39;false sign or small&#39; s-values (T=1)</span></code></pre></div>
<p>Note that <em>testing</em> against an LFC threshold is not equivalent to testing against a null hypothesis of 0 and then filtering on LFC values. We prefer the former, as discussed in <span class="citation">Love, Huber, and Anders (2014)</span> and <span class="citation">Zhu, Ibrahim, and Love (2018)</span>.</p>
<p>The <em>apeglm</em> method provides s-values <span class="citation">(Stephens 2016)</span> when <code>svalue=TRUE</code> or when we supply a minimum effect size as above. These are analogous to q-values or adjusted p-values, in that the genes with s-values less than <span class="math inline">\(\alpha\)</span> should have an aggregate rate of false sign or being smaller in absolute value than our given LFC threshold, which is bounded by <span class="math inline">\(\alpha\)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">plotMA</span>(res.lfc, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">5</span>,<span class="dv">5</span>), <span class="dt">main=</span><span class="st">&quot;No shrinkage, LFC test&quot;</span>)
<span class="kw">plotMA</span>(res.lfc2, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">5</span>,<span class="dv">5</span>), <span class="dt">main=</span><span class="st">&quot;apeglm, LFC test&quot;</span>, <span class="dt">alpha=</span><span class="fl">0.01</span>)</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
</div>
</div>
<div id="annotationhub-1" class="section level2">
<h2><span class="header-section-number">7.6</span> <em>AnnotationHub</em></h2>
<div id="querying-annotationhub-1" class="section level3">
<h3><span class="header-section-number">7.6.1</span> Querying <em>AnnotationHub</em></h3>
<p>We will use the <em>AnnotationHub</em> package to attach additional information to the results table. <em>AnnotationHub</em> provides an easy-to-use interface to more than 40,000 annotation records. A record may be peaks from a ChIP-seq experiment from ENCODE, the sequence of the human genome, a <em>TxDb</em> containing information about transcripts and genes, or an <em>OrgDb</em> containing general information about biological identifiers for a particular organism.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;AnnotationHub&quot;</span>)
ah &lt;-<span class="st"> </span><span class="kw">AnnotationHub</span>()
<span class="co">#&gt; snapshotDate(): 2018-06-27</span></code></pre></div>
<p>The following code chunk, un-evaluated here, launches a browser for navigating all the records available through <em>AnnotationHub</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">display</span>(ah)</code></pre></div>
<p>We can also query using keywords with the <em>query</em> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">query</span>(ah, <span class="kw">c</span>(<span class="st">&quot;OrgDb&quot;</span>,<span class="st">&quot;Homo sapiens&quot;</span>))
<span class="co">#&gt; AnnotationHub with 1 record</span>
<span class="co">#&gt; # snapshotDate(): 2018-06-27 </span>
<span class="co">#&gt; # names(): AH61777</span>
<span class="co">#&gt; # $dataprovider: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/</span>
<span class="co">#&gt; # $species: Homo sapiens</span>
<span class="co">#&gt; # $rdataclass: OrgDb</span>
<span class="co">#&gt; # $rdatadateadded: 2018-04-19</span>
<span class="co">#&gt; # $title: org.Hs.eg.db.sqlite</span>
<span class="co">#&gt; # $description: NCBI gene ID based annotations about Homo sapiens</span>
<span class="co">#&gt; # $taxonomyid: 9606</span>
<span class="co">#&gt; # $genome: NCBI genomes</span>
<span class="co">#&gt; # $sourcetype: NCBI/ensembl</span>
<span class="co">#&gt; # $sourceurl: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/, ftp://ftp.ensembl....</span>
<span class="co">#&gt; # $sourcesize: NA</span>
<span class="co">#&gt; # $tags: c(&quot;NCBI&quot;, &quot;Gene&quot;, &quot;Annotation&quot;) </span>
<span class="co">#&gt; # retrieve record with &#39;object[[&quot;AH61777&quot;]]&#39;</span></code></pre></div>
<p>To pull down a particular record we use double brackets and the <em>name</em> of the record:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hs &lt;-<span class="st"> </span>ah[[<span class="st">&quot;AH61777&quot;</span>]]
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/ubuntu//.AnnotationHub/68523&#39;</span>
hs
<span class="co">#&gt; OrgDb object:</span>
<span class="co">#&gt; | DBSCHEMAVERSION: 2.1</span>
<span class="co">#&gt; | Db type: OrgDb</span>
<span class="co">#&gt; | Supporting package: AnnotationDbi</span>
<span class="co">#&gt; | DBSCHEMA: HUMAN_DB</span>
<span class="co">#&gt; | ORGANISM: Homo sapiens</span>
<span class="co">#&gt; | SPECIES: Human</span>
<span class="co">#&gt; | EGSOURCEDATE: 2018-Apr4</span>
<span class="co">#&gt; | EGSOURCENAME: Entrez Gene</span>
<span class="co">#&gt; | EGSOURCEURL: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA</span>
<span class="co">#&gt; | CENTRALID: EG</span>
<span class="co">#&gt; | TAXID: 9606</span>
<span class="co">#&gt; | GOSOURCENAME: Gene Ontology</span>
<span class="co">#&gt; | GOSOURCEURL: ftp://ftp.geneontology.org/pub/go/godatabase/archive/latest-lite/</span>
<span class="co">#&gt; | GOSOURCEDATE: 2018-Mar28</span>
<span class="co">#&gt; | GOEGSOURCEDATE: 2018-Apr4</span>
<span class="co">#&gt; | GOEGSOURCENAME: Entrez Gene</span>
<span class="co">#&gt; | GOEGSOURCEURL: ftp://ftp.ncbi.nlm.nih.gov/gene/DATA</span>
<span class="co">#&gt; | KEGGSOURCENAME: KEGG GENOME</span>
<span class="co">#&gt; | KEGGSOURCEURL: ftp://ftp.genome.jp/pub/kegg/genomes</span>
<span class="co">#&gt; | KEGGSOURCEDATE: 2011-Mar15</span>
<span class="co">#&gt; | GPSOURCENAME: UCSC Genome Bioinformatics (Homo sapiens)</span>
<span class="co">#&gt; | GPSOURCEURL: </span>
<span class="co">#&gt; | GPSOURCEDATE: 2018-Mar26</span>
<span class="co">#&gt; | ENSOURCEDATE: 2017-Dec04</span>
<span class="co">#&gt; | ENSOURCENAME: Ensembl</span>
<span class="co">#&gt; | ENSOURCEURL: ftp://ftp.ensembl.org/pub/current_fasta</span>
<span class="co">#&gt; | UPSOURCENAME: Uniprot</span>
<span class="co">#&gt; | UPSOURCEURL: http://www.UniProt.org/</span>
<span class="co">#&gt; | UPSOURCEDATE: Mon Apr  9 20:58:54 2018</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Please see: help(&#39;select&#39;) for usage information</span></code></pre></div>
</div>
<div id="mapping-ids" class="section level3">
<h3><span class="header-section-number">7.6.2</span> Mapping IDs</h3>
<p>The <em>rownames</em> of the results table are Ensembl IDs, and most of these are entries in <em>OrgDb</em> (although thousands are not).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">columns</span>(hs)
<span class="co">#&gt;  [1] &quot;ACCNUM&quot;       &quot;ALIAS&quot;        &quot;ENSEMBL&quot;      &quot;ENSEMBLPROT&quot; </span>
<span class="co">#&gt;  [5] &quot;ENSEMBLTRANS&quot; &quot;ENTREZID&quot;     &quot;ENZYME&quot;       &quot;EVIDENCE&quot;    </span>
<span class="co">#&gt;  [9] &quot;EVIDENCEALL&quot;  &quot;GENENAME&quot;     &quot;GO&quot;           &quot;GOALL&quot;       </span>
<span class="co">#&gt; [13] &quot;IPI&quot;          &quot;MAP&quot;          &quot;OMIM&quot;         &quot;ONTOLOGY&quot;    </span>
<span class="co">#&gt; [17] &quot;ONTOLOGYALL&quot;  &quot;PATH&quot;         &quot;PFAM&quot;         &quot;PMID&quot;        </span>
<span class="co">#&gt; [21] &quot;PROSITE&quot;      &quot;REFSEQ&quot;       &quot;SYMBOL&quot;       &quot;UCSCKG&quot;      </span>
<span class="co">#&gt; [25] &quot;UNIGENE&quot;      &quot;UNIPROT&quot;</span>
<span class="kw">table</span>(<span class="kw">rownames</span>(res) <span class="op">%in%</span><span class="st"> </span><span class="kw">keys</span>(hs, <span class="st">&quot;ENSEMBL&quot;</span>))
<span class="co">#&gt; </span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt;  3492 14540</span></code></pre></div>
<p>We can use the <em>mapIds</em> function to add gene symbols, using <code>ENSEMBL</code> as the keytype, and requesting the column <code>SYMBOL</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res<span class="op">$</span>symbol &lt;-<span class="st"> </span><span class="kw">mapIds</span>(hs, <span class="kw">rownames</span>(res), <span class="dt">column=</span><span class="st">&quot;SYMBOL&quot;</span>, <span class="dt">keytype=</span><span class="st">&quot;ENSEMBL&quot;</span>)
<span class="co">#&gt; &#39;select()&#39; returned 1:many mapping between keys and columns</span>
<span class="kw">head</span>(res)
<span class="co">#&gt; log2 fold change (MLE): dex trt vs untrt </span>
<span class="co">#&gt; Wald test p-value: dex trt vs untrt </span>
<span class="co">#&gt; DataFrame with 6 rows and 7 columns</span>
<span class="co">#&gt;                         baseMean      log2FoldChange              lfcSE</span>
<span class="co">#&gt;                        &lt;numeric&gt;           &lt;numeric&gt;          &lt;numeric&gt;</span>
<span class="co">#&gt; ENSG00000000003  708.84032163709  -0.381888900525941  0.100800595893129</span>
<span class="co">#&gt; ENSG00000000419 520.444343803335   0.206203578138316   0.11134065494833</span>
<span class="co">#&gt; ENSG00000000457 237.237392013978  0.0373231179429596  0.140524882318114</span>
<span class="co">#&gt; ENSG00000000460 57.9518862998956 -0.0907678445512801  0.276878121357344</span>
<span class="co">#&gt; ENSG00000000971 5819.01711439455   0.425781621579747  0.089731469359872</span>
<span class="co">#&gt; ENSG00000001036 1282.59042750161  -0.241675199658791 0.0898743244142716</span>
<span class="co">#&gt;                              stat               pvalue</span>
<span class="co">#&gt;                         &lt;numeric&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt; ENSG00000000003 -3.78855796577658 0.000151524241379562</span>
<span class="co">#&gt; ENSG00000000419  1.85200615385287   0.0640249392376881</span>
<span class="co">#&gt; ENSG00000000457  0.26559793061039    0.790548879507716</span>
<span class="co">#&gt; ENSG00000000460 -0.32782599111229    0.743043234271692</span>
<span class="co">#&gt; ENSG00000000971  4.74506463136284 2.08439771237453e-06</span>
<span class="co">#&gt; ENSG00000001036 -2.68903495223842   0.0071658915709543</span>
<span class="co">#&gt;                                 padj      symbol</span>
<span class="co">#&gt;                            &lt;numeric&gt; &lt;character&gt;</span>
<span class="co">#&gt; ENSG00000000003  0.00121584173684748      TSPAN6</span>
<span class="co">#&gt; ENSG00000000419    0.184546856289308        DPM1</span>
<span class="co">#&gt; ENSG00000000457    0.903592643011972       SCYL3</span>
<span class="co">#&gt; ENSG00000000460    0.878207188925546    C1orf112</span>
<span class="co">#&gt; ENSG00000000971 2.54952583986418e-05         CFH</span>
<span class="co">#&gt; ENSG00000001036   0.0334768821314771       FUCA2</span></code></pre></div>
</div>
</div>
<div id="building-reports" class="section level2">
<h2><span class="header-section-number">7.7</span> Building reports</h2>
<div id="reportingtools" class="section level3">
<h3><span class="header-section-number">7.7.1</span> <em>ReportingTools</em></h3>
<p>There are many packages for building interactive reports from Bioconductor. Two of these are <em>ReportingTools</em> and <em>Glimma</em>, which both provide HTML reports that allow for collaborators to examine the top genes (or whatever features of interest) from a genomic analysis.</p>
<p>The code for compiling a <em>ReportingTools</em> report is:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;ReportingTools&quot;</span>)
tmp &lt;-<span class="st"> </span><span class="kw">tempdir</span>() <span class="co"># you would instead use a meaningful path here</span>
rep &lt;-<span class="st"> </span><span class="kw">HTMLReport</span>(<span class="dt">shortName=</span><span class="st">&quot;airway&quot;</span>, <span class="dt">title=</span><span class="st">&quot;Airway DGE&quot;</span>,
                  <span class="dt">basePath=</span>tmp, <span class="dt">reportDirectory=</span><span class="st">&quot;report&quot;</span>)
<span class="kw">publish</span>(res, rep, dds, <span class="dt">n=</span><span class="dv">20</span>, <span class="dt">make.plots=</span><span class="ot">TRUE</span>, <span class="dt">factor=</span>dds<span class="op">$</span>dex)
<span class="kw">finish</span>(rep)
<span class="co">#&gt; [1] &quot;/tmp/RtmpBov3oy/report/airway.html&quot;</span></code></pre></div>
<p>This last line, un-evaluated would launch the report in a web browser:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">browseURL</span>(<span class="kw">file.path</span>(tmp,<span class="st">&quot;report&quot;</span>,<span class="st">&quot;airway.html&quot;</span>))</code></pre></div>
</div>
<div id="glimma" class="section level3">
<h3><span class="header-section-number">7.7.2</span> <em>Glimma</em></h3>
<p>Another package which can generate interactive reports is <em>Glimma</em>. The <em>glMDPlot</em> constructs an interactive MA-plot where hovering over a gene in the MA-plot on the left side will display the counts for the samples on the right hand side. Clicking will bring up the genes information in a tooltip and in a list at the bottom of the screen. Hovering on a sample on the right hand side will give the sample ID in a tooltip.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;Glimma&quot;</span>)
status &lt;-<span class="st"> </span><span class="kw">as.numeric</span>(res<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>.<span class="dv">1</span>)
anno &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">GeneID=</span><span class="kw">rownames</span>(res), <span class="dt">symbol=</span>res<span class="op">$</span>symbol)
<span class="kw">glMDPlot</span>(res2, <span class="dt">status=</span>status, <span class="dt">counts=</span><span class="kw">counts</span>(dds,<span class="dt">normalized=</span><span class="ot">TRUE</span>),
         <span class="dt">groups=</span>dds<span class="op">$</span>dex, <span class="dt">transform=</span><span class="ot">FALSE</span>,
         <span class="dt">samples=</span><span class="kw">colnames</span>(dds), <span class="dt">anno=</span>anno,
         <span class="dt">path=</span>tmp, <span class="dt">folder=</span><span class="st">&quot;glimma&quot;</span>, <span class="dt">launch=</span><span class="ot">FALSE</span>)</code></pre></div>
<p>This last line would launch the report in a web browser:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">browseURL</span>(<span class="kw">file.path</span>(tmp,<span class="st">&quot;glimma&quot;</span>,<span class="st">&quot;MD-Plot.html&quot;</span>))</code></pre></div>
</div>
</div>
<div id="integration-with-zinb-wave" class="section level2">
<h2><span class="header-section-number">7.8</span> Integration with <em>ZINB-WaVE</em></h2>
<div id="simulate-with-splatter" class="section level3">
<h3><span class="header-section-number">7.8.1</span> Simulate with <em>splatter</em></h3>
<p>In this last section, we show that <em>DESeq2</em> can be integrated with another Bioconductor package <em>zinbwave</em> <span class="citation">(Risso et al. 2018a)</span> in order to model and account for additional zeros (more than expected by the Negative Binomial model). This can be useful for single cell RNA-seq experiments.</p>
<p>Here we use the <em>splatter</em> package to simulate single-cell RNA-seq data <span class="citation">(Zappia, Phipson, and Oshlack 2017)</span>. We then use the methods defined in <span class="citation">Van den Berge et al. (2018)</span> to combine <em>zinbwave</em> observation weights with <em>DESeq2</em> modeling of negative binomial counts.</p>
<p>From <span class="citation">Van den Berge et al. (2018)</span>:</p>
<blockquote>
<p>It is important to note that while methods such as ZINB-WaVE and ZINGER can successfully identify excess zeros, they cannot, however, readily discriminate between their underlying causes, i.e., between technical (e.g., dropout) and biological (e.g., bursting) zeros.</p>
</blockquote>
<p>The above note implies that the zero-inflation weighting approach outlined below can be used when the interesting signal is not in the zero component. That is, if you wanted to find biological differences in transcriptional bursting across groups of cells, the below approach would not help you find these differences. It instead helps to uncover differences in counts besides the zero component (whether those zeros be biological or technical).</p>
</div>
<div id="simulate-single-cell-count-data-with-splatter" class="section level3">
<h3><span class="header-section-number">7.8.2</span> Simulate single-cell count data with <em>splatter</em></h3>
<p>The following chunks of code create a <em>splatter</em> simulation:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;splatter&quot;</span>)
params &lt;-<span class="st"> </span><span class="kw">newSplatParams</span>()
params &lt;-<span class="st"> </span><span class="kw">setParam</span>(params, <span class="st">&quot;de.facLoc&quot;</span>, <span class="dv">1</span>) 
params &lt;-<span class="st"> </span><span class="kw">setParam</span>(params, <span class="st">&quot;de.facScale&quot;</span>, .<span class="dv">25</span>)
params &lt;-<span class="st"> </span><span class="kw">setParam</span>(params, <span class="st">&quot;dropout.type&quot;</span>, <span class="st">&quot;experiment&quot;</span>)
params &lt;-<span class="st"> </span><span class="kw">setParam</span>(params, <span class="st">&quot;dropout.mid&quot;</span>, <span class="dv">3</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">1</span>)
sim &lt;-<span class="st"> </span><span class="kw">splatSimulate</span>(params, <span class="dt">group.prob=</span><span class="kw">c</span>(.<span class="dv">5</span>,.<span class="dv">5</span>), <span class="dt">method=</span><span class="st">&quot;groups&quot;</span>)
<span class="co">#&gt; Getting parameters...</span>
<span class="co">#&gt; Creating simulation object...</span>
<span class="co">#&gt; Simulating library sizes...</span>
<span class="co">#&gt; Simulating gene means...</span>
<span class="co">#&gt; Simulating group DE...</span>
<span class="co">#&gt; Simulating cell means...</span>
<span class="co">#&gt; Simulating BCV...</span>
<span class="co">#&gt; Simulating counts...</span>
<span class="co">#&gt; Simulating dropout (if needed)...</span>
<span class="co">#&gt; Done!</span></code></pre></div>
<p>We can plot the amount of dropouts over the true counts:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(<span class="kw">log10</span>(<span class="kw">rowMeans</span>(<span class="kw">assays</span>(sim)[[<span class="st">&quot;TrueCounts&quot;</span>]])),
     <span class="kw">rowMeans</span>(<span class="kw">assays</span>(sim)[[<span class="st">&quot;Dropout&quot;</span>]]))</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-42-1.png" width="672" /></p>
<p>We will store the true log2 fold change for comparison:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rowData</span>(sim)<span class="op">$</span>log2FC &lt;-<span class="st"> </span><span class="kw">with</span>(<span class="kw">rowData</span>(sim), <span class="kw">log2</span>(DEFacGroup2<span class="op">/</span>DEFacGroup1))</code></pre></div>
<p>The true dispersion for the Negative Binomial component, over the mean:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rowData</span>(sim)<span class="op">$</span>trueDisp &lt;-<span class="st"> </span><span class="kw">rowMeans</span>(<span class="kw">assays</span>(sim)[[<span class="st">&quot;BCV&quot;</span>]])<span class="op">^</span><span class="dv">2</span>
gridlines &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">1e-2</span>,<span class="fl">1e-1</span>,<span class="dv">1</span>); cols &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;blue&quot;</span>,<span class="st">&quot;red&quot;</span>,<span class="st">&quot;darkgreen&quot;</span>)
<span class="kw">with</span>(<span class="kw">rowData</span>(sim)[<span class="kw">rowData</span>(sim)<span class="op">$</span>GeneMean<span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>,],
     <span class="kw">plot</span>(GeneMean, trueDisp, <span class="dt">log=</span><span class="st">&quot;xy&quot;</span>, <span class="dt">xlim=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">300</span>), <span class="dt">ylim=</span><span class="kw">c</span>(.<span class="dv">01</span>,<span class="dv">5</span>)))
<span class="kw">abline</span>(<span class="dt">h=</span>gridlines, <span class="dt">col=</span>cols)
<span class="kw">text</span>(<span class="dv">300</span>, gridlines, <span class="dt">labels=</span>gridlines, <span class="dt">col=</span>cols, <span class="dt">pos=</span><span class="dv">3</span>)</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-44-1.png" width="672" /></p>
</div>
<div id="model-zeros-with-zinbwave" class="section level3">
<h3><span class="header-section-number">7.8.3</span> Model zeros with <em>zinbwave</em></h3>
<p>The following code subsets the dataset and creates a <code>condition</code> variable that we will use to test for differential expression:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(zinbwave)
keep &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">counts</span>(sim) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">5</span>) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">25</span>
<span class="kw">table</span>(keep)
<span class="co">#&gt; keep</span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt;  9020   980</span>
zinb &lt;-<span class="st"> </span>sim[keep,]
zinb<span class="op">$</span>condition &lt;-<span class="st"> </span><span class="kw">factor</span>(zinb<span class="op">$</span>Group)</code></pre></div>
<p>We need to re-arrange the assays in the <code>zinb</code> object such that <code>&quot;counts&quot;</code> is the first assay:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">nms &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;counts&quot;</span>, <span class="kw">setdiff</span>(<span class="kw">assayNames</span>(zinb), <span class="st">&quot;counts&quot;</span>))
<span class="kw">assays</span>(zinb) &lt;-<span class="st"> </span><span class="kw">assays</span>(zinb)[nms]</code></pre></div>
<p>Finally we fit the <em>ZINB-WaVE</em> model. See <code>?zinbwave</code> and the <em>zinbwave</em> vignette for more details, including options on parallelization. It runs in less than a minute on this simulated dataset (with not so many cells).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">zinb &lt;-<span class="st"> </span><span class="kw">zinbwave</span>(zinb, <span class="dt">K=</span><span class="dv">0</span>, <span class="dt">BPPARAM=</span><span class="kw">SerialParam</span>(), <span class="dt">epsilon=</span><span class="fl">1e12</span>)</code></pre></div>
</div>
<div id="model-non-zeros-with-deseq2" class="section level3">
<h3><span class="header-section-number">7.8.4</span> Model non-zeros with <em>DESeq2</em></h3>
<p>Now we import the <code>zinb</code> object using <em>DESeqDataSet</em> (which works because the <em>SingleCellExperiment</em> object builds on top of the <em>SummarizedExperiment</em>). All of the simulation information comes along in the metadata columns of the object.</p>
<p><span class="citation">Van den Berge et al. (2018)</span> and others have shown the LRT may perform better for null hypothesis testing, so we use the LRT. In order to use the Wald test, it is recommended to set <code>useT=TRUE</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">zdds &lt;-<span class="st"> </span><span class="kw">DESeqDataSet</span>(zinb, <span class="dt">design=</span><span class="op">~</span>condition)
zdds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(zdds, <span class="dt">test=</span><span class="st">&quot;LRT&quot;</span>, <span class="dt">reduced=</span><span class="op">~</span><span class="dv">1</span>,
              <span class="dt">sfType=</span><span class="st">&quot;poscounts&quot;</span>, <span class="dt">minmu=</span><span class="fl">1e-6</span>, <span class="dt">minRep=</span><span class="ot">Inf</span>)
<span class="co">#&gt; estimating size factors</span>
<span class="co">#&gt; estimating dispersions</span>
<span class="co">#&gt; gene-wise dispersion estimates</span>
<span class="co">#&gt; mean-dispersion relationship</span>
<span class="co">#&gt; final dispersion estimates</span>
<span class="co">#&gt; fitting model and testing</span></code></pre></div>
</div>
<div id="plot-dispersion-estimates" class="section level3">
<h3><span class="header-section-number">7.8.5</span> Plot dispersion estimates</h3>
<p>It is recommended to plot the dispersion estimates for <em>DESeq2</em> on single-cell data. As discussed in the <em>DESeq2</em> paper, it becomes difficult to accurately estimate the dispersion when the counts are very small, because the Poisson component of the variance is dominant. Therefore we see some very low dispersion estimates here, although the trend is still accurately capturing the upper proportion. So here everything looks good.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotDispEsts</span>(zdds)</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-49-1.png" width="672" /></p>
<p>If the parametric trend fails to fit (there would be a warning in this case), one should check the dispersion plot as above. If it looks like the dispersion fit is being thrown off by the low count genes with low dispersion estimates at the bottom of the plot, there is a relatively easy solution: one can filter out more of the low count genes only for the dispersion estimation step, so that the trend still captures the upper portion. This is pretty easy to do in <em>DESeq2</em>, to filter genes solely for the dispersion trend estimation, but to use a larger set for the rest of the analysis. An example of how this can be done:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">keepForDispTrend &lt;-<span class="st"> </span><span class="kw">rowSums</span>(<span class="kw">counts</span>(zdds) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">10</span>) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">25</span>
zdds2 &lt;-<span class="st"> </span><span class="kw">estimateDispersionsFit</span>(zdds[keepForDispTrend,])
<span class="kw">plotDispEsts</span>(zdds2)</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-50-1.png" width="672" /></p>
<p>One would then assign the dispersion function to the original dataset, re-estimate final dispersions, check <code>plotDispEsts</code>, and then either re-run the Wald or LRT function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dispersionFunction</span>(zdds) &lt;-<span class="st"> </span><span class="kw">dispersionFunction</span>(zdds2)
zdds &lt;-<span class="st"> </span><span class="kw">estimateDispersionsMAP</span>(zdds)
<span class="co">#&gt; found already estimated dispersions, removing these</span>
zdds &lt;-<span class="st"> </span><span class="kw">nbinomLRT</span>(zdds, <span class="dt">reduced=</span><span class="op">~</span><span class="dv">1</span>, <span class="dt">minmu=</span><span class="fl">1e-6</span>)
<span class="co">#&gt; found results columns, replacing these</span></code></pre></div>
</div>
<div id="evaluation-against-truth" class="section level3">
<h3><span class="header-section-number">7.8.6</span> Evaluation against truth</h3>
<p>Compare dispersion on the non-zero-component counts to the true value used for simulation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">with</span>(<span class="kw">mcols</span>(zdds), <span class="kw">plot</span>(trueDisp, dispMAP, <span class="dt">log=</span><span class="st">&quot;xy&quot;</span>))
<span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col=</span><span class="st">&quot;red&quot;</span>)</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-52-1.png" width="672" /></p>
<p>Extract results table:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">zres &lt;-<span class="st"> </span><span class="kw">results</span>(zdds, <span class="dt">independentFiltering=</span><span class="ot">FALSE</span>)
<span class="kw">plot</span>(<span class="kw">mcols</span>(zdds)<span class="op">$</span>log2FC, zres<span class="op">$</span>log2FoldChange, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">4</span>,<span class="dv">4</span>)); <span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col=</span><span class="st">&quot;red&quot;</span>)</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/unnamed-chunk-53-1.png" width="672" /></p>
<p>Below we show that the simple LFC does not work - it over-estimates the true DE LFC because of the dropout zeros in the group with the smaller mean. It also has a lot of noise for the null genes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ncts &lt;-<span class="st"> </span><span class="kw">counts</span>(zdds, <span class="dt">normalized=</span><span class="ot">TRUE</span>)
simple.lfc &lt;-<span class="st"> </span><span class="kw">log2</span>(<span class="kw">rowMeans</span>(ncts[,zdds<span class="op">$</span>condition <span class="op">==</span><span class="st"> &quot;Group2&quot;</span>])<span class="op">/</span>
<span class="st">                   </span><span class="kw">rowMeans</span>(ncts[,zdds<span class="op">$</span>condition <span class="op">==</span><span class="st"> &quot;Group1&quot;</span>]))
<span class="kw">plot</span>(<span class="kw">mcols</span>(zdds)<span class="op">$</span>log2FC, simple.lfc, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">4</span>,<span class="dv">4</span>)); <span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>,<span class="dt">col=</span><span class="st">&quot;red&quot;</span>)</code></pre></div>
<p><img src="201_Love_DESeq2_files/figure-html/trueLFCVsSimple-1.png" width="672" /></p>
<p>How well do we do in null hypothesis testing:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tab &lt;-<span class="st"> </span><span class="kw">table</span>(<span class="dt">sig=</span>zres<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span>.<span class="dv">05</span>, <span class="dt">DE.status=</span><span class="kw">mcols</span>(zdds)<span class="op">$</span>log2FC <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>)
tab
<span class="co">#&gt;        DE.status</span>
<span class="co">#&gt; sig     FALSE TRUE</span>
<span class="co">#&gt;   FALSE   750   15</span>
<span class="co">#&gt;   TRUE     11  204</span>
<span class="kw">round</span>(<span class="kw">prop.table</span>(tab, <span class="dv">1</span>), <span class="dv">3</span>)
<span class="co">#&gt;        DE.status</span>
<span class="co">#&gt; sig     FALSE  TRUE</span>
<span class="co">#&gt;   FALSE 0.980 0.020</span>
<span class="co">#&gt;   TRUE  0.051 0.949</span></code></pre></div>

</div>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="12">
<li id="fn12"><p>UNC-Chapel Hill, NC, US<a href="rna-seq-data-analysis-with-deseq2.html#fnref12"></a></p></li>
<li id="fn13"><p>ZMBH Heidelberg, Germany<a href="rna-seq-data-analysis-with-deseq2.html#fnref13"></a></p></li>
<li id="fn14"><p>EMBL Heidelberg, Germany<a href="rna-seq-data-analysis-with-deseq2.html#fnref14"></a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
},
"split_bib": true
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
