<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html">
<link rel="next" href="fluent-genomic-data-analysis-with-plyranges.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#for-everyone"><i class="fa fa-check"></i><b>1.1</b> For Everyone</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#for-workshop-authors"><i class="fa fa-check"></i><b>1.2</b> For Workshop Authors</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#description"><i class="fa fa-check"></i><b>1.2.1</b> DESCRIPTION</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#classifying-your-workshop"><i class="fa fa-check"></i><b>1.2.2</b> Classifying your workshop</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#bookdown"><i class="fa fa-check"></i><b>1.2.3</b> Bookdown</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#deadlines-for-bioc2018"><i class="fa fa-check"></i><b>1.3</b> Deadlines for Bioc2018</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>2</b> RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-1"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>2.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>2.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="2.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>2.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="2.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>2.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="2.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>2.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>2.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>2.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="2.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>2.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="2.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>2.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>2.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="2.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>2.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="2.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>2.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="2.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>2.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis"><i class="fa fa-check"></i><b>2.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="2.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>2.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="2.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>2.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub"><i class="fa fa-check"></i><b>2.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="2.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub"><i class="fa fa-check"></i><b>2.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="2.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>2.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>2.7</b> Building reports</a><ul>
<li class="chapter" data-level="2.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>2.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="2.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>2.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>2.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="2.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>2.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="2.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>2.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="2.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>2.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="2.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>2.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="2.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>2.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="2.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>2.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>3</b> Cytoscape Automation in R using Rcy3</a><ul>
<li class="chapter" data-level="3.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-1"><i class="fa fa-check"></i><b>3.1</b> Overview</a><ul>
<li class="chapter" data-level="3.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructors-names-and-contact-information"><i class="fa fa-check"></i><b>3.1.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="3.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description"><i class="fa fa-check"></i><b>3.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="3.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>3.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="3.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>3.1.4</b> Background:</a></li>
<li class="chapter" data-level="3.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation"><i class="fa fa-check"></i><b>3.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="3.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="3.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-1"><i class="fa fa-check"></i><b>3.1.7</b> Time outline</a></li>
<li class="chapter" data-level="3.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>3.2</b> Background</a><ul>
<li class="chapter" data-level="3.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>3.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="3.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>3.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="3.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>3.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>3.3</b> Translating biological data into Cytoscape using RCy3</a><ul>
<li class="chapter" data-level="3.3.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>3.3.1</b> Set up</a></li>
<li class="chapter" data-level="3.3.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started"><i class="fa fa-check"></i><b>3.3.2</b> Getting started</a></li>
<li class="chapter" data-level="3.3.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>3.3.3</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="3.3.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>3.3.4</b> Example Data Set</a></li>
<li class="chapter" data-level="3.3.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>3.3.5</b> Finding Network Data</a></li>
<li class="chapter" data-level="3.3.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>3.3.6</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="3.3.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>3.3.7</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="3.3.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>3.3.8</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#references"><i class="fa fa-check"></i><b>3.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><i class="fa fa-check"></i><b>4</b> Variant Functional Annotation using StatePaintR, FunciVar &amp; MotifBreakR</a><ul>
<li class="chapter" data-level="4.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i><b>4.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="4.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-description-1"><i class="fa fa-check"></i><b>4.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="4.2.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#pre-requisites-1"><i class="fa fa-check"></i><b>4.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.2.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-participation-1"><i class="fa fa-check"></i><b>4.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="4.2.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.2.4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#time-outline-2"><i class="fa fa-check"></i><b>4.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="4.3.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#key-concepts"><i class="fa fa-check"></i><b>4.3.1</b> Key Concepts</a></li>
<li class="chapter" data-level="4.3.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#learning-objectives"><i class="fa fa-check"></i><b>4.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>5</b> Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="5.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructors-names-and-contact-information-2"><i class="fa fa-check"></i><b>5.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="5.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-2"><i class="fa fa-check"></i><b>5.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="5.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-2"><i class="fa fa-check"></i><b>5.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="5.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-2"><i class="fa fa-check"></i><b>5.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>5.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="5.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-3"><i class="fa fa-check"></i><b>5.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>5.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="5.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop"><i class="fa fa-check"></i><b>5.4</b> Workshop</a></li>
<li class="chapter" data-level="5.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>5.5</b> Where does it all come from?</a></li>
<li class="chapter" data-level="5.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>5.6</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="5.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>5.7</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="5.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>5.8</b> Data types</a></li>
<li class="chapter" data-level="5.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>5.9</b> Differential expression analysis</a></li>
<li class="chapter" data-level="5.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>5.10</b> Gene sets</a></li>
<li class="chapter" data-level="5.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>5.11</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="5.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>5.12</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="5.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>5.13</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="5.14" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>5.14</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>6</b> RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>6.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-3"><i class="fa fa-check"></i><b>6.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="6.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-3"><i class="fa fa-check"></i><b>6.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="6.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>6.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-4"><i class="fa fa-check"></i><b>6.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>6.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="6.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals"><i class="fa fa-check"></i><b>6.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="6.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-1"><i class="fa fa-check"></i><b>6.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-1"><i class="fa fa-check"></i><b>6.4</b> Introduction</a></li>
<li class="chapter" data-level="6.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>6.5</b> Data packaging</a><ul>
<li class="chapter" data-level="6.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>6.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="6.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>6.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="6.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>6.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>6.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="6.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>6.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="6.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>6.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="6.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>6.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="6.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>6.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>6.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="6.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>6.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="6.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>6.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="6.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>6.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="6.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>6.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="6.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>6.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="6.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>6.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>7</b> Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="7.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructors-names-and-contact-information-3"><i class="fa fa-check"></i><b>7.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="7.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-4"><i class="fa fa-check"></i><b>7.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="7.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-4"><i class="fa fa-check"></i><b>7.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="7.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>7.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-5"><i class="fa fa-check"></i><b>7.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>7.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="7.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-1"><i class="fa fa-check"></i><b>7.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="7.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-2"><i class="fa fa-check"></i><b>7.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>7.4</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="7.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>7.5</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="7.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>7.6</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="7.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>7.7</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="7.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>7.8</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="7.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#grangeslist-groups-of-genomic-ranges"><i class="fa fa-check"></i><b>7.9</b> <em>GRangesList</em>: Groups of Genomic Ranges</a></li>
<li class="chapter" data-level="7.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-grangeslist-accessors"><i class="fa fa-check"></i><b>7.10</b> Basic <em>GRangesList</em> accessors</a></li>
<li class="chapter" data-level="7.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#combining-grangeslist-objects"><i class="fa fa-check"></i><b>7.11</b> Combining <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="7.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-grangeslist-objects"><i class="fa fa-check"></i><b>7.12</b> Basic interval operations for <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="7.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-grangeslist-objects"><i class="fa fa-check"></i><b>7.13</b> Subsetting <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="7.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#looping-over-grangeslist-objects"><i class="fa fa-check"></i><b>7.14</b> Looping over <em>GRangesList</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>8</b> Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructors-names-and-contact-information-4"><i class="fa fa-check"></i><b>8.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-5"><i class="fa fa-check"></i><b>8.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="8.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-5"><i class="fa fa-check"></i><b>8.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="8.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>8.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-6"><i class="fa fa-check"></i><b>8.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>8.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="8.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-2"><i class="fa fa-check"></i><b>8.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="8.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-3"><i class="fa fa-check"></i><b>8.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-1"><i class="fa fa-check"></i><b>8.4</b> Workshop</a></li>
<li class="chapter" data-level="8.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-2"><i class="fa fa-check"></i><b>8.5</b> Introduction</a><ul>
<li class="chapter" data-level="8.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>8.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="8.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>8.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="8.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>8.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup"><i class="fa fa-check"></i><b>8.6</b> Setup</a></li>
<li class="chapter" data-level="8.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>8.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="8.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>8.8</b> The Grammar</a><ul>
<li class="chapter" data-level="8.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>8.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="8.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>8.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="8.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises"><i class="fa fa-check"></i><b>8.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="8.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>8.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="8.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>8.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="8.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>8.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="8.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>8.10</b> Whatâ€™s next?</a></li>
<li class="chapter" data-level="8.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>8.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>9</b> Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="9.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>9.1</b> Instructors</a></li>
<li class="chapter" data-level="9.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description-6"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>9.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation-6"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>9.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="9.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-3"><i class="fa fa-check"></i><b>9.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="9.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-4"><i class="fa fa-check"></i><b>9.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>9.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="9.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>9.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="9.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>9.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="9.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>9.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="9.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>9.4.4</b> Data containers</a></li>
<li class="chapter" data-level="9.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>9.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="9.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>9.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="9.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>9.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="9.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>9.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="9.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>9.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="9.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>9.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="9.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>9.4.11</b> Simple example</a></li>
<li class="chapter" data-level="9.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>9.4.12</b> Questions!</a></li>
<li class="chapter" data-level="9.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>9.4.13</b> More questions!</a></li>
<li class="chapter" data-level="9.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>9.4.14</b> Another example</a></li>
<li class="chapter" data-level="9.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>9.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="9.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>9.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="9.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>9.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="9.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>9.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="9.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>9.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="9.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>9.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="9.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>9.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="9.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>9.4.22</b> GRanges</a></li>
<li class="chapter" data-level="9.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>9.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="9.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>9.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="9.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>9.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="9.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>9.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="9.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>9.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="9.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>9.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="9.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>9.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="9.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>9.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="9.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>9.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="9.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>9.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="9.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dply-example"><i class="fa fa-check"></i><b>9.4.33</b> Organism.dply example</a></li>
<li class="chapter" data-level="9.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>9.4.34</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="9.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>9.4.35</b> BSgenome packages</a></li>
<li class="chapter" data-level="9.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>9.4.36</b> BSgenome packages</a></li>
<li class="chapter" data-level="9.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>9.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="9.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>9.4.38</b> BSgenome exercises</a></li>
<li class="chapter" data-level="9.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-1"><i class="fa fa-check"></i><b>9.4.39</b> AnnotationHub</a></li>
<li class="chapter" data-level="9.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>9.4.40</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="9.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>9.4.41</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="9.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>9.4.42</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="9.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>9.4.43</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="9.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>9.4.44</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="9.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>9.4.45</b> AnnotationHub query</a></li>
<li class="chapter" data-level="9.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>9.4.46</b> AnnotationHub query</a></li>
<li class="chapter" data-level="9.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>9.4.47</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="9.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>9.4.48</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="9.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>9.4.49</b> biomaRt</a></li>
<li class="chapter" data-level="9.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>9.4.50</b> biomaRt data sets</a></li>
<li class="chapter" data-level="9.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>9.4.51</b> biomaRt queries</a></li>
<li class="chapter" data-level="9.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>9.4.52</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="9.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>9.4.53</b> biomaRt query</a></li>
<li class="chapter" data-level="9.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>9.4.54</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>10</b> Workflow for Multi-omics Analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="10.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>10.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-7"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-6"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-7"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>10.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-7"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-4"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-5"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>10.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="10.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="10.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>10.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="10.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>10.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>10.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="10.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>10.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="10.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>10.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="10.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>10.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>10.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="10.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>10.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="10.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>10.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="10.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>10.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="10.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>10.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="10.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>10.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>10.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="10.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>10.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="10.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>10.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="10.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>10.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>10.9</b> Complete cases</a></li>
<li class="chapter" data-level="10.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>10.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="10.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>10.11</b> Extraction</a><ul>
<li class="chapter" data-level="10.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>10.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="10.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>10.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="10.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>10.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="10.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>10.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="10.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>10.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="10.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>10.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="10.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>10.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="10.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>10.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="10.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>10.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>10.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="10.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>10.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="10.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>10.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="10.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>10.16.1</b> â€œSimplificationâ€ of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="10.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>10.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="10.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>10.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="10.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>10.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="10.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>10.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="10.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>10.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="10.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>10.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="10.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>10.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="10.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>10.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="10.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>10.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="10.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>10.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="10.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>10.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>11</b> Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="11.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>11.1</b> Instructors:</a></li>
<li class="chapter" data-level="11.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-8"><i class="fa fa-check"></i><b>11.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="11.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-7"><i class="fa fa-check"></i><b>11.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="11.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-8"><i class="fa fa-check"></i><b>11.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>11.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="11.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-8"><i class="fa fa-check"></i><b>11.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="11.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-5"><i class="fa fa-check"></i><b>11.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="11.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-6"><i class="fa fa-check"></i><b>11.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>12</b> Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="12.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>12.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="12.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-9"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-8"><i class="fa fa-check"></i><b>12.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-1"><i class="fa fa-check"></i><b>12.2.2</b> Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>12.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="12.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-4"><i class="fa fa-check"></i><b>12.3</b> Introduction</a></li>
<li class="chapter" data-level="12.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>12.4</b> Why maintain your package</a></li>
<li class="chapter" data-level="12.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>12.5</b> Infrastructure used for maintenance</a></li>
<li class="chapter" data-level="12.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>12.6</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="12.6.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>12.6.1</b> Build report</a></li>
<li class="chapter" data-level="12.6.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>12.6.2</b> Landing page badges</a></li>
<li class="chapter" data-level="12.6.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>12.6.3</b> RSS feeds</a></li>
<li class="chapter" data-level="12.6.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>12.6.4</b> Download statistics</a></li>
<li class="chapter" data-level="12.6.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-github-issues-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>12.6.5</b> Support site, Github issues, Bioc-devel mailing list</a></li>
</ul></li>
<li class="chapter" data-level="12.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#sync-package-with-github-bioconductor-before-every-change"><i class="fa fa-check"></i><b>12.7</b> Sync package with Github / Bioconductor before every change</a><ul>
<li class="chapter" data-level="12.7.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github-webhooks"><i class="fa fa-check"></i><b>12.7.1</b> GitHub + webhooks</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>12.8</b> Round up of resources available</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>13</b> Public Data Resources and Bioconductor</a><ul>
<li class="chapter" data-level="13.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>13.1</b> Instructor name(s) and contact information</a></li>
<li class="chapter" data-level="13.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>13.2</b> Syllabus</a><ul>
<li class="chapter" data-level="13.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-10"><i class="fa fa-check"></i><b>13.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="13.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-9"><i class="fa fa-check"></i><b>13.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-9"><i class="fa fa-check"></i><b>13.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>13.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="13.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-9"><i class="fa fa-check"></i><b>13.2.5</b> Time outline</a></li>
<li class="chapter" data-level="13.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.2.6</b> workshop goals and objectives</a></li>
<li class="chapter" data-level="13.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-6"><i class="fa fa-check"></i><b>13.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="13.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-7"><i class="fa fa-check"></i><b>13.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-2"><i class="fa fa-check"></i><b>13.3</b> Overview</a></li>
<li class="chapter" data-level="13.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>13.4</b> GEOquery</a><ul>
<li class="chapter" data-level="13.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>13.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="13.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>13.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="13.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>13.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="13.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>13.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>13.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="13.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>13.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="13.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>13.5.2</b> Basic design</a></li>
<li class="chapter" data-level="13.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>13.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>13.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="13.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>13.6.1</b> Authentication</a></li>
<li class="chapter" data-level="13.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>13.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="13.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>13.7</b> Sequence Read Archive</a></li>
<li class="chapter" data-level="13.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>13.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="13.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>13.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="13.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>13.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>13.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="13.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>13.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="13.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>13.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="13.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>13.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="13.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>13.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="13.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>13.12</b> Pharmacogenomics</a></li>
<li class="chapter" data-level="13.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>13.13</b> Bibliography</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="solving-common-bioinformatic-challenges-using-genomicranges" class="section level1">
<h1><span class="header-section-number">7</span> Solving common bioinformatic challenges using GenomicRanges</h1>
<div id="instructors-names-and-contact-information-3" class="section level2">
<h2><span class="header-section-number">7.1</span> Instructor(s) name(s) and contact information</h2>
<ul>
<li>Michael Lawrence (<a href="mailto:michafla@gene.com">michafla@gene.com</a>)</li>
</ul>
</div>
<div id="workshop-description-4" class="section level2">
<h2><span class="header-section-number">7.2</span> Workshop Description</h2>
<p>We will introduce the fundamental concepts underlying the GenomicRanges package and related infrastructure. After a structured introduction, we will follow a realistic workflow, along the way exploring the central data structures, including GRanges and SummarizedExperiment, and useful operations in the ranges algebra. Topics will include data import/export, computing and summarizing data on genomic features, overlap detection, integration with reference annotations, scaling strategies, and visualization. Students can follow along, and there will be plenty of time for students to ask questions about how to apply the infrastructure to their particular use case. Michael Lawrence (Genentech).</p>
<div id="pre-requisites-4" class="section level3">
<h3><span class="header-section-number">7.2.1</span> Pre-requisites</h3>
<ul>
<li>Solid understanding of R</li>
<li>Basic familiarity with GRanges objects</li>
<li>Basic familiarity with packages like S4Vectors, IRanges, GenomicRanges, rtracklayer, etc.</li>
</ul>
</div>
<div id="workshop-participation-4" class="section level3">
<h3><span class="header-section-number">7.2.2</span> Workshop Participation</h3>
<p>Describe how students will be expected to participate in the workshop.</p>
</div>
<div id="r-bioconductor-packages-used-5" class="section level3">
<h3><span class="header-section-number">7.2.3</span> <em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li>S4Vectors</li>
<li>IRanges</li>
<li>GenomicRanges</li>
<li>rtracklayer</li>
<li>GenomicFeatures</li>
<li>SummarizedExperiment</li>
<li>VariantAnnotation</li>
<li>GenomicAlignments</li>
</ul>
</div>
<div id="time-outline-5" class="section level3">
<h3><span class="header-section-number">7.2.4</span> Time outline</h3>
<table>
<thead>
<tr class="header">
<th>Activity</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Intro slides</td>
<td>30m</td>
</tr>
<tr class="even">
<td>Workflow(s)</td>
<td>1hr</td>
</tr>
<tr class="odd">
<td>Remaining questions</td>
<td>30m</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="workshop-goals-and-objectives-4" class="section level2">
<h2><span class="header-section-number">7.3</span> Workshop goals and objectives</h2>
<div id="learning-goals-1" class="section level3">
<h3><span class="header-section-number">7.3.1</span> Learning goals</h3>
<ul>
<li>Understand how to apply the *Ranges infrastructure to real-world problems</li>
<li>Gain insight into the design principles of the infrastructure and how it was meant to be used</li>
</ul>
</div>
<div id="learning-objectives-2" class="section level3">
<h3><span class="header-section-number">7.3.2</span> Learning objectives</h3>
<ul>
<li>Manipulate GRanges and related objects</li>
<li>Use the ranges algebra to analyze genomic ranges</li>
<li>Implement efficient workflows based on the *Ranges infrastructure</li>
</ul>
</div>
</div>
<div id="granges-genomic-ranges" class="section level2">
<h2><span class="header-section-number">7.4</span> <em>GRanges</em>: Genomic Ranges</h2>
<p>The <em>GRanges</em> class represents a collection of genomic ranges that each have a single start and end location on the genome. It can be used to store the location of genomic features such as contiguous binding sites, transcripts, and exons. These objects can be created by using the <code>GRanges</code> constructor function. For example,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>({
 <span class="kw">library</span>(BiocStyle)
 <span class="kw">library</span>(GenomicRanges)
})</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gr &lt;-<span class="st"> </span><span class="kw">GRanges</span>(
    <span class="dt">seqnames =</span> <span class="kw">Rle</span>(<span class="kw">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr2&quot;</span>, <span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr3&quot;</span>), <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">4</span>)),
    <span class="dt">ranges =</span> <span class="kw">IRanges</span>(<span class="dv">101</span><span class="op">:</span><span class="dv">110</span>, <span class="dt">end =</span> <span class="dv">111</span><span class="op">:</span><span class="dv">120</span>, <span class="dt">names =</span> <span class="kw">head</span>(letters, <span class="dv">10</span>)),
    <span class="dt">strand =</span> <span class="kw">Rle</span>(<span class="kw">strand</span>(<span class="kw">c</span>(<span class="st">&quot;-&quot;</span>, <span class="st">&quot;+&quot;</span>, <span class="st">&quot;*&quot;</span>, <span class="st">&quot;+&quot;</span>, <span class="st">&quot;-&quot;</span>)), <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">2</span>)),
    <span class="dt">score =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,
    <span class="dt">GC =</span> <span class="kw">seq</span>(<span class="dv">1</span>, <span class="dv">0</span>, <span class="dt">length=</span><span class="dv">10</span>))
gr
<span class="co">#&gt; GRanges object with 10 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   101-111      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   102-112      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   103-113      + |         3 0.777777777777778</span>
<span class="co">#&gt;   d     chr2   104-114      * |         4 0.666666666666667</span>
<span class="co">#&gt;   e     chr1   105-115      * |         5 0.555555555555556</span>
<span class="co">#&gt;   f     chr1   106-116      + |         6 0.444444444444444</span>
<span class="co">#&gt;   g     chr3   107-117      + |         7 0.333333333333333</span>
<span class="co">#&gt;   h     chr3   108-118      + |         8 0.222222222222222</span>
<span class="co">#&gt;   i     chr3   109-119      - |         9 0.111111111111111</span>
<span class="co">#&gt;   j     chr3   110-120      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">options</span>(<span class="dt">warn=</span><span class="dv">2</span>)</code></pre></div>
<p>creates a <em>GRanges</em> object with 10 genomic ranges. The output of the <em>GRanges</em> <code>show</code> method separates the information into a left and right hand region that are separated by <code>|</code> symbols. The genomic coordinates (seqnames, ranges, and strand) are located on the left-hand side and the metadata columns (annotation) are located on the right. For this example, the metadata is comprised of <code>score</code> and <code>GC</code> information, but almost anything can be stored in the metadata portion of a <em>GRanges</em> object.</p>
<p>The components of the genomic coordinates within a <em>GRanges</em> object can be extracted using the <code>seqnames</code>, <code>ranges</code>, and <code>strand</code> accessor functions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">seqnames</span>(gr)
<span class="co">#&gt; factor-Rle of length 10 with 4 runs</span>
<span class="co">#&gt;   Lengths:    1    3    2    4</span>
<span class="co">#&gt;   Values : chr1 chr2 chr1 chr3</span>
<span class="co">#&gt; Levels(3): chr1 chr2 chr3</span>
<span class="kw">ranges</span>(gr)
<span class="co">#&gt; IRanges object with 10 ranges and 0 metadata columns:</span>
<span class="co">#&gt;         start       end     width</span>
<span class="co">#&gt;     &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   a       101       111        11</span>
<span class="co">#&gt;   b       102       112        11</span>
<span class="co">#&gt;   c       103       113        11</span>
<span class="co">#&gt;   d       104       114        11</span>
<span class="co">#&gt;   e       105       115        11</span>
<span class="co">#&gt;   f       106       116        11</span>
<span class="co">#&gt;   g       107       117        11</span>
<span class="co">#&gt;   h       108       118        11</span>
<span class="co">#&gt;   i       109       119        11</span>
<span class="co">#&gt;   j       110       120        11</span>
<span class="kw">strand</span>(gr)
<span class="co">#&gt; factor-Rle of length 10 with 5 runs</span>
<span class="co">#&gt;   Lengths: 1 2 2 3 2</span>
<span class="co">#&gt;   Values : - + * + -</span>
<span class="co">#&gt; Levels(3): + - *</span></code></pre></div>
<p>The genomic ranges can be extracted without corresponding metadata with <code>granges</code></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">granges</span>(gr)
<span class="co">#&gt; GRanges object with 10 ranges and 0 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   a     chr1   101-111      -</span>
<span class="co">#&gt;   b     chr2   102-112      +</span>
<span class="co">#&gt;   c     chr2   103-113      +</span>
<span class="co">#&gt;   d     chr2   104-114      *</span>
<span class="co">#&gt;   e     chr1   105-115      *</span>
<span class="co">#&gt;   f     chr1   106-116      +</span>
<span class="co">#&gt;   g     chr3   107-117      +</span>
<span class="co">#&gt;   h     chr3   108-118      +</span>
<span class="co">#&gt;   i     chr3   109-119      -</span>
<span class="co">#&gt;   j     chr3   110-120      -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Annotations for these coordinates can be extracted as a <em>DataFrame</em> object using the <code>mcols</code> accessor.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">mcols</span>(gr)
<span class="co">#&gt; DataFrame with 10 rows and 2 columns</span>
<span class="co">#&gt;       score                GC</span>
<span class="co">#&gt;   &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt; a         1                 1</span>
<span class="co">#&gt; b         2 0.888888888888889</span>
<span class="co">#&gt; c         3 0.777777777777778</span>
<span class="co">#&gt; d         4 0.666666666666667</span>
<span class="co">#&gt; e         5 0.555555555555556</span>
<span class="co">#&gt; f         6 0.444444444444444</span>
<span class="co">#&gt; g         7 0.333333333333333</span>
<span class="co">#&gt; h         8 0.222222222222222</span>
<span class="co">#&gt; i         9 0.111111111111111</span>
<span class="co">#&gt; j        10                 0</span>
<span class="kw">mcols</span>(gr)<span class="op">$</span>score
<span class="co">#&gt;  [1]  1  2  3  4  5  6  7  8  9 10</span></code></pre></div>
<p>Information about the lengths of the various sequences that the ranges are aligned to can also be stored in the <em>GRanges</em> object. So if this is data from <em>Homo sapiens</em>, we can set the values as:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">seqlengths</span>(gr) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">249250621</span>, <span class="dv">243199373</span>, <span class="dv">198022430</span>)</code></pre></div>
<p>And then retrieves as:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">seqlengths</span>(gr)
<span class="co">#&gt;      chr1      chr2      chr3 </span>
<span class="co">#&gt; 249250621 243199373 198022430</span></code></pre></div>
<p>Methods for accessing the <code>length</code> and <code>names</code> have also been defined.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(gr)
<span class="co">#&gt;  [1] &quot;a&quot; &quot;b&quot; &quot;c&quot; &quot;d&quot; &quot;e&quot; &quot;f&quot; &quot;g&quot; &quot;h&quot; &quot;i&quot; &quot;j&quot;</span>
<span class="kw">length</span>(gr)
<span class="co">#&gt; [1] 10</span></code></pre></div>
</div>
<div id="splitting-and-combining-granges-objects" class="section level2">
<h2><span class="header-section-number">7.5</span> Splitting and combining <em>GRanges</em> objects</h2>
<p><em>GRanges</em> objects can be devided into groups using the <code>split</code> method. This produces a <em>GRangesList</em> object, a class that will be discussed in detail in the next section.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sp &lt;-<span class="st"> </span><span class="kw">split</span>(gr, <span class="kw">rep</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>, <span class="dt">each=</span><span class="dv">5</span>))
sp
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $1 </span>
<span class="co">#&gt; GRanges object with 5 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   101-111      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   102-112      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   103-113      + |         3 0.777777777777778</span>
<span class="co">#&gt;   d     chr2   104-114      * |         4 0.666666666666667</span>
<span class="co">#&gt;   e     chr1   105-115      * |         5 0.555555555555556</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $2 </span>
<span class="co">#&gt; GRanges object with 5 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames  ranges strand | score                GC</span>
<span class="co">#&gt;   f     chr1 106-116      + |     6 0.444444444444444</span>
<span class="co">#&gt;   g     chr3 107-117      + |     7 0.333333333333333</span>
<span class="co">#&gt;   h     chr3 108-118      + |     8 0.222222222222222</span>
<span class="co">#&gt;   i     chr3 109-119      - |     9 0.111111111111111</span>
<span class="co">#&gt;   j     chr3 110-120      - |    10                 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>Separate <em>GRanges</em> instances can be concatenated by using the <code>c</code> and <code>append</code> methods.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">c</span>(sp[[<span class="dv">1</span>]], sp[[<span class="dv">2</span>]])
<span class="co">#&gt; GRanges object with 10 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   101-111      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   102-112      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   103-113      + |         3 0.777777777777778</span>
<span class="co">#&gt;   d     chr2   104-114      * |         4 0.666666666666667</span>
<span class="co">#&gt;   e     chr1   105-115      * |         5 0.555555555555556</span>
<span class="co">#&gt;   f     chr1   106-116      + |         6 0.444444444444444</span>
<span class="co">#&gt;   g     chr3   107-117      + |         7 0.333333333333333</span>
<span class="co">#&gt;   h     chr3   108-118      + |         8 0.222222222222222</span>
<span class="co">#&gt;   i     chr3   109-119      - |         9 0.111111111111111</span>
<span class="co">#&gt;   j     chr3   110-120      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
</div>
<div id="subsetting-granges-objects" class="section level2">
<h2><span class="header-section-number">7.6</span> Subsetting <em>GRanges</em> objects</h2>
<p><em>GRanges</em> objects act like vectors of ranges, with the expected vector-like subsetting operations available</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gr[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>]
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   b     chr2   102-112      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   103-113      + |         3 0.777777777777778</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>A second argument to the <code>[</code> subset operator can be used to specify which metadata columns to extract from the <em>GRanges</em> object. For example,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gr[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>, <span class="st">&quot;GC&quot;</span>]
<span class="co">#&gt; GRanges object with 2 ranges and 1 metadata column:</span>
<span class="co">#&gt;     seqnames    ranges strand |                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |         &lt;numeric&gt;</span>
<span class="co">#&gt;   b     chr2   102-112      + | 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   103-113      + | 0.777777777777778</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>Elements can also be assigned to the <em>GRanges</em> object. Here is an example where the second row of a <em>GRanges</em> object is replaced with the first row of <code>gr</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">singles &lt;-<span class="st"> </span><span class="kw">split</span>(gr, <span class="kw">names</span>(gr))
grMod &lt;-<span class="st"> </span>gr
grMod[<span class="dv">2</span>] &lt;-<span class="st"> </span>singles[[<span class="dv">1</span>]]
<span class="kw">head</span>(grMod, <span class="dt">n=</span><span class="dv">3</span>)
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   101-111      - |         1                 1</span>
<span class="co">#&gt;   b     chr1   101-111      - |         1                 1</span>
<span class="co">#&gt;   c     chr2   103-113      + |         3 0.777777777777778</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>There are methods to repeat, reverse, or select specific portions of <em>GRanges</em> objects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rep</span>(singles[[<span class="dv">2</span>]], <span class="dt">times =</span> <span class="dv">3</span>)
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   b     chr2   102-112      + |         2 0.888888888888889</span>
<span class="co">#&gt;   b     chr2   102-112      + |         2 0.888888888888889</span>
<span class="co">#&gt;   b     chr2   102-112      + |         2 0.888888888888889</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span>
<span class="kw">rev</span>(gr)
<span class="co">#&gt; GRanges object with 10 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   j     chr3   110-120      - |        10                 0</span>
<span class="co">#&gt;   i     chr3   109-119      - |         9 0.111111111111111</span>
<span class="co">#&gt;   h     chr3   108-118      + |         8 0.222222222222222</span>
<span class="co">#&gt;   g     chr3   107-117      + |         7 0.333333333333333</span>
<span class="co">#&gt;   f     chr1   106-116      + |         6 0.444444444444444</span>
<span class="co">#&gt;   e     chr1   105-115      * |         5 0.555555555555556</span>
<span class="co">#&gt;   d     chr2   104-114      * |         4 0.666666666666667</span>
<span class="co">#&gt;   c     chr2   103-113      + |         3 0.777777777777778</span>
<span class="co">#&gt;   b     chr2   102-112      + |         2 0.888888888888889</span>
<span class="co">#&gt;   a     chr1   101-111      - |         1                 1</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span>
<span class="kw">head</span>(gr,<span class="dt">n=</span><span class="dv">2</span>)
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   101-111      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   102-112      + |         2 0.888888888888889</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span>
<span class="kw">tail</span>(gr,<span class="dt">n=</span><span class="dv">2</span>)
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   i     chr3   109-119      - |         9 0.111111111111111</span>
<span class="co">#&gt;   j     chr3   110-120      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span>
<span class="kw">window</span>(gr, <span class="dt">start=</span><span class="dv">2</span>,<span class="dt">end=</span><span class="dv">4</span>)
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   b     chr2   102-112      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   103-113      + |         3 0.777777777777778</span>
<span class="co">#&gt;   d     chr2   104-114      * |         4 0.666666666666667</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span>
gr[<span class="kw">IRanges</span>(<span class="dt">start=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">7</span>), <span class="dt">end=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">9</span>))]
<span class="co">#&gt; GRanges object with 5 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   b     chr2   102-112      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   103-113      + |         3 0.777777777777778</span>
<span class="co">#&gt;   g     chr3   107-117      + |         7 0.333333333333333</span>
<span class="co">#&gt;   h     chr3   108-118      + |         8 0.222222222222222</span>
<span class="co">#&gt;   i     chr3   109-119      - |         9 0.111111111111111</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
</div>
<div id="basic-interval-operations-for-granges-objects" class="section level2">
<h2><span class="header-section-number">7.7</span> Basic interval operations for <em>GRanges</em> objects</h2>
<p>Basic interval characteristics of <em>GRanges</em> objects can be extracted using the <code>start</code>, <code>end</code>, <code>width</code>, and <code>range</code> methods.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">g &lt;-<span class="st"> </span>gr[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]
g &lt;-<span class="st"> </span><span class="kw">append</span>(g, singles[[<span class="dv">10</span>]])
<span class="kw">start</span>(g)
<span class="co">#&gt; [1] 101 102 103 110</span>
<span class="kw">end</span>(g)
<span class="co">#&gt; [1] 111 112 113 120</span>
<span class="kw">width</span>(g)
<span class="co">#&gt; [1] 11 11 11 11</span>
<span class="kw">range</span>(g)
<span class="co">#&gt; GRanges object with 3 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr1   101-111      -</span>
<span class="co">#&gt;   [2]     chr2   102-113      +</span>
<span class="co">#&gt;   [3]     chr3   110-120      -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>The <em>GRanges</em> class also has many methods for manipulating the ranges. The methods can be classified as <em>intra-range methods</em>, <em>inter-range methods</em>, and <em>between-range methods</em>.</p>
<p><em>Intra-range methods</em> operate on each element of a <em>GRanges</em> object independent of the other ranges in the object. For example, the <code>flank</code> method can be used to recover regions flanking the set of ranges represented by the <em>GRanges</em> object. So to get a <em>GRanges</em> object containing the ranges that include the 10 bases upstream of the ranges:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">flank</span>(g, <span class="dv">10</span>)
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   112-121      - |         1                 1</span>
<span class="co">#&gt;   b     chr2    92-101      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2    93-102      + |         3 0.777777777777778</span>
<span class="co">#&gt;   j     chr3   121-130      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>And to include the downstream bases:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">flank</span>(g, <span class="dv">10</span>, <span class="dt">start=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1    91-100      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   113-122      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   114-123      + |         3 0.777777777777778</span>
<span class="co">#&gt;   j     chr3   100-109      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>Other examples of intra-range methods include <code>resize</code> and <code>shift</code>. The <code>shift</code> method will move the ranges by a specific number of base pairs, and the <code>resize</code> method will extend the ranges by a specified width.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">shift</span>(g, <span class="dv">5</span>)
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   106-116      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   107-117      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   108-118      + |         3 0.777777777777778</span>
<span class="co">#&gt;   j     chr3   115-125      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span>
<span class="kw">resize</span>(g, <span class="dv">30</span>)
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1    82-111      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   102-131      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   103-132      + |         3 0.777777777777778</span>
<span class="co">#&gt;   j     chr3    91-120      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>The <em><a href="http://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></em> help page <code>?&quot;intra-range-methods&quot;</code> summarizes these methods.</p>
<p><em>Inter-range methods</em> involve comparisons between ranges in a single <em>GRanges</em> object. For instance, the <code>reduce</code> method will align the ranges and merge overlapping ranges to produce a simplified set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">reduce</span>(g)
<span class="co">#&gt; GRanges object with 3 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr1   101-111      -</span>
<span class="co">#&gt;   [2]     chr2   102-113      +</span>
<span class="co">#&gt;   [3]     chr3   110-120      -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>Sometimes one is interested in the gaps or the qualities of the gaps between the ranges represented by your <em>GRanges</em> object. The <code>gaps</code> method provides this information: reduced version of your ranges:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">gaps</span>(g)
<span class="co">#&gt; GRanges object with 12 ranges and 0 metadata columns:</span>
<span class="co">#&gt;        seqnames        ranges strand</span>
<span class="co">#&gt;           &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;    [1]     chr1   1-249250621      +</span>
<span class="co">#&gt;    [2]     chr1         1-100      -</span>
<span class="co">#&gt;    [3]     chr1 112-249250621      -</span>
<span class="co">#&gt;    [4]     chr1   1-249250621      *</span>
<span class="co">#&gt;    [5]     chr2         1-101      +</span>
<span class="co">#&gt;    ...      ...           ...    ...</span>
<span class="co">#&gt;    [8]     chr2   1-243199373      *</span>
<span class="co">#&gt;    [9]     chr3   1-198022430      +</span>
<span class="co">#&gt;   [10]     chr3         1-109      -</span>
<span class="co">#&gt;   [11]     chr3 121-198022430      -</span>
<span class="co">#&gt;   [12]     chr3   1-198022430      *</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>The <code>disjoin</code> method represents a <em>GRanges</em> object as a collection of non-overlapping ranges:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">disjoin</span>(g)
<span class="co">#&gt; GRanges object with 5 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr1   101-111      -</span>
<span class="co">#&gt;   [2]     chr2       102      +</span>
<span class="co">#&gt;   [3]     chr2   103-112      +</span>
<span class="co">#&gt;   [4]     chr2       113      +</span>
<span class="co">#&gt;   [5]     chr3   110-120      -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>The <code>coverage</code> method quantifies the degree of overlap for all the ranges in a <em>GRanges</em> object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">coverage</span>(g)
<span class="co">#&gt; RleList of length 3</span>
<span class="co">#&gt; $chr1</span>
<span class="co">#&gt; integer-Rle of length 249250621 with 3 runs</span>
<span class="co">#&gt;   Lengths:       100        11 249250510</span>
<span class="co">#&gt;   Values :         0         1         0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $chr2</span>
<span class="co">#&gt; integer-Rle of length 243199373 with 5 runs</span>
<span class="co">#&gt;   Lengths:       101         1        10         1 243199260</span>
<span class="co">#&gt;   Values :         0         1         2         1         0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $chr3</span>
<span class="co">#&gt; integer-Rle of length 198022430 with 3 runs</span>
<span class="co">#&gt;   Lengths:       109        11 198022310</span>
<span class="co">#&gt;   Values :         0         1         0</span></code></pre></div>
<p>See the <em><a href="http://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></em> help page <code>?&quot;inter-range-methods&quot;</code> for additional help.</p>
<p><em>Between-range methods</em> involve operations between two <em>GRanges</em> objects; some of these are summarized in the next section.</p>
</div>
<div id="interval-set-operations-for-granges-objects" class="section level2">
<h2><span class="header-section-number">7.8</span> Interval set operations for <em>GRanges</em> objects</h2>
<p><em>Between-range methods</em> calculate relationships between different <em>GRanges</em> objects. Of central importance are <code>findOverlaps</code> and related operations; these are discussed below. Additional operations treat <em>GRanges</em> as mathematical sets of coordinates; <code>union(g, g2)</code> is the union of the coordinates in <code>g</code> and <code>g2</code>. Here are examples for calculating the <code>union</code>, the <code>intersect</code> and the asymmetric difference (using <code>setdiff</code>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">g2 &lt;-<span class="st"> </span><span class="kw">head</span>(gr, <span class="dt">n=</span><span class="dv">2</span>)
<span class="kw">union</span>(g, g2)
<span class="co">#&gt; GRanges object with 3 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr1   101-111      -</span>
<span class="co">#&gt;   [2]     chr2   102-113      +</span>
<span class="co">#&gt;   [3]     chr3   110-120      -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span>
<span class="kw">intersect</span>(g, g2)
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr1   101-111      -</span>
<span class="co">#&gt;   [2]     chr2   102-112      +</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span>
<span class="kw">setdiff</span>(g, g2)
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr2       113      +</span>
<span class="co">#&gt;   [2]     chr3   110-120      -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>Related methods are available when the structure of the <em>GRanges</em> objects are â€˜parallelâ€™ to one another, i.e., element 1 of object 1 is related to element 1 of object 2, and so on. These operations all begin with a <code>p</code>, which is short for parallel. The methods then perform element-wise, e.g., the union of element 1 of object 1 with element 1 of object 2, etc. A requirement for these operations is that the number of elements in each <em>GRanges</em> object is the same, and that both of the objects have the same seqnames and strand assignments throughout.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">g3 &lt;-<span class="st"> </span>g[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]
<span class="kw">ranges</span>(g3[<span class="dv">1</span>]) &lt;-<span class="st"> </span><span class="kw">IRanges</span>(<span class="dt">start=</span><span class="dv">105</span>, <span class="dt">end=</span><span class="dv">112</span>)
<span class="kw">punion</span>(g2, g3)
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   a     chr1   101-112      -</span>
<span class="co">#&gt;   b     chr2   102-112      +</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span>
<span class="kw">pintersect</span>(g2, g3)
<span class="co">#&gt; GRanges object with 2 ranges and 3 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC       hit</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt; &lt;logical&gt;</span>
<span class="co">#&gt;   a     chr1   105-111      - |         1                 1      TRUE</span>
<span class="co">#&gt;   b     chr2   102-112      + |         2 0.888888888888889      TRUE</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span>
<span class="kw">psetdiff</span>(g2, g3)
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   a     chr1   101-104      -</span>
<span class="co">#&gt;   b     chr2   102-101      +</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>For more information on the <code>GRanges</code> classes be sure to consult the manual page.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">?GRanges</code></pre></div>
<p>A relatively comprehensive list of available methods is discovered with</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">methods</span>(<span class="dt">class=</span><span class="st">&quot;GRanges&quot;</span>)</code></pre></div>
</div>
<div id="grangeslist-groups-of-genomic-ranges" class="section level2">
<h2><span class="header-section-number">7.9</span> <em>GRangesList</em>: Groups of Genomic Ranges</h2>
<p>Some important genomic features, such as spliced transcripts that are are comprised of exons, are inherently compound structures. Such a feature makes much more sense when expressed as a compound object such as a <em>GRangesList</em>. Whenever genomic features consist of multiple ranges that are grouped by a parent feature, they can be represented as a <em>GRangesList</em> object. Consider the simple example of the two transcript <code>GRangesList</code> below created using the <code>GRangesList</code> constructor.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gr1 &lt;-<span class="st"> </span><span class="kw">GRanges</span>(
    <span class="dt">seqnames =</span> <span class="st">&quot;chr2&quot;</span>, 
    <span class="dt">ranges =</span> <span class="kw">IRanges</span>(<span class="dv">103</span>, <span class="dv">106</span>),
    <span class="dt">strand =</span> <span class="st">&quot;+&quot;</span>, 
    <span class="dt">score =</span> 5L, <span class="dt">GC =</span> <span class="fl">0.45</span>)
gr2 &lt;-<span class="st"> </span><span class="kw">GRanges</span>(
    <span class="dt">seqnames =</span> <span class="kw">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr1&quot;</span>),
    <span class="dt">ranges =</span> <span class="kw">IRanges</span>(<span class="kw">c</span>(<span class="dv">107</span>, <span class="dv">113</span>), <span class="dt">width =</span> <span class="dv">3</span>),
    <span class="dt">strand =</span> <span class="kw">c</span>(<span class="st">&quot;+&quot;</span>, <span class="st">&quot;-&quot;</span>), 
    <span class="dt">score =</span> <span class="dv">3</span><span class="op">:</span><span class="dv">4</span>, <span class="dt">GC =</span> <span class="kw">c</span>(<span class="fl">0.3</span>, <span class="fl">0.5</span>))
grl &lt;-<span class="st"> </span><span class="kw">GRangesList</span>(<span class="st">&quot;txA&quot;</span> =<span class="st"> </span>gr1, <span class="st">&quot;txB&quot;</span> =<span class="st"> </span>gr2)
grl
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames  ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1 107-109      + |     3 0.3</span>
<span class="co">#&gt;   [2]     chr1 113-115      - |     4 0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>The <code>show</code> method for a <em>GRangesList</em> object displays it as a named list of <em>GRanges</em> objects, where the names of this list are considered to be the names of the grouping feature. In the example above, the groups of individual exon ranges are represented as separate <em>GRanges</em> objects which are further organized into a list structure where each element name is a transcript name. Many other combinations of grouped and labeled <em>GRanges</em> objects are possible of course, but this example is expected to be a common arrangement.</p>
</div>
<div id="basic-grangeslist-accessors" class="section level2">
<h2><span class="header-section-number">7.10</span> Basic <em>GRangesList</em> accessors</h2>
<p>Just as with <em>GRanges</em> object, the components of the genomic coordinates within a <em>GRangesList</em> object can be extracted using simple accessor methods. Not surprisingly, the <em>GRangesList</em> objects have many of the same accessors as <em>GRanges</em> objects. The difference is that many of these methods return a list since the input is now essentially a list of <em>GRanges</em> objects. Here are a few examples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">seqnames</span>(grl)
<span class="co">#&gt; RleList of length 2</span>
<span class="co">#&gt; $txA</span>
<span class="co">#&gt; factor-Rle of length 1 with 1 run</span>
<span class="co">#&gt;   Lengths:    1</span>
<span class="co">#&gt;   Values : chr2</span>
<span class="co">#&gt; Levels(2): chr2 chr1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB</span>
<span class="co">#&gt; factor-Rle of length 2 with 1 run</span>
<span class="co">#&gt;   Lengths:    2</span>
<span class="co">#&gt;   Values : chr1</span>
<span class="co">#&gt; Levels(2): chr2 chr1</span>
<span class="kw">ranges</span>(grl)
<span class="co">#&gt; IRangesList of length 2</span>
<span class="co">#&gt; $txA</span>
<span class="co">#&gt; IRanges object with 1 range and 0 metadata columns:</span>
<span class="co">#&gt;           start       end     width</span>
<span class="co">#&gt;       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   [1]       103       106         4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB</span>
<span class="co">#&gt; IRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;           start       end     width</span>
<span class="co">#&gt;       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   [1]       107       109         3</span>
<span class="co">#&gt;   [2]       113       115         3</span>
<span class="kw">strand</span>(grl)
<span class="co">#&gt; RleList of length 2</span>
<span class="co">#&gt; $txA</span>
<span class="co">#&gt; factor-Rle of length 1 with 1 run</span>
<span class="co">#&gt;   Lengths: 1</span>
<span class="co">#&gt;   Values : +</span>
<span class="co">#&gt; Levels(3): + - *</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB</span>
<span class="co">#&gt; factor-Rle of length 2 with 2 runs</span>
<span class="co">#&gt;   Lengths: 1 1</span>
<span class="co">#&gt;   Values : + -</span>
<span class="co">#&gt; Levels(3): + - *</span></code></pre></div>
<p>The <code>length</code> and <code>names</code> methods will return the length or names of the list and the <code>seqlengths</code> method will return the set of sequence lengths.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">length</span>(grl)
<span class="co">#&gt; [1] 2</span>
<span class="kw">names</span>(grl)
<span class="co">#&gt; [1] &quot;txA&quot; &quot;txB&quot;</span>
<span class="kw">seqlengths</span>(grl)
<span class="co">#&gt; chr2 chr1 </span>
<span class="co">#&gt;   NA   NA</span></code></pre></div>
<p>The <code>elementNROWS</code> method returns a list of integers corresponding to the result of calling <code>NROW</code> on each individual <em>GRanges</em> object contained by the <em>GRangesList</em>. This is a faster alternative to calling <code>lapply</code> on the <em>GRangesList</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">elementNROWS</span>(grl)
<span class="co">#&gt; txA txB </span>
<span class="co">#&gt;   1   2</span></code></pre></div>
<p><code>isEmpty</code> tests if a <em>GRangesList</em> object contains anything.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">isEmpty</span>(grl)
<span class="co">#&gt; [1] FALSE</span></code></pre></div>
<p>In the context of a <em>GRangesList</em> object, the <code>mcols</code> method performs a similar operation to what it does on a <em>GRanges</em> object. However, this metadata now refers to information at the list level instead of the level of the individual <em>GRanges</em> objects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">mcols</span>(grl) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Transcript A&quot;</span>,<span class="st">&quot;Transcript B&quot;</span>)
<span class="kw">mcols</span>(grl)
<span class="co">#&gt; DataFrame with 2 rows and 1 column</span>
<span class="co">#&gt;            value</span>
<span class="co">#&gt;      &lt;character&gt;</span>
<span class="co">#&gt; txA Transcript A</span>
<span class="co">#&gt; txB Transcript B</span></code></pre></div>
<p>Element-level metadata can be retrieved by unlisting the <code>GRangesList</code>, and extracting the metadata</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">mcols</span>(<span class="kw">unlist</span>(grl))
<span class="co">#&gt; DataFrame with 3 rows and 2 columns</span>
<span class="co">#&gt;         score        GC</span>
<span class="co">#&gt;     &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; txA         5      0.45</span>
<span class="co">#&gt; txB         3       0.3</span>
<span class="co">#&gt; txB         4       0.5</span></code></pre></div>
</div>
<div id="combining-grangeslist-objects" class="section level2">
<h2><span class="header-section-number">7.11</span> Combining <em>GRangesList</em> objects</h2>
<p><em>GRangesList</em> objects can be unlisted to combine the separate <em>GRanges</em> objects that they contain as an expanded <em>GRanges</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ul &lt;-<span class="st"> </span><span class="kw">unlist</span>(grl)
ul
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   txA     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   txB     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   txB     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Append lists using <code>append</code> or <code>c</code>.</p>
<p>A  had two <em>GRangesList</em> objects with â€˜parallelâ€™ elements, and wanted to combined these element-wise into a single <em>GRangesList</em>. One solution is to use <code>pc()</code> â€“ parallel (element-wise) <code>c()</code>. A more general solution is to concatenate the lists and then re-group by some factor, in this case the names of the elements.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">grl1 &lt;-<span class="st"> </span><span class="kw">GRangesList</span>(
    <span class="dt">gr1 =</span> <span class="kw">GRanges</span>(<span class="st">&quot;chr2&quot;</span>, <span class="kw">IRanges</span>(<span class="dv">3</span>, <span class="dv">6</span>)),
    <span class="dt">gr2 =</span> <span class="kw">GRanges</span>(<span class="st">&quot;chr1&quot;</span>, <span class="kw">IRanges</span>(<span class="kw">c</span>(<span class="dv">7</span>,<span class="dv">13</span>), <span class="dt">width =</span> <span class="dv">3</span>)))
grl2 &lt;-<span class="st"> </span><span class="kw">GRangesList</span>(
    <span class="dt">gr1 =</span> <span class="kw">GRanges</span>(<span class="st">&quot;chr2&quot;</span>, <span class="kw">IRanges</span>(<span class="dv">9</span>, <span class="dv">12</span>)),
    <span class="dt">gr2 =</span> <span class="kw">GRanges</span>(<span class="st">&quot;chr1&quot;</span>, <span class="kw">IRanges</span>(<span class="kw">c</span>(<span class="dv">25</span>,<span class="dv">38</span>), <span class="dt">width =</span> <span class="dv">3</span>)))

<span class="kw">pc</span>(grl1, grl2)
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $gr1 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr2       3-6      *</span>
<span class="co">#&gt;   [2]     chr2      9-12      *</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr2 </span>
<span class="co">#&gt; GRanges object with 4 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand</span>
<span class="co">#&gt;   [1]     chr1    7-9      *</span>
<span class="co">#&gt;   [2]     chr1  13-15      *</span>
<span class="co">#&gt;   [3]     chr1  25-27      *</span>
<span class="co">#&gt;   [4]     chr1  38-40      *</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>

grl3 &lt;-<span class="st"> </span><span class="kw">c</span>(grl1, grl2)
<span class="kw">regroup</span>(grl3, <span class="kw">names</span>(grl3))
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $gr1 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr2       3-6      *</span>
<span class="co">#&gt;   [2]     chr2      9-12      *</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr2 </span>
<span class="co">#&gt; GRanges object with 4 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand</span>
<span class="co">#&gt;   [1]     chr1    7-9      *</span>
<span class="co">#&gt;   [2]     chr1  13-15      *</span>
<span class="co">#&gt;   [3]     chr1  25-27      *</span>
<span class="co">#&gt;   [4]     chr1  38-40      *</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
</div>
<div id="basic-interval-operations-for-grangeslist-objects" class="section level2">
<h2><span class="header-section-number">7.12</span> Basic interval operations for <em>GRangesList</em> objects</h2>
<p>For interval operations, many of the same methods exist for <em>GRangesList</em> objects that exist for <em>GRanges</em> objects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">start</span>(grl)
<span class="co">#&gt; IntegerList of length 2</span>
<span class="co">#&gt; [[&quot;txA&quot;]] 103</span>
<span class="co">#&gt; [[&quot;txB&quot;]] 107 113</span>
<span class="kw">end</span>(grl)
<span class="co">#&gt; IntegerList of length 2</span>
<span class="co">#&gt; [[&quot;txA&quot;]] 106</span>
<span class="co">#&gt; [[&quot;txB&quot;]] 109 115</span>
<span class="kw">width</span>(grl)
<span class="co">#&gt; IntegerList of length 2</span>
<span class="co">#&gt; [[&quot;txA&quot;]] 4</span>
<span class="co">#&gt; [[&quot;txB&quot;]] 3 3</span></code></pre></div>
<p>These operations return a data structure representing, e.g., <em>IntegerList</em>, a list where all elements are integers; it can be convenient to use mathematical and other operations on **List* objects that work on each element, e.g.,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sum</span>(<span class="kw">width</span>(grl))  <span class="co"># sum of widths of each grl element</span>
<span class="co">#&gt; txA txB </span>
<span class="co">#&gt;   4   6</span></code></pre></div>
<p>Most of the intra-, inter- and between-range methods operate on <em>GRangesList</em> objects, e.g., to shift all the <em>GRanges</em> objects in a <em>GRangesList</em> object, or calculate the coverage. Both of these operations are also carried out across each <em>GRanges</em> list member.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">shift</span>(grl, <span class="dv">20</span>)
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   123-126      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames  ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1 127-129      + |     3 0.3</span>
<span class="co">#&gt;   [2]     chr1 133-135      - |     4 0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">coverage</span>(grl)
<span class="co">#&gt; RleList of length 2</span>
<span class="co">#&gt; $chr2</span>
<span class="co">#&gt; integer-Rle of length 106 with 2 runs</span>
<span class="co">#&gt;   Lengths: 102   4</span>
<span class="co">#&gt;   Values :   0   1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $chr1</span>
<span class="co">#&gt; integer-Rle of length 115 with 4 runs</span>
<span class="co">#&gt;   Lengths: 106   3   3   3</span>
<span class="co">#&gt;   Values :   0   1   0   1</span></code></pre></div>
</div>
<div id="subsetting-grangeslist-objects" class="section level2">
<h2><span class="header-section-number">7.13</span> Subsetting <em>GRangesList</em> objects</h2>
<p>A <em>GRangesList</em> object is behaves like a <code>list</code>: <code>[</code> returns a <em>GRangesList</em> containing a subset of the original object; <code>[[</code> or <code>\$</code> returns the <em>GRanges</em> object at that location in the list.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">grl[<span class="dv">1</span>]
grl[[<span class="dv">1</span>]]
grl[<span class="st">&quot;txA&quot;</span>]
grl<span class="op">$</span>txB</code></pre></div>
<p>In addition, subsetting a <em>GRangesList</em> also accepts a second parameter to specify which of the metadata columns you wish to select.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">grl[<span class="dv">1</span>, <span class="st">&quot;score&quot;</span>]
<span class="co">#&gt; GRangesList object of length 1:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 1 metadata column:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
grl[<span class="st">&quot;txB&quot;</span>, <span class="st">&quot;GC&quot;</span>]
<span class="co">#&gt; GRangesList object of length 1:</span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 1 metadata column:</span>
<span class="co">#&gt;       seqnames    ranges strand |        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1   107-109      + |       0.3</span>
<span class="co">#&gt;   [2]     chr1   113-115      - |       0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>The <code>head</code>, <code>tail</code>, <code>rep</code>, <code>rev</code>, and <code>window</code> methods all behave as you would expect them to for a list object. For example, the elements referred to by <code>window</code> are now list elements instead of <em>GRanges</em> elements.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rep</span>(grl[[<span class="dv">1</span>]], <span class="dt">times =</span> <span class="dv">3</span>)
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   [2]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   [3]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">rev</span>(grl)
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   [2]     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames  ranges strand | score   GC</span>
<span class="co">#&gt;   [1]     chr2 103-106      + |     5 0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">head</span>(grl, <span class="dt">n=</span><span class="dv">1</span>)
<span class="co">#&gt; GRangesList object of length 1:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">tail</span>(grl, <span class="dt">n=</span><span class="dv">1</span>)
<span class="co">#&gt; GRangesList object of length 1:</span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   [2]     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">window</span>(grl, <span class="dt">start=</span><span class="dv">1</span>, <span class="dt">end=</span><span class="dv">1</span>)
<span class="co">#&gt; GRangesList object of length 1:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
grl[<span class="kw">IRanges</span>(<span class="dt">start=</span><span class="dv">2</span>, <span class="dt">end=</span><span class="dv">2</span>)]
<span class="co">#&gt; GRangesList object of length 1:</span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   [2]     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
</div>
<div id="looping-over-grangeslist-objects" class="section level2">
<h2><span class="header-section-number">7.14</span> Looping over <em>GRangesList</em> objects</h2>
<p>For <em>GRangesList</em> objects there is also a family of <code>apply</code> methods. These include <code>lapply</code>, <code>sapply</code>, <code>mapply</code>, <code>endoapply</code>, <code>mendoapply</code>, <code>Map</code>, and <code>Reduce</code>.</p>
<p>The different looping methods defined for <em>GRangesList</em> objects are useful for returning different kinds of results. The standard <code>lapply</code> and <code>sapply</code> behave according to convention, with the <code>lapply</code> method returning a list and <code>sapply</code> returning a more simplified output.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">lapply</span>(grl, length)
<span class="co">#&gt; $txA</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB</span>
<span class="co">#&gt; [1] 2</span>
<span class="kw">sapply</span>(grl, length)
<span class="co">#&gt; txA txB </span>
<span class="co">#&gt;   1   2</span></code></pre></div>
<p>As with <em>IRanges</em> objects, there is also a multivariate version of <code>sapply</code>, called <code>mapply</code>, defined for <em>GRangesList</em> objects. And, if you donâ€™t want the results simplified, you can call the <code>Map</code> method, which does the same things as <code>mapply</code> but without simplifying the output.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">grl2 &lt;-<span class="st"> </span><span class="kw">shift</span>(grl, <span class="dv">10</span>)
<span class="kw">names</span>(grl2) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;shiftTxA&quot;</span>, <span class="st">&quot;shiftTxB&quot;</span>)

<span class="kw">mapply</span>(c, grl, grl2)
<span class="co">#&gt; $txA</span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   [2]     chr2   113-116      + |         5      0.45</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB</span>
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   [2]     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt;   [3]     chr1   117-119      + |         3       0.3</span>
<span class="co">#&gt;   [4]     chr1   123-125      - |         4       0.5</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">Map</span>(c, grl, grl2)
<span class="co">#&gt; $txA</span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   [2]     chr2   113-116      + |         5      0.45</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB</span>
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   [2]     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt;   [3]     chr1   117-119      + |         3       0.3</span>
<span class="co">#&gt;   [4]     chr1   123-125      - |         4       0.5</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Sometimes you will want to get back a modified version of the <em>GRangesList</em> that you originally passed in.</p>
<p>An endomorphism is a transformation of an object to another instance of the same class . This is achieved using the <code>endoapply</code> method, which will return the results as a <em>GRangesList</em> object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">endoapply</span>(grl, rev)
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames  ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1 113-115      - |     4 0.5</span>
<span class="co">#&gt;   [2]     chr1 107-109      + |     3 0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">mendoapply</span>(c, grl, grl2)
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   [2]     chr2   113-116      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames  ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1 107-109      + |     3 0.3</span>
<span class="co">#&gt;   [2]     chr1 113-115      - |     4 0.5</span>
<span class="co">#&gt;   [3]     chr1 117-119      + |     3 0.3</span>
<span class="co">#&gt;   [4]     chr1 123-125      - |     4 0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>The <code>Reduce</code> method will allow the <em>GRanges</em> objects to be collapsed across the whole of the <em>GRangesList</em> object. % Again, this seems like a sub-optimal example to me.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">Reduce</span>(c, grl)
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2   103-106      + |         5      0.45</span>
<span class="co">#&gt;   [2]     chr1   107-109      + |         3       0.3</span>
<span class="co">#&gt;   [3]     chr1   113-115      - |         4       0.5</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Explicit element-wise operations (<code>lapply()</code> and friends) on <em>GRangesList</em> objects with many elements can be slow. It is therefore beneficial to explore operations that work on <code>*List</code> objects directly (e.g., many of the <code>group generic' operators, see</code>?S4groupGeneric<code>, and the set and parallel set operators (e.g.,</code>union<code>,</code>punion<code>). A useful and fast strategy is to</code>unlist<code>the *GRangesList* to a *GRanges* object, operate on the *GRanges* object, then</code>relist` the result, e.g.,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gr &lt;-<span class="st"> </span><span class="kw">unlist</span>(grl)
gr<span class="op">$</span>log_score &lt;-<span class="st"> </span><span class="kw">log</span>(gr<span class="op">$</span>score)
grl &lt;-<span class="st"> </span><span class="kw">relist</span>(gr, grl)
grl
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $txA </span>
<span class="co">#&gt; GRanges object with 1 range and 3 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC       log_score</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt;   txA     chr2   103-106      + |         5      0.45 1.6094379124341</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $txB </span>
<span class="co">#&gt; GRanges object with 2 ranges and 3 metadata columns:</span>
<span class="co">#&gt;       seqnames  ranges strand | score  GC        log_score</span>
<span class="co">#&gt;   txB     chr1 107-109      + |     3 0.3 1.09861228866811</span>
<span class="co">#&gt;   txB     chr1 113-115      - |     4 0.5 1.38629436111989</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>See also <code>?extractList</code>.</p>
<p>For more information on the <code>GRangesList</code> classes be sure to consult the manual page and available methods</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">?GRangesList
<span class="kw">methods</span>(<span class="dt">class=</span><span class="st">&quot;GRangesList&quot;</span>)   <span class="co"># _partial_ list</span></code></pre></div>

<p>Interval overlapping is the process of comparing the ranges in two objects to determine if and when they overlap. As such, it is perhaps the most common operation performed on <em>GRanges</em> and <em>GRangesList</em> objects. To this end, the <em><a href="http://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></em> package provides a family of interval overlap functions. The most general of these functions is <code>findOverlaps</code>, which takes a query and a subject as inputs and returns a <em>Hits</em> object containing the index pairings for the overlapping elements.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mtch &lt;-<span class="st"> </span><span class="kw">findOverlaps</span>(gr, grl)
<span class="kw">as.matrix</span>(mtch)
<span class="co">#&gt;      queryHits subjectHits</span>
<span class="co">#&gt; [1,]         1           1</span>
<span class="co">#&gt; [2,]         2           2</span>
<span class="co">#&gt; [3,]         3           2</span></code></pre></div>
<p>As suggested in the sections discussing the nature of the <em>GRanges</em> and <em>GRangesList</em> classes, the index in the above matrix of hits for a <em>GRanges</em> object is a single range while for a <em>GRangesList</em> object it is the set of ranges that define a â€œfeatureâ€.</p>
<p>Another function in the overlaps family is <code>countOverlaps</code>, which tabulates the number of overlaps for each element in the query.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">countOverlaps</span>(gr, grl)
<span class="co">#&gt; txA txB txB </span>
<span class="co">#&gt;   1   1   1</span></code></pre></div>
<p>A third function in this family is <code>subsetByOverlaps</code>, which extracts the elements in the query that overlap at least one element in the subject.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">subsetByOverlaps</span>(gr,grl)
<span class="co">#&gt; GRanges object with 3 ranges and 3 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC        log_score</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;        &lt;numeric&gt;</span>
<span class="co">#&gt;   txA     chr2   103-106      + |         5      0.45  1.6094379124341</span>
<span class="co">#&gt;   txB     chr1   107-109      + |         3       0.3 1.09861228866811</span>
<span class="co">#&gt;   txB     chr1   113-115      - |         4       0.5 1.38629436111989</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Finally, you can use the <code>select</code> argument to get the index of the first overlapping element in the subject for each element in the query.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">findOverlaps</span>(gr, grl, <span class="dt">select=</span><span class="st">&quot;first&quot;</span>)
<span class="co">#&gt; [1] 1 2 2</span>
<span class="kw">findOverlaps</span>(grl, gr, <span class="dt">select=</span><span class="st">&quot;first&quot;</span>)
<span class="co">#&gt; [1] 1 2</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="fluent-genomic-data-analysis-with-plyranges.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
