<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html">
<link rel="next" href="functional-enrichment-analysis-of-high-throughput-omics-data.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#for-everyone"><i class="fa fa-check"></i><b>1.1</b> For Everyone</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#for-workshop-authors"><i class="fa fa-check"></i><b>1.2</b> For Workshop Authors</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#description"><i class="fa fa-check"></i><b>1.2.1</b> DESCRIPTION</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#classifying-your-workshop"><i class="fa fa-check"></i><b>1.2.2</b> Classifying your workshop</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#bookdown"><i class="fa fa-check"></i><b>1.2.3</b> Bookdown</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#deadlines-for-bioc2018"><i class="fa fa-check"></i><b>1.3</b> Deadlines for Bioc2018</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html"><i class="fa fa-check"></i><b>2</b> <em>R</em> and <em>Bioconductor</em> for everyone: an introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#description-1"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="2.1.7" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-goals"><i class="fa fa-check"></i><b>2.1.7</b> Learning goals</a></li>
<li class="chapter" data-level="2.1.8" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-objectives"><i class="fa fa-check"></i><b>2.1.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-r"><i class="fa fa-check"></i><b>2.2</b> Introduction to <em>R</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#rstudio-orientation"><i class="fa fa-check"></i><b>2.2.1</b> <em>RStudio</em> orientation</a></li>
<li class="chapter" data-level="2.2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#very-basics"><i class="fa fa-check"></i><b>2.2.2</b> Very basics</a></li>
<li class="chapter" data-level="2.2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#data-input-and-manipulation"><i class="fa fa-check"></i><b>2.2.3</b> Data input and manipulation</a></li>
<li class="chapter" data-level="2.2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#writing-r-scripts"><i class="fa fa-check"></i><b>2.2.4</b> Writing <em>R</em> scripts</a></li>
<li class="chapter" data-level="2.2.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#using-packages"><i class="fa fa-check"></i><b>2.2.5</b> Using packages</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-bioconductor"><i class="fa fa-check"></i><b>2.3</b> Introduction to <em>Bioconductor</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#discovering-installing-and-learning-how-to-use-bioconductor-packages"><i class="fa fa-check"></i><b>2.3.1</b> Discovering, installing, and learning how to use <em>Bioconductor</em> packages</a></li>
<li class="chapter" data-level="2.3.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-genomic-ranges"><i class="fa fa-check"></i><b>2.3.2</b> Working with Genomic Ranges</a></li>
<li class="chapter" data-level="2.3.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-summarized-experimental-data"><i class="fa fa-check"></i><b>2.3.3</b> Working with summarized experimental data</a></li>
<li class="chapter" data-level="2.3.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#down-stream-analysis"><i class="fa fa-check"></i><b>2.3.4</b> Down-stream analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>3</b> RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview-1"><i class="fa fa-check"></i><b>3.1</b> Overview</a><ul>
<li class="chapter" data-level="3.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-2"><i class="fa fa-check"></i><b>3.1.1</b> Description</a></li>
<li class="chapter" data-level="3.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites-1"><i class="fa fa-check"></i><b>3.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="3.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation-1"><i class="fa fa-check"></i><b>3.1.3</b> Participation</a></li>
<li class="chapter" data-level="3.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="3.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline-1"><i class="fa fa-check"></i><b>3.1.5</b> Time outline</a></li>
<li class="chapter" data-level="3.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>3.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="3.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>3.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="3.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>3.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="3.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>3.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="3.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>3.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>3.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="3.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>3.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="3.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>3.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="3.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>3.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>3.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="3.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>3.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="3.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>3.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="3.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>3.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis"><i class="fa fa-check"></i><b>3.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="3.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>3.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="3.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>3.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub"><i class="fa fa-check"></i><b>3.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="3.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub"><i class="fa fa-check"></i><b>3.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="3.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>3.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>3.7</b> Building reports</a><ul>
<li class="chapter" data-level="3.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>3.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="3.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>3.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>3.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="3.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>3.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="3.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>3.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="3.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>3.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="3.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>3.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="3.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>3.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="3.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>3.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>4</b> Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-2"><i class="fa fa-check"></i><b>4.1</b> Overview</a><ul>
<li class="chapter" data-level="4.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-3"><i class="fa fa-check"></i><b>4.1.1</b> Description</a></li>
<li class="chapter" data-level="4.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-2"><i class="fa fa-check"></i><b>4.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-2"><i class="fa fa-check"></i><b>4.1.3</b> Participation</a></li>
<li class="chapter" data-level="4.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-2"><i class="fa fa-check"></i><b>4.1.5</b> Time outline</a></li>
<li class="chapter" data-level="4.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started"><i class="fa fa-check"></i><b>4.2</b> Getting started</a><ul>
<li class="chapter" data-level="4.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>4.2.1</b> The data</a></li>
<li class="chapter" data-level="4.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>4.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>4.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="4.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>4.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="4.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>4.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>4.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="4.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>4.5.1</b> Normalization</a></li>
<li class="chapter" data-level="4.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>4.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>4.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="4.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>4.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="4.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>4.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>5</b> Cytoscape Automation in R using Rcy3</a><ul>
<li class="chapter" data-level="5.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-3"><i class="fa fa-check"></i><b>5.1</b> Overview</a><ul>
<li class="chapter" data-level="5.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructors-names-and-contact-information"><i class="fa fa-check"></i><b>5.1.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="5.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description"><i class="fa fa-check"></i><b>5.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="5.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>5.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="5.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>5.1.4</b> Background:</a></li>
<li class="chapter" data-level="5.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation"><i class="fa fa-check"></i><b>5.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>5.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="5.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-3"><i class="fa fa-check"></i><b>5.1.7</b> Time outline</a></li>
<li class="chapter" data-level="5.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>5.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>5.2</b> Background</a><ul>
<li class="chapter" data-level="5.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>5.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="5.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>5.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="5.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>5.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>5.3</b> Translating biological data into Cytoscape using RCy3</a></li>
<li class="chapter" data-level="5.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>5.4</b> Set up</a><ul>
<li class="chapter" data-level="5.4.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started-1"><i class="fa fa-check"></i><b>5.4.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>5.5</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="5.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>5.6</b> Example Data Set</a></li>
<li class="chapter" data-level="5.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>5.7</b> Finding Network Data</a></li>
<li class="chapter" data-level="5.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>5.8</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="5.9" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>5.9</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="5.10" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>5.10</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
<li class="chapter" data-level="5.11" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#references"><i class="fa fa-check"></i><b>5.11</b> References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><i class="fa fa-check"></i><b>6</b> Effectively using the DelayedArray framework to support the analysis of large datasets</a><ul>
<li class="chapter" data-level="6.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-4"><i class="fa fa-check"></i><b>6.1</b> Overview</a><ul>
<li class="chapter" data-level="6.1.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#description-4"><i class="fa fa-check"></i><b>6.1.1</b> Description</a></li>
<li class="chapter" data-level="6.1.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.1.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#participation-3"><i class="fa fa-check"></i><b>6.1.3</b> Participation</a></li>
<li class="chapter" data-level="6.1.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>6.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.1.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#time-outline-4"><i class="fa fa-check"></i><b>6.1.5</b> Time outline</a></li>
<li class="chapter" data-level="6.1.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>6.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#introductory-material"><i class="fa fa-check"></i><b>6.2</b> Introductory material</a><ul>
<li class="chapter" data-level="6.2.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#discussion"><i class="fa fa-check"></i><b>6.2.1</b> Discussion</a></li>
<li class="chapter" data-level="6.2.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#why-learn-delayedarray"><i class="fa fa-check"></i><b>6.2.2</b> Why learn DelayedArray?</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-of-delayedarray-framework"><i class="fa fa-check"></i><b>6.3</b> Overview of DelayedArray framework</a><ul>
<li class="chapter" data-level="6.3.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-package"><i class="fa fa-check"></i><b>6.3.1</b> The <strong>DelayedArray</strong> package</a></li>
<li class="chapter" data-level="6.3.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#packages-that-extend-delayedarray"><i class="fa fa-check"></i><b>6.3.2</b> Packages that extend <strong>DelayedArray</strong></a></li>
<li class="chapter" data-level="6.3.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#packages-that-use-delayedarray"><i class="fa fa-check"></i><b>6.3.3</b> Packages that use <strong>DelayedArray</strong></a></li>
<li class="chapter" data-level="6.3.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-class"><i class="fa fa-check"></i><b>6.3.4</b> The <em>DelayedArray</em> class</a></li>
<li class="chapter" data-level="6.3.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fundamental-concepts"><i class="fa fa-check"></i><b>6.3.5</b> Fundamental concepts</a></li>
<li class="chapter" data-level="6.3.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#whats-out-there-already"><i class="fa fa-check"></i><b>6.3.6</b> Whatâ€™s out there already?</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#incoporating-delayedarray-into-a-package"><i class="fa fa-check"></i><b>6.4</b> Incoporating DelayedArray into a package</a><ul>
<li class="chapter" data-level="6.4.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#writing-algorithms-to-process-delayedarray-instances"><i class="fa fa-check"></i><b>6.4.1</b> Writing algorithms to process <em>DelayedArray</em> instances</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#questions-and-discussion"><i class="fa fa-check"></i><b>6.5</b> Questions and discussion</a></li>
<li class="chapter" data-level="6.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#todos"><i class="fa fa-check"></i><b>6.6</b> TODOs</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><i class="fa fa-check"></i><b>7</b> Variant Functional Annotation using StatePaintR, FunciVar &amp; MotifBreakR</a><ul>
<li class="chapter" data-level="7.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i><b>7.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="7.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-description-1"><i class="fa fa-check"></i><b>7.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="7.2.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.2.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-participation-1"><i class="fa fa-check"></i><b>7.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="7.2.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>7.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.2.4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#time-outline-5"><i class="fa fa-check"></i><b>7.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>7.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="7.3.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#key-concepts"><i class="fa fa-check"></i><b>7.3.1</b> Key Concepts</a></li>
<li class="chapter" data-level="7.3.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#learning-objectives-4"><i class="fa fa-check"></i><b>7.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>8</b> Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="8.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructors-names-and-contact-information-2"><i class="fa fa-check"></i><b>8.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="8.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-2"><i class="fa fa-check"></i><b>8.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="8.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-2"><i class="fa fa-check"></i><b>8.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="8.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>8.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-6"><i class="fa fa-check"></i><b>8.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>8.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="8.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop"><i class="fa fa-check"></i><b>8.4</b> Workshop</a></li>
<li class="chapter" data-level="8.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>8.5</b> Where does it all come from?</a></li>
<li class="chapter" data-level="8.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>8.6</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="8.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>8.7</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="8.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>8.8</b> Data types</a></li>
<li class="chapter" data-level="8.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>8.9</b> Differential expression analysis</a></li>
<li class="chapter" data-level="8.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>8.10</b> Gene sets</a></li>
<li class="chapter" data-level="8.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>8.11</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="8.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>8.12</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="8.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>8.13</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="8.14" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>8.14</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>9</b> RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="9.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>9.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-3"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-6"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-3"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-7"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>9.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="9.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals-4"><i class="fa fa-check"></i><b>9.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="9.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-5"><i class="fa fa-check"></i><b>9.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-1"><i class="fa fa-check"></i><b>9.4</b> Introduction</a></li>
<li class="chapter" data-level="9.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>9.5</b> Data packaging</a><ul>
<li class="chapter" data-level="9.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>9.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="9.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>9.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="9.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>9.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>9.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="9.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>9.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="9.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>9.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="9.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>9.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="9.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>9.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>9.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="9.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>9.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="9.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>9.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="9.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>9.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="9.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>9.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="9.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>9.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="9.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>9.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>10</b> Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="10.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructors-names-and-contact-information-3"><i class="fa fa-check"></i><b>10.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-4"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-7"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-4"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>10.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-8"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-5"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-6"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>10.4</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="10.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>10.5</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>10.6</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>10.7</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>10.8</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#grangeslist-groups-of-genomic-ranges"><i class="fa fa-check"></i><b>10.9</b> <em>GRangesList</em>: Groups of Genomic Ranges</a></li>
<li class="chapter" data-level="10.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-grangeslist-accessors"><i class="fa fa-check"></i><b>10.10</b> Basic <em>GRangesList</em> accessors</a></li>
<li class="chapter" data-level="10.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#combining-grangeslist-objects"><i class="fa fa-check"></i><b>10.11</b> Combining <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-grangeslist-objects"><i class="fa fa-check"></i><b>10.12</b> Basic interval operations for <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-grangeslist-objects"><i class="fa fa-check"></i><b>10.13</b> Subsetting <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#looping-over-grangeslist-objects"><i class="fa fa-check"></i><b>10.14</b> Looping over <em>GRangesList</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>11</b> Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="11.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructors-names-and-contact-information-4"><i class="fa fa-check"></i><b>11.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="11.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-5"><i class="fa fa-check"></i><b>11.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="11.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-8"><i class="fa fa-check"></i><b>11.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="11.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-5"><i class="fa fa-check"></i><b>11.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-9"><i class="fa fa-check"></i><b>11.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="11.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-9"><i class="fa fa-check"></i><b>11.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="11.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-6"><i class="fa fa-check"></i><b>11.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="11.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-7"><i class="fa fa-check"></i><b>11.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-1"><i class="fa fa-check"></i><b>11.4</b> Workshop</a></li>
<li class="chapter" data-level="11.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-2"><i class="fa fa-check"></i><b>11.5</b> Introduction</a><ul>
<li class="chapter" data-level="11.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>11.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="11.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>11.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="11.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>11.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup"><i class="fa fa-check"></i><b>11.6</b> Setup</a></li>
<li class="chapter" data-level="11.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>11.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="11.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>11.8</b> The Grammar</a><ul>
<li class="chapter" data-level="11.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>11.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="11.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>11.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="11.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises"><i class="fa fa-check"></i><b>11.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>11.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="11.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>11.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="11.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>11.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="11.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>11.10</b> Whatâ€™s next?</a></li>
<li class="chapter" data-level="11.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>11.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>12</b> Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="12.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>12.1</b> Instructors</a></li>
<li class="chapter" data-level="12.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description-6"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>12.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation-6"><i class="fa fa-check"></i><b>12.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-10"><i class="fa fa-check"></i><b>12.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-7"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-8"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>12.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="12.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>12.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="12.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>12.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="12.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>12.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="12.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>12.4.4</b> Data containers</a></li>
<li class="chapter" data-level="12.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>12.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="12.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>12.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="12.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>12.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="12.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>12.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="12.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>12.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="12.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>12.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="12.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>12.4.11</b> Simple example</a></li>
<li class="chapter" data-level="12.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>12.4.12</b> Questions!</a></li>
<li class="chapter" data-level="12.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>12.4.13</b> More questions!</a></li>
<li class="chapter" data-level="12.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>12.4.14</b> Another example</a></li>
<li class="chapter" data-level="12.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>12.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="12.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>12.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="12.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>12.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="12.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>12.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="12.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>12.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="12.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>12.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="12.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>12.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="12.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>12.4.22</b> GRanges</a></li>
<li class="chapter" data-level="12.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>12.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="12.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>12.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="12.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>12.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="12.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>12.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="12.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>12.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="12.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>12.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="12.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>12.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="12.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>12.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="12.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>12.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="12.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>12.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="12.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr"><i class="fa fa-check"></i><b>12.4.33</b> Organism.dplyr</a></li>
<li class="chapter" data-level="12.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-1"><i class="fa fa-check"></i><b>12.4.34</b> Organism.dplyr</a></li>
<li class="chapter" data-level="12.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-2"><i class="fa fa-check"></i><b>12.4.35</b> Organism.dplyr</a></li>
<li class="chapter" data-level="12.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>12.4.36</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="12.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>12.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="12.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>12.4.38</b> BSgenome packages</a></li>
<li class="chapter" data-level="12.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>12.4.39</b> BSgenome packages</a></li>
<li class="chapter" data-level="12.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>12.4.40</b> BSgenome exercises</a></li>
<li class="chapter" data-level="12.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-1"><i class="fa fa-check"></i><b>12.4.41</b> AnnotationHub</a></li>
<li class="chapter" data-level="12.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>12.4.42</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="12.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>12.4.43</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="12.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>12.4.44</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="12.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>12.4.45</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="12.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>12.4.46</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="12.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>12.4.47</b> AnnotationHub query</a></li>
<li class="chapter" data-level="12.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>12.4.48</b> AnnotationHub query</a></li>
<li class="chapter" data-level="12.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>12.4.49</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="12.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>12.4.50</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="12.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>12.4.51</b> biomaRt</a></li>
<li class="chapter" data-level="12.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>12.4.52</b> biomaRt data sets</a></li>
<li class="chapter" data-level="12.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>12.4.53</b> biomaRt queries</a></li>
<li class="chapter" data-level="12.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>12.4.54</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="12.4.55" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>12.4.55</b> biomaRt query</a></li>
<li class="chapter" data-level="12.4.56" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>12.4.56</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>13</b> Workflow for Multi-omics Analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>13.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-7"><i class="fa fa-check"></i><b>13.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="13.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-9"><i class="fa fa-check"></i><b>13.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-7"><i class="fa fa-check"></i><b>13.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>13.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="13.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-10"><i class="fa fa-check"></i><b>13.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="13.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-8"><i class="fa fa-check"></i><b>13.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="13.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-9"><i class="fa fa-check"></i><b>13.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>13.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="13.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>13.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="13.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>13.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="13.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>13.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>13.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="13.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>13.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="13.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>13.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="13.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>13.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>13.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="13.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>13.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="13.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>13.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="13.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>13.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="13.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>13.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="13.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>13.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="13.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>13.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>13.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="13.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>13.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="13.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>13.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="13.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>13.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>13.9</b> Complete cases</a></li>
<li class="chapter" data-level="13.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>13.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="13.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>13.11</b> Extraction</a><ul>
<li class="chapter" data-level="13.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>13.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="13.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>13.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="13.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>13.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="13.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>13.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="13.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>13.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="13.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>13.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="13.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>13.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="13.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>13.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="13.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>13.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="13.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>13.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="13.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>13.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="13.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>13.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="13.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>13.16.1</b> â€œSimplificationâ€ of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="13.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>13.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="13.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>13.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="13.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>13.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="13.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>13.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="13.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>13.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="13.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>13.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="13.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>13.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="13.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>13.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="13.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>13.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="13.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>13.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="13.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>13.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>14</b> Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="14.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>14.1</b> Instructors:</a></li>
<li class="chapter" data-level="14.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-8"><i class="fa fa-check"></i><b>14.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-10"><i class="fa fa-check"></i><b>14.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-8"><i class="fa fa-check"></i><b>14.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="14.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-11"><i class="fa fa-check"></i><b>14.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="14.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-11"><i class="fa fa-check"></i><b>14.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-11"><i class="fa fa-check"></i><b>14.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="14.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-9"><i class="fa fa-check"></i><b>14.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="14.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-10"><i class="fa fa-check"></i><b>14.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>15</b> Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="15.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>15.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="15.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-9"><i class="fa fa-check"></i><b>15.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="15.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-11"><i class="fa fa-check"></i><b>15.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="15.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-4"><i class="fa fa-check"></i><b>15.2.2</b> Participation</a></li>
<li class="chapter" data-level="15.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>15.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="15.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-12"><i class="fa fa-check"></i><b>15.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-4"><i class="fa fa-check"></i><b>15.3</b> Introduction</a></li>
<li class="chapter" data-level="15.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>15.4</b> Why maintain your package</a></li>
<li class="chapter" data-level="15.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>15.5</b> Infrastructure used for maintenance</a></li>
<li class="chapter" data-level="15.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>15.6</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="15.6.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>15.6.1</b> Build report</a></li>
<li class="chapter" data-level="15.6.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>15.6.2</b> Landing page badges</a></li>
<li class="chapter" data-level="15.6.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>15.6.3</b> Download statistics</a></li>
<li class="chapter" data-level="15.6.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>15.6.4</b> RSS feeds</a></li>
<li class="chapter" data-level="15.6.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>15.6.5</b> Support site, Bioc-devel mailing list</a></li>
<li class="chapter" data-level="15.6.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github"><i class="fa fa-check"></i><b>15.6.6</b> GitHub</a></li>
</ul></li>
<li class="chapter" data-level="15.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>15.7</b> Round up of resources available</a></li>
<li class="chapter" data-level="15.8" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#acknowledgements"><i class="fa fa-check"></i><b>15.8</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>16</b> Public Data Resources and Bioconductor</a><ul>
<li class="chapter" data-level="16.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>16.1</b> Instructor name(s) and contact information</a></li>
<li class="chapter" data-level="16.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>16.2</b> Syllabus</a><ul>
<li class="chapter" data-level="16.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-10"><i class="fa fa-check"></i><b>16.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="16.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-12"><i class="fa fa-check"></i><b>16.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="16.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-9"><i class="fa fa-check"></i><b>16.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="16.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>16.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="16.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-12"><i class="fa fa-check"></i><b>16.2.5</b> Time outline</a></li>
<li class="chapter" data-level="16.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-13"><i class="fa fa-check"></i><b>16.2.6</b> workshop goals and objectives</a></li>
<li class="chapter" data-level="16.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-10"><i class="fa fa-check"></i><b>16.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="16.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-11"><i class="fa fa-check"></i><b>16.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-5"><i class="fa fa-check"></i><b>16.3</b> Overview</a></li>
<li class="chapter" data-level="16.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>16.4</b> GEOquery</a><ul>
<li class="chapter" data-level="16.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>16.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="16.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>16.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="16.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>16.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="16.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>16.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>16.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="16.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>16.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="16.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>16.5.2</b> Basic design</a></li>
<li class="chapter" data-level="16.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>16.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="16.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>16.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="16.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>16.6.1</b> Authentication</a></li>
<li class="chapter" data-level="16.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>16.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>16.7</b> Sequence Read Archive</a></li>
<li class="chapter" data-level="16.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>16.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="16.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>16.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="16.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>16.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="16.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>16.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="16.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>16.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="16.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>16.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="16.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>16.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="16.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>16.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="16.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>16.12</b> Pharmacogenomics</a></li>
<li class="chapter" data-level="16.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>16.13</b> Bibliography</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr" class="section level1">
<h1><span class="header-section-number">7</span> Variant Functional Annotation using StatePaintR, FunciVar &amp; MotifBreakR</h1>
<div id="instructors-names-and-contact-information-1" class="section level2">
<h2><span class="header-section-number">7.1</span> Instructor(s) name(s) and contact information</h2>
<ul>
<li>Simon G. Coetzee (<a href="mailto:scoetzee@gmail.com">scoetzee@gmail.com</a>)</li>
<li>Dennis J. Hazelett (<a href="mailto:dennis.hazelett@csmc.edu">dennis.hazelett@csmc.edu</a>)</li>
</ul>
</div>
<div id="workshop-description-1" class="section level2">
<h2><span class="header-section-number">7.2</span> Workshop Description</h2>
<p>This workshop will entail lecture and live demo of StateHub/ StatePaintR and funciVar bioconductor packages.</p>
<div id="pre-requisites-4" class="section level3">
<h3><span class="header-section-number">7.2.1</span> Pre-requisites</h3>
<p>It is recommended that workshop participants have: * Basic concepts in epigenomics and GWAS * Intermediate R skills, familiarity with GRanges</p>
<p>Relevant background reading for the workshop.</p>
<ul>
<li><a href="https://f1000research.com/articles/7-214/v1">StateHub/StatePaintR</a></li>
<li><a href="https://www.simoncoetzee.com/bioc2017.html">BioC2017 Workshop</a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/motifbreakR.html">MotifBreakR</a></li>
</ul>
</div>
<div id="workshop-participation-1" class="section level3">
<h3><span class="header-section-number">7.2.2</span> Workshop Participation</h3>
<p>Describe how students will be expected to participate in the workshop.</p>
</div>
<div id="r-bioconductor-packages-used-5" class="section level3">
<h3><span class="header-section-number">7.2.3</span> <em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li>StatePaintR</li>
<li>FunciVar</li>
<li>MotifBreakR</li>
</ul>
</div>
<div id="time-outline-5" class="section level3">
<h3><span class="header-section-number">7.2.4</span> Time outline</h3>
<p>1 hour Workshop:</p>
<table>
<thead>
<tr class="header">
<th>Activity</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>StatePaintR</strong></td>
<td>30m</td>
</tr>
<tr class="even">
<td>* intro and theory</td>
<td>(10m)</td>
</tr>
<tr class="odd">
<td>* Generate Decision Matrix</td>
<td>(10m)</td>
</tr>
<tr class="even">
<td>* live demo</td>
<td>(10m)</td>
</tr>
<tr class="odd">
<td><strong>FunciVar</strong></td>
<td>15m</td>
</tr>
<tr class="even">
<td>* intro</td>
<td>(10m)</td>
</tr>
<tr class="odd">
<td>* live demo</td>
<td>(5m)</td>
</tr>
<tr class="even">
<td><strong>MotifBreakR</strong></td>
<td>15m</td>
</tr>
<tr class="odd">
<td>* intro &amp; demo</td>
<td>(10m)</td>
</tr>
<tr class="even">
<td>* Shiny webapp demo</td>
<td>(5m)</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="workshop-goals-and-objectives-5" class="section level2">
<h2><span class="header-section-number">7.3</span> Workshop goals and objectives</h2>
<p>This workshop focuses on bioconductor based tools for non-coding variant annotation. I will present a high level overview of these concepts and the bioconductor tools our group uses to address them <em>in silico</em>. These tools can be used for germline or somatic variants, but as we will see, can also be used for any other type of feature represented as GRanges objects.</p>
<div id="key-concepts" class="section level3">
<h3><span class="header-section-number">7.3.1</span> Key Concepts</h3>
<p>I will provide brief overview with examples of:</p>
<ul>
<li>Genome segmentation and Chromatin State</li>
<li>Feature enrichment analysis</li>
<li>Transcription factors and Motif Disruption</li>
</ul>
</div>
<div id="learning-objectives-4" class="section level3">
<h3><span class="header-section-number">7.3.2</span> Learning objectives</h3>
<p>At the end of this workshop students will understand the basic inputs and outputs of all these analyses, as well as an intuitive understanding of the tools and where to find them:</p>
<ul>
<li><a href="http://statehub.org">The StateHub website</a></li>
<li>StatePaintR genome segmentation software <a href="https://github.com/Simon-Coetzee/StatePaintR/blob/master/vignettes/StatePaintR_Analysis.Rmd">Bioconductor</a></li>
<li>FunciVar for variant annotation and enrichment</li>
<li>MotifBreakR Transcription Factor motif disruption <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/motifbreakR/inst/doc/motifbreakR-vignette.html">Bioconductor</a></li>
</ul>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="functional-enrichment-analysis-of-high-throughput-omics-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
