<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7.13 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="introduction-to-bioconductor-annotation-resources.html">
<link rel="next" href="maintaining-your-bioconductor-package.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#for-everyone"><i class="fa fa-check"></i><b>1.1</b> For Everyone</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#for-workshop-authors"><i class="fa fa-check"></i><b>1.2</b> For Workshop Authors</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#description"><i class="fa fa-check"></i><b>1.2.1</b> DESCRIPTION</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#classifying-your-workshop"><i class="fa fa-check"></i><b>1.2.2</b> Classifying your workshop</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#bookdown"><i class="fa fa-check"></i><b>1.2.3</b> Bookdown</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#deadlines-for-bioc2018"><i class="fa fa-check"></i><b>1.3</b> Deadlines for Bioc2018</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>2</b> RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-1"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>2.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>2.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="2.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>2.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="2.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>2.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="2.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>2.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>2.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>2.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="2.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>2.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="2.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>2.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>2.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="2.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>2.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="2.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>2.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="2.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>2.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis"><i class="fa fa-check"></i><b>2.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="2.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>2.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="2.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>2.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub"><i class="fa fa-check"></i><b>2.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="2.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub"><i class="fa fa-check"></i><b>2.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="2.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>2.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>2.7</b> Building reports</a><ul>
<li class="chapter" data-level="2.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>2.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="2.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>2.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>2.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="2.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>2.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="2.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>2.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="2.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>2.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="2.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>2.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="2.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>2.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="2.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>2.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>3</b> Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="3.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-1"><i class="fa fa-check"></i><b>3.1</b> Overview</a><ul>
<li class="chapter" data-level="3.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-2"><i class="fa fa-check"></i><b>3.1.1</b> Description</a></li>
<li class="chapter" data-level="3.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-1"><i class="fa fa-check"></i><b>3.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="3.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-1"><i class="fa fa-check"></i><b>3.1.3</b> Participation</a></li>
<li class="chapter" data-level="3.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="3.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-1"><i class="fa fa-check"></i><b>3.1.5</b> Time outline</a></li>
<li class="chapter" data-level="3.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started"><i class="fa fa-check"></i><b>3.2</b> Getting started</a><ul>
<li class="chapter" data-level="3.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>3.2.1</b> The data</a></li>
<li class="chapter" data-level="3.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>3.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>3.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="3.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>3.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="3.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>3.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>3.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="3.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>3.5.1</b> Normalization</a></li>
<li class="chapter" data-level="3.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>3.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>3.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="3.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>3.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="3.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>3.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>4</b> Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="4.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructors-names-and-contact-information"><i class="fa fa-check"></i><b>4.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="4.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description"><i class="fa fa-check"></i><b>4.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="4.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-2"><i class="fa fa-check"></i><b>4.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation"><i class="fa fa-check"></i><b>4.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="4.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-2"><i class="fa fa-check"></i><b>4.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>4.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="4.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop"><i class="fa fa-check"></i><b>4.4</b> Workshop</a></li>
<li class="chapter" data-level="4.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>4.5</b> Where does it all come from?</a></li>
<li class="chapter" data-level="4.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>4.6</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="4.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>4.7</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="4.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>4.8</b> Data types</a></li>
<li class="chapter" data-level="4.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>4.9</b> Differential expression analysis</a></li>
<li class="chapter" data-level="4.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>4.10</b> Gene sets</a></li>
<li class="chapter" data-level="4.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>4.11</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="4.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>4.12</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="4.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>4.13</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="4.14" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>4.14</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>5</b> RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i><b>5.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-1"><i class="fa fa-check"></i><b>5.2</b> Introduction</a></li>
<li class="chapter" data-level="5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>5.3</b> Data packaging</a><ul>
<li class="chapter" data-level="5.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>5.3.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="5.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>5.3.2</b> Organising sample information</a></li>
<li class="chapter" data-level="5.3.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>5.3.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>5.4</b> Data pre-processing</a><ul>
<li class="chapter" data-level="5.4.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>5.4.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="5.4.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>5.4.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="5.4.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>5.4.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="5.4.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>5.4.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>5.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="5.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>5.5.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="5.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>5.5.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="5.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>5.5.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="5.5.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>5.5.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="5.5.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>5.5.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="5.5.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>5.5.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>6</b> Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="6.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructors-names-and-contact-information-2"><i class="fa fa-check"></i><b>6.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="6.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-1"><i class="fa fa-check"></i><b>6.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="6.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-1"><i class="fa fa-check"></i><b>6.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="6.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>6.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-3"><i class="fa fa-check"></i><b>6.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>6.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="6.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals"><i class="fa fa-check"></i><b>6.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="6.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives"><i class="fa fa-check"></i><b>6.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>6.4</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="6.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>6.5</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="6.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>6.6</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="6.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>6.7</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="6.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>6.8</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="6.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#grangeslist-groups-of-genomic-ranges"><i class="fa fa-check"></i><b>6.9</b> <em>GRangesList</em>: Groups of Genomic Ranges</a></li>
<li class="chapter" data-level="6.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-grangeslist-accessors"><i class="fa fa-check"></i><b>6.10</b> Basic <em>GRangesList</em> accessors</a></li>
<li class="chapter" data-level="6.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#combining-grangeslist-objects"><i class="fa fa-check"></i><b>6.11</b> Combining <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="6.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-grangeslist-objects"><i class="fa fa-check"></i><b>6.12</b> Basic interval operations for <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="6.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-grangeslist-objects"><i class="fa fa-check"></i><b>6.13</b> Subsetting <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="6.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#looping-over-grangeslist-objects"><i class="fa fa-check"></i><b>6.14</b> Looping over <em>GRangesList</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>7</b> Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="7.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructors-names-and-contact-information-3"><i class="fa fa-check"></i><b>7.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="7.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-2"><i class="fa fa-check"></i><b>7.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="7.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-2"><i class="fa fa-check"></i><b>7.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="7.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>7.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-4"><i class="fa fa-check"></i><b>7.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>7.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="7.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-1"><i class="fa fa-check"></i><b>7.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="7.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-1"><i class="fa fa-check"></i><b>7.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-1"><i class="fa fa-check"></i><b>7.4</b> Workshop</a></li>
<li class="chapter" data-level="7.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-2"><i class="fa fa-check"></i><b>7.5</b> Introduction</a><ul>
<li class="chapter" data-level="7.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>7.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="7.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>7.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="7.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>7.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup"><i class="fa fa-check"></i><b>7.6</b> Setup</a></li>
<li class="chapter" data-level="7.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>7.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="7.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>7.8</b> The Grammar</a><ul>
<li class="chapter" data-level="7.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>7.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="7.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>7.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="7.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises"><i class="fa fa-check"></i><b>7.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>7.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="7.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>7.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="7.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>7.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="7.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>7.10</b> Whatâ€™s next?</a></li>
<li class="chapter" data-level="7.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>7.11</b> Appendix</a></li>
<li class="chapter" data-level="7.12" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#references"><i class="fa fa-check"></i><b>7.12</b> References</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>8</b> Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="8.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>8.1</b> Instructors</a></li>
<li class="chapter" data-level="8.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description-3"><i class="fa fa-check"></i><b>8.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="8.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>8.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="8.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation-3"><i class="fa fa-check"></i><b>8.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="8.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>8.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>8.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="8.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-2"><i class="fa fa-check"></i><b>8.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="8.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-2"><i class="fa fa-check"></i><b>8.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>8.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="8.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>8.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="8.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>8.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="8.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>8.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="8.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>8.4.4</b> Data containers</a></li>
<li class="chapter" data-level="8.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>8.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="8.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>8.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="8.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>8.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="8.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>8.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="8.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>8.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="8.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>8.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="8.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>8.4.11</b> Simple example</a></li>
<li class="chapter" data-level="8.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>8.4.12</b> Questions!</a></li>
<li class="chapter" data-level="8.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>8.4.13</b> More questions!</a></li>
<li class="chapter" data-level="8.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>8.4.14</b> Another example</a></li>
<li class="chapter" data-level="8.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>8.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="8.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>8.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="8.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>8.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="8.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>8.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="8.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>8.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="8.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>8.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="8.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>8.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="8.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>8.4.22</b> GRanges</a></li>
<li class="chapter" data-level="8.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>8.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="8.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>8.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="8.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>8.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="8.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>8.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="8.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>8.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="8.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>8.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="8.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>8.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="8.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>8.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="8.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>8.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="8.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>8.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="8.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dply-example"><i class="fa fa-check"></i><b>8.4.33</b> Organism.dply example</a></li>
<li class="chapter" data-level="8.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>8.4.34</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="8.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>8.4.35</b> BSgenome packages</a></li>
<li class="chapter" data-level="8.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>8.4.36</b> BSgenome packages</a></li>
<li class="chapter" data-level="8.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>8.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="8.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>8.4.38</b> BSgenome exercises</a></li>
<li class="chapter" data-level="8.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-1"><i class="fa fa-check"></i><b>8.4.39</b> AnnotationHub</a></li>
<li class="chapter" data-level="8.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>8.4.40</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="8.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>8.4.41</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="8.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>8.4.42</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="8.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>8.4.43</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="8.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>8.4.44</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="8.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>8.4.45</b> AnnotationHub query</a></li>
<li class="chapter" data-level="8.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>8.4.46</b> AnnotationHub query</a></li>
<li class="chapter" data-level="8.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>8.4.47</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="8.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>8.4.48</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="8.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>8.4.49</b> biomaRt</a></li>
<li class="chapter" data-level="8.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>8.4.50</b> biomaRt data sets</a></li>
<li class="chapter" data-level="8.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>8.4.51</b> biomaRt queries</a></li>
<li class="chapter" data-level="8.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>8.4.52</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="8.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>8.4.53</b> biomaRt query</a></li>
<li class="chapter" data-level="8.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>8.4.54</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>9</b> Workflow for Multi-omics Analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="9.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>9.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-4"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-5"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-4"><i class="fa fa-check"></i><b>9.2.2</b> workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>9.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-5"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>9.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="9.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-3"><i class="fa fa-check"></i><b>9.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="9.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-3"><i class="fa fa-check"></i><b>9.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>9.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="9.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summarizedexperiment-raggedexperiment"><i class="fa fa-check"></i><b>9.4.1</b> <code>SummarizedExperiment</code> / <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="9.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>9.4.2</b> <code>MultiAssayExperiment</code></a></li>
<li class="chapter" data-level="9.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>9.4.3</b> <code>RaggedExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>9.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="9.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>9.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="9.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#using-granges-objects"><i class="fa fa-check"></i><b>9.5.2</b> Using <code>GRanges</code> objects</a></li>
<li class="chapter" data-level="9.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>9.5.3</b> *Assay functions</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#qreduceassay"><i class="fa fa-check"></i><b>9.6</b> qreduceAssay</a></li>
<li class="chapter" data-level="9.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>9.7</b> Working with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="9.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>9.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="9.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>9.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="9.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>9.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="9.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>9.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="9.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>9.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>9.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="9.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>9.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="9.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>9.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="9.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>9.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="9.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#patients-with-complete-data"><i class="fa fa-check"></i><b>9.9</b> Patients with complete data</a></li>
<li class="chapter" data-level="9.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>9.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="9.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>9.11</b> Extraction</a><ul>
<li class="chapter" data-level="9.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>9.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="9.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>9.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="9.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>9.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="9.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>9.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="9.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>9.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="9.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>9.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="9.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>9.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="9.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>9.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="9.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>9.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="9.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>9.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="9.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiments"><i class="fa fa-check"></i><b>9.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiments</a></li>
<li class="chapter" data-level="9.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>9.16</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="9.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>9.16.1</b> How many samples have data for each combination of assays?</a></li>
</ul></li>
<li class="chapter" data-level="9.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#tcgautils-package"><i class="fa fa-check"></i><b>9.17</b> <code>TCGAutils</code> package</a><ul>
<li class="chapter" data-level="9.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>9.17.1</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="9.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#is-there-subtype-data-available-in-the-multiassayexperiment-obtained-from"><i class="fa fa-check"></i><b>9.17.2</b> Is there subtype data available in the <code>MultiAssayExperiment</code> obtained from</a></li>
<li class="chapter" data-level="9.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>9.17.3</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="9.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>9.17.4</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="9.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>9.17.5</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="9.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>9.17.6</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="9.17.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>9.17.7</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>10</b> Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="10.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#authors"><i class="fa fa-check"></i><b>10.1</b> Authors:</a></li>
<li class="chapter" data-level="10.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-5"><i class="fa fa-check"></i><b>10.2</b> Workshop Description:</a></li>
<li class="chapter" data-level="10.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-6"><i class="fa fa-check"></i><b>10.3</b> Pre-requisites:</a></li>
<li class="chapter" data-level="10.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-2"><i class="fa fa-check"></i><b>10.4</b> Participation:</a></li>
<li class="chapter" data-level="10.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>10.5</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="10.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>10.6</b> Workshop goals and objectives:</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>11</b> Public Data Resources and Bioconductor</a><ul>
<li class="chapter" data-level="11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-6"><i class="fa fa-check"></i><b>11.1</b> Workshop Description</a><ul>
<li class="chapter" data-level="11.1.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-7"><i class="fa fa-check"></i><b>11.1.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="11.1.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-5"><i class="fa fa-check"></i><b>11.1.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.1.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>11.1.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="11.1.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-6"><i class="fa fa-check"></i><b>11.1.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>11.2</b> workshop goals and objectives</a><ul>
<li class="chapter" data-level="11.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-4"><i class="fa fa-check"></i><b>11.2.1</b> Learning goals</a></li>
<li class="chapter" data-level="11.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-4"><i class="fa fa-check"></i><b>11.2.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-2"><i class="fa fa-check"></i><b>11.3</b> Overview</a></li>
<li class="chapter" data-level="11.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>11.4</b> GEOquery</a></li>
<li class="chapter" data-level="11.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>11.5</b> GenomicDataCommons</a></li>
<li class="chapter" data-level="11.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>11.6</b> Sequence Read Archive</a></li>
<li class="chapter" data-level="11.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>11.7</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="11.7.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata"><i class="fa fa-check"></i><b>11.7.1</b> curatedTCGAData</a></li>
</ul></li>
<li class="chapter" data-level="11.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#gbm-dataset-example"><i class="fa fa-check"></i><b>11.8</b> GBM dataset example</a><ul>
<li class="chapter" data-level="11.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#subtype-information"><i class="fa fa-check"></i><b>11.8.1</b> Subtype information</a></li>
<li class="chapter" data-level="11.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>11.8.2</b> TCGABiolinks</a></li>
<li class="chapter" data-level="11.8.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons-1"><i class="fa fa-check"></i><b>11.8.3</b> GenomicDataCommons</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>11.9</b> Microbiome data</a></li>
<li class="chapter" data-level="11.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>11.10</b> Pharmacogenomics</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="workflow-for-multi-omics-analysis-with-multiassayexperiment" class="section level1">
<h1><span class="header-section-number">9</span> Workflow for Multi-omics Analysis with MultiAssayExperiment</h1>
<div id="instructor-names-and-contact-information" class="section level2">
<h2><span class="header-section-number">9.1</span> Instructor names and contact information</h2>
<ul>
<li>Marcel Ramos</li>
<li>Ludwig Geistlinger</li>
<li>Levi Waldron</li>
</ul>
</div>
<div id="workshop-description-4" class="section level2">
<h2><span class="header-section-number">9.2</span> Workshop Description</h2>
<p>This workshop demonstrates data management and analyses of multiple assays associated with a single set of biological specimens, using the <code>MultiAssayExperiment</code> data class and methods. It introduces the <code>RaggedExperiment</code> data class, which provides efficient and powerful operations for representation of copy number and mutation and variant data that are represented by different genomic ranges for each specimen.</p>
<div id="pre-requisites-5" class="section level3">
<h3><span class="header-section-number">9.2.1</span> Pre-requisites</h3>
<p>List any workshop prerequisites, for example:</p>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Familiarity with the GRanges and SummarizedExperiment classes</li>
<li>Familiarity with â€™omics data types including copy number and gene expression</li>
</ul>
</div>
<div id="workshop-participation-4" class="section level3">
<h3><span class="header-section-number">9.2.2</span> workshop Participation</h3>
<p>Students will have a chance to build a <code>MultiAssayExperiment</code> object from scratch, and will also work with more complex objects provided by the <code>curatedTCGAData</code> package.</p>
</div>
<div id="rbioconductor-packages-used" class="section level3">
<h3><span class="header-section-number">9.2.3</span> R/Bioconductor packages used</h3>
<ul>
<li><a href="https://bioconductor.org/packages/MultiAssayExperiment">MultiAssayExperiment</a></li>
<li><a href="https://bioconductor.org/packages/RaggedExperiment">RaggedExperiment</a></li>
<li><a href="https://bioconductor.org/packages/curatedTCGAData">curatedTCGAData</a></li>
<li><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></li>
<li><a href="https://bioconductor.org/packages/TCGAutils">TCGAutils</a></li>
<li><a href="https://bioconductor.org/packages/UpSetR">UpSetR</a></li>
<li><a href="https://bioconductor.org/packages/EnsDb.Hsapiens.v86">EnsDb.Hsapiens.v86</a></li>
<li><a href="https://cran.r-project.org/package=survminer">survminer</a></li>
<li><a href="https://cran.r-project.org/package=pheatmap">pheatmap</a></li>
</ul>
</div>
<div id="time-outline-5" class="section level3">
<h3><span class="header-section-number">9.2.4</span> Time outline</h3>
<p>1h 45m total</p>
<table>
<thead>
<tr class="header">
<th>Activity</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Overview of key data classes</td>
<td>25m</td>
</tr>
<tr class="even">
<td>Working with RaggedExperiment</td>
<td>20m</td>
</tr>
<tr class="odd">
<td>Building a MultiAssayExperiment from scratch</td>
<td>10m</td>
</tr>
<tr class="even">
<td>TCGA multi-assay dataset</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>Subsetting and reshaping multi-assay data</td>
<td>20m</td>
</tr>
<tr class="even">
<td>Plotting, correlation, and other analyses</td>
<td>20m</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="workshop-goals-and-objectives-5" class="section level2">
<h2><span class="header-section-number">9.3</span> Workshop goals and objectives</h2>
<div id="learning-goals-3" class="section level3">
<h3><span class="header-section-number">9.3.1</span> Learning goals</h3>
<ul>
<li>identify appropriate data structures for different â€™omics data types</li>
<li>gain familiarity with GRangesList and RaggedExperiment</li>
</ul>
</div>
<div id="learning-objectives-3" class="section level3">
<h3><span class="header-section-number">9.3.2</span> Learning objectives</h3>
<ul>
<li>use curatedTCGAData to create custom TCGA MultiAssayExperiment objects</li>
<li>create a MultiAssayExperiment for TCGA or other multiâ€™omics data</li>
<li>perform subsetting, reshaping, growing, and extraction of a MultiAssayExperiment</li>
<li>link MultiAssayExperiment data with packages for differential expression, machine learning, and plotting</li>
</ul>
</div>
</div>
<div id="overview-of-key-data-classes" class="section level2">
<h2><span class="header-section-number">9.4</span> Overview of key data classes</h2>
<div id="summarizedexperiment-raggedexperiment" class="section level3">
<h3><span class="header-section-number">9.4.1</span> <code>SummarizedExperiment</code> / <code>RaggedExperiment</code></h3>
<div class="figure">
<img src="Ramos_MultiAssayExperiment/SummarizedExperiment.svg" alt="SummarizedExperiment_Schematic" />
<p class="caption">SummarizedExperiment_Schematic</p>
</div>
<p><em>The</em> standard Bioconductor class convenient for storing experimental results produced by sequencing and microarry experiments. Each object can store multiple experiments with associated metadata such as features and samples, as well as phenotype/clinical data. Supersedes the use of <code>ExpressionSet</code>.</p>
</div>
<div id="multiassayexperiment" class="section level3">
<h3><span class="header-section-number">9.4.2</span> <code>MultiAssayExperiment</code></h3>
<p>An integrative container for coordinating multi-omics experiment data on a set of biological specimens.</p>
<div class="figure">
<img src="Ramos_MultiAssayExperiment/MultiAssayExperiment.png" alt="MultiAssayExperiment_Schematic" />
<p class="caption">MultiAssayExperiment_Schematic</p>
</div>
</div>
<div id="raggedexperiment" class="section level3">
<h3><span class="header-section-number">9.4.3</span> <code>RaggedExperiment</code></h3>
<p>A flexible data representation for copy number, mutation, and other ragged array schema for genomic location data.</p>
<div class="figure">
<img src="Ramos_MultiAssayExperiment/RaggedExperiment.png" alt="RaggedExperiment_Schematic" />
<p class="caption">RaggedExperiment_Schematic</p>
</div>
</div>
</div>
<div id="working-with-raggedexperiment" class="section level2">
<h2><span class="header-section-number">9.5</span> Working with RaggedExperiment</h2>
<div id="constructing-a-raggedexperiment-object" class="section level3">
<h3><span class="header-section-number">9.5.1</span> Constructing a <code>RaggedExperiment</code> object</h3>
<p>We start with a couple of <code>GRanges</code> objects, each representing an individual sample:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(GenomicRanges)
sample1 &lt;-<span class="st"> </span><span class="kw">GRanges</span>(
    <span class="kw">c</span>(<span class="dt">A =</span> <span class="st">&quot;chr1:1-10:-&quot;</span>, <span class="dt">B =</span> <span class="st">&quot;chr1:8-14:+&quot;</span>, <span class="dt">C =</span> <span class="st">&quot;chr2:15-18:+&quot;</span>),
    <span class="dt">score =</span> <span class="dv">3</span><span class="op">:</span><span class="dv">5</span>)
sample2 &lt;-<span class="st"> </span><span class="kw">GRanges</span>(
    <span class="kw">c</span>(<span class="dt">D =</span> <span class="st">&quot;chr1:1-10:-&quot;</span>, <span class="dt">E =</span> <span class="st">&quot;chr2:11-18:+&quot;</span>),
    <span class="dt">score =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)</code></pre></div>
<p>Include column data <code>colData</code> to describe the samples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">colDat &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(<span class="dt">id =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)</code></pre></div>
</div>
<div id="using-granges-objects" class="section level3">
<h3><span class="header-section-number">9.5.2</span> Using <code>GRanges</code> objects</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(RaggedExperiment)

ragexp &lt;-<span class="st"> </span><span class="kw">RaggedExperiment</span>(
    <span class="dt">sample1 =</span> sample1,
    <span class="dt">sample2 =</span> sample2,
    <span class="dt">colData =</span> colDat)

ragexp
<span class="co">#&gt; class: RaggedExperiment </span>
<span class="co">#&gt; dim: 5 2 </span>
<span class="co">#&gt; assays(1): score</span>
<span class="co">#&gt; rownames(5): A B C D E</span>
<span class="co">#&gt; colnames(2): sample1 sample2</span>
<span class="co">#&gt; colData names(1): id</span></code></pre></div>
<p>It can also be constructed from a <code>GRangesList</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(GenomicRanges)
<span class="kw">example</span>(GRangesList)
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; ## Construction with GRangesList():</span>
<span class="co">#&gt; GRngsL&gt; gr1 &lt;-</span>
<span class="co">#&gt; GRngsL+   GRanges(seqnames = &quot;chr2&quot;, ranges = IRanges(3, 6),</span>
<span class="co">#&gt; GRngsL+           strand = &quot;+&quot;, score = 5L, GC = 0.45)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; gr2 &lt;-</span>
<span class="co">#&gt; GRngsL+   GRanges(seqnames = c(&quot;chr1&quot;, &quot;chr1&quot;),</span>
<span class="co">#&gt; GRngsL+           ranges = IRanges(c(7,13), width = 3),</span>
<span class="co">#&gt; GRngsL+           strand = c(&quot;+&quot;, &quot;-&quot;), score = 3:4, GC = c(0.3, 0.5))</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; gr3 &lt;-</span>
<span class="co">#&gt; GRngsL+   GRanges(seqnames = c(&quot;chr1&quot;, &quot;chr2&quot;),</span>
<span class="co">#&gt; GRngsL+           ranges = IRanges(c(1, 4), c(3, 9)),</span>
<span class="co">#&gt; GRngsL+           strand = c(&quot;-&quot;, &quot;-&quot;), score = c(6L, 2L), GC = c(0.4, 0.1))</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; grl &lt;- GRangesList(&quot;gr1&quot; = gr1, &quot;gr2&quot; = gr2, &quot;gr3&quot; = gr3)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; grl</span>
<span class="co">#&gt; GRangesList object of length 3:</span>
<span class="co">#&gt; $gr1 </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]     chr2       3-6      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr2 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1    7-9      + |     3 0.3</span>
<span class="co">#&gt;   [2]     chr1  13-15      - |     4 0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr3 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1    1-3      - |     6 0.4</span>
<span class="co">#&gt;   [2]     chr2    4-9      - |     2 0.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; ## Summarizing elements:</span>
<span class="co">#&gt; GRngsL&gt; elementNROWS(grl)</span>
<span class="co">#&gt; gr1 gr2 gr3 </span>
<span class="co">#&gt;   1   2   2 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; table(seqnames(grl))</span>
<span class="co">#&gt;     chr2 chr1</span>
<span class="co">#&gt; gr1    1    0</span>
<span class="co">#&gt; gr2    0    2</span>
<span class="co">#&gt; gr3    1    1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; ## Extracting subsets:</span>
<span class="co">#&gt; GRngsL&gt; grl[seqnames(grl) == &quot;chr1&quot;, ]</span>
<span class="co">#&gt; GRangesList object of length 3:</span>
<span class="co">#&gt; $gr1 </span>
<span class="co">#&gt; GRanges object with 0 ranges and 2 metadata columns:</span>
<span class="co">#&gt;    seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;       &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr2 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1    7-9      + |     3 0.3</span>
<span class="co">#&gt;   [2]     chr1  13-15      - |     4 0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr3 </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1    1-3      - |     6 0.4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; grl[seqnames(grl) == &quot;chr1&quot; &amp; strand(grl) == &quot;+&quot;, ]</span>
<span class="co">#&gt; GRangesList object of length 3:</span>
<span class="co">#&gt; $gr1 </span>
<span class="co">#&gt; GRanges object with 0 ranges and 2 metadata columns:</span>
<span class="co">#&gt;    seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;       &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr2 </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand | score  GC</span>
<span class="co">#&gt;   [1]     chr1    7-9      + |     3 0.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr3 </span>
<span class="co">#&gt; GRanges object with 0 ranges and 2 metadata columns:</span>
<span class="co">#&gt;      seqnames ranges strand | score GC</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; ## Renaming the underlying sequences:</span>
<span class="co">#&gt; GRngsL&gt; seqlevels(grl)</span>
<span class="co">#&gt; [1] &quot;chr2&quot; &quot;chr1&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; seqlevels(grl) &lt;- sub(&quot;chr&quot;, &quot;Chrom&quot;, seqlevels(grl))</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; grl</span>
<span class="co">#&gt; GRangesList object of length 3:</span>
<span class="co">#&gt; $gr1 </span>
<span class="co">#&gt; GRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]   Chrom2       3-6      + |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr2 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand | score  GC</span>
<span class="co">#&gt;   [1]   Chrom1    7-9      + |     3 0.3</span>
<span class="co">#&gt;   [2]   Chrom1  13-15      - |     4 0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr3 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand | score  GC</span>
<span class="co">#&gt;   [1]   Chrom1    1-3      - |     6 0.4</span>
<span class="co">#&gt;   [2]   Chrom2    4-9      - |     2 0.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; ## Coerce to IRangesList (seqnames and strand information is lost):</span>
<span class="co">#&gt; GRngsL&gt; as(grl, &quot;IRangesList&quot;)</span>
<span class="co">#&gt; IRangesList of length 3</span>
<span class="co">#&gt; $gr1</span>
<span class="co">#&gt; IRanges object with 1 range and 2 metadata columns:</span>
<span class="co">#&gt;           start       end     width |     score        GC</span>
<span class="co">#&gt;       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]         3         6         4 |         5      0.45</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr2</span>
<span class="co">#&gt; IRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;           start       end     width |     score        GC</span>
<span class="co">#&gt;       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]         7         9         3 |         3       0.3</span>
<span class="co">#&gt;   [2]        13        15         3 |         4       0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr3</span>
<span class="co">#&gt; IRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;           start       end     width |     score        GC</span>
<span class="co">#&gt;       &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]         1         3         3 |         6       0.4</span>
<span class="co">#&gt;   [2]         4         9         6 |         2       0.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; ## isDisjoint():</span>
<span class="co">#&gt; GRngsL&gt; isDisjoint(grl)</span>
<span class="co">#&gt;  gr1  gr2  gr3 </span>
<span class="co">#&gt; TRUE TRUE TRUE </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; ## disjoin():</span>
<span class="co">#&gt; GRngsL&gt; disjoin(grl)  # metadata columns and order NOT preserved</span>
<span class="co">#&gt; GRangesList object of length 3:</span>
<span class="co">#&gt; $gr1 </span>
<span class="co">#&gt; GRanges object with 1 range and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]   Chrom2       3-6      +</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr2 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand</span>
<span class="co">#&gt;   [1]   Chrom1    7-9      +</span>
<span class="co">#&gt;   [2]   Chrom1  13-15      -</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $gr3 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames ranges strand</span>
<span class="co">#&gt;   [1]   Chrom2    4-9      -</span>
<span class="co">#&gt;   [2]   Chrom1    1-3      -</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; ## Construction with makeGRangesListFromFeatureFragments():</span>
<span class="co">#&gt; GRngsL&gt; filepath &lt;- system.file(&quot;extdata&quot;, &quot;feature_frags.txt&quot;,</span>
<span class="co">#&gt; GRngsL+                         package=&quot;GenomicRanges&quot;)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; featfrags &lt;- read.table(filepath, header=TRUE, stringsAsFactors=FALSE)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; grl2 &lt;- with(featfrags,</span>
<span class="co">#&gt; GRngsL+              makeGRangesListFromFeatureFragments(seqnames=targetName,</span>
<span class="co">#&gt; GRngsL+                                                  fragmentStarts=targetStart,</span>
<span class="co">#&gt; GRngsL+                                                  fragmentWidths=blockSizes,</span>
<span class="co">#&gt; GRngsL+                                                  strand=strand))</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; names(grl2) &lt;- featfrags$RefSeqID</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; GRngsL&gt; grl2</span>
<span class="co">#&gt; GRangesList object of length 3:</span>
<span class="co">#&gt; $XM_001065892.1 </span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames              ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr4 124513961-124514087      +</span>
<span class="co">#&gt;   [2]     chr4 124514706-124515691      +</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $XM_578205.2 </span>
<span class="co">#&gt; GRanges object with 3 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames              ranges strand</span>
<span class="co">#&gt;   [1]     chr2 155875533-155876067      -</span>
<span class="co">#&gt;   [2]     chr2 155879894-155880030      -</span>
<span class="co">#&gt;   [3]     chr2 155895543-155895690      -</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $NM_012543.2 </span>
<span class="co">#&gt; GRanges object with 4 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames            ranges strand</span>
<span class="co">#&gt;   [1]     chr1 96173572-96174077      +</span>
<span class="co">#&gt;   [2]     chr1 96174920-96175330      +</span>
<span class="co">#&gt;   [3]     chr1 96176574-96176785      +</span>
<span class="co">#&gt;   [4]     chr1 96177991-96178484      +</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 3 sequences from an unspecified genome; no seqlengths</span>
rgx &lt;-<span class="st"> </span><span class="kw">RaggedExperiment</span>(grl)</code></pre></div>
</div>
<div id="assay-functions" class="section level3">
<h3><span class="header-section-number">9.5.3</span> *Assay functions</h3>
<div id="sparseassay" class="section level4">
<h4><span class="header-section-number">9.5.3.1</span> sparseAssay</h4>
<p>The most straightforward matrix representation of a <code>RaggedExperiment</code> will return a matrix of dimensions equal to the product of the number of ranges and samples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(ragexp)
<span class="co">#&gt; [1] 5 2</span>
<span class="kw">Reduce</span>(<span class="st">`</span><span class="dt">*</span><span class="st">`</span>, <span class="kw">dim</span>(ragexp))
<span class="co">#&gt; [1] 10</span>
<span class="kw">sparseAssay</span>(ragexp)
<span class="co">#&gt;   sample1 sample2</span>
<span class="co">#&gt; A       3      NA</span>
<span class="co">#&gt; B       4      NA</span>
<span class="co">#&gt; C       5      NA</span>
<span class="co">#&gt; D      NA       1</span>
<span class="co">#&gt; E      NA       2</span>
<span class="kw">length</span>(<span class="kw">sparseAssay</span>(ragexp))
<span class="co">#&gt; [1] 10</span></code></pre></div>
</div>
<div id="compactassay" class="section level4">
<h4><span class="header-section-number">9.5.3.2</span> compactAssay</h4>
<p>Samples with identical ranges are placed in the same row. Non-disjoint ranges are <strong>not</strong> collapsed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">compactAssay</span>(ragexp)
<span class="co">#&gt;              sample1 sample2</span>
<span class="co">#&gt; chr1:8-14:+        4      NA</span>
<span class="co">#&gt; chr1:1-10:-        3       1</span>
<span class="co">#&gt; chr2:11-18:+      NA       2</span>
<span class="co">#&gt; chr2:15-18:+       5      NA</span></code></pre></div>
</div>
<div id="disjoinassay" class="section level4">
<h4><span class="header-section-number">9.5.3.3</span> disjoinAssay</h4>
<p>This function returns a matrix of disjoint ranges across all samples. Elements of the matrix are summarized by applying the <code>simplifyDisjoin</code> functional argument to assay values of overlapping ranges.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">disjoinAssay</span>(ragexp, <span class="dt">simplifyDisjoin =</span> mean)
<span class="co">#&gt;              sample1 sample2</span>
<span class="co">#&gt; chr1:8-14:+        4      NA</span>
<span class="co">#&gt; chr1:1-10:-        3       1</span>
<span class="co">#&gt; chr2:11-14:+      NA       2</span>
<span class="co">#&gt; chr2:15-18:+       5       2</span></code></pre></div>
</div>
</div>
</div>
<div id="qreduceassay" class="section level2">
<h2><span class="header-section-number">9.6</span> qreduceAssay</h2>
<p>The <code>qreduceAssay</code> function works with a <code>query</code> parameter that highlights a window of ranges for the resulting matrix. The returned matrix will have dimensions <code>length(query)</code> by <code>ncol(x)</code>. Elements contain assay values for the <em>i</em> th query range and the <em>j</em> th sample, summarized according to the <code>simplifyReduce</code> functional argument.</p>
<p>For demonstration purposes, we can have a look at the original <code>GRangesList</code> and the associated scores from which the current <code>ragexp</code> object is derived:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">unlist</span>(grl, <span class="dt">use.names =</span> <span class="ot">FALSE</span>)
<span class="co">#&gt; GRanges object with 5 ranges and 2 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand |     score        GC</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;   [1]   Chrom2       3-6      + |         5      0.45</span>
<span class="co">#&gt;   [2]   Chrom1       7-9      + |         3       0.3</span>
<span class="co">#&gt;   [3]   Chrom1     13-15      - |         4       0.5</span>
<span class="co">#&gt;   [4]   Chrom1       1-3      - |         6       0.4</span>
<span class="co">#&gt;   [5]   Chrom2       4-9      - |         2       0.1</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>This data is represented as <code>rowRanges</code> and <code>assays</code> in <code>RaggedExperiment</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rowRanges</span>(ragexp)
<span class="co">#&gt; GRanges object with 5 ranges and 0 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   A     chr1      1-10      -</span>
<span class="co">#&gt;   B     chr1      8-14      +</span>
<span class="co">#&gt;   C     chr2     15-18      +</span>
<span class="co">#&gt;   D     chr1      1-10      -</span>
<span class="co">#&gt;   E     chr2     11-18      +</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span>
<span class="kw">assay</span>(ragexp, <span class="st">&quot;score&quot;</span>)
<span class="co">#&gt;   sample1 sample2</span>
<span class="co">#&gt; A       3      NA</span>
<span class="co">#&gt; B       4      NA</span>
<span class="co">#&gt; C       5      NA</span>
<span class="co">#&gt; D      NA       1</span>
<span class="co">#&gt; E      NA       2</span></code></pre></div>
<p>Here we provide the â€œqueryâ€ region of interest:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(query &lt;-<span class="st"> </span><span class="kw">GRanges</span>(<span class="kw">c</span>(<span class="st">&quot;chr1:1-14:-&quot;</span>, <span class="st">&quot;chr2:11-18:+&quot;</span>)))
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr1      1-14      -</span>
<span class="co">#&gt;   [2]     chr2     11-18      +</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>The <code>simplifyReduce</code> argument in <code>qreduceAssay</code> allows the user to summarize overlapping regions with a custom method for the given â€œqueryâ€ region of interest. We provide one for calculating a weighted average score per query range, where the weight is proportional to the overlap widths between overlapping ranges and a query range.</p>
<p><em>Note</em> that there are three arguments to this function. See the documentation for additional details.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">weightedmean &lt;-<span class="st"> </span><span class="cf">function</span>(scores, ranges, qranges)
{
    isects &lt;-<span class="st"> </span><span class="kw">pintersect</span>(ranges, qranges)
    <span class="kw">sum</span>(scores <span class="op">*</span><span class="st"> </span><span class="kw">width</span>(isects)) <span class="op">/</span><span class="st"> </span><span class="kw">sum</span>(<span class="kw">width</span>(isects))
}</code></pre></div>
<p>A call to <code>qreduceAssay</code> involves the <code>RaggedExperiment</code>, the <code>GRanges</code> query and the <code>simplifyReduce</code> functional argument.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">qreduceAssay</span>(ragexp, query, <span class="dt">simplifyReduce =</span> weightedmean)
<span class="co">#&gt;              sample1 sample2</span>
<span class="co">#&gt; chr1:1-14:-        3       1</span>
<span class="co">#&gt; chr2:11-18:+       5       2</span></code></pre></div>
<p><strong>Note</strong>: Coercion is possible to <code>SummarizedExperiment</code>.</p>
</div>
<div id="working-with-multiassayexperiment" class="section level2">
<h2><span class="header-section-number">9.7</span> Working with MultiAssayExperiment</h2>
<div id="the-multiassayexperiment-miniacc-demo-object" class="section level3">
<h3><span class="header-section-number">9.7.1</span> The MultiAssayExperiment miniACC demo object</h3>
<p>Get started by trying out <code>MultiAssayExperiment</code> using a subset of the TCGA adrenocortical carcinoma (ACC) dataset provided with the package. This dataset provides five assays on 92 patients, although all five assays were not performed for every patient:</p>
<ol style="list-style-type: decimal">
<li><strong>RNASeq2GeneNorm</strong>: gene mRNA abundance by RNA-seq</li>
<li><strong>gistict</strong>: GISTIC genomic copy number by gene</li>
<li><strong>RPPAArray</strong>: protein abundance by Reverse Phase Protein Array</li>
<li><strong>Mutations</strong>: non-silent somatic mutations by gene</li>
<li><strong>miRNASeqGene</strong>: microRNA abundance by microRNA-seq.</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>({
    <span class="kw">library</span>(MultiAssayExperiment)
    <span class="kw">library</span>(S4Vectors)
})
<span class="kw">data</span>(miniACC)
miniACC
<span class="co">#&gt; A MultiAssayExperiment object of 5 listed</span>
<span class="co">#&gt;  experiments with user-defined names and respective classes. </span>
<span class="co">#&gt;  Containing an ExperimentList class object of length 5: </span>
<span class="co">#&gt;  [1] RNASeq2GeneNorm: SummarizedExperiment with 198 rows and 79 columns </span>
<span class="co">#&gt;  [2] gistict: SummarizedExperiment with 198 rows and 90 columns </span>
<span class="co">#&gt;  [3] RPPAArray: SummarizedExperiment with 33 rows and 46 columns </span>
<span class="co">#&gt;  [4] Mutations: matrix with 97 rows and 90 columns </span>
<span class="co">#&gt;  [5] miRNASeqGene: SummarizedExperiment with 471 rows and 80 columns </span>
<span class="co">#&gt; Features: </span>
<span class="co">#&gt;  experiments() - obtain the ExperimentList instance </span>
<span class="co">#&gt;  colData() - the primary/phenotype DataFrame </span>
<span class="co">#&gt;  sampleMap() - the sample availability DataFrame </span>
<span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment </span>
<span class="co">#&gt;  *Format() - convert into a long or wide DataFrame </span>
<span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></code></pre></div>
</div>
<div id="coldata---information-biological-units" class="section level3">
<h3><span class="header-section-number">9.7.2</span> colData - information biological units</h3>
<p>This slot is a <code>DataFrame</code> describing the characteristics of biological units, for example clinical data for patients. In the prepared datasets from [The Cancer Genome Atlas][], each row is one patient and each column is a clinical, pathological, subtype, or other variable. The <code>$</code> function provides a shortcut for accessing or setting <code>colData</code> columns.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">colData</span>(miniACC)[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]
<span class="co">#&gt; DataFrame with 4 rows and 4 columns</span>
<span class="co">#&gt;                 patientID years_to_birth vital_status days_to_death</span>
<span class="co">#&gt;               &lt;character&gt;      &lt;integer&gt;    &lt;integer&gt;     &lt;integer&gt;</span>
<span class="co">#&gt; TCGA-OR-A5J1 TCGA-OR-A5J1             58            1          1355</span>
<span class="co">#&gt; TCGA-OR-A5J2 TCGA-OR-A5J2             44            1          1677</span>
<span class="co">#&gt; TCGA-OR-A5J3 TCGA-OR-A5J3             23            0            NA</span>
<span class="co">#&gt; TCGA-OR-A5J4 TCGA-OR-A5J4             23            1           423</span>
<span class="kw">table</span>(miniACC<span class="op">$</span>race)
<span class="co">#&gt; </span>
<span class="co">#&gt;                     asian black or african american </span>
<span class="co">#&gt;                         2                         1 </span>
<span class="co">#&gt;                     white </span>
<span class="co">#&gt;                        78</span></code></pre></div>
<p><em>Key points:</em></p>
<ul>
<li>One row per patient</li>
<li>Each row maps to zero or more observations in each experiment in the <code>ExperimentList</code>, below.</li>
</ul>
</div>
<div id="experimentlist---experiment-data" class="section level3">
<h3><span class="header-section-number">9.7.3</span> ExperimentList - experiment data</h3>
<p>A base <code>list</code> or <code>ExperimentList</code> object containing the experimental datasets for the set of samples collected. This gets converted into a class <code>ExperimentList</code> during construction.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">experiments</span>(miniACC)
<span class="co">#&gt; ExperimentList class object of length 5: </span>
<span class="co">#&gt;  [1] RNASeq2GeneNorm: SummarizedExperiment with 198 rows and 79 columns </span>
<span class="co">#&gt;  [2] gistict: SummarizedExperiment with 198 rows and 90 columns </span>
<span class="co">#&gt;  [3] RPPAArray: SummarizedExperiment with 33 rows and 46 columns </span>
<span class="co">#&gt;  [4] Mutations: matrix with 97 rows and 90 columns </span>
<span class="co">#&gt;  [5] miRNASeqGene: SummarizedExperiment with 471 rows and 80 columns</span></code></pre></div>
<p><em>Key points:</em></p>
<ul>
<li>One matrix-like dataset per list element (although they do not even need to be matrix-like, see for example the <code>RaggedExperiment</code> package)</li>
<li>One matrix column per assayed specimen. Each matrix column must correspond to exactly one row of <code>colData</code>: in other words, you must know which patient or cell line the observation came from. However, multiple columns can come from the same patient, or there can be no data for that patient.</li>
<li>Matrix rows correspond to variables, e.g.Â genes or genomic ranges</li>
<li><code>ExperimentList</code> elements can be genomic range-based (e.g. <code>SummarizedExperiment::RangedSummarizedExperiment-class</code> or <code>RaggedExperiment::RaggedExperiment-class</code>) or ID-based data (e.g. <code>SummarizedExperiment::SummarizedExperiment-class</code>, <code>Biobase::eSet-class</code> <code>base::matrix-class</code>, <code>DelayedArray::DelayedArray-class</code>, and derived classes)</li>
<li>Any data class can be included in the <code>ExperimentList</code>, as long as it supports: single-bracket subsetting (<code>[</code>), <code>dimnames</code>, and <code>dim</code>. Most data classes defined in Bioconductor meet these requirements.</li>
</ul>
</div>
<div id="samplemap---relationship-graph" class="section level3">
<h3><span class="header-section-number">9.7.4</span> sampleMap - relationship graph</h3>
<p><code>sampleMap</code> is a graph representation of the relationship between biological units and experimental results. In simple cases where the column names of <code>ExperimentList</code> data matrices match the row names of <code>colData</code>, the user wonâ€™t need to specify or think about a sample map, it can be created automatically by the <code>MultiAssayExperiment</code> constructor. <code>sampleMap</code> is a simple three-column <code>DataFrame</code>:</p>
<ol style="list-style-type: decimal">
<li><code>assay</code> column: the name of the assay, and found in the names of <code>ExperimentList</code> list names</li>
<li><code>primary</code> column: identifiers of patients or biological units, and found in the row names of <code>colData</code></li>
<li><code>colname</code> column: identifiers of assay results, and found in the column names of <code>ExperimentList</code> elements Helper functions are available for creating a map from a list. See <code>?listToMap</code></li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sampleMap</span>(miniACC)
<span class="co">#&gt; DataFrame with 385 rows and 3 columns</span>
<span class="co">#&gt;               assay      primary                      colname</span>
<span class="co">#&gt;            &lt;factor&gt;  &lt;character&gt;                  &lt;character&gt;</span>
<span class="co">#&gt; 1   RNASeq2GeneNorm TCGA-OR-A5J1 TCGA-OR-A5J1-01A-11R-A29S-07</span>
<span class="co">#&gt; 2   RNASeq2GeneNorm TCGA-OR-A5J2 TCGA-OR-A5J2-01A-11R-A29S-07</span>
<span class="co">#&gt; 3   RNASeq2GeneNorm TCGA-OR-A5J3 TCGA-OR-A5J3-01A-11R-A29S-07</span>
<span class="co">#&gt; 4   RNASeq2GeneNorm TCGA-OR-A5J5 TCGA-OR-A5J5-01A-11R-A29S-07</span>
<span class="co">#&gt; 5   RNASeq2GeneNorm TCGA-OR-A5J6 TCGA-OR-A5J6-01A-31R-A29S-07</span>
<span class="co">#&gt; ...             ...          ...                          ...</span>
<span class="co">#&gt; 381    miRNASeqGene TCGA-PA-A5YG TCGA-PA-A5YG-01A-11R-A29W-13</span>
<span class="co">#&gt; 382    miRNASeqGene TCGA-PK-A5H8 TCGA-PK-A5H8-01A-11R-A29W-13</span>
<span class="co">#&gt; 383    miRNASeqGene TCGA-PK-A5H9 TCGA-PK-A5H9-01A-11R-A29W-13</span>
<span class="co">#&gt; 384    miRNASeqGene TCGA-PK-A5HA TCGA-PK-A5HA-01A-11R-A29W-13</span>
<span class="co">#&gt; 385    miRNASeqGene TCGA-PK-A5HB TCGA-PK-A5HB-01A-11R-A29W-13</span></code></pre></div>
<p><em>Key points:</em></p>
<ul>
<li>relates experimental observations (<code>colnames</code>) to <code>colData</code></li>
<li>permits experiment-specific sample naming, missing, and replicate observations</li>
</ul>
<p style="text-align: right;">
<a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes">back to top</a>
</p>
</div>
<div id="metadata" class="section level3">
<h3><span class="header-section-number">9.7.5</span> metadata</h3>
<p>Metadata can be used to keep additional information about patients, assays performed on individuals or on the entire cohort, or features such as genes, proteins, and genomic ranges. There are many options available for storing metadata. First, <code>MultiAssayExperiment</code> has its own metadata for describing the entire experiment:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">metadata</span>(miniACC)
<span class="co">#&gt; $title</span>
<span class="co">#&gt; [1] &quot;Comprehensive Pan-Genomic Characterization of Adrenocortical Carcinoma&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $PMID</span>
<span class="co">#&gt; [1] &quot;27165744&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $sourceURL</span>
<span class="co">#&gt; [1] &quot;http://s3.amazonaws.com/multiassayexperiments/accMAEO.rds&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $RPPAfeatureDataURL</span>
<span class="co">#&gt; [1] &quot;http://genomeportal.stanford.edu/pan-tcga/show_target_selection_file?filename=Allprotein.txt&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $colDataExtrasURL</span>
<span class="co">#&gt; [1] &quot;http://www.cell.com/cms/attachment/2062093088/2063584534/mmc3.xlsx&quot;</span></code></pre></div>
<p>Additionally, the <code>DataFrame</code> class used by <code>sampleMap</code> and <code>colData</code>, as well as the <code>ExperimentList</code> class, similarly support metadata. Finally, many experimental data objects that can be used in the <code>ExperimentList</code> support metadata. These provide flexible options to users and to developers of derived classes.</p>
</div>
</div>
<div id="multiassayexperiment-subsetting" class="section level2">
<h2><span class="header-section-number">9.8</span> MultiAssayExperiment Subsetting</h2>
<div id="single-bracket" class="section level3">
<h3><span class="header-section-number">9.8.1</span> Single bracket <code>[</code></h3>
<p>In pseudo code below, the subsetting operations work on the rows of the following indices: 1. <em>i</em> experimental data rows 2. <em>j</em> the primary names or the column names (entered as a <code>list</code> or <code>List</code>) 3. <em>k</em> assay</p>
<pre><code>multiassayexperiment[i = rownames, j = primary or colnames, k = assay]</code></pre>
<p>Subsetting operations always return another <code>MultiAssayExperiment</code>. For example, the following will return any rows named â€œMAPK14â€ or â€œIGFBP2â€, and remove any assays where no rows match:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">miniACC[<span class="kw">c</span>(<span class="st">&quot;MAPK14&quot;</span>, <span class="st">&quot;IGFBP2&quot;</span>), , ]</code></pre></div>
<p>The following will keep only patients of pathological stage iv, and all their associated assays:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">miniACC[, miniACC<span class="op">$</span>pathologic_stage <span class="op">==</span><span class="st"> &quot;stage iv&quot;</span>, ]
<span class="co">#&gt; harmonizing input:</span>
<span class="co">#&gt;   removing 311 sampleMap rows with &#39;colname&#39; not in colnames of experiments</span>
<span class="co">#&gt;   removing 74 colData rownames not in sampleMap &#39;primary&#39;</span></code></pre></div>
<p>And the following will keep only the RNA-seq dataset, and only patients for which this assay is available:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">miniACC[, , <span class="st">&quot;RNASeq2GeneNorm&quot;</span>]
<span class="co">#&gt; harmonizing input:</span>
<span class="co">#&gt;   removing 13 colData rownames not in sampleMap &#39;primary&#39;</span></code></pre></div>
</div>
<div id="subsetting-by-genomic-ranges" class="section level3">
<h3><span class="header-section-number">9.8.2</span> Subsetting by genomic ranges</h3>
<p>If any ExperimentList objects have features represented by genomic ranges (e.g. <code>RangedSummarizedExperiment</code>, <code>RaggedExperiment</code>), then a <code>GRanges</code> object in the first subsetting position will subset these objects as in <code>GenomicRanges::findOverlaps()</code>.</p>
</div>
<div id="double-bracket" class="section level3">
<h3><span class="header-section-number">9.8.3</span> Double bracket <code>[[</code></h3>
<p>The â€œdouble bracketâ€ method (<code>[[</code>) is a convenience function for extracting a single element of the <code>MultiAssayExperiment</code> <code>ExperimentList</code>. It avoids the use of <code>experiments(mae)[[1L]]</code>. For example, both of the following extract the <code>ExpressionSet</code> object containing RNA-seq data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">miniACC[[1L]]  <span class="co">#or equivalently, miniACC[[&quot;RNASeq2GeneNorm&quot;]]</span>
<span class="co">#&gt; class: SummarizedExperiment </span>
<span class="co">#&gt; dim: 198 79 </span>
<span class="co">#&gt; metadata(3): experimentData annotation protocolData</span>
<span class="co">#&gt; assays(1): exprs</span>
<span class="co">#&gt; rownames(198): DIRAS3 MAPK14 ... SQSTM1 KCNJ13</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(79): TCGA-OR-A5J1-01A-11R-A29S-07</span>
<span class="co">#&gt;   TCGA-OR-A5J2-01A-11R-A29S-07 ... TCGA-PK-A5HA-01A-11R-A29S-07</span>
<span class="co">#&gt;   TCGA-PK-A5HB-01A-11R-A29S-07</span>
<span class="co">#&gt; colData names(0):</span></code></pre></div>
</div>
</div>
<div id="patients-with-complete-data" class="section level2">
<h2><span class="header-section-number">9.9</span> Patients with complete data</h2>
<p><code>complete.cases()</code> shows which patients have complete data for all assays:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(<span class="kw">complete.cases</span>(miniACC))
<span class="co">#&gt;    Mode   FALSE    TRUE </span>
<span class="co">#&gt; logical      49      43</span></code></pre></div>
<p>The above logical vector could be used for patient subsetting. More simply, <code>intersectColumns()</code> will select complete cases and rearrange each <code>ExperimentList</code> element so its columns correspond exactly to rows of <code>colData</code> in the same order:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">accmatched =<span class="st"> </span><span class="kw">intersectColumns</span>(miniACC)
<span class="co">#&gt; harmonizing input:</span>
<span class="co">#&gt;   removing 170 sampleMap rows with &#39;colname&#39; not in colnames of experiments</span>
<span class="co">#&gt;   removing 49 colData rownames not in sampleMap &#39;primary&#39;</span></code></pre></div>
<p>Note, the column names of the assays in <code>accmatched</code> are not the same because of assay-specific identifiers, but they have been automatically re-arranged to correspond to the same patients. In these TCGA assays, the first three <code>-</code> delimited positions correspond to patient, ie the first patient is <em>TCGA-OR-A5J2</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">colnames</span>(accmatched)
<span class="co">#&gt; CharacterList of length 5</span>
<span class="co">#&gt; [[&quot;RNASeq2GeneNorm&quot;]] TCGA-OR-A5J2-01A-11R-A29S-07 ...</span>
<span class="co">#&gt; [[&quot;gistict&quot;]] TCGA-OR-A5J2-01A-11D-A29H-01 ...</span>
<span class="co">#&gt; [[&quot;RPPAArray&quot;]] TCGA-OR-A5J2-01A-21-A39K-20 ...</span>
<span class="co">#&gt; [[&quot;Mutations&quot;]] TCGA-OR-A5J2-01A-11D-A29I-10 ...</span>
<span class="co">#&gt; [[&quot;miRNASeqGene&quot;]] TCGA-OR-A5J2-01A-11R-A29W-13 ...</span></code></pre></div>
</div>
<div id="row-names-that-are-common-across-assays" class="section level2">
<h2><span class="header-section-number">9.10</span> Row names that are common across assays</h2>
<p><code>intersectRows()</code> keeps only rows that are common to each assay, and aligns them in identical order. For example, to keep only genes where data are available for RNA-seq, GISTIC copy number, and somatic mutations:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">accmatched2 &lt;-<span class="st"> </span><span class="kw">intersectRows</span>(miniACC[, , <span class="kw">c</span>(<span class="st">&quot;RNASeq2GeneNorm&quot;</span>, <span class="st">&quot;gistict&quot;</span>, <span class="st">&quot;Mutations&quot;</span>)])
<span class="kw">rownames</span>(accmatched2)
<span class="co">#&gt; CharacterList of length 3</span>
<span class="co">#&gt; [[&quot;RNASeq2GeneNorm&quot;]] DIRAS3 G6PD KDR ERBB3 ... RET CDKN2A MACC1 CHGA</span>
<span class="co">#&gt; [[&quot;gistict&quot;]] DIRAS3 G6PD KDR ERBB3 AKT1S1 ... PREX1 RET CDKN2A MACC1 CHGA</span>
<span class="co">#&gt; [[&quot;Mutations&quot;]] DIRAS3 G6PD KDR ERBB3 AKT1S1 ... RET CDKN2A MACC1 CHGA</span></code></pre></div>
<p style="text-align: right;">
<a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes">back to top</a>
</p>
</div>
<div id="extraction" class="section level2">
<h2><span class="header-section-number">9.11</span> Extraction</h2>
<div id="assay-and-assays" class="section level3">
<h3><span class="header-section-number">9.11.1</span> assay and assays</h3>
<p>The <code>assay</code> and <code>assays</code> methods follow <code>SummarizedExperiment</code> convention. The <code>assay</code> (singular) method will extract the first element of the <code>ExperimentList</code> and will return a <code>matrix</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(<span class="kw">assay</span>(miniACC))
<span class="co">#&gt; [1] &quot;matrix&quot;</span></code></pre></div>
<p>The <code>assays</code> (plural) method will return a <code>SimpleList</code> of the data with each element being a <code>matrix</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">assays</span>(miniACC)
<span class="co">#&gt; List of length 5</span>
<span class="co">#&gt; names(5): RNASeq2GeneNorm gistict RPPAArray Mutations miRNASeqGene</span></code></pre></div>
<p><em>Key point:</em></p>
<ul>
<li>Whereas the <code>[[</code> returned an assay as its original class, <code>assay()</code> and <code>assays()</code> convert the assay data to matrix form.</li>
</ul>
<p style="text-align: right;">
<a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes">back to top</a>
</p>
</div>
</div>
<div id="summary-of-slots-and-accessors" class="section level2">
<h2><span class="header-section-number">9.12</span> Summary of slots and accessors</h2>
<p>Slot in the <code>MultiAssayExperiment</code> can be accessed or set using their accessor functions:</p>
<table>
<thead>
<tr class="header">
<th>Slot</th>
<th>Accessor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>ExperimentList</code></td>
<td><code>experiments()</code></td>
</tr>
<tr class="even">
<td><code>colData</code></td>
<td><code>colData()</code> and <code>$</code> *</td>
</tr>
<tr class="odd">
<td><code>sampleMap</code></td>
<td><code>sampleMap()</code></td>
</tr>
<tr class="even">
<td><code>metadata</code></td>
<td><code>metadata()</code></td>
</tr>
</tbody>
</table>
<p>__*__ The <code>$</code> operator on a <code>MultiAssayExperiment</code> returns a single column of the <code>colData</code>.</p>
</div>
<div id="transformation-reshaping" class="section level2">
<h2><span class="header-section-number">9.13</span> Transformation / reshaping</h2>
<p>The <code>longFormat</code> or <code>wideFormat</code> functions will â€œreshapeâ€ and combine experiments with each other and with <code>colData</code> into one <code>DataFrame</code>. These functions provide compatibility with most of the common R/Bioconductor functions for regression, machine learning, and visualization.</p>
<div id="longformat" class="section level3">
<h3><span class="header-section-number">9.13.1</span> <code>longFormat</code></h3>
<p>In <em>long</em> format a single column provides all assay results, with additional optional <code>colData</code> columns whose values are repeated as necessary. Here <em>assay</em> is the name of the ExperimentList element, <em>primary</em> is the patient identifier (rowname of colData), <em>rowname</em> is the assay rowname (in this case genes), <em>colname</em> is the assay-specific identifier (column name), <em>value</em> is the numeric measurement (gene expression, copy number, presence of a non-silent mutation, etc), and following these are the <em>vital_status</em> and <em>days_to_death</em> colData columns that have been added:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">longFormat</span>(miniACC[<span class="kw">c</span>(<span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;CTNNB1&quot;</span>), , ],
           <span class="dt">colDataCols =</span> <span class="kw">c</span>(<span class="st">&quot;vital_status&quot;</span>, <span class="st">&quot;days_to_death&quot;</span>))
<span class="co">#&gt; DataFrame with 518 rows and 7 columns</span>
<span class="co">#&gt;               assay      primary     rowname                      colname</span>
<span class="co">#&gt;         &lt;character&gt;  &lt;character&gt; &lt;character&gt;                  &lt;character&gt;</span>
<span class="co">#&gt; 1   RNASeq2GeneNorm TCGA-OR-A5J1        TP53 TCGA-OR-A5J1-01A-11R-A29S-07</span>
<span class="co">#&gt; 2   RNASeq2GeneNorm TCGA-OR-A5J1      CTNNB1 TCGA-OR-A5J1-01A-11R-A29S-07</span>
<span class="co">#&gt; 3   RNASeq2GeneNorm TCGA-OR-A5J2        TP53 TCGA-OR-A5J2-01A-11R-A29S-07</span>
<span class="co">#&gt; 4   RNASeq2GeneNorm TCGA-OR-A5J2      CTNNB1 TCGA-OR-A5J2-01A-11R-A29S-07</span>
<span class="co">#&gt; 5   RNASeq2GeneNorm TCGA-OR-A5J3        TP53 TCGA-OR-A5J3-01A-11R-A29S-07</span>
<span class="co">#&gt; ...             ...          ...         ...                          ...</span>
<span class="co">#&gt; 514       Mutations TCGA-PK-A5HA      CTNNB1 TCGA-PK-A5HA-01A-11D-A29I-10</span>
<span class="co">#&gt; 515       Mutations TCGA-PK-A5HB        TP53 TCGA-PK-A5HB-01A-11D-A29I-10</span>
<span class="co">#&gt; 516       Mutations TCGA-PK-A5HB      CTNNB1 TCGA-PK-A5HB-01A-11D-A29I-10</span>
<span class="co">#&gt; 517       Mutations TCGA-PK-A5HC        TP53 TCGA-PK-A5HC-01A-11D-A30A-10</span>
<span class="co">#&gt; 518       Mutations TCGA-PK-A5HC      CTNNB1 TCGA-PK-A5HC-01A-11D-A30A-10</span>
<span class="co">#&gt;          value vital_status days_to_death</span>
<span class="co">#&gt;      &lt;numeric&gt;    &lt;integer&gt;     &lt;integer&gt;</span>
<span class="co">#&gt; 1     563.4006            1          1355</span>
<span class="co">#&gt; 2    5634.4669            1          1355</span>
<span class="co">#&gt; 3     165.4811            1          1677</span>
<span class="co">#&gt; 4   62658.3913            1          1677</span>
<span class="co">#&gt; 5     956.3028            0            NA</span>
<span class="co">#&gt; ...        ...          ...           ...</span>
<span class="co">#&gt; 514          0            0            NA</span>
<span class="co">#&gt; 515          0            0            NA</span>
<span class="co">#&gt; 516          0            0            NA</span>
<span class="co">#&gt; 517          0            0            NA</span>
<span class="co">#&gt; 518          0            0            NA</span></code></pre></div>
</div>
<div id="wideformat" class="section level3">
<h3><span class="header-section-number">9.13.2</span> <code>wideFormat</code></h3>
<p>In <em>wide</em> format, each feature from each assay goes in a separate column, with one row per primary identifier (patient). Here, each variable becomes a new column:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">wideFormat</span>(miniACC[<span class="kw">c</span>(<span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;CTNNB1&quot;</span>), , ],
           <span class="dt">colDataCols =</span> <span class="kw">c</span>(<span class="st">&quot;vital_status&quot;</span>, <span class="st">&quot;days_to_death&quot;</span>))
<span class="co">#&gt; DataFrame with 92 rows and 9 columns</span>
<span class="co">#&gt;          primary vital_status days_to_death RNASeq2GeneNorm_CTNNB1</span>
<span class="co">#&gt;      &lt;character&gt;    &lt;integer&gt;     &lt;integer&gt;              &lt;numeric&gt;</span>
<span class="co">#&gt; 1   TCGA-OR-A5J1            1          1355              5634.4669</span>
<span class="co">#&gt; 2   TCGA-OR-A5J2            1          1677             62658.3913</span>
<span class="co">#&gt; 3   TCGA-OR-A5J3            0            NA              6337.4256</span>
<span class="co">#&gt; 4   TCGA-OR-A5J4            1           423                     NA</span>
<span class="co">#&gt; 5   TCGA-OR-A5J5            1           365               5979.055</span>
<span class="co">#&gt; ...          ...          ...           ...                    ...</span>
<span class="co">#&gt; 88  TCGA-PK-A5H9            0            NA              5258.9863</span>
<span class="co">#&gt; 89  TCGA-PK-A5HA            0            NA              8120.1654</span>
<span class="co">#&gt; 90  TCGA-PK-A5HB            0            NA              5257.8148</span>
<span class="co">#&gt; 91  TCGA-PK-A5HC            0            NA                     NA</span>
<span class="co">#&gt; 92  TCGA-P6-A5OG            1           383              6390.0997</span>
<span class="co">#&gt;     RNASeq2GeneNorm_TP53 gistict_CTNNB1 gistict_TP53 Mutations_CTNNB1</span>
<span class="co">#&gt;                &lt;numeric&gt;      &lt;numeric&gt;    &lt;numeric&gt;        &lt;numeric&gt;</span>
<span class="co">#&gt; 1               563.4006              0            0                0</span>
<span class="co">#&gt; 2               165.4811              1            0                1</span>
<span class="co">#&gt; 3               956.3028              0            0                0</span>
<span class="co">#&gt; 4                     NA              0            1                0</span>
<span class="co">#&gt; 5              1169.6359              0            0                0</span>
<span class="co">#&gt; ...                  ...            ...          ...              ...</span>
<span class="co">#&gt; 88              890.8663              0            0                0</span>
<span class="co">#&gt; 89              683.5722              0           -1                0</span>
<span class="co">#&gt; 90              237.3697             -1           -1                0</span>
<span class="co">#&gt; 91                    NA              1            1                0</span>
<span class="co">#&gt; 92              815.3446              1           -1               NA</span>
<span class="co">#&gt;     Mutations_TP53</span>
<span class="co">#&gt;          &lt;numeric&gt;</span>
<span class="co">#&gt; 1                0</span>
<span class="co">#&gt; 2                1</span>
<span class="co">#&gt; 3                0</span>
<span class="co">#&gt; 4                0</span>
<span class="co">#&gt; 5                0</span>
<span class="co">#&gt; ...            ...</span>
<span class="co">#&gt; 88               0</span>
<span class="co">#&gt; 89               0</span>
<span class="co">#&gt; 90               0</span>
<span class="co">#&gt; 91               0</span>
<span class="co">#&gt; 92              NA</span></code></pre></div>
</div>
</div>
<div id="multiassayexperiment-class-construction-and-concatenation" class="section level2">
<h2><span class="header-section-number">9.14</span> MultiAssayExperiment class construction and concatenation</h2>
<div id="multiassayexperiment-constructor-function" class="section level3">
<h3><span class="header-section-number">9.14.1</span> MultiAssayExperiment constructor function</h3>
<p>The <code>MultiAssayExperiment</code> constructor function can take three arguments:</p>
<ol style="list-style-type: decimal">
<li><code>experiments</code> - An <code>ExperimentList</code> or <code>list</code> of data</li>
<li><code>colData</code> - A <code>DataFrame</code> describing the patients (or cell lines, or other biological units)</li>
<li><code>sampleMap</code> - A <code>DataFrame</code> of <code>assay</code>, <code>primary</code>, and <code>colname</code> identifiers</li>
</ol>
<p>The miniACC object can be reconstructed as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">MultiAssayExperiment</span>(<span class="dt">experiments=</span><span class="kw">experiments</span>(miniACC),
                     <span class="dt">colData=</span><span class="kw">colData</span>(miniACC),
                     <span class="dt">sampleMap=</span><span class="kw">sampleMap</span>(miniACC),
                     <span class="dt">metadata=</span><span class="kw">metadata</span>(miniACC))
<span class="co">#&gt; A MultiAssayExperiment object of 5 listed</span>
<span class="co">#&gt;  experiments with user-defined names and respective classes. </span>
<span class="co">#&gt;  Containing an ExperimentList class object of length 5: </span>
<span class="co">#&gt;  [1] RNASeq2GeneNorm: SummarizedExperiment with 198 rows and 79 columns </span>
<span class="co">#&gt;  [2] gistict: SummarizedExperiment with 198 rows and 90 columns </span>
<span class="co">#&gt;  [3] RPPAArray: SummarizedExperiment with 33 rows and 46 columns </span>
<span class="co">#&gt;  [4] Mutations: matrix with 97 rows and 90 columns </span>
<span class="co">#&gt;  [5] miRNASeqGene: SummarizedExperiment with 471 rows and 80 columns </span>
<span class="co">#&gt; Features: </span>
<span class="co">#&gt;  experiments() - obtain the ExperimentList instance </span>
<span class="co">#&gt;  colData() - the primary/phenotype DataFrame </span>
<span class="co">#&gt;  sampleMap() - the sample availability DataFrame </span>
<span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment </span>
<span class="co">#&gt;  *Format() - convert into a long or wide DataFrame </span>
<span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></code></pre></div>
</div>
<div id="prepmultiassay---constructor-function-helper" class="section level3">
<h3><span class="header-section-number">9.14.2</span> <code>prepMultiAssay</code> - Constructor function helper</h3>
<p>The <code>prepMultiAssay</code> function allows the user to diagnose typical problems when creating a <code>MultiAssayExperiment</code> object. See <code>?prepMultiAssay</code> for more details.</p>
</div>
<div id="c---concatenate-to-multiassayexperiment" class="section level3">
<h3><span class="header-section-number">9.14.3</span> <code>c</code> - concatenate to MultiAssayExperiment</h3>
<p>The <code>c</code> function allows the user to concatenate an additional experiment to an existing <code>MultiAssayExperiment</code>. The optional <code>sampleMap</code> argument allows concatenating an assay whose column names do not match the row names of <code>colData</code>. For convenience, the <em>mapFrom</em> argument allows the user to map from a particular experiment <strong>provided</strong> that the <strong>order</strong> of the colnames is in the <strong>same</strong>. A <code>warning</code> will be issued to make the user aware of this assumption. For example, to concatenate a matrix of log2-transformed RNA-seq results:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">miniACC2 &lt;-<span class="st"> </span><span class="kw">c</span>(miniACC, <span class="dt">log2rnaseq =</span> <span class="kw">log2</span>(<span class="kw">assays</span>(miniACC)<span class="op">$</span>RNASeq2GeneNorm), <span class="dt">mapFrom=</span>1L)
<span class="co">#&gt; Warning in .local(x, ...): Assuming column order in the data provided </span>
<span class="co">#&gt;  matches the order in &#39;mapFrom&#39; experiment(s) colnames</span>
<span class="kw">experiments</span>(miniACC2)
<span class="co">#&gt; ExperimentList class object of length 6: </span>
<span class="co">#&gt;  [1] RNASeq2GeneNorm: SummarizedExperiment with 198 rows and 79 columns </span>
<span class="co">#&gt;  [2] gistict: SummarizedExperiment with 198 rows and 90 columns </span>
<span class="co">#&gt;  [3] RPPAArray: SummarizedExperiment with 33 rows and 46 columns </span>
<span class="co">#&gt;  [4] Mutations: matrix with 97 rows and 90 columns </span>
<span class="co">#&gt;  [5] miRNASeqGene: SummarizedExperiment with 471 rows and 80 columns </span>
<span class="co">#&gt;  [6] log2rnaseq: matrix with 198 rows and 79 columns</span></code></pre></div>
<p style="text-align: right;">
<a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes">back to top</a>
</p>
</div>
<div id="building-a-multiassayexperiment-from-scratch" class="section level3">
<h3><span class="header-section-number">9.14.4</span> Building a MultiAssayExperiment from scratch</h3>
<p>To start from scratch building your own MultiAssayExperiment, see the package <a href="https://bioconductor.org/packages/release/bioc/vignettes/MultiAssayExperiment/inst/doc/MultiAssayExperiment.html">Coordinating Analysis of Multi-Assay Experiments vignette</a>. The package <a href="https://bioconductor.org/packages/release/bioc/vignettes/MultiAssayExperiment/inst/doc/MultiAssayExperiment_cheatsheet.pdf">cheat sheet</a> is also helpful.</p>
<p>If anything is unclear, please ask a question at <a href="https://support.bioconductor.org/" class="uri">https://support.bioconductor.org/</a> or create an issue on the <a href="https://github.com/waldronlab/MultiAssayExperiment/issues">MultiAssayExperiment issue tracker</a>.</p>
</div>
</div>
<div id="the-cancer-genome-atlas-tcga-as-multiassayexperiments" class="section level2">
<h2><span class="header-section-number">9.15</span> The Cancer Genome Atlas (TCGA) as MultiAssayExperiments</h2>
<p>Most unrestricted TCGA data are available as MultiAssayExperiment objects from the <code>curatedTCGAData</code> package. This represents a lot of harmonization!</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(curatedTCGAData)
<span class="kw">curatedTCGAData</span>(<span class="st">&quot;ACC&quot;</span>)
<span class="co">#&gt;                                  ACC_CNASNP </span>
<span class="co">#&gt;                   &quot;ACC_CNASNP-20160128.rda&quot; </span>
<span class="co">#&gt;                                  ACC_CNVSNP </span>
<span class="co">#&gt;                   &quot;ACC_CNVSNP-20160128.rda&quot; </span>
<span class="co">#&gt;                        ACC_GISTIC_AllByGene </span>
<span class="co">#&gt;         &quot;ACC_GISTIC_AllByGene-20160128.rda&quot; </span>
<span class="co">#&gt;                ACC_GISTIC_ThresholdedByGene </span>
<span class="co">#&gt; &quot;ACC_GISTIC_ThresholdedByGene-20160128.rda&quot; </span>
<span class="co">#&gt;                             ACC_Methylation </span>
<span class="co">#&gt;              &quot;ACC_Methylation-20160128.rda&quot; </span>
<span class="co">#&gt;                            ACC_miRNASeqGene </span>
<span class="co">#&gt;             &quot;ACC_miRNASeqGene-20160128.rda&quot; </span>
<span class="co">#&gt;                                ACC_Mutation </span>
<span class="co">#&gt;                 &quot;ACC_Mutation-20160128.rda&quot; </span>
<span class="co">#&gt;                         ACC_RNASeq2GeneNorm </span>
<span class="co">#&gt;          &quot;ACC_RNASeq2GeneNorm-20160128.rda&quot; </span>
<span class="co">#&gt;                               ACC_RPPAArray </span>
<span class="co">#&gt;                &quot;ACC_RPPAArray-20160128.rda&quot;</span>
acc &lt;-<span class="st"> </span><span class="kw">curatedTCGAData</span>(<span class="st">&quot;ACC&quot;</span>,
    <span class="dt">assays=</span><span class="kw">c</span>(<span class="st">&quot;miRNASeqGene&quot;</span>, <span class="st">&quot;RPPAArray&quot;</span>, <span class="st">&quot;Mutation&quot;</span>, <span class="st">&quot;RNASeq2GeneNorm&quot;</span>),
    <span class="dt">dry.run=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; snapshotDate(): 2018-06-29</span>
<span class="co">#&gt; snapshotDate(): 2018-06-29</span>
<span class="co">#&gt; see ?curatedTCGAData and browseVignettes(&#39;curatedTCGAData&#39;) for documentation</span>
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.ExperimentHub/565&#39;</span>
<span class="co">#&gt; snapshotDate(): 2018-06-29</span>
<span class="co">#&gt; see ?curatedTCGAData and browseVignettes(&#39;curatedTCGAData&#39;) for documentation</span>
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.ExperimentHub/566&#39;</span>
<span class="co">#&gt; snapshotDate(): 2018-06-29</span>
<span class="co">#&gt; see ?curatedTCGAData and browseVignettes(&#39;curatedTCGAData&#39;) for documentation</span>
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.ExperimentHub/567&#39;</span>
<span class="co">#&gt; snapshotDate(): 2018-06-29</span>
<span class="co">#&gt; see ?curatedTCGAData and browseVignettes(&#39;curatedTCGAData&#39;) for documentation</span>
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.ExperimentHub/568&#39;</span>
<span class="co">#&gt; snapshotDate(): 2018-06-29</span>
<span class="co">#&gt; see ?curatedTCGAData and browseVignettes(&#39;curatedTCGAData&#39;) for documentation</span>
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.ExperimentHub/560&#39;</span>
<span class="co">#&gt; snapshotDate(): 2018-06-29</span>
<span class="co">#&gt; see ?curatedTCGAData and browseVignettes(&#39;curatedTCGAData&#39;) for documentation</span>
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.ExperimentHub/563&#39;</span>
<span class="co">#&gt; snapshotDate(): 2018-06-29</span>
<span class="co">#&gt; see ?curatedTCGAData and browseVignettes(&#39;curatedTCGAData&#39;) for documentation</span>
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.ExperimentHub/569&#39;</span>
<span class="co">#&gt; harmonizing input:</span>
<span class="co">#&gt;   removing 620 sampleMap rows not in names(experiments)</span>
acc
<span class="co">#&gt; A MultiAssayExperiment object of 4 listed</span>
<span class="co">#&gt;  experiments with user-defined names and respective classes. </span>
<span class="co">#&gt;  Containing an ExperimentList class object of length 4: </span>
<span class="co">#&gt;  [1] ACC_miRNASeqGene-20160128: SummarizedExperiment with 1046 rows and 80 columns </span>
<span class="co">#&gt;  [2] ACC_Mutation-20160128: RaggedExperiment with 20166 rows and 90 columns </span>
<span class="co">#&gt;  [3] ACC_RNASeq2GeneNorm-20160128: SummarizedExperiment with 20501 rows and 79 columns </span>
<span class="co">#&gt;  [4] ACC_RPPAArray-20160128: SummarizedExperiment with 192 rows and 46 columns </span>
<span class="co">#&gt; Features: </span>
<span class="co">#&gt;  experiments() - obtain the ExperimentList instance </span>
<span class="co">#&gt;  colData() - the primary/phenotype DataFrame </span>
<span class="co">#&gt;  sampleMap() - the sample availability DataFrame </span>
<span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment </span>
<span class="co">#&gt;  *Format() - convert into a long or wide DataFrame </span>
<span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></code></pre></div>
<p>These objects contain most unrestricted TCGA assay and clinical / pathological data, as well as material curated from the supplements of published TCGA primary papers at the end of the colData columns:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">tail</span>(<span class="kw">colnames</span>(<span class="kw">colData</span>(acc)), <span class="dv">10</span>)
<span class="co">#&gt;  [1] &quot;MethyLevel&quot;       &quot;miRNA.cluster&quot;    &quot;SCNA.cluster&quot;    </span>
<span class="co">#&gt;  [4] &quot;protein.cluster&quot;  &quot;COC&quot;              &quot;OncoSign&quot;        </span>
<span class="co">#&gt;  [7] &quot;purity&quot;           &quot;ploidy&quot;           &quot;genome_doublings&quot;</span>
<span class="co">#&gt; [10] &quot;ADS&quot;</span></code></pre></div>
</div>
<div id="plotting-correlation-and-other-analyses" class="section level2">
<h2><span class="header-section-number">9.16</span> Plotting, correlation, and other analyses</h2>
<div id="how-many-samples-have-data-for-each-combination-of-assays" class="section level3">
<h3><span class="header-section-number">9.16.1</span> How many samples have data for each combination of assays?</h3>
<p><strong>Solution</strong></p>
<p>The built-in <code>upsetSamples</code> creates an â€œupsetâ€ Venn diagram to answer this question:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">upsetSamples</span>(miniACC)
<span class="co">#&gt; Loading required namespace: UpSetR</span></code></pre></div>
<p><img src="Ramos_MultiAssayExperiment_files/figure-html/unnamed-chunk-36-1.png" width="672" /></p>
<p>In this dataset only 43 samples have all 5 assays, 32 are missing reverse-phase protein (RPPAArray), 2 are missing Mutations, 1 is missing gistict, 12 have only mutations and gistict, etc.</p>
</div>
</div>
<div id="tcgautils-package" class="section level2">
<h2><span class="header-section-number">9.17</span> <code>TCGAutils</code> package</h2>
<p>Aside from the available reshaping functions already included in the <code>MultiAssayExperiment</code> package, the <code>TCGAutils</code> package provides a suite of helper functions for working with TCGA data.</p>
<div id="what-types-of-samples-are-in-the-data" class="section level3">
<h3><span class="header-section-number">9.17.1</span> What types of samples are in the data?</h3>
<p><strong>Solution</strong></p>
<p>The <code>sampleTables</code> function gives you an overview of samples in each assay:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(TCGAutils)
<span class="kw">sampleTables</span>(acc)
<span class="co">#&gt; $`ACC_miRNASeqGene-20160128`</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 01 </span>
<span class="co">#&gt; 80 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`ACC_Mutation-20160128`</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 01 </span>
<span class="co">#&gt; 90 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`ACC_RNASeq2GeneNorm-20160128`</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 01 </span>
<span class="co">#&gt; 79 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`ACC_RPPAArray-20160128`</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 01 </span>
<span class="co">#&gt; 46</span>

<span class="kw">head</span>(sampleTypes)
<span class="co">#&gt;   Code                                      Definition Short.Letter.Code</span>
<span class="co">#&gt; 1   01                             Primary Solid Tumor                TP</span>
<span class="co">#&gt; 2   02                           Recurrent Solid Tumor                TR</span>
<span class="co">#&gt; 3   03 Primary Blood Derived Cancer - Peripheral Blood                TB</span>
<span class="co">#&gt; 4   04    Recurrent Blood Derived Cancer - Bone Marrow              TRBM</span>
<span class="co">#&gt; 5   05                        Additional - New Primary               TAP</span>
<span class="co">#&gt; 6   06                                      Metastatic                TM</span></code></pre></div>
</div>
<div id="is-there-subtype-data-available-in-the-multiassayexperiment-obtained-from" class="section level3">
<h3><span class="header-section-number">9.17.2</span> Is there subtype data available in the <code>MultiAssayExperiment</code> obtained from</h3>
<p><code>curatedTCGAData</code>?</p>
<p><strong>Solution</strong></p>
<p>The <code>getSubtypeMap</code> function will show actual variable names found in <code>colData</code> that contain subtype information. This can only be obtained from <code>MultiAssayExperiment</code> objects provided by <code>curatedTCGAData</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getSubtypeMap</span>(acc)
<span class="co">#&gt;          ACC_annotations     ACC_subtype</span>
<span class="co">#&gt; 1             Patient_ID          SAMPLE</span>
<span class="co">#&gt; 2  histological_subtypes       Histology</span>
<span class="co">#&gt; 3          mrna_subtypes         C1A/C1B</span>
<span class="co">#&gt; 4          mrna_subtypes         mRNA_K4</span>
<span class="co">#&gt; 5                   cimp      MethyLevel</span>
<span class="co">#&gt; 6      microrna_subtypes   miRNA cluster</span>
<span class="co">#&gt; 7          scna_subtypes    SCNA cluster</span>
<span class="co">#&gt; 8       protein_subtypes protein cluster</span>
<span class="co">#&gt; 9   integrative_subtypes             COC</span>
<span class="co">#&gt; 10     mutation_subtypes        OncoSign</span>
<span class="kw">head</span>(<span class="kw">colData</span>(acc)<span class="op">$</span>Histology)
<span class="co">#&gt; [1] &quot;Usual Type&quot; &quot;Usual Type&quot; &quot;Usual Type&quot; &quot;Usual Type&quot; &quot;Usual Type&quot;</span>
<span class="co">#&gt; [6] &quot;Usual Type&quot;</span></code></pre></div>
<p>A number of other utility functions are available in <code>TCGAutils</code> including TCGA identifier translation from TCGA barcodes to UUIDs and vice versa and helper functions to create Bioconductor objects from TCGA exon files (legacy archive), copy number and GISTIC objects from <code>RTCGAToolbox</code>.</p>
</div>
<div id="kaplan-meier-plot-stratified-by-pathology_n_stage" class="section level3">
<h3><span class="header-section-number">9.17.3</span> Kaplan-meier plot stratified by pathology_N_stage</h3>
<p>Create a Kaplan-meier plot, using pathology_N_stage as a stratifying variable.</p>
<p><strong>Solution</strong></p>
<p>The colData provides clinical data for things like a Kaplan-Meier plot for overall survival stratified by nodal stage.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>({
  <span class="kw">library</span>(survival)
  <span class="kw">library</span>(survminer)
})
<span class="kw">Surv</span>(miniACC<span class="op">$</span>days_to_death, miniACC<span class="op">$</span>vital_status)
<span class="co">#&gt;  [1] 1355  1677    NA+  423   365    NA+  490   579    NA+  922   551 </span>
<span class="co">#&gt; [12] 1750    NA+ 2105    NA+  541    NA+   NA+  490    NA+   NA+  562 </span>
<span class="co">#&gt; [23]   NA+   NA+   NA+   NA+   NA+   NA+  289    NA+   NA+   NA+  552 </span>
<span class="co">#&gt; [34]   NA+   NA+   NA+  994    NA+   NA+  498    NA+   NA+  344    NA+</span>
<span class="co">#&gt; [45]   NA+   NA+   NA+   NA+   NA+   NA+   NA+   NA+   NA+  391   125 </span>
<span class="co">#&gt; [56]   NA+ 1852    NA+   NA+   NA+   NA+   NA+   NA+   NA+ 1204   159 </span>
<span class="co">#&gt; [67] 1197   662   445    NA+ 2385   436  1105    NA+ 1613    NA+   NA+</span>
<span class="co">#&gt; [78] 2405    NA+   NA+   NA+   NA+   NA+  207     0    NA+   NA+   NA+</span>
<span class="co">#&gt; [89]   NA+   NA+   NA+  383</span></code></pre></div>
<p>And remove any patients missing overall survival information:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">miniACCsurv &lt;-<span class="st"> </span>miniACC[, <span class="kw">complete.cases</span>(miniACC<span class="op">$</span>days_to_death, miniACC<span class="op">$</span>vital_status), ]
<span class="co">#&gt; harmonizing input:</span>
<span class="co">#&gt;   removing 248 sampleMap rows with &#39;colname&#39; not in colnames of experiments</span>
<span class="co">#&gt;   removing 58 colData rownames not in sampleMap &#39;primary&#39;</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fit &lt;-<span class="st"> </span><span class="kw">survfit</span>(<span class="kw">Surv</span>(days_to_death, vital_status) <span class="op">~</span><span class="st"> </span>pathology_N_stage, <span class="dt">data =</span> <span class="kw">colData</span>(miniACCsurv))
<span class="kw">ggsurvplot</span>(fit, <span class="dt">data =</span> <span class="kw">colData</span>(miniACCsurv), <span class="dt">risk.table =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="Ramos_MultiAssayExperiment_files/figure-html/unnamed-chunk-41-1.png" width="672" /></p>
</div>
<div id="multivariate-cox-regression-including-rna-seq-copy-number-and-pathology" class="section level3">
<h3><span class="header-section-number">9.17.4</span> Multivariate Cox regression including RNA-seq, copy number, and pathology</h3>
<p>Choose the <em>EZH2</em> gene for demonstration. This subsetting will drop assays with no row named EZH2:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">wideacc =<span class="st"> </span><span class="kw">wideFormat</span>(miniACC[<span class="st">&quot;EZH2&quot;</span>, , ],
    <span class="dt">colDataCols=</span><span class="kw">c</span>(<span class="st">&quot;vital_status&quot;</span>, <span class="st">&quot;days_to_death&quot;</span>, <span class="st">&quot;pathology_N_stage&quot;</span>))
wideacc<span class="op">$</span>y =<span class="st"> </span><span class="kw">Surv</span>(wideacc<span class="op">$</span>days_to_death, wideacc<span class="op">$</span>vital_status)
<span class="kw">head</span>(wideacc)
<span class="co">#&gt; DataFrame with 6 rows and 7 columns</span>
<span class="co">#&gt;        primary vital_status days_to_death pathology_N_stage</span>
<span class="co">#&gt;    &lt;character&gt;    &lt;integer&gt;     &lt;integer&gt;       &lt;character&gt;</span>
<span class="co">#&gt; 1 TCGA-OR-A5J1            1          1355                n0</span>
<span class="co">#&gt; 2 TCGA-OR-A5J2            1          1677                n0</span>
<span class="co">#&gt; 3 TCGA-OR-A5J3            0            NA                n0</span>
<span class="co">#&gt; 4 TCGA-OR-A5J4            1           423                n1</span>
<span class="co">#&gt; 5 TCGA-OR-A5J5            1           365                n0</span>
<span class="co">#&gt; 6 TCGA-OR-A5J6            0            NA                n0</span>
<span class="co">#&gt;   RNASeq2GeneNorm_EZH2 gistict_EZH2      y</span>
<span class="co">#&gt;              &lt;numeric&gt;    &lt;numeric&gt; &lt;Surv&gt;</span>
<span class="co">#&gt; 1              75.8886            0 1355:1</span>
<span class="co">#&gt; 2             326.5332            1 1677:1</span>
<span class="co">#&gt; 3              190.194            1   NA:0</span>
<span class="co">#&gt; 4                   NA           -2  423:1</span>
<span class="co">#&gt; 5             366.3826            1  365:1</span>
<span class="co">#&gt; 6              30.7371            1   NA:0</span></code></pre></div>
<p>Perform a multivariate Cox regression with <em>EZH2</em> copy number (gistict), log2-transformed <em>EZH2</em> expression (RNASeq2GeneNorm), and nodal status (pathology_N_stage) as predictors:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">coxph</span>(<span class="kw">Surv</span>(days_to_death, vital_status) <span class="op">~</span><span class="st"> </span>gistict_EZH2 <span class="op">+</span><span class="st"> </span><span class="kw">log2</span>(RNASeq2GeneNorm_EZH2) <span class="op">+</span><span class="st"> </span>pathology_N_stage,
      <span class="dt">data=</span>wideacc)
<span class="co">#&gt; Call:</span>
<span class="co">#&gt; coxph(formula = Surv(days_to_death, vital_status) ~ gistict_EZH2 + </span>
<span class="co">#&gt;     log2(RNASeq2GeneNorm_EZH2) + pathology_N_stage, data = wideacc)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;                               coef exp(coef) se(coef)     z       p</span>
<span class="co">#&gt; gistict_EZH2               -0.0372    0.9635   0.2821 -0.13 0.89499</span>
<span class="co">#&gt; log2(RNASeq2GeneNorm_EZH2)  0.9773    2.6573   0.2811  3.48 0.00051</span>
<span class="co">#&gt; pathology_N_stagen1         0.3775    1.4586   0.5699  0.66 0.50774</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Likelihood ratio test=16.28  on 3 df, p=0.001</span>
<span class="co">#&gt; n= 26, number of events= 26 </span>
<span class="co">#&gt;    (66 observations deleted due to missingness)</span></code></pre></div>
<p>We see that <em>EZH2</em> expression is significantly associated with overal survival (p &lt; 0.001), but <em>EZH2</em> copy number and nodal status are not. This analysis could easily be extended to the whole genome for discovery of prognostic features by repeated univariate regressions over columns, penalized multivariate regression, etc.</p>
<p>For further detail, see the main MultiAssayExperiment vignette.</p>
<p style="text-align: right;">
<a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes">back to top</a>
</p>
</div>
<div id="correlation-between-rna-seq-and-copy-number" class="section level3">
<h3><span class="header-section-number">9.17.5</span> Correlation between RNA-seq and copy number</h3>
<p><strong>Part 1</strong></p>
<p>For all genes where there is both recurrent copy number (gistict assay) and RNA-seq, calculate the correlation between log2(RNAseq + 1) and copy number. Create a histogram of these correlations. Compare this with the histogram of correlations between all <em>unmatched</em> gene - copy number pairs.</p>
<p><strong>Solution</strong></p>
<p>First, narrow down <code>miniACC</code> to only the assays needed:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">subacc &lt;-<span class="st"> </span>miniACC[, , <span class="kw">c</span>(<span class="st">&quot;RNASeq2GeneNorm&quot;</span>, <span class="st">&quot;gistict&quot;</span>)]</code></pre></div>
<p>Align the rows and columns, keeping only samples with both assays available:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">subacc &lt;-<span class="st"> </span><span class="kw">intersectColumns</span>(subacc)
<span class="co">#&gt; harmonizing input:</span>
<span class="co">#&gt;   removing 15 sampleMap rows with &#39;colname&#39; not in colnames of experiments</span>
<span class="co">#&gt;   removing 15 colData rownames not in sampleMap &#39;primary&#39;</span>
subacc &lt;-<span class="st"> </span><span class="kw">intersectRows</span>(subacc)</code></pre></div>
<p>Create a list of numeric matrices:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">subacc.list &lt;-<span class="st"> </span><span class="kw">assays</span>(subacc)</code></pre></div>
<p>Log-transform the RNA-seq assay:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">subacc.list[[<span class="dv">1</span>]] &lt;-<span class="st"> </span><span class="kw">log2</span>(subacc.list[[<span class="dv">1</span>]] <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)</code></pre></div>
<p>Transpose both, so genes are in columns:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">subacc.list &lt;-<span class="st"> </span><span class="kw">lapply</span>(subacc.list, t)</code></pre></div>
<p>Calculate the correlation between columns in the first matrix and columns in the second matrix:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">corres &lt;-<span class="st"> </span><span class="kw">cor</span>(subacc.list[[<span class="dv">1</span>]], subacc.list[[<span class="dv">2</span>]])</code></pre></div>
<p>And finally, create the histograms:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(<span class="kw">diag</span>(corres))</code></pre></div>
<p><img src="Ramos_MultiAssayExperiment_files/figure-html/unnamed-chunk-50-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(corres[<span class="kw">upper.tri</span>(corres)])</code></pre></div>
<p><img src="Ramos_MultiAssayExperiment_files/figure-html/unnamed-chunk-50-2.png" width="672" /></p>
<p><strong>Part 2</strong></p>
<p>For the gene with highest correlation to copy number, make a box plot of log2 expression against copy number.</p>
<p><strong>Solution</strong></p>
<p>First, identify the gene with highest correlation between expression and copy number:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">which.max</span>(<span class="kw">diag</span>(corres))
<span class="co">#&gt; EIF4E </span>
<span class="co">#&gt;    91</span></code></pre></div>
<p>You could now make the plot by taking the EIF4E columns from each element of the list subacc.list <em>list</em> that was extracted from the subacc <em>MultiAssayExperiment</em>, but letâ€™s do it by subsetting and extracting from the <em>MultiAssayExperiment</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span><span class="kw">wideFormat</span>(subacc[<span class="st">&quot;EIF4E&quot;</span>, , ])
<span class="kw">head</span>(df)
<span class="co">#&gt; DataFrame with 6 rows and 3 columns</span>
<span class="co">#&gt;        primary RNASeq2GeneNorm_EIF4E gistict_EIF4E</span>
<span class="co">#&gt;    &lt;character&gt;             &lt;numeric&gt;     &lt;numeric&gt;</span>
<span class="co">#&gt; 1 TCGA-OR-A5J1              460.6148             0</span>
<span class="co">#&gt; 2 TCGA-OR-A5J2              371.2252             0</span>
<span class="co">#&gt; 3 TCGA-OR-A5J3              516.0717             0</span>
<span class="co">#&gt; 4 TCGA-OR-A5J5             1129.3571             1</span>
<span class="co">#&gt; 5 TCGA-OR-A5J6              822.0782             0</span>
<span class="co">#&gt; 6 TCGA-OR-A5J7              344.5648            -1</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(RNASeq2GeneNorm_EIF4E <span class="op">~</span><span class="st"> </span>gistict_EIF4E,
        <span class="dt">data=</span>df, <span class="dt">varwidth=</span><span class="ot">TRUE</span>,
        <span class="dt">xlab=</span><span class="st">&quot;GISTIC Relative Copy Number Call&quot;</span>,
        <span class="dt">ylab=</span><span class="st">&quot;RNA-seq counts&quot;</span>)</code></pre></div>
<p><img src="Ramos_MultiAssayExperiment_files/figure-html/unnamed-chunk-53-1.png" width="672" /></p>
<p style="text-align: right;">
<a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes">back to top</a>
</p>
</div>
<div id="identifying-correlated-principal-components" class="section level3">
<h3><span class="header-section-number">9.17.6</span> Identifying correlated principal components</h3>
<p>Perform Principal Components Analysis of each of the five assays, using samples available on each assay, log-transforming RNA-seq data first. Using the first 10 components, calculate Pearson correlation between all scores and plot these correlations as a heatmap to identify correlated components across assays.</p>
<p><strong>Solution</strong></p>
<p>Hereâ€™s a function to simplify doing the PCAs:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">getLoadings &lt;-<span class="st"> </span><span class="cf">function</span>(x, <span class="dt">ncomp=</span><span class="dv">10</span>, <span class="dt">dolog=</span><span class="ot">FALSE</span>, <span class="dt">center=</span><span class="ot">TRUE</span>, <span class="dt">scale.=</span><span class="ot">TRUE</span>){
  <span class="cf">if</span>(dolog){
    x &lt;-<span class="st"> </span><span class="kw">log2</span>(x <span class="op">+</span><span class="st"> </span><span class="dv">1</span>)
  }
  pc =<span class="st"> </span><span class="kw">prcomp</span>(x, <span class="dt">center=</span>center, <span class="dt">scale.=</span>scale.)
  <span class="kw">return</span>(<span class="kw">t</span>(pc<span class="op">$</span>rotation[, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]))
}</code></pre></div>
<p>Although it would be possible to do the following with a loop, the different data types do require different options for PCA (e.g.Â mutations are a 0/1 matrix with 1 meaning there is a somatic mutation, and gistict varies between -2 for homozygous loss and 2 for a genome doubling, so neither make sense to scale and center). So it is just as well to do the following one by one, concatenating each PCA results to the MultiAssayExperiment:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">miniACC2 &lt;-<span class="st"> </span><span class="kw">intersectColumns</span>(miniACC)
<span class="co">#&gt; harmonizing input:</span>
<span class="co">#&gt;   removing 170 sampleMap rows with &#39;colname&#39; not in colnames of experiments</span>
<span class="co">#&gt;   removing 49 colData rownames not in sampleMap &#39;primary&#39;</span>
miniACC2 &lt;-<span class="st"> </span><span class="kw">c</span>(miniACC2, <span class="dt">rnaseqPCA=</span><span class="kw">getLoadings</span>(<span class="kw">assays</span>(miniACC2)[[<span class="dv">1</span>]], <span class="dt">dolog=</span><span class="ot">TRUE</span>), <span class="dt">mapFrom=</span>1L)
<span class="co">#&gt; Warning in .local(x, ...): Assuming column order in the data provided </span>
<span class="co">#&gt;  matches the order in &#39;mapFrom&#39; experiment(s) colnames</span>
miniACC2 &lt;-<span class="st"> </span><span class="kw">c</span>(miniACC2, <span class="dt">gistictPCA=</span><span class="kw">getLoadings</span>(<span class="kw">assays</span>(miniACC2)[[<span class="dv">2</span>]], <span class="dt">center=</span><span class="ot">FALSE</span>, <span class="dt">scale.=</span><span class="ot">FALSE</span>), <span class="dt">mapFrom=</span>2L)
<span class="co">#&gt; Warning in .local(x, ...): Assuming column order in the data provided </span>
<span class="co">#&gt;  matches the order in &#39;mapFrom&#39; experiment(s) colnames</span>
miniACC2 &lt;-<span class="st"> </span><span class="kw">c</span>(miniACC2, <span class="dt">proteinPCA=</span><span class="kw">getLoadings</span>(<span class="kw">assays</span>(miniACC2)[[<span class="dv">3</span>]]), <span class="dt">mapFrom=</span>3L)
<span class="co">#&gt; Warning in .local(x, ...): Assuming column order in the data provided </span>
<span class="co">#&gt;  matches the order in &#39;mapFrom&#39; experiment(s) colnames</span>
miniACC2 &lt;-<span class="st"> </span><span class="kw">c</span>(miniACC2, <span class="dt">mutationsPCA=</span><span class="kw">getLoadings</span>(<span class="kw">assays</span>(miniACC2)[[<span class="dv">4</span>]], <span class="dt">center=</span><span class="ot">FALSE</span>, <span class="dt">scale.=</span><span class="ot">FALSE</span>), <span class="dt">mapFrom=</span>4L)
<span class="co">#&gt; Warning in .local(x, ...): Assuming column order in the data provided </span>
<span class="co">#&gt;  matches the order in &#39;mapFrom&#39; experiment(s) colnames</span>
miniACC2 &lt;-<span class="st"> </span><span class="kw">c</span>(miniACC2, <span class="dt">miRNAPCA=</span><span class="kw">getLoadings</span>(<span class="kw">assays</span>(miniACC2)[[<span class="dv">5</span>]]), <span class="dt">mapFrom=</span>5L)
<span class="co">#&gt; Warning in .local(x, ...): Assuming column order in the data provided </span>
<span class="co">#&gt;  matches the order in &#39;mapFrom&#39; experiment(s) colnames</span></code></pre></div>
<p>Now subset to keep <em>only</em> the PCA results:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">miniACC2 &lt;-<span class="st"> </span>miniACC2[, , <span class="dv">6</span><span class="op">:</span><span class="dv">10</span>]
<span class="kw">experiments</span>(miniACC2)
<span class="co">#&gt; ExperimentList class object of length 5: </span>
<span class="co">#&gt;  [1] rnaseqPCA: matrix with 10 rows and 43 columns </span>
<span class="co">#&gt;  [2] gistictPCA: matrix with 10 rows and 43 columns </span>
<span class="co">#&gt;  [3] proteinPCA: matrix with 10 rows and 43 columns </span>
<span class="co">#&gt;  [4] mutationsPCA: matrix with 10 rows and 43 columns </span>
<span class="co">#&gt;  [5] miRNAPCA: matrix with 10 rows and 43 columns</span></code></pre></div>
<p>Note, it would be equally easy (and maybe better) to do PCA on all samples available for each assay, then do intersectColumns at this point instead.</p>
<p>Now, steps for calculating the correlations and plotting a heatmap: * Use <em>wideFormat</em> to paste these together, which has the nice property of adding assay names to the column names. * The first column always contains the sample identifier, so remove it. * Coerce to a matrix * Calculate the correlations, and take the absolute value (since signs of principal components are arbitrary) * Set the diagonals to NA (each variable has a correlation of 1 to itself).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span><span class="kw">wideFormat</span>(miniACC2)[, <span class="op">-</span><span class="dv">1</span>]
mycors &lt;-<span class="st"> </span><span class="kw">cor</span>(<span class="kw">as.matrix</span>(df))
mycors &lt;-<span class="st"> </span><span class="kw">abs</span>(mycors)
<span class="kw">diag</span>(mycors) &lt;-<span class="st"> </span><span class="ot">NA</span></code></pre></div>
<p>To simplify the heatmap, show only components that have at least one correlation greater than 0.5.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">has.high.cor &lt;-<span class="st"> </span><span class="kw">apply</span>(mycors, <span class="dv">2</span>, max, <span class="dt">na.rm=</span><span class="ot">TRUE</span>) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>
mycors &lt;-<span class="st"> </span>mycors[has.high.cor, has.high.cor]
pheatmap<span class="op">::</span><span class="kw">pheatmap</span>(mycors)</code></pre></div>
<p><img src="Ramos_MultiAssayExperiment_files/figure-html/unnamed-chunk-58-1.png" width="672" /></p>
<p>The highest correlation present is between PC2 of the RNA-seq assay, and PC1 of the protein assay.</p>
<p style="text-align: right;">
<a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes">back to top</a>
</p>
</div>
<div id="annotating-with-ranges" class="section level3">
<h3><span class="header-section-number">9.17.7</span> Annotating with ranges</h3>
<p>Convert all the <code>ExperimentList</code> elements in <code>miniACC</code> to <code>RangedSummarizedExperiment</code> objects. Then use <code>rowRanges</code> to annotate these objects with genomic ranges. For the microRNA assay, annotate instead with the genomic coordinates of predicted targets.</p>
<p><strong>Solution</strong></p>
<p>The following shortcut function takes a list of human gene symbols and uses <code>AnnotationFilter</code> and <code>EnsDb.Hsapiens.v86</code> to look up the ranges, and return these as a GRangesList which can be used to replace the rowRanges of the SummarizedExperiment objects:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">getrr &lt;-<span class="st"> </span><span class="cf">function</span>(identifiers, <span class="dt">EnsDbFilterFunc=</span><span class="st">&quot;SymbolFilter&quot;</span>){
  <span class="kw">suppressPackageStartupMessages</span>({
    <span class="kw">library</span>(AnnotationFilter)
    <span class="kw">library</span>(EnsDb.Hsapiens.v86)
  })
    FUN &lt;-<span class="st"> </span><span class="kw">get</span>(EnsDbFilterFunc)
    edb &lt;-<span class="st"> </span>EnsDb.Hsapiens.v86
    afl &lt;-<span class="st"> </span><span class="kw">AnnotationFilterList</span>(<span class="kw">FUN</span>(identifiers),
                                <span class="kw">SeqNameFilter</span>(<span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">21</span>, <span class="st">&quot;X&quot;</span>, <span class="st">&quot;Y&quot;</span>)),
                                <span class="kw">TxBiotypeFilter</span>(<span class="st">&quot;protein_coding&quot;</span>))
    gr &lt;-<span class="st"> </span><span class="kw">genes</span>(edb, <span class="dt">filter=</span>afl)
    grl &lt;-<span class="st"> </span><span class="kw">split</span>(gr, <span class="kw">factor</span>(identifiers))
    grl &lt;-<span class="st"> </span>grl[<span class="kw">match</span>(identifiers, <span class="kw">names</span>(grl))]
    <span class="kw">stopifnot</span>(<span class="kw">identical</span>(<span class="kw">names</span>(grl), identifiers))
    <span class="kw">return</span>(grl)
}</code></pre></div>
<p>For example:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">getrr</span>(<span class="kw">rownames</span>(miniACC)[[<span class="dv">1</span>]])
<span class="co">#&gt; GRangesList object of length 198:</span>
<span class="co">#&gt; $DIRAS3 </span>
<span class="co">#&gt; GRanges object with 1 range and 7 metadata columns:</span>
<span class="co">#&gt;                   seqnames          ranges strand |         gene_id</span>
<span class="co">#&gt;                      &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;</span>
<span class="co">#&gt;   ENSG00000116288        1 7954291-7985505      + | ENSG00000116288</span>
<span class="co">#&gt;                     gene_name   gene_biotype seq_coord_system      symbol</span>
<span class="co">#&gt;                   &lt;character&gt;    &lt;character&gt;      &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt;   ENSG00000116288       PARK7 protein_coding       chromosome       PARK7</span>
<span class="co">#&gt;                   entrezid     tx_biotype</span>
<span class="co">#&gt;                     &lt;list&gt;    &lt;character&gt;</span>
<span class="co">#&gt;   ENSG00000116288    11315 protein_coding</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $MAPK14 </span>
<span class="co">#&gt; GRanges object with 1 range and 7 metadata columns:</span>
<span class="co">#&gt;                   seqnames          ranges strand |         gene_id</span>
<span class="co">#&gt;   ENSG00000116285        1 8004404-8026308      - | ENSG00000116285</span>
<span class="co">#&gt;                   gene_name   gene_biotype seq_coord_system symbol</span>
<span class="co">#&gt;   ENSG00000116285    ERRFI1 protein_coding       chromosome ERRFI1</span>
<span class="co">#&gt;                   entrezid     tx_biotype</span>
<span class="co">#&gt;   ENSG00000116285    54206 protein_coding</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $YAP1 </span>
<span class="co">#&gt; GRanges object with 1 range and 7 metadata columns:</span>
<span class="co">#&gt;                   seqnames            ranges strand |         gene_id</span>
<span class="co">#&gt;   ENSG00000198793        1 11106535-11262507      - | ENSG00000198793</span>
<span class="co">#&gt;                   gene_name   gene_biotype seq_coord_system symbol</span>
<span class="co">#&gt;   ENSG00000198793      MTOR protein_coding       chromosome   MTOR</span>
<span class="co">#&gt;                   entrezid     tx_biotype</span>
<span class="co">#&gt;   ENSG00000198793     2475 protein_coding</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ...</span>
<span class="co">#&gt; &lt;195 more elements&gt;</span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 22 sequences from GRCh38 genome</span></code></pre></div>
<p>Use this to set the rowRanges of experiments 1-4 with these GRangesList objects</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">rseACC &lt;-<span class="st"> </span>miniACC
withRSE &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">5</span>)
<span class="cf">for</span> (i <span class="cf">in</span> withRSE){
  <span class="kw">rowRanges</span>(rseACC[[i]]) &lt;-<span class="st"> </span><span class="kw">getrr</span>(<span class="kw">rownames</span>(rseACC[[i]]))
}</code></pre></div>
<p>Note that the class of experiments 1-4 is now <code>RangedSummarizedExperiment</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">experiments</span>(rseACC)
<span class="co">#&gt; ExperimentList class object of length 5: </span>
<span class="co">#&gt;  [1] RNASeq2GeneNorm: RangedSummarizedExperiment with 198 rows and 79 columns </span>
<span class="co">#&gt;  [2] gistict: RangedSummarizedExperiment with 198 rows and 90 columns </span>
<span class="co">#&gt;  [3] RPPAArray: RangedSummarizedExperiment with 33 rows and 46 columns </span>
<span class="co">#&gt;  [4] Mutations: matrix with 97 rows and 90 columns </span>
<span class="co">#&gt;  [5] miRNASeqGene: RangedSummarizedExperiment with 471 rows and 80 columns</span></code></pre></div>
<p>With ranged objects in the MultiAssayExperiment, you can then do subsetting by ranges. For example, select all genes on chromosome 1 for the four <em>rangedSummarizedExperiment</em> objects:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">rseACC[<span class="kw">GRanges</span>(<span class="dt">seqnames=</span><span class="st">&quot;1:1-1e9&quot;</span>), , withRSE]
<span class="co">#&gt; A MultiAssayExperiment object of 3 listed</span>
<span class="co">#&gt;  experiments with user-defined names and respective classes. </span>
<span class="co">#&gt;  Containing an ExperimentList class object of length 3: </span>
<span class="co">#&gt;  [1] RNASeq2GeneNorm: RangedSummarizedExperiment with 22 rows and 79 columns </span>
<span class="co">#&gt;  [2] gistict: RangedSummarizedExperiment with 22 rows and 90 columns </span>
<span class="co">#&gt;  [3] RPPAArray: RangedSummarizedExperiment with 3 rows and 46 columns </span>
<span class="co">#&gt; Features: </span>
<span class="co">#&gt;  experiments() - obtain the ExperimentList instance </span>
<span class="co">#&gt;  colData() - the primary/phenotype DataFrame </span>
<span class="co">#&gt;  sampleMap() - the sample availability DataFrame </span>
<span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment </span>
<span class="co">#&gt;  *Format() - convert into a long or wide DataFrame </span>
<span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span></code></pre></div>
<p><em>Note about microRNA</em>: You can set ranges for the microRNA assay according to the genomic location of those microRNA, or the locations of their predicted targets, but we donâ€™t do it here. Assigning genomic ranges of microRNA targets would be an easy way to subset them according to their targets.</p>
<p style="text-align: right;">
<a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes">back to top</a>
</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction-to-bioconductor-annotation-resources.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="maintaining-your-bioconductor-package.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
